--- WindowMaker-0.92.0/po/Makefile.am.vns	2004-10-27 01:03:42 +0400
+++ WindowMaker-0.92.0/po/Makefile.am	2005-09-06 08:55:14 +0400
@@ -7,7 +7,7 @@
 # Please keep this sorted in alphabetic order!
 EXTRA_DIST = be.po bg.po bs.po ca.po cs.po da.po de.po el.po es.po et.po \
 	fi.po fr.po gl.po hr.po hu.po it.po ja.po ko.po ms.po nl.po no.po \
-	pl.po pt.po ro.po ru.po sk.po sv.po tr.po zh_CN.po zh_TW.po
+	pl.po pt.po ro.po ru.po sk.po sv.po tr.po uk.po zh_CN.po zh_TW.po
 
 POTFILES  = \
 	$(top_builddir)/src/appicon.c \
--- WindowMaker-0.92.0/configure.ac.vns	2005-09-06 08:55:13 +0400
+++ WindowMaker-0.92.0/configure.ac	2005-09-06 08:57:02 +0400
@@ -289,8 +289,8 @@
 
 dnl List of supported locales
 dnl -------------------------
-supported_locales="be bg bs ca cs da de el es et fi fr gl hr hu it ja ko ms nl no pl pt ro ru sk sv tr zh_CN zh_TW"
-supported_wprefs_locales="bg ca cs de es et fi fr hr hu it ja ko pt ru sk zh_CN zh_TW"
+supported_locales="be bg bs ca cs da de el es et fi fr gl hr hu it ja ko ms nl no pl pt ro ru sk sv tr uk zh_CN zh_TW"
+supported_wprefs_locales="bg ca cs de es et fi fr hr hu it ja ko pt ru sk uk zh_CN zh_TW"
 supported_wings_locales="bg ca cs de fr sk"
 
 for lang in $LINGUAS; do
