--- WindowMaker-0.91.0/src/session.c.vns	2005-05-18 12:44:44 +0400
+++ WindowMaker-0.91.0/src/session.c	2005-05-18 12:46:21 +0400
@@ -331,7 +331,8 @@
 
         if ((wwin->transient_for==None
              || wwin->transient_for==wwin->screen_ptr->root_win)
-            && WMGetFirstInArray(wapp_list, (void*)appId)==WANotFound
+            && (WMGetFirstInArray(wapp_list, (void*)appId)==WANotFound
+		|| WFLAGP(wwin, shared_appicon))
             && !WFLAGP(wwin, dont_save_session)) {
             /* A entry for this application was not yet saved. Save one. */
             if ((win_info = makeWindowState(wwin, wapp))!=NULL) {
