diff -Naur netcat-1.10-23.orig/netcat.c netcat-1.10-23/netcat.c
--- netcat-1.10.orig/netcat.c   2004-08-28 00:16:10.000000000 -0500
+++ netcat-1.10/netcat.c        2004-08-28 20:09:33.000000000 -0500
@@ -1360,6 +1360,28 @@
   return (0);
 } /* readwrite */
 
+/* unescape :
+   translate \-'s into -'s, returns start */
+char * unescape(start)
+     char * start;
+{
+  char * end;
+  char * next;
+  char * p;
+
+  end = start + strlen(start);
+  next = start;
+
+  while (next = strstr (next+1, "\\-")) {
+    p = next;
+    while (p < end) /* copy string back one char, overwriting backslash */
+      *(p++) = *(p+1);
+    end--;
+  }
+
+  return start;
+} /* unescape */
+
 /* main :
    now we pull it all together... */
 main (argc, argv)
@@ -1627,13 +1649,21 @@
    argument, so we can control the pattern somewhat. */
   while (argv[optind]) {
     hiport = loport = 0;
+    /* I know it's ugly to have this test twice, but I'd rather not have
+       it do all of the dash code if there aren't any dashes at all */
     cp = strchr (argv[optind], '-');	/* nn-mm range? */
     if (cp) {
-      *cp = '\0';
-      cp++;
-      hiport = getportpoop (cp, 0);
-      if (hiport == 0)
-	bail ("invalid port %s", cp);
+      while (cp && *(cp-1) == '\\') /* if dash escaped by backslash */
+        cp = strchr (cp+1, '-');
+
+      if (cp) { /* it's a range */
+        *cp = '\0';
+        unescape(++cp); /* turn \-'s into -'s */
+        hiport = getportpoop (cp, 0);
+        if (hiport == 0)
+          bail ("invalid port %s", cp);
+      }
+      unescape(argv[optind]); /* turn \-'s into -'s */
     } /* if found a dash */
     loport = getportpoop (argv[optind], 0);
     if (loport == 0)
@@ -1807,7 +1807,8 @@
 	-v			verbose [use twice to be more verbose]\n\
 	-w secs			timeout for connects and final net reads\n\
 	-z			zero-I/O mode [used for scanning]");
-  bail ("port numbers can be individual or ranges: lo-hi [inclusive]");
+  bail ("port numbers can be individual or ranges: lo-hi [inclusive];\n\
+hyphens in port names must be backslash escaped (e.g. 'ftp\\-data').");
 } /* helpme */
 #endif /* HAVE_HELP */

diff -Naur netcat-1.10.orig/nc.1 netcat-1.10/nc.1
--- netcat-1.10.orig/nc.1	2004-11-23 16:59:15.000000000 -0600
+++ netcat-1.10/nc.1	2004-11-23 16:59:02.000000000 -0600
@@ -131,6 +131,10 @@
 mail me a check.  If you are affiliated in any way with Microsoft
 Network, get a life.  Always ski in control.  Comments, questions, and
 patches to hobbit@avian.org.
+.SH NOTES
+Some port names in /etc/services contain hyphens -- netcat currently
+will not correctly parse those unless you escape the hyphens with
+backslashes (e.g. "netcat localhost 'ftp\\\-data'").
 .SH BUGS
 Efforts have been made to have netcat "do the right thing" in all its
 various modes.  If you believe that it is doing the wrong thing under
@@ -141,10 +145,6 @@
 and I certainly encourage people to make custom mods and send in any
 improvements they make to it. Continued feedback from the Internet
 community is always welcome!
-.P
-Some port names in /etc/services contain hyphens -- netcat currently
-will not correctly parse those, so specify ranges using numbers if you
-can.
 .SH "SEE ALSO"
 /usr/share/doc/netcat/README.gz
 .SH AUTHOR
