--- emacs-21.4-orig/src/editfns.c	2003-01-20 12:03:33.000000000 +0100
+++ emacs-21.4/src/editfns.c	2007-11-25 17:02:54.000000000 +0100
@@ -3266,7 +3266,7 @@
 		  && *format != 'i' && *format != 'X' && *format != 'c')
 		error ("Invalid format operation %%%c", *format);
 
-	    thissize = 30;
+	    thissize = 30 + (precision > 0 ? precision : 0);
 	    if (*format == 'c'
 		&& (! SINGLE_BYTE_CHAR_P (XINT (args[n]))
 		    || XINT (args[n]) == 0))
