Update configure.in to work with autoconf 2.61.

--- emacs-21.4-orig/Makefile.in
+++ emacs-21.4/Makefile.in
@@ -273,7 +273,7 @@
 config.status: ${srcdir}/configure
 	./config.status --recheck
 
-${srcdir}/configure: configure.in aclocal.m4
+${srcdir}/configure: configure.in
 	cd ${srcdir} && autoconf
 
 src/Makefile: $(srcdir)/src/Makefile.in config.status
--- emacs-21.4-orig/configure.in
+++ emacs-21.4/configure.in
@@ -23,7 +23,7 @@
 dnl  Free Software Foundation, Inc., 59 Temple Place - Suite 330,
 dnl  Boston, MA 02111-1307, USA.
 
-AC_PREREQ(2.8)dnl
+AC_PREREQ(2.61)dnl
 AC_INIT(src/lisp.h)
 AC_CONFIG_HEADER(src/config.h:src/config.in)
 
@@ -1345,6 +1345,9 @@
   CPPFLAGS="$c_switch_system $c_switch_machine $CPPFLAGS"
 fi
 
+dnl For AC_FUNC_GETLOADAVG, at least:
+AC_CONFIG_LIBOBJ_DIR(src)
+
 dnl Do this early because it can frob feature test macros for Unix-98 &c.
 AC_SYS_LARGEFILE
 
@@ -2069,11 +2072,11 @@
     AC_CHECK_HEADERS(krb5.h)
   else
     AC_CHECK_HEADERS(des.h,,
-		     AC_CHECK_HEADERS(kerberosIV/des.h,,
-				      AC_CHECK_HEADERS(kerberos/des.h)))
+		     [AC_CHECK_HEADERS(kerberosIV/des.h,,
+				      [AC_CHECK_HEADERS(kerberos/des.h)])])
     AC_CHECK_HEADERS(krb.h,,
-		     AC_CHECK_HEADERS(kerberosIV/krb.h,,
-				      AC_CHECK_HEADERS(kerberos/krb.h)))
+		     [AC_CHECK_HEADERS(kerberosIV/krb.h,,
+				      [AC_CHECK_HEADERS(kerberos/krb.h)])])
   fi
   AC_CHECK_HEADERS(com_err.h)
 fi
