http://bugs.gentoo.org/165466

--- emacs-21.4-orig/Makefile.in
+++ emacs-21.4/Makefile.in
@@ -448,7 +448,7 @@
 	-chmod -R a+r ${datadir}/emacs ${COPYDESTS} ${infodir}
 	thisdir=`/bin/pwd`; \
 	cd ${srcdir}/etc; \
-	for page in emacs etags ctags gfdl ; do \
+	for page in emacs emacsclient etags ctags ; do \
 	  (cd $${thisdir}; \
 	   ${INSTALL_DATA} ${srcdir}/etc/$${page}.1 ${man1dir}/$${page}${manext}; \
 	   chmod a+r ${man1dir}/$${page}${manext}); \
