Index: xawtv-3.95/console/Subdir.mk
===================================================================
--- xawtv-3.95.orig/console/Subdir.mk
+++ xawtv-3.95/console/Subdir.mk
@@ -84,12 +84,12 @@ console/record   : LDLIBS  += $(CURSES_L
 console/v4l-conf : LDLIBS  += $(ATHENA_LIBS)
 
 # linker flags
-console/fbtv     : LDFLAGS := $(DLFLAGS)
-console/ttv      : LDFLAGS := $(DLFLAGS)
-console/scantv   : LDFLAGS := $(DLFLAGS)
-console/streamer : LDFLAGS := $(DLFLAGS)
-console/webcam   : LDFLAGS := $(DLFLAGS)
-console/v4l-conf : LDFLAGS := $(BINDNOW_FLAGS)
+console/fbtv     : LDLIBS += $(DLFLAGS)
+console/ttv      : LDLIBS += $(DLFLAGS)
+console/scantv   : LDLIBS += $(DLFLAGS)
+console/streamer : LDLIBS += $(DLFLAGS)
+console/webcam   : LDLIBS += $(DLFLAGS)
+console/v4l-conf : LDFLAGS += $(BINDNOW_FLAGS)
 
 # global targets
 all:: $(TARGETS-console) $(TARGETS-v4l-conf)
Index: xawtv-3.95/x11/Subdir.mk
===================================================================
--- xawtv-3.95.orig/x11/Subdir.mk
+++ xawtv-3.95/x11/Subdir.mk
@@ -110,11 +110,11 @@ x11/xawtv-remote : LDLIBS  += $(ATHENA_L
 x11/propwatch    : LDLIBS  += $(ATHENA_LIBS)
 
 # linker flags
-x11/xawtv        : LDFLAGS := $(DLFLAGS)
-x11/motv         : LDFLAGS := $(DLFLAGS)
-x11/mtt          : LDFLAGS := $(DLFLAGS)
-x11/v4lctl       : LDFLAGS := $(DLFLAGS)
-x11/pia          : LDFLAGS := $(DLFLAGS)
+x11/xawtv        : LDLIBS += $(DLFLAGS)
+x11/motv         : LDLIBS += $(DLFLAGS)
+x11/mtt          : LDLIBS += $(DLFLAGS)
+x11/v4lctl       : LDLIBS += $(DLFLAGS)
+x11/pia          : LDLIBS += $(DLFLAGS)
 
 # compile flags
 x11/complete-xaw.o   : CFLAGS += -DATHENA=1
