diff -Naur virtinst-0.300.3.orig/virtinst/FullVirtGuest.py virtinst-0.300.3/virtinst/FullVirtGuest.py
--- virtinst-0.300.3.orig/virtinst/FullVirtGuest.py	2008-08-11 16:49:47.000000000 +0200
+++ virtinst-0.300.3/virtinst/FullVirtGuest.py	2008-08-11 16:53:09.000000000 +0200
@@ -52,6 +52,8 @@
                                           "generic24": { "label": "Generic 2.4.x kernel" }, \
                                           "generic26": { "label": "Generic 2.6.x kernel" }, \
                                           "ubuntuHardy": { "label": "Ubuntu Hardy", "distro": "ubuntu", "virtio_blk" : False, "virtio_net" : True }, \
+                                          "gentoostable": { "label": "Gentoo Stable", "distro": "gentoo", "virtio_blk" : False, "virtio_net" : True }, \
+                                          "gentoounstable": { "label": "Gentoo Unstable (kernel >=2.6.26)", "distro": "gentoo", "virtio_blk" : True, "virtio_net" : True }, \
                                           }, \
                             }, \
                  "windows": { "label": "Windows", \
@@ -80,6 +82,7 @@
                                          "solaris10": { "label": "Sun Solaris 10" }, \
                                          "freebsd6": { "label": "Free BSD 6.x" }, \
                                          "openbsd4": { "label": "Open BSD 4.x" }, \
+                                         "gentoofreebsd": { "label": "Gentoo/FreeBSD" }, \
                                          }, \
                            }, \
                  "other": { "label": "Other", \
