Index: xine-lib-1.1.1/m4/ffmpeg.m4
===================================================================
--- xine-lib-1.1.1.orig/m4/ffmpeg.m4
+++ xine-lib-1.1.1/m4/ffmpeg.m4
@@ -65,10 +65,10 @@ if test x"$external_ffmpeg" != "xno"; th
         dnl look for libpostproc inside libavcodec
         AC_CHECK_LIB(avcodec, pp_get_context,
           [external_ffmpeg_found=yes
-          FFMPEG_LIBS="${FFMPEG_LIBS} -lavcodec"
+          FFMPEG_LIBS="${FFMPEG_LIBS} -lavcodec -lm -lavutil"
           break],
           ,
-          []
+          [-lm -lavutil]
         )
 
         dnl look for shared libpostproc and avcodec
@@ -76,13 +76,14 @@ if test x"$external_ffmpeg" != "xno"; th
           AC_CHECK_LIB(avcodec, register_avcodec,
             [external_ffmpeg_found=yes
             FFMPEG_POSTPROC_LIBS="${FFMPEG_LIBS} -lpostproc"
-            FFMPEG_LIBS="${FFMPEG_LIBS} -lavcodec"
-            break]
+            FFMPEG_LIBS="${FFMPEG_LIBS} -lavcodec -lm -lavutil"
+            break],,
+	    [-lm -lavutil]
           ),,
           [-lavcodec]
         )
       done
-      if test x$FFMPEG_POSTPROC_LIBS = "x"; then
+      if test "x$FFMPEG_POSTPROC_LIBS" = "x"; then
         FFMPEG_POSTPROC_LIBS="${FFMPEG_LIBS}"
       fi
 
