Index: xine-lib-1.1.2cvs20060328/src/demuxers/Makefile.am
===================================================================
--- xine-lib-1.1.2cvs20060328.orig/src/demuxers/Makefile.am
+++ xine-lib-1.1.2cvs20060328/src/demuxers/Makefile.am
@@ -19,10 +19,7 @@ if HAVE_LIBMNG
 mng_module = xineplug_dmx_mng.la
 endif
 
-if HAVE_WAND
-image_module = xineplug_dmx_image.la
-endif
-if HAVE_GDK_PIXBUF
+if BUILD_DMX_IMAGE
 image_module = xineplug_dmx_image.la
 endif
 
Index: xine-lib-1.1.2cvs20060328/src/libxinevdec/Makefile.am
===================================================================
--- xine-lib-1.1.2cvs20060328.orig/src/libxinevdec/Makefile.am
+++ xine-lib-1.1.2cvs20060328/src/libxinevdec/Makefile.am
@@ -38,5 +38,5 @@ xineplug_decode_image_la_LDFLAGS = $(WAN
 xineplug_decode_gdk_pixbuf_la_SOURCES = gdkpixbuf.c
 xineplug_decode_gdk_pixbuf_la_CFLAGS = $(GDK_PIXBUF_CFLAGS)
 xineplug_decode_gdk_pixbuf_la_LIBADD = $(XINE_LIB) $(DYNAMIC_LD_LIBS) $(GDK_PIXBUF_LIBS)
-xineplug_decode_gdk_pixbuf_la_LDFLAGS = $(GDK_PIXBUF_LDFLAGS) -avoid-version -module @XINE_PLUGIN_MIN_SYMS@
+xineplug_decode_gdk_pixbuf_la_LDFLAGS = -avoid-version -module @XINE_PLUGIN_MIN_SYMS@
 
Index: xine-lib-1.1.2cvs20060328/configure.ac
===================================================================
--- xine-lib-1.1.2cvs20060328.orig/configure.ac
+++ xine-lib-1.1.2cvs20060328/configure.ac
@@ -2251,6 +2251,8 @@ dnl as long as neither autoconf nor auto
 dnl check we will have to call it
 _AM_DEPENDENCIES([OBJC])
 
+AM_CONDITIONAL([BUILD_DMX_IMAGE], [test x"$have_imagemagick" = "xyes" -o x"$no_gdkpixbuf" != "xyes"])
+
 
 dnl ---------------------------------------------
 dnl Output configuration files
