Index: xine-lib-1.1.2cvs20060328/src/demuxers/demux_qt.c
===================================================================
--- xine-lib-1.1.2cvs20060328.orig/src/demuxers/demux_qt.c
+++ xine-lib-1.1.2cvs20060328/src/demuxers/demux_qt.c
@@ -3048,13 +3048,13 @@ static char *get_identifier (demux_class
 }
 
 static char *get_extensions (demux_class_t *this_gen) {
-  return "mov qt mp4 m4a";
+  return "mov qt mp4 m4a m4b";
 }
 
 static char *get_mimetypes (demux_class_t *this_gen) {
   return "video/quicktime: mov,qt: Quicktime animation;"
          "video/x-quicktime: mov,qt: Quicktime animation;"
-         "audio/x-m4a: m4a: MPEG-4 audio;"
+         "audio/x-m4a: m4a,m4b: MPEG-4 audio;"
          "application/x-quicktimeplayer: qtl: Quicktime list;";
 }
 
