Index: xine-lib-1.1.2cvs20060328/src/input/input_http.c
===================================================================
--- xine-lib-1.1.2cvs20060328.orig/src/input/input_http.c
+++ xine-lib-1.1.2cvs20060328/src/input/input_http.c
@@ -300,9 +300,13 @@ static int http_plugin_read_metainf (htt
     lprintf ("http_plugin_read_metainf: %s\n", metadata_buf);
 
     /* Extract the title of the current song */
-    if ((songtitle = strstr(metadata_buf, "StreamTitle='"))) {
-      songtitle += 13; /* skip "StreamTitle='" */
-      if ((title_end = strchr(songtitle, '\''))) {
+    if ((songtitle = strstr(metadata_buf, "StreamTitle="))) {
+      songtitle += 12; /* skip "StreamTitle=" */
+      char terminator[] = { ';', '\0', '\0' };
+      if ( songtitle[0] == '\'' || songtitle[0] == '"' )
+      { terminator[0] = songtitle[0]; terminator[1] = ';'; songtitle++; }
+
+      if ((title_end = strstr(songtitle, terminator))) {
         *title_end = '\0';
         
         if ((!this->shoutcast_songtitle ||
