Index: xine-lib-1.1.2cvs20060328/src/libmpeg2/idct_altivec.c
===================================================================
--- xine-lib-1.1.2cvs20060328.orig/src/libmpeg2/idct_altivec.c
+++ xine-lib-1.1.2cvs20060328/src/libmpeg2/idct_altivec.c
@@ -489,7 +489,7 @@ void mpeg2_idct_add_altivec (int16_t * b
 	 );
 }
 
-static const int16_t constants[5][8] ATTR_ALIGN(16) = {
+static int16_t constants[5][8] ATTR_ALIGN(16) = {
     {23170, 13573, 6518, 21895, -23170, -21895, 32, 31},
     {16384, 22725, 21407, 19266, 16384, 19266, 21407, 22725},
     {22725, 31521, 29692, 26722, 22725, 26722, 29692, 31521},
