http://bugs.gentoo.org/293573

--- openMotif-2.2.3-orig/clients/uil/Makefile.am
+++ openMotif-2.2.3/clients/uil/Makefile.am
@@ -6,7 +6,7 @@
 #
 bin_PROGRAMS = uil
 
-LDFLAGS = -version-info @CURRENT@:@REVISION@:@AGE@
+AM_LDFLAGS = -version-info @CURRENT@:@REVISION@:@AGE@
 
 UilParser.lo: UilParser.c
 	$(LTCOMPILE) -DCALLABLE -c UilParser.c
--- openMotif-2.2.3-orig/lib/Mrm/Makefile.am
+++ openMotif-2.2.3/lib/Mrm/Makefile.am
@@ -8,7 +8,7 @@
 
 libMrm_la_LIBADD = ../Xm/libXm.la ${X_LIBS} -lXt ${X_PRE_LIBS} -lX11 ${X_EXTRA_LIBS}
 
-LDFLAGS = -version-info @CURRENT@:@REVISION@:@AGE@
+AM_LDFLAGS = -version-info @CURRENT@:@REVISION@:@AGE@
 
 lib_LTLIBRARIES = libMrm.la
 
--- openMotif-2.2.3-orig/lib/Xm/Makefile.am
+++ openMotif-2.2.3/lib/Xm/Makefile.am
@@ -9,7 +9,7 @@
            ${X_CFLAGS}
 xmdir = $(includedir)/Xm
 
-LDFLAGS = -version-info @CURRENT@:@REVISION@:@AGE@
+AM_LDFLAGS = -version-info @CURRENT@:@REVISION@:@AGE@
 
 lib_LTLIBRARIES = libXm.la
 
