--- resin-3.2.1.vanilla/Makefile.am	2009-01-10 14:55:35.000000000 +0000
+++ resin-3.2.1/Makefile.am	2009-02-02 19:37:41.000000000 +0000
@@ -9,21 +9,21 @@
 
 install	:
 	(cd modules/c/src; $(MAKE) install)
-	if test $(PREFIX) != `pwd`; then \
-	  mkdir -p $(PREFIX)/lib; \
-	  mkdir -p $(PREFIX)/$(LIBEXEC); \
-	  cp -r $(LIBEXEC)/* $(PREFIX)/$(LIBEXEC); \
-	  cp lib/*.jar $(PREFIX)/lib; \
-	  mkdir -p $(PREFIX)/bin; \
-	  cp bin/* $(PREFIX)/bin; \
-	  mkdir -p $(PREFIX)/webapps; \
-	  cp -r webapps/* $(PREFIX)/webapps; \
-	  mkdir -p $(PREFIX)/doc; \
-	  cp -r doc/* $(PREFIX)/doc; \
-	  mkdir -p $(PREFIX)/conf; \
-	  cp conf/app-default.xml $(PREFIX)/conf/app-default.xml.orig; \
-	  if test ! -r $(PREFIX)/conf/resin.xml; then \
-	    cp conf/resin.xml $(PREFIX)/conf/resin.xml; \
-	    cp conf/app-default.xml $(PREFIX)/conf/app-default.xml; \
+	if test $(DESTDIR)/$(PREFIX) != `pwd`; then \
+	  mkdir -p $(DESTDIR)/$(PREFIX)/lib; \
+	  mkdir -p $(DESTDIR)/$(PREFIX)/$(LIBEXEC); \
+	  cp -r $(LIBEXEC)/* $(DESTDIR)/$(PREFIX)/$(LIBEXEC); \
+	  cp lib/*.jar $(DESTDIR)/$(PREFIX)/lib; \
+	  mkdir -p $(DESTDIR)/$(PREFIX)/bin; \
+	  cp bin/* $(DESTDIR)/$(PREFIX)/bin; \
+	  mkdir -p $(DESTDIR)/$(PREFIX)/webapps; \
+	  cp -r webapps/* $(DESTDIR)/$(PREFIX)/webapps; \
+	  mkdir -p $(DESTDIR)/$(PREFIX)/doc; \
+	  cp -r doc/* $(DESTDIR)/$(PREFIX)/doc; \
+	  mkdir -p $(DESTDIR)/$(PREFIX)/conf; \
+	  cp conf/app-default.xml $(DESTDIR)/$(PREFIX)/conf/app-default.xml.orig; \
+	  if test ! -r $(DESTDIR)/$(PREFIX)/conf/resin.xml; then \
+	    cp conf/resin.xml $(DESTDIR)/$(PREFIX)/conf/resin.xml; \
+	    cp conf/app-default.xml $(DESTDIR)/$(PREFIX)/conf/app-default.xml; \
 	  fi; \
 	fi
