diff -Nru resin-3.2.1.vanilla/modules/resin/src/boot-manifest resin-3.2.1/modules/resin/src/boot-manifest
--- resin-3.2.1.vanilla/modules/resin/src/boot-manifest	2009-05-03 15:22:11.000000000 +0100
+++ resin-3.2.1/modules/resin/src/boot-manifest	2009-05-03 15:26:04.000000000 +0100
@@ -1,5 +1,5 @@
 Main-Class: com.caucho.server.bootstrap.ResinBoot
-Class-Path: resin.jar jta-101.jar isorelax.jar jsdk-24.jar jmx-12.jar jca-15.jar jstl-11.jar
+Class-Path: resin.jar jta-101.jar isorelax.jar resin-servlet-api.jar jmx-12.jar jca-15.jar jstl-11.jar
 Extension-Name: com.caucho.server.bootstrap
 Implementation-Vendor: Caucho Technology, Inc.
 Implementation-Version: 3.0.1
diff -Nru resin-3.2.1.vanilla/modules/resin/src/manifest resin-3.2.1/modules/resin/src/manifest
--- resin-3.2.1.vanilla/modules/resin/src/manifest	2009-05-03 15:22:11.000000000 +0100
+++ resin-3.2.1/modules/resin/src/manifest	2009-05-03 15:27:41.000000000 +0100
@@ -1,4 +1,4 @@
-Class-Path: javaee-16.jar pro.jar license.jar hessian.jar quercus.jar resin-util.jar osgi-14.jar
+Class-Path: javaee-16.jar pro.jar license.jar hessian.jar quercus.jar resin-util.jar osgi-14.jar resin-servlet-api.jar
 Implementation-Vendor: Caucho Technology, Inc.
 Implementation-Version: 3.2.1
 J2EE-DeploymentFactory-Implementation-Class: com.caucho.j2ee.deployclient.DeploymentFactoryImpl
diff -Nru resin-3.2.1.vanilla/modules/resin/src/manifest.in resin-3.2.1/modules/resin/src/manifest.in
--- resin-3.2.1.vanilla/modules/resin/src/manifest.in	2009-05-03 15:22:11.000000000 +0100
+++ resin-3.2.1/modules/resin/src/manifest.in	2009-05-03 15:45:14.000000000 +0100
@@ -1,4 +1,4 @@
-Class-Path: javaee-16.jar pro.jar license.jar hessian.jar quercus.jar resin-util.jar osgi-14.jar
+Class-Path: javaee-16.jar pro.jar license.jar hessian.jar quercus.jar resin-util.jar osgi-14.jar resin-servlet-api.jar
 Implementation-Vendor: Caucho Technology, Inc.
 Implementation-Version: @VERSION@
 J2EE-DeploymentFactory-Implementation-Class: com.caucho.j2ee.deployclient.DeploymentFactoryImpl
