--- mozilla/security/coreconf/FreeBSD.mk.old	2009-01-25 14:34:25 +0000
+++ mozilla/security/coreconf/FreeBSD.mk	2009-01-25 14:34:42 +0000
@@ -65,7 +65,7 @@
 
 ARCH			= freebsd
 
-MOZ_OBJFORMAT		:= $(shell test -x /usr/bin/objformat && /usr/bin/objformat || echo aout)
+MOZ_OBJFORMAT		:= $(shell test -x /usr/bin/objformat && /usr/bin/objformat || echo elf)
 
 ifeq ($(MOZ_OBJFORMAT),elf)
 DLL_SUFFIX		= so
--- mozilla/nsprpub/configure.in.old	2009-01-25 14:34:52 +0000
+++ mozilla/nsprpub/configure.in	2009-01-25 14:35:04 +0000
@@ -1137,7 +1137,7 @@
     AC_DEFINE(HAVE_BSD_FLOCK)
     AC_DEFINE(HAVE_SOCKLEN_T)
     CFLAGS="$CFLAGS $(DSO_CFLAGS) -ansi -Wall"
-    MOZ_OBJFORMAT=`test -x /usr/bin/objformat && /usr/bin/objformat || echo aout`
+    MOZ_OBJFORMAT=`test -x /usr/bin/objformat && /usr/bin/objformat || echo elf`
     if test "$MOZ_OBJFORMAT" = "elf"; then
         DLL_SUFFIX=so
     else
--- mozilla/directory/c-sdk/config/FreeBSD.mk.old	2009-01-25 14:35:25 +0000
+++ mozilla/directory/c-sdk/config/FreeBSD.mk	2009-01-25 14:35:41 +0000
@@ -70,7 +70,7 @@
 
 ARCH			= freebsd
 
-MOZ_OBJFORMAT          := $(shell test -x /usr/bin/objformat && /usr/bin/objformat || echo aout)
+MOZ_OBJFORMAT          := $(shell test -x /usr/bin/objformat && /usr/bin/objformat || echo elf)
 
 ifeq ($(MOZ_OBJFORMAT),elf)
 DLL_SUFFIX		= so
--- mozilla/directory/c-sdk/configure.in.old	2009-01-25 14:35:16 +0000
+++ mozilla/directory/c-sdk/configure.in	2009-01-25 14:35:34 +0000
@@ -1037,7 +1037,7 @@
     AC_DEFINE(FREEBSD)
     AC_DEFINE(HAVE_BSD_FLOCK)
     CFLAGS="$CFLAGS $(DSO_CFLAGS) -ansi -Wall"
-    MOZ_OBJFORMAT=`test -x /usr/bin/objformat && /usr/bin/objformat || echo aout`
+    MOZ_OBJFORMAT=`test -x /usr/bin/objformat && /usr/bin/objformat || echo elf`
     if test "$MOZ_OBJFORMAT" = "elf"; then
         DLL_SUFFIX=so
     else
--- mozilla/configure.in.old	2009-01-25 13:53:32 +0000
+++ mozilla/configure.in	2009-01-25 13:53:39 +0000
@@ -1556,7 +1556,7 @@
     ;;
 
 *-freebsd*)
-    if test `test -x /usr/bin/objformat && /usr/bin/objformat || echo aout` != "elf"; then
+    if test `test -x /usr/bin/objformat && /usr/bin/objformat || echo elf` != "elf"; then
 	DLL_SUFFIX=".so.1.0"
 	DSO_LDOPTS="-shared"
     fi
