--- depscan.sh	2003-05-12 23:23:05.000000000 +0200
+++ ../sbin.new/depscan.sh	2003-12-22 01:31:42.000000000 +0100
@@ -3,16 +3,18 @@
 # Distributed under the terms of the GNU General Public License v2
 # Author:  Martin Schlemmer <azarah@gentoo.org>
 # $Header: /var/cvsroot/gentoo/users/beejay/baselayout/work/patches-sbin/depscan.sh.patch,v 1.1 2003/12/31 20:35:54 beejay Exp $
+TEXTDOMAINDIR=/etc/gentoo/locale
+TEXTDOMAIN=baselayout
 
 source /etc/init.d/functions.sh
 
-ebegin "Caching service dependencies"
+ebegin $"Caching service dependencies"
 
 if [ ! -d "${svcdir}" ]
 then
 	if ! install -d -m0755 "${svcdir}" 2>/dev/null
 	then
-		eerror " Could not create needed directory '${svcdir}'!"
+		eerror $" Could not create needed directory '${svcdir}'!"
 	fi
 fi
 
@@ -22,7 +24,7 @@
 	then
 		if ! install -d -m0755 "${svcdir}/${x}" 2>/dev/null
 		then
-			eerror " Could not create needed directory '${svcdir}/${x}'!"
+			eerror $" Could not create needed directory '${svcdir}/${x}'!"
 		fi
 	fi
 done
@@ -52,7 +54,7 @@
 	-f /lib/rcscripts/awk/functions.awk \
 	-f /lib/rcscripts/awk/gendepends.awk
 
-eend $? "Failed to cache service dependencies"
+eend $? $"Failed to cache service dependencies"
 
 
 # vim:ts=4
