--- functions.sh	2003-08-11 01:48:38.000000000 +0200
+++ ../sbin.new/functions.sh	2003-12-22 14:20:03.000000000 +0100
@@ -1,6 +1,10 @@
 # Copyright 1999-2003 Gentoo Technologies, Inc.
 # Distributed under the terms of the GNU General Public License v2
 # $Header: /var/cvsroot/gentoo/users/beejay/baselayout/work/patches-sbin/functions.sh.patch,v 1.1 2003/12/31 20:35:54 beejay Exp $
+export LANG=$(/bin/cat /etc/gentoo/locale/locale)
+export LC_ALL=$(/bin/cat /etc/gentoo/locale/locale)
+TEXTDOMAINDIR=/etc/gentoo/locale
+TEXTDOMAIN=baselayout
 
 
 set -a
