--- /home/benni/gentoo_ev/baselayout/etc/init.d.old/bootmisc	2003-07-22 23:07:08.000000000 +0200
+++ /home/benni/gentoo_ev/baselayout/etc/init.d.new/bootmisc	2003-12-22 21:18:15.000000000 +0100
@@ -2,6 +2,16 @@
 # Copyright 1999-2003 Gentoo Technologies, Inc.
 # Distributed under the terms of the GNU General Public License v2
 # $Header: /var/cvsroot/gentoo/users/beejay/baselayout/work/patches-init.d/bootmisc.patch,v 1.1 2003/12/31 20:35:54 beejay Exp $
+set -a
+
+export LANG=$(/bin/cat /etc/gentoo/locale/locale)
+export LC_ALL=$(/bin/cat /etc/gentoo/locale/locale)
+
+export TEXTDOMAINDIR=/etc/gentoo/locale
+export TEXTDOMAIN=baselayout
+
+#TEXTDOMAINDIR=/etc/gentoo/locale
+#TEXTDOMAIN=baselayout
 
 
 depend() {
@@ -16,20 +26,20 @@
 		# system startup is complete.
 		if [ "${DELAYLOGIN}" = "yes" ]
 		then
-			echo "System bootup in progress - please wait" > /etc/nologin
+			echo $"System bootup in progress - please wait" > /etc/nologin
 			cp /etc/nologin /etc/nologin.boot &>/dev/null
 		fi
 
 		if [ -e /etc/sysctl.conf ]
 		then
-			ebegin "Configuring kernel parameters"
+			ebegin $"Configuring kernel parameters"
 			/sbin/sysctl -p /etc/sysctl.conf &>/dev/null
 			eend 0
 		fi
 
 		if [ -x /sbin/env-update.sh ]
 		then
-			ebegin "Updating environment"
+			ebegin $"Updating environment"
 			/sbin/env-update.sh >/dev/null
 			eend 0
 		fi
@@ -40,7 +50,7 @@
 		#
 		# Clean up any stale locks.
 		#
-		ebegin "Cleaning /var/lock, /var/run"
+		ebegin $"Cleaning /var/lock, /var/run"
 		( find /var/lock -type f -exec rm -f -- {} \; )
 		
 		#
@@ -76,7 +86,7 @@
 		#
 		# Clean up /tmp directory
 		#
-		ebegin "Cleaning /tmp directory"
+		ebegin $"Cleaning /tmp directory"
 		rm -f /tmp/.X*-lock /tmp/esrv* /tmp/kio* /tmp/jpsock.* /tmp/.fam* &>/dev/null
 		rm -rf /tmp/.esd* /tmp/orbit-* /tmp/ssh-* /tmp/ksocket-* /tmp/.*-unix &>/dev/null
 		# Make sure our X11 stuff have the correct permissions
