--- /home/benni/gentoo_ev/baselayout/etc/init.d.old/checkroot	2003-07-31 23:08:24.000000000 +0200
+++ /home/benni/gentoo_ev/baselayout/etc/init.d.new/checkroot	2003-12-22 15:20:38.000000000 +0100
@@ -3,6 +3,12 @@
 # Distributed under the terms of the GNU General Public License v2
 # $Header: /var/cvsroot/gentoo/users/beejay/baselayout/work/patches-init.d/checkroot.patch,v 1.1 2003/12/31 20:35:54 beejay Exp $
 
+export LANG=$(/bin/cat /etc/gentoo/locale/locale)
+export LC_ALL=$(/bin/cat /etc/gentoo/locale/locale)
+
+export TEXTDOMAINDIR=/etc/gentoo/locale
+export TEXTDOMAIN=baselayout
+
 
 depend() {
 	before *
@@ -10,17 +16,17 @@
 
 start() {
 	local retval=0
-	
+
 	if [ ! -f /fastboot ]
 	then
-		ebegin "Remounting root filesystem read-only (if necessary)"
+		ebegin $"Remounting root filesystem read-only (if necessary)"
 		mount / -n -o remount,ro &>/dev/null
 		eend $?
 
-		ebegin "Checking root filesystem"
+		ebegin $"Checking root filesystem"
 		if [ -f /forcefsck ]
 		then
-			ewarn "A full fsck has been forced"
+			ewarn $"A full fsck has been forced"
 			fsck -C -a -f /
 			# /forcefsck isn't deleted because checkfs needs it.
 			# it'll be deleted in that script.
@@ -34,22 +40,22 @@
 			eend 0
 		elif [ "${retval}" -eq 1 ]
 		then
-			ewend 1 "Filesystem repaired"
+			ewend 1 $"Filesystem repaired"
 		elif [ "${retval}" -eq 2 -o "${retval}" -eq 3 ]
 		then
-			ewend 1 "Filesystem repaired, but reboot needed!"
+			ewend 1 $"Filesystem repaired, but reboot needed!"
 			echo -ne "\a"; sleep 1; echo -ne "\a"; sleep 1
 			echo -ne "\a"; sleep 1; echo -ne "\a"; sleep 1
-			ewarn "Rebooting in 10 seconds ..."
+			ewarn $"Rebooting in 10 seconds ..."
 			sleep 10
-			einfo "Rebooting"
+			einfo $"Rebooting"
 			/sbin/reboot -f
 		else
-			eend 2 "Filesystem couldn't be fixed :("
+			eend 2 $"Filesystem couldn't be fixed :("
 			/sbin/sulogin ${CONSOLE}
-			einfo "Unmounting filesystems"
+			einfo $"Unmounting filesystems"
 			/bin/mount -a -o remount,ro &>/dev/null
-			einfo "Rebooting"
+			einfo $"Rebooting"
 			/sbin/reboot -f
 		fi
 	fi
@@ -58,11 +64,11 @@
 	if mount -vf -o remount / 2> /dev/null | \
 			awk '{ if ($6 ~ /rw/) exit 0; else exit 1; }'
 	then
-		ebegin "Remounting root filesystem read/write"
+		ebegin $"Remounting root filesystem read/write"
 		mount / -n -o remount,rw &>/dev/null
 		if [ "$?" -ne 0 ]
 		then
-			eend 2 "Root filesystem could not be mounted read/write:("
+			eend 2 $"Root filesystem could not be mounted read/write:("
 			/sbin/sulogin ${CONSOLE}
 		else
 			eend 0
