--- /home/benni/gentoo_ev/baselayout/etc/init.d.old/checkfs	2003-08-04 21:41:15.000000000 +0200
+++ /home/benni/gentoo_ev/baselayout/etc/init.d.new/checkfs	2003-12-21 22:54:06.000000000 +0100
@@ -2,7 +2,8 @@
 # Copyright 1999-2003 Gentoo Technologies, Inc.
 # Distributed under the terms of the GNU General Public License v2
 # $Header: /var/cvsroot/gentoo/users/beejay/baselayout/work/patches-init.d/checkfs.patch,v 1.1 2003/12/31 20:35:54 beejay Exp $
-
+TEXTDOMAINDIR=/etc/gentoo/locale
+TEXTDOMAIN=baselayout
 
 depend() {
 	need checkroot modules
@@ -16,7 +17,7 @@
 	#
 	if [ -f /sbin/evms_activate ]
 	then
-		ebegin "Activating EVMS2"
+		ebegin $"Activating EVMS2"
 		evms_activate
 		retval = $?	
 		eend ${retval}
@@ -32,7 +33,7 @@
 	if [ -x /sbin/vgscan ] && \
 	   [ -d /proc/lvm -o "$(grep device-mapper /proc/misc 2>/dev/null)" ]
 	then
-		ebegin "Setting up the Logical Volume Manager"
+		ebegin $"Setting up the Logical Volume Manager"
 		#still echo stderr for debugging
 		/sbin/vgscan >/dev/null 
 		if [ "$?" -eq 0 ] && [ -x /sbin/vgchange ] && \
@@ -40,7 +41,7 @@
 		then
 			/sbin/vgchange -a y >/dev/null
 		fi
-		eend $? "Failed to setup the LVM"
+		eend $? $"Failed to setup the LVM"
 	fi
 
 	# Start software raid.
@@ -62,7 +63,7 @@
 		then
 			raidtools_devices=$(awk '/^[[:space:]]*raiddev/ { print $2 }' /etc/raidtab)
 		fi
-		ebegin "Starting up RAID devices: " 
+		ebegin $"Starting up RAID devices: " 
 
 		local rc=0
 		local retval=0
@@ -83,7 +84,7 @@
 				local retval=1
 				local noauto="$(egrep "^${i}" /etc/fstab | grep -c 'noauto')"
 
-				einfon "  Trying ${raiddev}..."
+				einfon $"  Trying ${raiddev}..."
 
 				raiddev=""
 				
@@ -126,7 +127,7 @@
 				local retval=1
 				local noauto="$(egrep "^${i}" /etc/fstab | grep -c 'noauto')"
 
-				einfon "  Trying ${raiddev}..."
+				einfon $"  Trying ${raiddev}..."
 
 				raiddev=""
 				
@@ -168,14 +169,14 @@
 		if [ "${rc}" -gt 0 ]
 		then
 			echo
-			eerror "An error occurred during the RAID startup"
-			eerror "Dropping you to a shell; the system will reboot"
-			eerror "when you leave the shell."
+			eerror $"An error occurred during the RAID startup"
+			eerror $"Dropping you to a shell; the system will reboot"
+			eerror $"when you leave the shell."
 			echo; echo
 			/sbin/sulogin ${CONSOLE}
-			einfo "Unmounting filesystems"
+			einfo $"Unmounting filesystems"
 			/bin/mount -a -o remount,ro &>/dev/null
-			einfo "Rebooting"
+			einfo $"Rebooting"
 			/sbin/reboot -f
 		fi
 	fi
@@ -184,10 +185,10 @@
 	then
 		rm -f /fastboot
 	else
-		ebegin "Checking all filesystems"
+		ebegin $"Checking all filesystems"
 		if [ -f /forcefsck ]
 		then
-			ewarn "A full fsck has been forced"
+			ewarn $"A full fsck has been forced"
 			fsck -C -R -A -a -f
 			retval=$?
 			rm -f /forcefsck
@@ -200,11 +201,11 @@
 			eend 0
 		elif [ "${retval}" -gt 1 -a "${retval}" -lt 4 ]
 		then
-			ewend 1 "Filesystem errors corrected."
+			ewend 1 $"Filesystem errors corrected."
 			# Everything should be ok, so return a pass
 			return 0
 		else
-			eend 2 "Fsck could not correct all errors, manual repair needed"
+			eend 2 $"Fsck could not correct all errors, manual repair needed"
 			/sbin/sulogin ${CONSOLE}
 		fi
 	fi
