--- /home/benni/gentoo_ev/baselayout/etc/init.d.old/clock	2003-07-16 20:27:55.000000000 +0200
+++ /home/benni/gentoo_ev/baselayout/etc/init.d.new/clock	2003-12-22 01:31:10.000000000 +0100
@@ -2,6 +2,8 @@
 # Copyright 1999-2003 Gentoo Technologies, Inc.
 # Distributed under the terms of the GNU General Public License v2
 # $Header: /var/cvsroot/gentoo/users/beejay/baselayout/work/patches-init.d/clock.patch,v 1.1 2003/12/31 20:35:54 beejay Exp $
+TEXTDOMAINDIR=/etc/gentoo/locale
+TEXTDOMAIN=baselayout
 
 
 opts="${opts} save"
@@ -29,26 +31,26 @@
 
 	if [ -n "$(grep 'UML' /proc/cpuinfo)" ]
 	then
-		ebegin "Setting system clock to hardware clock ${TBLURB}"
-		eend 0 "Detected UML Environment; ignoring"
+		ebegin $"Setting system clock to hardware clock ${TBLURB}"
+		eend 0 $"Detected UML Environment; ignoring"
 		return 0
 
 	elif [ -x /sbin/hwclock ]
 	then
-		ebegin "Setting system clock to hardware clock ${TBLURB}"
+		ebegin $"Setting system clock to hardware clock ${TBLURB}"
 		errstr="$(/sbin/hwclock --adjust ${myopts} 2>&1 >/dev/null)"
 		errstr="${errstr}$(/sbin/hwclock --hctosys ${myopts} 2>&1 >/dev/null)"
 		#hwclock do not return a fail status
 		if [ -n "${errstr}" ]
 		then
-			eend 1 "Failed to set system clock to hardware clock"
+			eend 1 $"Failed to set system clock to hardware clock"
 			return 1
 		else
 			eend 0
 			return 0
 		fi
 	else
-		eend 1 "hwclock executable not found"
+		eend 1 $"hwclock executable not found"
 	fi
 }
 
@@ -67,24 +69,24 @@
 
 	if [ -n "$(grep 'UML' /proc/cpuinfo)" ]
 	then
-		ebegin "Syncing hardware clock to system clock ${TBLURB}"
-		eend 0 "Detected UML Environment; ignoring"
+		ebegin $"Syncing hardware clock to system clock ${TBLURB}"
+		eend 0 $"Detected UML Environment; ignoring"
 		return 0
 		
 	elif [ -x /sbin/hwclock ]
 	then
-		ebegin "Syncing hardware clock to system clock ${TBLURB}"
+		ebegin $"Syncing hardware clock to system clock ${TBLURB}"
 		errstr="$(/sbin/hwclock --systohc ${myopts} 2>&1 >/dev/null)"
 		if [ -n "${errstr}" ]
 		then
-			eend 1 "Failed to set hardware clock to system clock"
+			eend 1 $"Failed to set hardware clock to system clock"
 			return 1
 		else
 			eend 0
 			return 0
 		fi
 	else
-		eend 1 "hwclock executable not found"
+		eend 1 $"hwclock executable not found"
 	fi
 }
 
