--- /home/benni/gentoo_ev/baselayout/etc/init.d.old/consolefont	2003-08-10 23:31:36.000000000 +0200
+++ /home/benni/gentoo_ev/baselayout/etc/init.d.new/consolefont	2003-12-22 01:31:26.000000000 +0100
@@ -2,6 +2,8 @@
 # Copyright 1999-2003 Gentoo Technologies, Inc.
 # Distributed under the terms of the GNU General Public License v2
 # $Header: /var/cvsroot/gentoo/users/beejay/baselayout/work/patches-init.d/consolefont.patch,v 1.1 2003/12/31 20:35:54 beejay Exp $
+TEXTDOMAINDIR=/etc/gentoo/locale
+TEXTDOMAIN=baselayout
 
 
 depend() {
@@ -27,7 +29,7 @@
 			# Console-Tools section:
 			#
 			
-			ebegin "Setting user font"
+			ebegin $"Setting user font"
 			for x in 1 2 3 4 5 6 7 8 9 10
 			do
 				# Using DEVFS ?
@@ -49,7 +51,7 @@
 			# Kbd Section
 			#
 			
-			ebegin "Setting user font"
+			ebegin $"Setting user font"
 			/usr/bin/setfont ${CONSOLEFONT} ${param} >/dev/null
 			retval=$?
 			
@@ -68,13 +70,13 @@
 	
 		else
 			# No console program installed!
-			eend 1 "The setfont/consolechars executable not found"
+			eend 1 $"The setfont/consolechars executable not found"
 			return 1
 		fi
 		
-		eend $retval "Failed to set user font"
+		eend $retval $"Failed to set user font"
 	else
-		ebegin "Using the default console font"
+		ebegin $"Using the default console font"
 		eend 0
 	fi
 	
