--- /home/benni/gentoo_ev/baselayout/etc/init.d.old/crypto-loop	2003-01-06 22:32:43.000000000 +0100
+++ /home/benni/gentoo_ev/baselayout/etc/init.d.new/crypto-loop	2003-12-22 01:31:34.000000000 +0100
@@ -3,6 +3,8 @@
 # Distributed under the terms of the GNU General Public License v2
 # Author: Matthew Kennedy <mkennedy@gentoo.org>
 # $Header: /var/cvsroot/gentoo/users/beejay/baselayout/work/patches-init.d/crypto-loop.patch,v 1.1 2003/12/31 20:35:54 beejay Exp $
+TEXTDOMAINDIR=/etc/gentoo/locale
+TEXTDOMAIN=baselayout
 
 loopconf="/etc/conf.d/crypto-loop"
 
@@ -14,7 +16,7 @@
 start() {
 	local status="0"
 
-	ebegin "Starting crypto loop devices"
+	ebegin $"Starting crypto loop devices"
 	
 	if [ -e ${loopconf} ]
 	then
@@ -28,18 +30,18 @@
 		
 			if [ "${configured}" != "yes" ]
 			then
-				einfo "  Loop ${loop} on device ${device} (cipher ${cipher}, key size ${keysize}): "
+				einfo $"  Loop ${loop} on device ${device} (cipher ${cipher}, key size ${keysize}): "
 				if ! /sbin/losetup -e ${cipher} -k ${keysize} ${loop} ${device} ${other}
 				then
-					ewarn "Failure configuring ${loop}.  Skipping."
+					ewarn $"Failure configuring ${loop}.  Skipping."
 					status=1
 				fi
 			else
-				ewarn "  Loop ${loop} on device ${device} are already configured"
+				ewarn $"  Loop ${loop} on device ${device} are already configured"
 			fi
 		done 
 	fi
-	ewend ${status} "Failed to start some loop devices."
+	ewend ${status} $"Failed to start some loop devices."
 
 	# We do not fail if some loop devices did not start ...
 	return 0
