--- /home/benni/gentoo_ev/baselayout/etc/init.d.old/domainname	2003-05-21 22:14:50.000000000 +0200
+++ /home/benni/gentoo_ev/baselayout/etc/init.d.new/domainname	2003-12-22 01:31:49.000000000 +0100
@@ -2,6 +2,8 @@
 # Copyright 1999-2003 Gentoo Technologies, Inc.
 # Distributed under the terms of the GNU General Public License v2 
 # $Header: /var/cvsroot/gentoo/users/beejay/baselayout/work/patches-init.d/domainname.patch,v 1.1 2003/12/31 20:35:54 beejay Exp $
+TEXTDOMAINDIR=/etc/gentoo/locale
+TEXTDOMAIN=baselayout
 
 
 depend() {
@@ -37,10 +39,10 @@
 	then
 		mynisdomain="$(< /etc/nisdomainname)"
 
-		ebegin "Setting NIS domainname to ${mynisdomain}"
+		ebegin $"Setting NIS domainname to ${mynisdomain}"
 		/bin/domainname "${mynisdomain}"
 		retval=$?
-		eend ${retval} "Failed to set the NIS domainname"
+		eend ${retval} $"Failed to set the NIS domainname"
 		
 	fi
 
@@ -49,7 +51,7 @@
 		mydnsdomain="$(< /etc/dnsdomainname)"
 
 		[ ! -f /etc/resolv.conf ] && touch /etc/resolv.conf
-		ebegin "Setting DNS domainname to ${mydnsdomain}"
+		ebegin $"Setting DNS domainname to ${mydnsdomain}"
 		gawk -v DOMAIN="${mydnsdomain}" \
 			'BEGIN { print "domain " DOMAIN }
 			 $0 !~ /^[[:space:]]*domain/ { print }' \
@@ -58,7 +60,7 @@
 		[ "${retval2}" -eq 0 ] \
 			&& (mv -f /etc/resolv.conf.new /etc/resolv.conf) \
 			|| (rm -f /etc/resolv.conf.new)
-		eend ${retval2} "Failed to set the DNS domainname"
+		eend ${retval2} $"Failed to set the DNS domainname"
 	fi
 
 	retval=$((retval + retval2))
