--- /home/benni/gentoo_ev/baselayout/etc/init.d.old/hostname	2003-05-05 01:16:31.000000000 +0200
+++ /home/benni/gentoo_ev/baselayout/etc/init.d.new/hostname	2003-12-22 01:32:19.000000000 +0100
@@ -2,6 +2,8 @@
 # Copyright 1999-2003 Gentoo Technologies, Inc.
 # Distributed under the terms of the GNU General Public License v2 
 # $Header: /var/cvsroot/gentoo/users/beejay/baselayout/work/patches-init.d/hostname.patch,v 1.1 2003/12/31 20:35:54 beejay Exp $
+TEXTDOMAINDIR=/etc/gentoo/locale
+TEXTDOMAIN=baselayout
 
 # This only set what /bin/hostname returns.  If you need to setup NIS, meaning
 # what /bin/domainname returns, please see:
@@ -19,7 +21,7 @@
 checkconfig() {
 	if [ ! -f /etc/hostname ] || [ -z "$(< /etc/hostname)" ]
 	then
-		eerror "You need to set /etc/hostname to a valid hostname"
+		eerror $"You need to set /etc/hostname to a valid hostname"
 		return 1
 	fi
 }
@@ -45,10 +47,10 @@
 		myhost="$(< /etc/hostname)"
 	fi
 	
-	ebegin "Setting hostname to ${myhost}"
+	ebegin $"Setting hostname to ${myhost}"
 	/bin/hostname "${myhost}"
 	retval=$?
-	eend ${retval} "Failed to set the hostname"
+	eend ${retval} $"Failed to set the hostname"
 	
 	if [ "${retval}" -eq 0 ]
 	then
