--- /home/benni/gentoo_ev/baselayout/etc/init.d.old/keymaps	2003-07-16 21:38:51.000000000 +0200
+++ /home/benni/gentoo_ev/baselayout/etc/init.d.new/keymaps	2003-12-22 01:32:29.000000000 +0100
@@ -2,6 +2,8 @@
 # Copyright 1999-2003 Gentoo Technologies, Inc.
 # Distributed under the terms of the GNU General Public License v2
 # $Header: /var/cvsroot/gentoo/users/beejay/baselayout/work/patches-init.d/keymaps.patch,v 1.1 2003/12/31 20:35:54 beejay Exp $
+TEXTDOMAINDIR=/etc/gentoo/locale
+TEXTDOMAIN=baselayout
 
 
 depend() {
@@ -11,7 +13,7 @@
 checkconfig() {
 	if [ -z "${KEYMAP}" ]
 	then
-		eerror "You need to setup \$KEYMAP in /etc/rc.conf first"
+		eerror $"You need to setup \$KEYMAP in /etc/rc.conf first"
 		return 1
 	fi
 }
@@ -27,7 +29,7 @@
 		echo 1 > /proc/sys/dev/mac_hid/keyboard_sends_linux_keycodes
 	fi
 	
-	ebegin "Loading key mappings"
+	ebegin $"Loading key mappings"
 	
 	# Enable unicode ...
 	set -- ${KEYMAP}
@@ -52,10 +54,10 @@
 		/usr/bin/loadkeys -q ${WINDOWKEYS_KEYMAP} ${KEYMAP} \
 		                     ${EXTENDED_KEYMAPS} >/dev/null
 	else
-		eend 1 "loadkeys executable not found"
+		eend 1 $"loadkeys executable not found"
 		return 1
 	fi
-	eend $? "Error loading key mappings"
+	eend $? $"Error loading key mappings"
 }
 
 
