--- /home/benni/gentoo_ev/baselayout/etc/init.d.old/localmount	2003-03-24 20:00:05.000000000 +0100
+++ /home/benni/gentoo_ev/baselayout/etc/init.d.new/localmount	2003-12-22 01:32:43.000000000 +0100
@@ -2,6 +2,8 @@
 # Copyright 1999-2003 Gentoo Technologies, Inc.
 # Distributed under the terms of the GNU General Public License v2
 # $Header: /var/cvsroot/gentoo/users/beejay/baselayout/work/patches-init.d/localmount.patch,v 1.1 2003/12/31 20:35:54 beejay Exp $
+TEXTDOMAINDIR=/etc/gentoo/locale
+TEXTDOMAIN=baselayout
 
 
 depend() {
@@ -10,9 +12,9 @@
 
 start() {
 	# Mount local filesystems in /etc/fstab.
-	ebegin "Mounting local filesystems"
+	ebegin $"Mounting local filesystems"
 	mount -at nocoda,nonfs,noproc,noncpfs,nosmbfs,noshm >/dev/null
-	eend $? "Some local filesystem failed to mount"
+	eend $? $"Some local filesystem failed to mount"
 
 	# Make sure we insert usbcore if its a module
 	if [ -f /proc/modules ]
@@ -44,13 +46,13 @@
 	if [ -n "${usbfs}" ] && \
 	   [ -e /proc/bus/usb -a ! -e /proc/bus/usb/devices ]
 	then
-		ebegin "Mounting USB device filesystem (${usbfs})"
+		ebegin $"Mounting USB device filesystem (${usbfs})"
 		mount -t ${usbfs} none /proc/bus/usb &>/dev/null
-		eend $? "Failed to mount USB device filesystem"
+		eend $? $"Failed to mount USB device filesystem"
 	fi
 
 	#swap on loopback devices, and other weirdnesses
-	ebegin "Activating (possibly) more swap"
+	ebegin $"Activating (possibly) more swap"
 	/sbin/swapon -a &>/dev/null
 	eend 0
 }
