--- /home/benni/gentoo_ev/baselayout/etc/init.d.old/local	2003-01-06 22:32:43.000000000 +0100
+++ /home/benni/gentoo_ev/baselayout/etc/init.d.new/local	2003-12-22 01:32:37.000000000 +0100
@@ -2,6 +2,8 @@
 # Copyright 1999-2003 Gentoo Technologies, Inc.
 # Distributed under the terms of the GNU General Public License v2
 # $Header: /var/cvsroot/gentoo/users/beejay/baselayout/work/patches-init.d/local.patch,v 1.1 2003/12/31 20:35:54 beejay Exp $
+TEXTDOMAINDIR=/etc/gentoo/locale
+TEXTDOMAIN=baselayout
 
 
 depend() {
@@ -9,7 +11,7 @@
 }
 
 start() {
-	ebegin "Starting local"
+	ebegin $"Starting local"
 
 	# Add any misc programs that should be started
 	# to /etc/conf.d/local.start
@@ -18,11 +20,11 @@
 		source /etc/conf.d/local.start
 	fi
 	
-	eend $? "Failed to start local."
+	eend $? $"Failed to start local."
 }
 
 stop() {
-	ebegin "Stopping local"
+	ebegin $"Stopping local"
 
 	# Add any misc programs that should be stopped
 	# to /etc/conf.d/local.stop
@@ -32,7 +34,7 @@
 		source /etc/conf.d/local.stop
 	fi
 	
-	eend $? "Failed to stop local."
+	eend $? $"Failed to stop local."
 }
 
 
