--- /home/benni/gentoo_ev/baselayout/etc/init.d.old/modules	2003-08-11 03:27:16.000000000 +0200
+++ /home/benni/gentoo_ev/baselayout/etc/init.d.new/modules	2003-12-22 13:35:40.000000000 +0100
@@ -1,7 +1,12 @@
 #!/sbin/runscript
 # Copyright 1999-2003 Gentoo Technologies, Inc.
 # Distributed under the terms of the GNU General Public License v2
-# $Header: /var/cvsroot/gentoo/users/beejay/baselayout/work/patches-init.d/modules.patch,v 1.1 2003/12/31 20:35:54 beejay Exp $
+# $Header: /home/cvsroot/gentoo-
+#src/rc-scripts/init.d/modules,v 1.28 2003/08/11 01:27:16 azarah Exp $
+
+export LANG=$(/bin/cat /etc/gentoo/locale/locale) 
+TEXTDOMAINDIR=/etc/gentoo/locale
+TEXTDOMAIN=baselayout
 
 
 depend() {
@@ -50,20 +55,20 @@
 
 	if [ "${modcount}" -gt 0 ]
 	then
-		einfo "Using ${config} as config:"
+		einfo $"Using ${config} as config:"
 		
 		for x in ${modules}
 		do
-			ebegin "  Loading module ${x}"
+			ebegin $"  Loading module ${x}"
 			modprobe -q ${x} ${modargs[${i}]} &>/dev/null
 			retval=$?
-			eend ${retval} "  Failed to load ${x}"
+			eend ${retval} $"  Failed to load ${x}"
 			
 			i=$((i+1))
 			[ "${retval}" -eq 0 ] || modcount=$((modcount-1))
 		done
 
-		einfo "Autoloaded ${modcount} module(s)"
+		einfo $"Autoloaded ${modcount} module(s)"
 	fi
 
 	return 0
@@ -78,13 +83,13 @@
 	# depmod command ...
 	if [ ! -x /sbin/depmod ]
 	then
-		eerror "ERROR:  system is missing /sbin/depmod !"
+		eerror $"ERROR:  system is missing /sbin/depmod !"
 		return 1
 	fi
 	
-	ebegin "Calculating module dependencies"
+	ebegin $"Calculating module dependencies"
 	/sbin/modules-update &>/dev/null
-	eend $? "Failed to calculate dependencies"
+	eend $? $"Failed to calculate dependencies"
 
 	if [ -f /etc/modules.autoload -a ! -L /etc/modules.autoload ]
 	then
@@ -103,7 +108,7 @@
 			
 		elif [ ! -f /etc/modules.autoload.d/kernel-"${KV_MAJOR}.${KV_MINOR}" ]
 		then
-			ewarn "Missing /etc/modules.autoload.d/kernel-${KV_MAJOR}.${KV_MINOR}"
+			ewarn $"Missing /etc/modules.autoload.d/kernel-${KV_MAJOR}.${KV_MINOR}"
 			load_modules /etc/modules.autoload.d/kernel-2.4
 		else
 			load_modules /etc/modules.autoload.d/kernel-2.4
