--- /home/benni/gentoo_ev/baselayout/etc/init.d.old/net.lo	2003-01-20 01:51:11.000000000 +0100
+++ /home/benni/gentoo_ev/baselayout/etc/init.d.new/net.lo	2003-12-22 01:33:03.000000000 +0100
@@ -2,10 +2,12 @@
 # Copyright 1999-2003 Gentoo Technologies, Inc.
 # Distributed under the terms of the GNU General Public License v2
 # $Header: /var/cvsroot/gentoo/users/beejay/baselayout/work/patches-init.d/net.lo.patch,v 1.1 2003/12/31 20:35:54 beejay Exp $
+TEXTDOMAINDIR=/etc/gentoo/locale
+TEXTDOMAIN=baselayout
 
 
 start() {
-	ebegin "Bringing ${IFACE} up"
+	ebegin $"Bringing ${IFACE} up"
 	/sbin/ifconfig lo 127.0.0.1 up 2>/dev/null
 	/sbin/route add -net 127.0.0.0 netmask 255.0.0.0 \
 		gw 127.0.0.1 dev lo 2> /dev/null
@@ -13,7 +15,7 @@
 }
 
 stop() {
-	ebegin "Bringing ${IFACE} down"
+	ebegin $"Bringing ${IFACE} down"
 	/sbin/ifconfig ${IFACE} down &>/dev/null
 	eend 0
 }
