--- /home/benni/gentoo_ev/baselayout/etc/init.d.old/netmount	2003-03-09 10:06:23.000000000 +0100
+++ /home/benni/gentoo_ev/baselayout/etc/init.d.new/netmount	2003-12-22 01:33:12.000000000 +0100
@@ -2,6 +2,8 @@
 # Copyright 1999-2003 Gentoo Technologies, Inc.
 # Distributed under the terms of the GNU General Public License v2
 # $Header: /var/cvsroot/gentoo/users/beejay/baselayout/work/patches-init.d/netmount.patch,v 1.1 2003/12/31 20:35:54 beejay Exp $
+TEXTDOMAINDIR=/etc/gentoo/locale
+TEXTDOMAIN=baselayout
 
 
 depend() {
@@ -38,12 +40,12 @@
 		rcfilesystems="coda,ncpfs,smbfs"
 	fi
 
-	ebegin "Mounting network filesystems"
+	ebegin $"Mounting network filesystems"
 	mount -at ${rcfilesystems} >/dev/null
 
 	if [ "$?" -ne 0 ]
 	then
-		ewend 1 "Could not mount all network filesystems!"
+		ewend 1 $"Could not mount all network filesystems!"
 	else
 		eend 0
 	fi
@@ -62,7 +64,7 @@
 	# just keep things nice and uniform
 	if [ -z "${remaining}" ]
 	then
-		ebegin "Unmounting network filesystems"
+		ebegin $"Unmounting network filesystems"
 		eend 0
 	else
 	        sig=
@@ -71,13 +73,13 @@
 	        do
 	                if [ "${retry}" -lt 3 ]
 			then
-	                        ebegin "Unmounting network filesystems (retry)"
+	                        ebegin $"Unmounting network filesystems (retry)"
 	                        umount ${remaining} &>/dev/null
-	                        eend $? "Failed to unmount filesystems this retry"
+	                        eend $? $"Failed to unmount filesystems this retry"
 	                else
-	                        ebegin "Unmounting network filesystems"
+	                        ebegin $"Unmounting network filesystems"
 	                        umount ${remaining} &>/dev/null
-	                        eend $? "Failed to unmount filesystems"
+	                        eend $? $"Failed to unmount filesystems"
 	                fi
 	                remaining="$(awk '$3 ~ /coda|nfs|ncpfs|smbfs/ { if ($2 != "/") print $2 }' /proc/mounts | sort -r)"
 	                [ -z "${remaining}" ] && break
