--- /home/benni/gentoo_ev/baselayout/etc/init.d.old/nscd	2003-04-27 22:17:15.000000000 +0200
+++ /home/benni/gentoo_ev/baselayout/etc/init.d.new/nscd	2003-12-22 01:33:23.000000000 +0100
@@ -2,10 +2,12 @@
 # Copyright 1999-2003 Gentoo Technologies, Inc.
 # Distributed under the terms of the GNU General Public License v2
 # $Header: /var/cvsroot/gentoo/users/beejay/baselayout/work/patches-init.d/nscd.patch,v 1.1 2003/12/31 20:35:54 beejay Exp $
+TEXTDOMAINDIR=/etc/gentoo/locale
+TEXTDOMAIN=baselayout
 
 
 start() {
-	ebegin "Starting Name Service Cache Daemon"
+	ebegin $"Starting Name Service Cache Daemon"
 	secure=`while read curline ; do
 		table=${curline%:*}
 		entries=${curline##$table:}
@@ -28,7 +30,7 @@
 }
 
 stop () {
-	ebegin "Shutting down Name Service Cache Daemon"
+	ebegin $"Shutting down Name Service Cache Daemon"
 	start-stop-daemon --stop --quiet --pid /var/run/nscd.pid
 	eend $?
 }
