--- /home/benni/gentoo_ev/baselayout/etc/init.d.old/numlock	2003-01-06 22:32:43.000000000 +0100
+++ /home/benni/gentoo_ev/baselayout/etc/init.d.new/numlock	2003-12-22 01:33:34.000000000 +0100
@@ -2,6 +2,8 @@
 # Copyright 1999-2003 Gentoo Technologies, Inc.
 # Distributed under the terms of the GNU General Public License v2
 # $Header: /var/cvsroot/gentoo/users/beejay/baselayout/work/patches-init.d/numlock.patch,v 1.1 2003/12/31 20:35:54 beejay Exp $
+TEXTDOMAINDIR=/etc/gentoo/locale
+TEXTDOMAIN=baselayout
 
 
 depend() {
@@ -9,21 +11,21 @@
 }
 
 start() {
-	ebegin "Enabling numlock on ttys"
+	ebegin $"Enabling numlock on ttys"
 	for tty in /dev/tty[1-8]
 	do
 		setleds -D +num < ${tty} &>/dev/null
 	done
-	eend $? "Failed to enable numlock"
+	eend $? $"Failed to enable numlock"
 }
 
 stop() {
-	ebegin "Disabling numlock on ttys"
+	ebegin $"Disabling numlock on ttys"
 	for tty in /dev/tty[1-8]
 	do
 		setleds -D -num < ${tty} &>/dev/null
 	done
-	eend $? "Failed to disable numlock"
+	eend $? $"Failed to disable numlock"
 }
 
 
