--- /home/benni/gentoo_ev/baselayout/etc/init.d.old/rmnologin	2003-01-06 22:32:43.000000000 +0100
+++ /home/benni/gentoo_ev/baselayout/etc/init.d.new/rmnologin	2003-12-22 01:33:58.000000000 +0100
@@ -2,6 +2,8 @@
 # Copyright 1999-2003 Gentoo Technologies, Inc.
 # Distributed under the terms of the GNU General Public License v2
 # $Header: /var/cvsroot/gentoo/users/beejay/baselayout/work/patches-init.d/rmnologin.patch,v 1.1 2003/12/31 20:35:54 beejay Exp $
+TEXTDOMAINDIR=/etc/gentoo/locale
+TEXTDOMAIN=baselayout
 
 
 depend() {
