--- /home/benni/gentoo_ev/baselayout/etc/init.d.old/serial	2003-07-17 21:16:28.000000000 +0200
+++ /home/benni/gentoo_ev/baselayout/etc/init.d.new/serial	2003-12-22 01:34:16.000000000 +0100
@@ -2,6 +2,8 @@
 # Copyright 1999-2003 Gentoo Technologies, Inc.
 # Distributed under the terms of the GNU General Public License v2
 # $Header: /var/cvsroot/gentoo/users/beejay/baselayout/work/patches-init.d/serial.patch,v 1.1 2003/12/31 20:35:54 beejay Exp $
+TEXTDOMAINDIR=/etc/gentoo/locale
+TEXTDOMAIN=baselayout
 
 
 moduleinserted() {
@@ -28,11 +30,11 @@
 		fi
 	fi
 
-	ebegin "Setting /dev/$1"
+	ebegin $"Setting /dev/$1"
 	${SETSERIAL} -b /dev/$1 irq $2 port $3 $4 1>&2
 	if [ "$?" -gt 0 ]
 	then
-		eend 1 "Error setting serial port $1 to irq $2 port $3 (args: $4)"
+		eend 1 $"Error setting serial port $1 to irq $2 port $3 (args: $4)"
 		no_errs=0
 	else
 		eend 0
