--- /home/benni/gentoo_ev/baselayout/etc/init.d.old/shutdown.sh	2003-01-06 22:32:43.000000000 +0100
+++ /home/benni/gentoo_ev/baselayout/etc/init.d.new/shutdown.sh	2003-12-22 01:34:24.000000000 +0100
@@ -1,6 +1,8 @@
 # Copyright 1999-2003 Gentoo Technologies, Inc.
 # Distributed under the terms of the GNU General Public License v2
 # $Header: /var/cvsroot/gentoo/users/beejay/baselayout/work/patches-init.d/shutdown.sh.patch,v 1.1 2003/12/31 20:35:54 beejay Exp $
+TEXTDOMAINDIR=/etc/gentoo/locale
+TEXTDOMAIN=baselayout
 
 
 /sbin/halt -idp
