--- /home/benni/gentoo_ev/baselayout/etc/init.d.old/urandom	2003-01-06 22:32:43.000000000 +0100
+++ /home/benni/gentoo_ev/baselayout/etc/init.d.new/urandom	2003-12-22 01:34:36.000000000 +0100
@@ -2,6 +2,8 @@
 # Copyright 1999-2003 Gentoo Technologies, Inc.
 # Distributed under the terms of the GNU General Public License v2
 # $Header: /var/cvsroot/gentoo/users/beejay/baselayout/work/patches-init.d/urandom.patch,v 1.1 2003/12/31 20:35:54 beejay Exp $
+TEXTDOMAINDIR=/etc/gentoo/locale
+TEXTDOMAIN=baselayout
 
 
 depend() {
@@ -10,7 +12,7 @@
 
 start() {
 	[ -c /dev/urandom ] || return
-	ebegin "Initializing random number generator"
+	ebegin $"Initializing random number generator"
 	if [ -f /var/run/random-seed ]
 	then
 	    cat /var/run/random-seed >/dev/urandom
@@ -18,17 +20,17 @@
 	rm -f /var/run/random-seed &>/dev/null
 	umask 077
 	dd if=/dev/urandom of=/var/run/random-seed count=1 &>/dev/null
-	eend $? "Error initializing random number generator"
+	eend $? $"Error initializing random number generator"
 	umask 022
 }
 
 stop() {
-	ebegin "Saving random seed"
+	ebegin $"Saving random seed"
 	# Carry a random seed from shut-down to start-up;
 	# see documentation in linux/drivers/char/random.c
 	umask 077
 	dd if=/dev/urandom of=/var/run/random-seed count=1 &>/dev/null
-	eend $? "Failed to save random seed."
+	eend $? $"Failed to save random seed."
 }
 
 
