Index: nfs-2.6/fs/Kconfig
diff -u nfs-2.6/fs/Kconfig:1.1.1.6 nfs-2.6/fs/Kconfig:1.6
--- nfs-2.6/fs/Kconfig:1.1.1.6	Mon Apr  5 11:55:01 2004
+++ nfs-2.6/fs/Kconfig	Mon Apr  5 16:31:39 2004
@@ -1307,6 +1307,24 @@
 
 	  If unsure, say Y.
 
+config NFS_XATTR
+	bool "Provide NFSv3 client support for extended attributes"
+	depends on NFS_FS && NFS_V3
+	help
+	 Say Y here if you want your NFS client to be able to use extended
+	 attributes.  This requires the server to be configured as well.
+
+	 If unsure, say N
+config NFS_SELINUX
+	bool "Provide NFSv3 client support for SELinux"
+	depends on NFS_FS && NFS_V3 && NFS_XATTR
+	help
+	 Say Y here for SELinux support for NFSv3.  This requires the
+	 SELinux security module on the client, and also that the server 
+	 is configured to use SELinux for NFSv3 as well.
+
+	 If unsure, say N
+	 	 
 config NFS_V4
 	bool "Provide NFSv4 client support (EXPERIMENTAL)"
 	depends on NFS_FS && EXPERIMENTAL
@@ -1378,6 +1396,24 @@
 	help
 	  If you would like to include the NFSv3 server as well as the NFSv2
 	  server, say Y here.  If unsure, say Y.
+
+config NFSD_XATTR
+	bool "Provide NFSv3 server support for extended attributes"
+	depends on NFSD && NFSD_V3
+	help
+	 Say Y here if you want your NFS server to be able to use extended
+	 attributes.
+
+	 If unsure, say N
+
+config NFSD_SELINUX
+	bool "Provide NFSv3 server support for SELinux"
+	depends on NFSD && NFSD_V3 && NFSD_XATTR
+	help
+	 Say Y here for SELinux support for NFSV3.  This requires the
+	 SELinux security module as well.
+
+	 If unsure, say N
 
 config NFSD_V4
 	bool "Provide NFSv4 server support (EXPERIMENTAL)"
Index: nfs-2.6/fs/lockd/svc.c
diff -u nfs-2.6/fs/lockd/svc.c:1.1.1.3 nfs-2.6/fs/lockd/svc.c:1.8
--- nfs-2.6/fs/lockd/svc.c:1.1.1.3	Fri Apr  2 14:13:44 2004
+++ nfs-2.6/fs/lockd/svc.c	Mon Apr  5 09:35:36 2004
@@ -472,6 +472,7 @@
 
 #define NLM_NRVERS	(sizeof(nlmsvc_version)/sizeof(nlmsvc_version[0]))
 struct svc_program	nlmsvc_program = {
+	.pg_next	= NULL,			/* last registered program */
 	.pg_prog	= NLM_PROGRAM,		/* program number */
 	.pg_nvers	= NLM_NRVERS,		/* number of entries in nlmsvc_version */
 	.pg_vers	= nlmsvc_version,	/* version table */
Index: nfs-2.6/fs/nfs/Makefile
diff -u nfs-2.6/fs/nfs/Makefile:1.1.1.1 nfs-2.6/fs/nfs/Makefile:1.4
--- nfs-2.6/fs/nfs/Makefile:1.1.1.1	Mon Aug 18 12:48:27 2003
+++ nfs-2.6/fs/nfs/Makefile	Wed Mar  3 15:40:36 2004
@@ -6,6 +6,8 @@
 
 nfs-y 			:= dir.o file.o inode.o nfs2xdr.o pagelist.o \
 			   proc.o read.o symlink.o unlink.o write.o
+
+nfs-$(CONFIG_NFS_XATTR) += xattr.o
 nfs-$(CONFIG_ROOT_NFS)	+= nfsroot.o mount_clnt.o      
 nfs-$(CONFIG_NFS_V3)	+= nfs3proc.o nfs3xdr.o
 nfs-$(CONFIG_NFS_V4)	+= nfs4proc.o nfs4xdr.o nfs4state.o nfs4renewd.o \
Index: nfs-2.6/fs/nfs/dir.c
diff -u nfs-2.6/fs/nfs/dir.c:1.1.1.4 nfs-2.6/fs/nfs/dir.c:1.10
--- nfs-2.6/fs/nfs/dir.c:1.1.1.4	Mon Apr  5 11:55:32 2004
+++ nfs-2.6/fs/nfs/dir.c	Mon Apr  5 16:31:44 2004
@@ -31,6 +31,7 @@
 #include <linux/pagemap.h>
 #include <linux/smp_lock.h>
 #include <linux/namei.h>
+#include <linux/sunrpc/xattr.h>
 
 #define NFS_PARANOIA 1
 /* #define NFS_DEBUG_VERBOSE 1 */
@@ -70,6 +71,10 @@
 	.permission	= nfs_permission,
 	.getattr	= nfs_getattr,
 	.setattr	= nfs_setattr,
+#ifdef CONFIG_NFS_XATTR
+	.getxattr	= nfs_getxattr,
+	.setxattr	= nfs_setxattr,
+#endif
 };
 
 #ifdef CONFIG_NFS_V4
@@ -1503,6 +1508,7 @@
 {
 	struct nfs_access_cache *cache = &NFS_I(inode)->cache_access;
 	struct rpc_cred *cred;
+	struct rpc_clnt *client = NFS_CLIENT(inode);
 	int mode = inode->i_mode;
 	int res;
 
@@ -1527,7 +1533,7 @@
 			return -EACCES;
 	}
 	/* Are we checking permissions on anything other than lookup/execute? */
-	if ((mask & MAY_EXEC) == 0) {
+	if (!client->cl_selinux && ((mask & MAY_EXEC) == 0)) {
 		/* We only need to check permissions on file open() and access() */
 		if (!nd || !(nd->flags & (LOOKUP_OPEN|LOOKUP_ACCESS)))
 			return 0;
Index: nfs-2.6/fs/nfs/file.c
diff -u nfs-2.6/fs/nfs/file.c:1.1.1.4 nfs-2.6/fs/nfs/file.c:1.7
--- nfs-2.6/fs/nfs/file.c:1.1.1.4	Mon Apr  5 11:55:33 2004
+++ nfs-2.6/fs/nfs/file.c	Mon Apr  5 16:31:44 2004
@@ -30,6 +30,7 @@
 
 #include <asm/uaccess.h>
 #include <asm/system.h>
+#include <linux/sunrpc/xattr.h>
 
 #define NFSDBG_FACILITY		NFSDBG_FILE
 
@@ -61,6 +62,10 @@
 	.permission	= nfs_permission,
 	.getattr	= nfs_getattr,
 	.setattr	= nfs_setattr,
+#ifdef CONFIG_NFS_XATTR
+	.getxattr	= nfs_getxattr,
+	.setxattr	= nfs_setxattr,
+#endif
 };
 
 /* Hack for future NFS swap support */
Index: nfs-2.6/fs/nfs/inode.c
diff -u nfs-2.6/fs/nfs/inode.c:1.1.1.5 nfs-2.6/fs/nfs/inode.c:1.16
--- nfs-2.6/fs/nfs/inode.c:1.1.1.5	Mon Apr  5 11:55:33 2004
+++ nfs-2.6/fs/nfs/inode.c	Mon Apr  5 16:31:44 2004
@@ -35,6 +35,8 @@
 #include <linux/mount.h>
 #include <linux/nfs_idmap.h>
 #include <linux/vfs.h>
+#include <linux/sunrpc/selinux.h>
+#include <linux/sunrpc/xattr.h>
 
 #include <asm/system.h>
 #include <asm/uaccess.h>
@@ -104,6 +106,54 @@
 	.pipe_dir_name		= "/nfs",
 };
 
+#ifdef CONFIG_NFS_SELINUX
+/* RPC cruft for selinuxnfs  */
+struct rpc_stat			selinux_nfs_rpcstat = {
+	.program		= &selinux_nfs_program
+};
+
+static struct rpc_version *	selinux_nfs_version[] = {
+	NULL,
+	NULL,
+	NULL,
+	&selinux_nfs_version3,
+};
+
+struct rpc_program		selinux_nfs_program = {
+	.name			= "selinuxnfs",
+	.number			= SELINUX_NFS_PROGRAM,
+	.nrvers			= sizeof(selinux_nfs_version) / 
+	                                sizeof(selinux_nfs_version[0]),
+	.version		= selinux_nfs_version,
+	.stats			= &selinux_nfs_rpcstat,
+	.pipe_dir_name		= "/selinuxnfs",
+};
+#endif /* CONFIG_NFS_SELINUX */
+
+#ifdef CONFIG_NFS_XATTR
+/* RPC cruft for xattrnfs  */
+struct rpc_stat			xattr_nfs_rpcstat = {
+	.program		= &xattr_nfs_program
+};
+
+static struct rpc_version *	xattr_nfs_version[] = {
+	NULL,
+	NULL,
+	NULL,
+	&xattr_nfs_version3,
+};
+
+struct rpc_program		xattr_nfs_program = {
+	.name			= "xattrnfs",
+	.number			= XATTR_NFS_PROGRAM,
+	.nrvers			= sizeof(xattr_nfs_version) / 
+	                                sizeof(xattr_nfs_version[0]),
+	.version		= xattr_nfs_version,
+	.stats			= &xattr_nfs_rpcstat,
+	.pipe_dir_name		= "/xattrnfs",
+};
+#endif /* CONFIG_NFS_XATTR */
+
 static inline unsigned long
 nfs_fattr_to_ino_t(struct nfs_fattr *fattr)
 {
@@ -352,6 +402,7 @@
 	struct rpc_timeout	timeparms;
 	struct rpc_xprt		*xprt = NULL;
 	struct rpc_clnt		*clnt = NULL;
+	struct rpc_program	*program;
 	int			tcp   = (data->flags & NFS_MOUNT_TCP);
 
 	/* Initialize timeout values */
@@ -372,7 +423,18 @@
 		printk(KERN_WARNING "NFS: cannot create RPC transport.\n");
 		return (struct rpc_clnt *)xprt;
 	}
-	clnt = rpc_create_client(xprt, server->hostname, &nfs_program,
+
+	program = &nfs_program;
+#ifdef CONFIG_NFS_XATTR
+	if (data->flags & NFS_MOUNT_XATTR)
+		program = &xattr_nfs_program;
+#endif
+#ifdef CONFIG_NFS_SELINUX
+	if (data->flags & NFS_MOUNT_SELINUX)
+		program = &selinux_nfs_program;
+#endif	
+
+	clnt = rpc_create_client(xprt, server->hostname, program,
 				 server->rpc_ops->version, data->pseudoflavor);
 	if (IS_ERR(clnt)) {
 		printk(KERN_WARNING "NFS: cannot create RPC client.\n");
@@ -383,6 +445,11 @@
 	clnt->cl_softrtry = (server->flags & NFS_MOUNT_SOFT) ? 1 : 0;
 	clnt->cl_droppriv = (server->flags & NFS_MOUNT_BROKEN_SUID) ? 1 : 0;
 	clnt->cl_chatty   = 1;
+#ifdef CONFIG_NFS_SELINUX
+	clnt->cl_selinux = (server->flags & NFS_MOUNT_SELINUX) ? 1 : 0;
+#else
+	clnt->cl_selinux = 0;
+#endif
 
 	return clnt;
 
@@ -431,6 +498,19 @@
 	if (server->flags & NFS_MOUNT_VER3) {
 #ifdef CONFIG_NFS_V3
 		server->rpc_ops = &nfs_v3_clientops;
+		if (data->flags & NFS_MOUNT_XATTR)
+#ifdef CONFIG_NFS_XATTR
+			server->rpc_ops = &nfs_v3ea_clientops;
+#else
+		        printk(KERN_NOTICE "NFS: NFSv3 extended attributes not supported.\n");
+#endif
+		if (data->flags & NFS_MOUNT_SELINUX)
+#ifdef CONFIG_NFS_SELINUX
+			server->rpc_ops = &nfs_v3ea_clientops;
+#else
+		        printk(KERN_NOTICE "NFS: NFSv3 with selinux not supported.\n");
+#endif
+			
 		server->caps |= NFS_CAP_READDIRPLUS;
 		if (data->version < 4) {
 			printk(KERN_NOTICE "NFS: NFSv3 not supported by mount program.\n");
@@ -554,6 +634,8 @@
 		{ NFS_MOUNT_SOFT, ",soft", ",hard" },
 		{ NFS_MOUNT_INTR, ",intr", "" },
 		{ NFS_MOUNT_POSIX, ",posix", "" },
+		{ NFS_MOUNT_XATTR, ",xattr", "" },
+		{ NFS_MOUNT_SELINUX, ",selinux", "" },
 		{ NFS_MOUNT_TCP, ",tcp", ",udp" },
 		{ NFS_MOUNT_NOCTO, ",nocto", "" },
 		{ NFS_MOUNT_NOAC, ",noac", "" },
@@ -1371,6 +1453,16 @@
 	.fs_flags	= FS_ODD_RENAME|FS_REVAL_DOT|FS_BINARY_MOUNTDATA,
 };
 
+#ifdef CONFIG_NFS_SELINUX
+static struct file_system_type selinux_nfs_fs_type = {
+	.owner		= THIS_MODULE,
+	.name		= "selinuxnfs",
+	.get_sb		= nfs_get_sb,
+	.kill_sb	= nfs_kill_super,
+	.fs_flags	= FS_ODD_RENAME|FS_REVAL_DOT|FS_BINARY_MOUNTDATA,
+};
+#endif
+
 #ifdef CONFIG_NFS_V4
 
 static void nfs4_clear_inode(struct inode *);
@@ -1790,15 +1882,32 @@
 
 #ifdef CONFIG_PROC_FS
 	rpc_proc_register(&nfs_rpcstat);
+#ifdef CONFIG_NFS_SELINUX
+	rpc_proc_register(&selinux_nfs_rpcstat);
+#endif
+#ifdef CONFIG_NFS_XATTR
+	rpc_proc_register(&xattr_nfs_rpcstat);
+#endif
 #endif
         err = register_filesystem(&nfs_fs_type);
 	if (err)
 		goto out;
+#ifdef CONFIG_NFS_SELINUX
+	err = register_filesystem(&selinux_nfs_fs_type);
+	if (err)
+		goto out;
+#endif
 	if ((err = register_nfs4fs()) != 0)
 		goto out;
 	return 0;
 out:
 	rpc_proc_unregister("nfs");
+#ifdef CONFIG_NFS_SELINUX
+	rpc_proc_unregister("selinuxnfs");
+#endif
+#ifdef CONFIG_NFS_XATTR
+	rpc_proc_unregister("xattrnfs");
+#endif
 	nfs_destroy_writepagecache();
 out1:
 	nfs_destroy_readpagecache();
@@ -1818,8 +1927,17 @@
 	nfs_destroy_nfspagecache();
 #ifdef CONFIG_PROC_FS
 	rpc_proc_unregister("nfs");
+#ifdef CONFIG_NFS_SELINUX
+	rpc_proc_unregister("selinuxnfs");
+#endif
+#ifdef CONFIG_NFS_XATTR
+	rpc_proc_unregister("xattrnfs");
+#endif
 #endif
 	unregister_filesystem(&nfs_fs_type);
+#ifdef CONFIG_NFS_SELINUX
+	unregister_filesystem(&selinux_nfs_fs_type);
+#endif
 	unregister_nfs4fs();
 }
 
Index: nfs-2.6/fs/nfs/nfs2xdr.c
diff -u nfs-2.6/fs/nfs/nfs2xdr.c:1.1.1.3 nfs-2.6/fs/nfs/nfs2xdr.c:1.3
--- nfs-2.6/fs/nfs/nfs2xdr.c:1.1.1.3	Mon Apr  5 11:55:33 2004
+++ nfs-2.6/fs/nfs/nfs2xdr.c	Mon Apr  5 16:31:44 2004
@@ -611,6 +611,10 @@
 	{ NFSERR_NOSPC,		ENOSPC		},
 	{ NFSERR_ROFS,		EROFS		},
 	{ NFSERR_MLINK,		EMLINK		},
+#ifdef CONFIG_NFSD_XATTR
+	{ NFSERR_ERANGE,	ERANGE		},
+	{ NFSERR_EOPNOTSUPP,	EOPNOTSUPP 	},
+#endif
 	{ NFSERR_NAMETOOLONG,	ENAMETOOLONG	},
 	{ NFSERR_NOTEMPTY,	ENOTEMPTY	},
 	{ NFSERR_DQUOT,		EDQUOT		},
Index: nfs-2.6/fs/nfs/nfs3proc.c
diff -u nfs-2.6/fs/nfs/nfs3proc.c:1.1.1.4 nfs-2.6/fs/nfs/nfs3proc.c:1.9
--- nfs-2.6/fs/nfs/nfs3proc.c:1.1.1.4	Mon Apr  5 11:55:33 2004
+++ nfs-2.6/fs/nfs/nfs3proc.c	Mon Apr  5 16:31:44 2004
@@ -17,11 +17,11 @@
 #include <linux/nfs_page.h>
 #include <linux/lockd/bind.h>
 #include <linux/smp_lock.h>
+#include <linux/limits.h>
+#include <linux/sunrpc/xattr.h>
 
 #define NFSDBG_FACILITY		NFSDBG_PROC
 
-extern struct rpc_procinfo nfs3_procedures[];
-
 /* A wrapper to handle the EJUKEBOX error message */
 static int
 nfs3_rpc_wrapper(struct rpc_clnt *clnt, struct rpc_message *msg, int flags)
@@ -45,7 +45,7 @@
 nfs3_rpc_call_wrapper(struct rpc_clnt *clnt, u32 proc, void *argp, void *resp, int flags)
 {
 	struct rpc_message msg = {
-		.rpc_proc	= &nfs3_procedures[proc],
+		.rpc_proc	= &clnt->cl_procinfo[proc],
 		.rpc_argp	= argp,
 		.rpc_resp	= resp,
 	};
@@ -167,6 +167,7 @@
 static int
 nfs3_proc_access(struct inode *inode, struct rpc_cred *cred, int mode)
 {
+	struct rpc_clnt		*clnt = NFS_CLIENT(inode);
 	struct nfs_fattr	fattr;
 	struct nfs3_accessargs	arg = {
 		.fh		= NFS_FH(inode),
@@ -175,7 +176,7 @@
 		.fattr		= &fattr,
 	};
 	struct rpc_message msg = {
-		.rpc_proc	= &nfs3_procedures[NFS3PROC_ACCESS],
+		.rpc_proc	= &clnt->cl_procinfo[NFS3PROC_ACCESS],
 		.rpc_argp	= &arg,
 		.rpc_resp	= &res,
 		.rpc_cred	= cred
@@ -198,7 +199,7 @@
 		if (mode & MAY_EXEC)
 			arg.access |= NFS3_ACCESS_EXECUTE;
 	}
-	status = rpc_call_sync(NFS_CLIENT(inode), &msg, 0);
+	status = rpc_call_sync(clnt, &msg, 0);
 	nfs_refresh_inode(inode, &fattr);
 	dprintk("NFS reply access\n");
 
@@ -232,9 +233,10 @@
 {
 	int			flags = rdata->flags;
 	struct inode *		inode = rdata->inode;
+	struct rpc_clnt	*	clnt = NFS_CLIENT(inode);
 	struct nfs_fattr *	fattr = rdata->res.fattr;
 	struct rpc_message	msg = {
-		.rpc_proc	= &nfs3_procedures[NFS3PROC_READ],
+		.rpc_proc	= &clnt->cl_procinfo[NFS3PROC_READ],
 		.rpc_argp	= &rdata->args,
 		.rpc_resp	= &rdata->res,
 	};
@@ -244,7 +246,7 @@
 			(long long) rdata->args.offset);
 	fattr->valid = 0;
 	msg.rpc_cred = nfs_cred(inode, filp);
-	status = rpc_call_sync(NFS_CLIENT(inode), &msg, flags);
+	status = rpc_call_sync(clnt, &msg, flags);
 	if (status >= 0)
 		nfs_refresh_inode(inode, fattr);
 	dprintk("NFS reply read: %d\n", status);
@@ -256,9 +258,10 @@
 {
 	int			rpcflags = wdata->flags;
 	struct inode *		inode = wdata->inode;
+	struct rpc_clnt	*	clnt = NFS_CLIENT(inode);
 	struct nfs_fattr *	fattr = wdata->res.fattr;
 	struct rpc_message	msg = {
-		.rpc_proc	= &nfs3_procedures[NFS3PROC_WRITE],
+		.rpc_proc	= &clnt->cl_procinfo[NFS3PROC_WRITE],
 		.rpc_argp	= &wdata->args,
 		.rpc_resp	= &wdata->res,
 	};
@@ -268,7 +271,7 @@
 			(long long) wdata->args.offset);
 	fattr->valid = 0;
 	msg.rpc_cred = nfs_cred(inode, filp);
-	status = rpc_call_sync(NFS_CLIENT(inode), &msg, rpcflags);
+	status = rpc_call_sync(clnt, &msg, rpcflags);
 	if (status >= 0)
 		nfs_refresh_inode(inode, fattr);
 	dprintk("NFS reply write: %d\n", status);
@@ -279,9 +282,10 @@
 nfs3_proc_commit(struct nfs_write_data *cdata, struct file *filp)
 {
 	struct inode *		inode = cdata->inode;
+	struct rpc_clnt	*	clnt = NFS_CLIENT(inode);
 	struct nfs_fattr *	fattr = cdata->res.fattr;
 	struct rpc_message	msg = {
-		.rpc_proc	= &nfs3_procedures[NFS3PROC_COMMIT],
+		.rpc_proc	= &clnt->cl_procinfo[NFS3PROC_COMMIT],
 		.rpc_argp	= &cdata->args,
 		.rpc_resp	= &cdata->res,
 	};
@@ -291,7 +295,7 @@
 			(long long) cdata->args.offset);
 	fattr->valid = 0;
 	msg.rpc_cred = nfs_cred(inode, filp);
-	status = rpc_call_sync(NFS_CLIENT(inode), &msg, 0);
+	status = rpc_call_sync(clnt, &msg, 0);
 	if (status >= 0)
 		nfs_refresh_inode(inode, fattr);
 	dprintk("NFS reply commit: %d\n", status);
@@ -396,6 +400,7 @@
 static int
 nfs3_proc_remove(struct inode *dir, struct qstr *name)
 {
+	struct rpc_clnt	*	clnt = NFS_CLIENT(dir);
 	struct nfs_fattr	dir_attr;
 	struct nfs3_diropargs	arg = {
 		.fh		= NFS_FH(dir),
@@ -403,7 +408,7 @@
 		.len		= name->len
 	};
 	struct rpc_message	msg = {
-		.rpc_proc	= &nfs3_procedures[NFS3PROC_REMOVE],
+		.rpc_proc	= &clnt->cl_procinfo[NFS3PROC_REMOVE],
 		.rpc_argp	= &arg,
 		.rpc_resp	= &dir_attr,
 	};
@@ -411,7 +416,7 @@
 
 	dprintk("NFS call  remove %s\n", name->name);
 	dir_attr.valid = 0;
-	status = rpc_call_sync(NFS_CLIENT(dir), &msg, 0);
+	status = rpc_call_sync(clnt, &msg, 0);
 	nfs_refresh_inode(dir, &dir_attr);
 	dprintk("NFS reply remove: %d\n", status);
 	return status;
@@ -422,16 +427,18 @@
 {
 	struct nfs3_diropargs	*arg;
 	struct nfs_fattr	*res;
+	struct inode *		inode = dir->d_inode;	
+	struct rpc_clnt	*	clnt = NFS_CLIENT(inode);
 
 	arg = (struct nfs3_diropargs *)kmalloc(sizeof(*arg)+sizeof(*res), GFP_KERNEL);
 	if (!arg)
 		return -ENOMEM;
 	res = (struct nfs_fattr*)(arg + 1);
-	arg->fh = NFS_FH(dir->d_inode);
+	arg->fh = NFS_FH(inode);
 	arg->name = name->name;
 	arg->len = name->len;
 	res->valid = 0;
-	msg->rpc_proc = &nfs3_procedures[NFS3PROC_REMOVE];
+	msg->rpc_proc = &clnt->cl_procinfo[NFS3PROC_REMOVE];
 	msg->rpc_argp = arg;
 	msg->rpc_resp = res;
 	return 0;
@@ -598,6 +605,7 @@
 		  u64 cookie, struct page *page, unsigned int count, int plus)
 {
 	struct inode		*dir = dentry->d_inode;
+	struct rpc_clnt		*clnt = NFS_CLIENT(dir);
 	struct nfs_fattr	dir_attr;
 	u32			*verf = NFS_COOKIEVERF(dir);
 	struct nfs3_readdirargs	arg = {
@@ -614,7 +622,7 @@
 		.plus		= plus
 	};
 	struct rpc_message	msg = {
-		.rpc_proc	= &nfs3_procedures[NFS3PROC_READDIR],
+		.rpc_proc	= &clnt->cl_procinfo[NFS3PROC_READDIR],
 		.rpc_argp	= &arg,
 		.rpc_resp	= &res,
 		.rpc_cred	= cred
@@ -624,13 +632,13 @@
 	lock_kernel();
 
 	if (plus)
-		msg.rpc_proc = &nfs3_procedures[NFS3PROC_READDIRPLUS];
+		msg.rpc_proc = &clnt->cl_procinfo[NFS3PROC_READDIRPLUS];
 
 	dprintk("NFS call  readdir%s %d\n",
 			plus? "plus" : "", (unsigned int) cookie);
 
 	dir_attr.valid = 0;
-	status = rpc_call_sync(NFS_CLIENT(dir), &msg, 0);
+	status = rpc_call_sync(clnt, &msg, 0);
 	nfs_refresh_inode(dir, &dir_attr);
 	dprintk("NFS reply readdir: %d\n", status);
 	unlock_kernel();
@@ -733,10 +741,11 @@
 {
 	struct rpc_task		*task = &data->task;
 	struct inode		*inode = data->inode;
+	struct rpc_clnt		*clnt = NFS_CLIENT(inode);
 	struct nfs_page		*req;
 	int			flags;
 	struct rpc_message	msg = {
-		.rpc_proc	= &nfs3_procedures[NFS3PROC_READ],
+		.rpc_proc	= &clnt->cl_procinfo[NFS3PROC_READ],
 		.rpc_argp	= &data->args,
 		.rpc_resp	= &data->res,
 		.rpc_cred	= data->cred,
@@ -756,7 +765,7 @@
 	flags = RPC_TASK_ASYNC | (IS_SWAPFILE(inode)? NFS_RPC_SWAPFLAGS : 0);
 
 	/* Finalize the task. */
-	rpc_init_task(task, NFS_CLIENT(inode), nfs3_read_done, flags);
+	rpc_init_task(task, clnt, nfs3_read_done, flags);
 	task->tk_calldata = data;
 	/* Release requests */
 	task->tk_release = nfs_readdata_release;
@@ -782,11 +791,12 @@
 {
 	struct rpc_task		*task = &data->task;
 	struct inode		*inode = data->inode;
+	struct rpc_clnt		*clnt = NFS_CLIENT(inode);
 	struct nfs_page		*req;
 	int			stable;
 	int			flags;
 	struct rpc_message	msg = {
-		.rpc_proc	= &nfs3_procedures[NFS3PROC_WRITE],
+		.rpc_proc	= &clnt->cl_procinfo[NFS3PROC_WRITE],
 		.rpc_argp	= &data->args,
 		.rpc_resp	= &data->res,
 		.rpc_cred	= data->cred,
@@ -815,7 +825,7 @@
 	flags = (how & FLUSH_SYNC) ? 0 : RPC_TASK_ASYNC;
 
 	/* Finalize the task. */
-	rpc_init_task(task, NFS_CLIENT(inode), nfs3_write_done, flags);
+	rpc_init_task(task, clnt, nfs3_write_done, flags);
 	task->tk_calldata = data;
 	/* Release requests */
 	task->tk_release = nfs_writedata_release;
@@ -841,15 +851,16 @@
 {
 	struct rpc_task		*task = &data->task;
 	struct inode		*inode = data->inode;
+	struct rpc_clnt		*clnt = NFS_CLIENT(inode);
 	int			flags;
 	struct rpc_message	msg = {
-		.rpc_proc	= &nfs3_procedures[NFS3PROC_COMMIT],
+		.rpc_proc	= &clnt->cl_procinfo[NFS3PROC_COMMIT],
 		.rpc_argp	= &data->args,
 		.rpc_resp	= &data->res,
 		.rpc_cred	= data->cred,
 	};
 
-	data->args.fh     = NFS_FH(data->inode);
+	data->args.fh     = NFS_FH(inode);
 	data->args.offset = start;
 	data->args.count  = len;
 	data->res.count   = len;
@@ -860,7 +871,7 @@
 	flags = (how & FLUSH_SYNC) ? 0 : RPC_TASK_ASYNC;
 
 	/* Finalize the task. */
-	rpc_init_task(task, NFS_CLIENT(inode), nfs3_commit_done, flags);
+	rpc_init_task(task, clnt, nfs3_commit_done, flags);
 	task->tk_calldata = data;
 	/* Release requests */
 	task->tk_release = nfs_commit_release;
@@ -895,6 +906,47 @@
 	return nlmclnt_proc(filp->f_dentry->d_inode, cmd, fl);
 }
 
+#ifdef CONFIG_NFS_XATTR
+static int
+nfs3_proc_getxattr(struct dentry *dentry, const char *name, void *buffer,
+		   unsigned int size, unsigned int *xattrlen)
+{
+	struct inode *inode = dentry->d_inode;
+	struct nfs3_getxattrargs	arg = {
+		.fh			= NFS_FH(inode),
+		.name			= name,
+		.namelen		= strnlen(name, XATTR_NAME_MAX+1),
+		.bufsize		= size,
+	};
+	struct nfs3_getxattrres		res = {	
+		.buffer			= buffer,
+		.xattrlen		= xattrlen,
+	};
+	if (arg.namelen > XATTR_NAME_MAX)
+		return -ERANGE;
+	return rpc_call(NFS_CLIENT(inode), NFS3PROC_GETXATTR, &arg, &res, 0);
+}
+
+static int
+nfs3_proc_setxattr(struct dentry *dentry, const char *name, const void *value, 
+			     unsigned int size, int flags)
+{
+	struct inode *inode = dentry->d_inode;
+	struct nfs3_setxattrargs	arg = {
+		.fh			= NFS_FH(inode),
+		.name			= name,
+		.namelen		= strnlen(name, XATTR_NAME_MAX+1),
+		.value			= value,
+		.size			= size,
+		.flags			= flags,
+	};
+	struct nfs3_setxattrres		res;
+	if (arg.namelen > XATTR_NAME_MAX)
+		return -ERANGE;
+	return rpc_call(NFS_CLIENT(inode), NFS3PROC_SETXATTR, &arg, &res, 0);
+}
+#endif /* CONFIG_NFS_XATTR */
+
 struct nfs_rpc_ops	nfs_v3_clientops = {
 	.version	= 3,			/* protocol version */
 	.dentry_ops	= &nfs_dentry_operations,
@@ -932,3 +984,47 @@
 	.request_compatible = nfs3_request_compatible,
 	.lock		= nfs3_proc_lock,
 };
+
+#ifdef CONFIG_NFS_XATTR
+struct nfs_rpc_ops	nfs_v3ea_clientops = {
+	.version	= 3,			/* protocol version */
+	.dentry_ops	= &nfs_dentry_operations,
+	.dir_inode_ops	= &nfs_dir_inode_operations,
+	.getroot	= nfs3_proc_get_root,
+	.getattr	= nfs3_proc_getattr,
+	.setattr	= nfs3_proc_setattr,
+	.lookup		= nfs3_proc_lookup,
+	.access		= nfs3_proc_access,
+	.readlink	= nfs3_proc_readlink,
+	.read		= nfs3_proc_read,
+	.write		= nfs3_proc_write,
+	.commit		= nfs3_proc_commit,
+	.create		= nfs3_proc_create,
+	.remove		= nfs3_proc_remove,
+	.unlink_setup	= nfs3_proc_unlink_setup,
+	.unlink_done	= nfs3_proc_unlink_done,
+	.rename		= nfs3_proc_rename,
+	.link		= nfs3_proc_link,
+	.symlink	= nfs3_proc_symlink,
+	.mkdir		= nfs3_proc_mkdir,
+	.rmdir		= nfs3_proc_rmdir,
+	.readdir	= nfs3_proc_readdir,
+	.mknod		= nfs3_proc_mknod,
+	.statfs		= nfs3_proc_statfs,
+	.fsinfo		= nfs3_proc_fsinfo,
+	.pathconf	= nfs3_proc_pathconf,
+	.decode_dirent	= nfs3_decode_dirent,
+	.read_setup	= nfs3_proc_read_setup,
+	.write_setup	= nfs3_proc_write_setup,
+	.commit_setup	= nfs3_proc_commit_setup,
+	.file_open	= nfs_open,
+	.file_release	= nfs_release,
+	.request_init	= nfs3_request_init,
+	.request_compatible = nfs3_request_compatible,
+	.lock		= nfs3_proc_lock,
+	.getxattr	= nfs3_proc_getxattr,
+	.setxattr	= nfs3_proc_setxattr,
+};
+#else
+#define nfs_v3ea_clientops nfs_v3_clientops
+#endif /* CONFIG_NFS_XATTR */
Index: nfs-2.6/fs/nfs/nfs3xdr.c
diff -u nfs-2.6/fs/nfs/nfs3xdr.c:1.1.1.3 nfs-2.6/fs/nfs/nfs3xdr.c:1.13
--- nfs-2.6/fs/nfs/nfs3xdr.c:1.1.1.3	Mon Apr  5 11:55:33 2004
+++ nfs-2.6/fs/nfs/nfs3xdr.c	Mon Apr  5 16:31:44 2004
@@ -21,6 +21,8 @@
 #include <linux/nfs.h>
 #include <linux/nfs3.h>
 #include <linux/nfs_fs.h>
+#include <linux/limits.h> 	/* for XATTR_NAME_MAX */
+#include <linux/sunrpc/xattr.h>
 
 #define NFSDBG_FACILITY		NFSDBG_XDR
 
@@ -33,6 +35,8 @@
  * Declare the space requirements for NFS arguments and replies as
  * number of 32bit-words
  */
+#define NFS3_xattr_name_sz	(1+XDR_QUADLEN(XATTR_NAME_MAX))
+#define NFS3_xattr_value_sz	(1+XDR_QUADLEN(NFS_XATTR_SIZE_MAX))
 #define NFS3_fhandle_sz		(1+16)
 #define NFS3_fh_sz		(NFS3_fhandle_sz)	/* shorthand */
 #define NFS3_sattr_sz		(15)
@@ -62,6 +66,8 @@
 #define NFS3_linkargs_sz		(NFS3_fh_sz+NFS3_diropargs_sz)
 #define NFS3_readdirargs_sz	(NFS3_fh_sz+2)
 #define NFS3_commitargs_sz	(NFS3_fh_sz+3)
+#define NFS3_getxattrargs_sz	(NFS3_fh_sz+NFS3_xattr_name_sz+1)
+#define NFS3_setxattrargs_sz	(NFS3_fh_sz+NFS3_xattr_name_sz+NFS3_xattr_value_sz+1)
 
 #define NFS3_attrstat_sz	(1+NFS3_fattr_sz)
 #define NFS3_wccstat_sz		(1+NFS3_wcc_data_sz)
@@ -78,6 +84,8 @@
 #define NFS3_fsinfores_sz	(1+NFS3_post_op_attr_sz+12)
 #define NFS3_pathconfres_sz	(1+NFS3_post_op_attr_sz+6)
 #define NFS3_commitres_sz	(1+NFS3_wcc_data_sz+2)
+#define NFS3_getxattrres_sz	1+1+NFS3_xattr_value_sz
+#define NFS3_setxattrres_sz	1
 
 /*
  * Map file type to S_IFMT bits
@@ -631,6 +639,37 @@
 	return 0;
 }
 
+#ifdef CONFIG_NFS_XATTR
+/*
+ * Encode GETXATTR request
+ */
+static int
+nfs3_xdr_getxattrargs(struct rpc_rqst *req, u32 *p, 
+		      struct nfs3_getxattrargs *args)
+{
+	p = xdr_encode_fhandle(p, args->fh);
+	p = xdr_encode_array(p, args->name, args->namelen+1);
+	*p++ = htonl(args->bufsize);
+	req->rq_slen = xdr_adjust_iovec(req->rq_svec, p);
+	return 0;
+}
+
+/*
+ * Encode SETXATTR request
+ */
+static int
+nfs3_xdr_setxattrargs(struct rpc_rqst *req, u32 *p, 
+		      struct nfs3_setxattrargs *args)
+{
+	p = xdr_encode_fhandle(p, args->fh);
+	p = xdr_encode_array(p, args->name, args->namelen+1);
+	p = xdr_encode_array(p, args->value, args->size);
+	*p++ = htonl(args->flags);
+	req->rq_slen = xdr_adjust_iovec(req->rq_svec, p);
+	return 0;
+}
+#endif /* CONFIG_NFS_XATTR */
+
 /*
  * NFS XDR decode functions
  */
@@ -973,6 +1012,44 @@
 	return 0;
 }
 
+#ifdef CONFIG_NFS_XATTR
+/*
+ * Decode GETXATTR result
+ */
+static int
+nfs3_xdr_getxattrres(struct rpc_rqst *req, u32 *p, struct nfs3_getxattrres *res)
+{
+	int status;
+	int len;
+
+	status = ntohl(*p++);
+	if (status != 0)
+		return -nfs_stat_to_errno(status);
+
+	*res->xattrlen = ntohl(*p++);
+	len = ntohl(*p++);
+	if (len) {
+		memcpy(res->buffer,p,len);
+		p += XDR_QUADLEN(len);
+	}
+	return 0;
+}
+
+/*
+ * Decode SETXATTR result
+ */
+static int
+nfs3_xdr_setxattrres(struct rpc_rqst *req, u32 *p, struct nfs3_setxattrres *res)
+{
+	int status;
+
+	status = ntohl(*p++);
+	if (status != 0)
+		return -nfs_stat_to_errno(status);
+	return 0;
+}
+#endif /* CONFIG_NFS_XATTR */
+
 #ifndef MAX
 # define MAX(a, b)	(((a) > (b))? (a) : (b))
 #endif
@@ -1012,7 +1089,77 @@
 
 struct rpc_version		nfs_version3 = {
 	.number			= 3,
-	.nrprocs		= sizeof(nfs3_procedures)/sizeof(nfs3_procedures[0]),
+	.nrprocs		= sizeof(nfs3_procedures)/
+	                                 sizeof(nfs3_procedures[0]),
 	.procs			= nfs3_procedures
 };
 
+#ifdef CONFIG_NFS_XATTR
+struct rpc_procinfo	xattr_nfs3_procedures[] = {
+  PROC(GETATTR,		fhandle,	attrstat, 1),
+  PROC(SETATTR, 	sattrargs,	wccstat, 0),
+  PROC(LOOKUP,		diropargs,	lookupres, 2),
+  PROC(ACCESS,		accessargs,	accessres, 1),
+  PROC(READLINK,	readlinkargs,	readlinkres, 3),
+  PROC(READ,		readargs,	readres, 3),
+  PROC(WRITE,		writeargs,	writeres, 4),
+  PROC(CREATE,		createargs,	createres, 0),
+  PROC(MKDIR,		mkdirargs,	createres, 0),
+  PROC(SYMLINK,		symlinkargs,	createres, 0),
+  PROC(MKNOD,		mknodargs,	createres, 0),
+  PROC(REMOVE,		diropargs,	wccstat, 0),
+  PROC(RMDIR,		diropargs,	wccstat, 0),
+  PROC(RENAME,		renameargs,	renameres, 0),
+  PROC(LINK,		linkargs,	linkres, 0),
+  PROC(READDIR,		readdirargs,	readdirres, 3),
+  PROC(READDIRPLUS,	readdirargs,	readdirres, 3),
+  PROC(FSSTAT,		fhandle,	fsstatres, 0),
+  PROC(FSINFO,  	fhandle,	fsinfores, 0),
+  PROC(PATHCONF,	fhandle,	pathconfres, 0),
+  PROC(COMMIT,		commitargs,	commitres, 5),
+  PROC(GETXATTR,	getxattrargs,	getxattrres, 1),
+  PROC(SETXATTR,	setxattrargs,	setxattrres, 0),
+};
+
+struct rpc_version		xattr_nfs_version3 = {
+	.number			= 3,
+	.nrprocs		= sizeof(nfs3_procedures)/
+	                                 sizeof(nfs3_procedures[0]),
+	.procs			= xattr_nfs3_procedures
+};
+#endif /* CONFIG_NFS_XATTR */
+
+#ifdef CONFIG_NFS_SELINUX
+struct rpc_procinfo	selinux_nfs3_procedures[] = {
+  PROC(GETATTR,		fhandle,	attrstat, 1),
+  PROC(SETATTR, 	sattrargs,	wccstat, 0),
+  PROC(LOOKUP,		diropargs,	lookupres, 2),
+  PROC(ACCESS,		accessargs,	accessres, 1),
+  PROC(READLINK,	readlinkargs,	readlinkres, 3),
+  PROC(READ,		readargs,	readres, 3),
+  PROC(WRITE,		writeargs,	writeres, 4),
+  PROC(CREATE,		createargs,	createres, 0),
+  PROC(MKDIR,		mkdirargs,	createres, 0),
+  PROC(SYMLINK,		symlinkargs,	createres, 0),
+  PROC(MKNOD,		mknodargs,	createres, 0),
+  PROC(REMOVE,		diropargs,	wccstat, 0),
+  PROC(RMDIR,		diropargs,	wccstat, 0),
+  PROC(RENAME,		renameargs,	renameres, 0),
+  PROC(LINK,		linkargs,	linkres, 0),
+  PROC(READDIR,		readdirargs,	readdirres, 3),
+  PROC(READDIRPLUS,	readdirargs,	readdirres, 3),
+  PROC(FSSTAT,		fhandle,	fsstatres, 0),
+  PROC(FSINFO,  	fhandle,	fsinfores, 0),
+  PROC(PATHCONF,	fhandle,	pathconfres, 0),
+  PROC(COMMIT,		commitargs,	commitres, 5),
+  PROC(GETXATTR,	getxattrargs,	getxattrres, 1),
+  PROC(SETXATTR,	setxattrargs,	setxattrres, 0),
+};
+
+struct rpc_version		selinux_nfs_version3 = {
+	.number			= 3,
+	.nrprocs		= sizeof(nfs3_procedures)/
+	                                 sizeof(nfs3_procedures[0]),
+	.procs			= selinux_nfs3_procedures
+};
+#endif /* CONFIG_NFS_SELINUX */
Index: nfs-2.6/fs/nfs/xattr.c
diff -u /dev/null nfs-2.6/fs/nfs/xattr.c:1.4
--- /dev/null	Wed Apr  7 15:15:24 2004
+++ nfs-2.6/fs/nfs/xattr.c	Wed Mar  3 15:40:37 2004
@@ -0,0 +1,62 @@
+/*
+ * File: fs/nfs/xattr.c
+ *
+ * Derived from fs/devpts/xattr.c
+ */
+
+#include <linux/fs.h>
+#include <linux/nfs_fs.h>
+#include <linux/sunrpc/xattr.h>
+
+/*
+ * Inode operation getxattr()
+ *
+ * dentry->d_inode->i_sem down
+ */
+
+ssize_t
+nfs_getxattr(struct dentry *dentry, const char *name, void *buffer, size_t size)
+{
+	struct inode *inode = dentry->d_inode;
+	int xattrlen;
+	int error;
+
+	error = -EOPNOTSUPP;
+	if (NFS_PROTO(inode)->getxattr) {
+		if (strcmp(name, "") == 0)
+			return -EINVAL;
+		error = NFS_PROTO(inode)->getxattr(dentry, name, buffer, size, 
+						   &xattrlen);
+	}
+	if (error < 0)
+		return (ssize_t)error;
+	
+	return (ssize_t)xattrlen;
+}
+
+/*
+ * Inode operation setxattr()
+ *
+ * dentry->d_inode->i_sem down
+ */
+int
+nfs_setxattr(struct dentry *dentry, const char *name,
+	      const void *value, size_t size, int flags)
+{
+	struct inode *inode = dentry->d_inode;
+	int error;
+
+	error = -EOPNOTSUPP;
+	if (NFS_PROTO(inode)->setxattr) {
+		if (strcmp(name, "") == 0)
+			return -EINVAL;
+		if (size == 0)
+			value = ""; /* empty EA, do not remove */
+		if (size > NFS_XATTR_SIZE_MAX)
+			return -EOPNOTSUPP;
+		error = NFS_PROTO(inode)->setxattr(dentry, name, value, size, 
+						   flags);
+	}
+
+	return error;
+}
Index: nfs-2.6/fs/nfsd/export.c
diff -u nfs-2.6/fs/nfsd/export.c:1.1.1.3 nfs-2.6/fs/nfsd/export.c:1.5
--- nfs-2.6/fs/nfsd/export.c:1.1.1.3	Fri Apr  2 14:13:48 2004
+++ nfs-2.6/fs/nfsd/export.c	Mon Apr  5 09:35:36 2004
@@ -1014,6 +1014,7 @@
 	{ NFSEXP_CROSSMOUNT, {"crossmnt", ""}},
 	{ NFSEXP_NOSUBTREECHECK, {"no_subtree_check", ""}},
 	{ NFSEXP_NOAUTHNLM, {"insecure_locks", ""}},
+	{ NFSEXP_SELINUX, {"selinux", ""}},
 #ifdef MSNFS
 	{ NFSEXP_MSNFS, {"msnfs", ""}},
 #endif
Index: nfs-2.6/fs/nfsd/nfs3proc.c
diff -u nfs-2.6/fs/nfsd/nfs3proc.c:1.1.1.2 nfs-2.6/fs/nfsd/nfs3proc.c:1.12
--- nfs-2.6/fs/nfsd/nfs3proc.c:1.1.1.2	Fri Apr  2 14:13:48 2004
+++ nfs-2.6/fs/nfsd/nfs3proc.c	Mon Apr  5 09:35:36 2004
@@ -24,6 +24,9 @@
 #include <linux/nfsd/cache.h>
 #include <linux/nfsd/xdr3.h>
 #include <linux/nfs3.h>
+#include <linux/sunrpc/xdr.h>
+#include <linux/sunrpc/xattr.h>
+#include <linux/security.h>
 
 #define NFSDDBG_FACILITY		NFSDDBG_PROC
 
@@ -57,12 +60,17 @@
 					   struct nfsd3_attrstat *resp)
 {
 	int	nfserr;
+	struct svc_fh *fhp;
 
 	dprintk("nfsd: GETATTR(3)  %s\n",
 				SVCFH_fmt(&argp->fh));
 
 	fh_copy(&resp->fh, &argp->fh);
 	nfserr = fh_verify(rqstp, &resp->fh, 0, MAY_NOP);
+	if (nfserr)
+		RETURN_STATUS(nfserr);
+	fhp = &resp->fh;
+	nfserr = security_inode_getattr(fhp->fh_export->ex_mnt, fhp->fh_dentry);
 	RETURN_STATUS(nfserr);
 }
 
@@ -623,6 +631,39 @@
 	RETURN_STATUS(nfserr);
 }
 
+#ifdef CONFIG_NFSD_XATTR
+/*
+ * Get the named Extended Attribute value for the given file
+ */
+static int
+nfsd3_proc_getxattr(struct svc_rqst *rqstp, struct nfsd3_getxattrargs *argp,
+		    struct nfsd3_getxattrres *resp)
+{
+	int nfserr;
+	fh_copy(&resp->fh, &argp->fh);
+	nfserr = nfsd_getxattr(rqstp, &resp->fh, argp->name, argp->size, 
+			       resp->buffer, &resp->xattrlen);
+	if (argp->size == 0)
+		resp->buflen = 0;
+	else
+		resp->buflen = resp->xattrlen;
+	RETURN_STATUS(nfserr);
+}
+
+/*
+ * Set the named Extended Attribute value for the given file
+ */
+static int
+nfsd3_proc_setxattr(struct svc_rqst *rqstp, struct nfsd3_setxattrargs *argp,
+		    struct nfsd3_setxattrres *resp)
+{
+	int nfserr;
+	fh_copy(&resp->fh, &argp->fh);
+	nfserr = nfsd_setxattr(rqstp, &resp->fh, argp->name, argp->value, 
+			       argp->size, argp->flags);
+	RETURN_STATUS(nfserr);
+}
+#endif /* CONFIG_NFSD_XATTR */
 
 /*
  * NFSv3 Server procedures.
@@ -660,8 +701,9 @@
 #define AT 21		/* attributes */
 #define pAT (1+AT)	/* post attributes - conditional */
 #define WC (7+pAT)	/* WCC attributes */
+#define EA (1+XDR_QUADLEN(NFS_XATTR_SIZE_MAX))	/* xattr */
 
-static struct svc_procedure		nfsd_procedures3[22] = {
+static struct svc_procedure		nfsd_procedures3[24] = {
   PROC(null,	 void,		void,		void,	  RC_NOCACHE, ST),
   PROC(getattr,	 fhandle,	attrstat,	fhandle,  RC_NOCACHE, ST+AT),
   PROC(setattr,  sattr,		wccstat,	fhandle,  RC_REPLBUFF, ST+WC),
@@ -693,3 +735,77 @@
 		.vs_dispatch	= nfsd_dispatch,
 		.vs_xdrsize	= NFS3_SVC_XDRSIZE,
 };
+
+#ifdef CONFIG_NFSD_SELINUX
+static struct svc_procedure		selinux_nfsd_procedures3[24] = {
+  PROC(null,	 void,		void,		void,	  RC_NOCACHE, ST),
+  PROC(getattr,	 fhandle,	attrstat,	fhandle,  RC_NOCACHE, ST+AT),
+  PROC(setattr,  sattr,		wccstat,	fhandle,  RC_REPLBUFF, ST+WC),
+  PROC(lookup,	 dirop,		dirop,		fhandle2, RC_NOCACHE, ST+FH+pAT+pAT),
+  PROC(access,	 access,	access,		fhandle,  RC_NOCACHE, ST+pAT+1),
+  PROC(readlink, readlink,	readlink,	fhandle,  RC_NOCACHE, ST+pAT+1+NFS3_MAXPATHLEN/4),
+  PROC(read,	 read,		read,		fhandle,  RC_NOCACHE, ST+pAT+4+NFSSVC_MAXBLKSIZE),
+  PROC(write,	 write,		write,		fhandle,  RC_REPLBUFF, ST+WC+4),
+  PROC(create,	 create,	create,		fhandle2, RC_REPLBUFF, ST+(1+FH+pAT)+WC),
+  PROC(mkdir,	 mkdir,		create,		fhandle2, RC_REPLBUFF, ST+(1+FH+pAT)+WC),
+  PROC(symlink,	 symlink,	create,		fhandle2, RC_REPLBUFF, ST+(1+FH+pAT)+WC),
+  PROC(mknod,	 mknod,		create,		fhandle2, RC_REPLBUFF, ST+(1+FH+pAT)+WC),
+  PROC(remove,	 dirop,		wccstat,	fhandle,  RC_REPLBUFF, ST+WC),
+  PROC(rmdir,	 dirop,		wccstat,	fhandle,  RC_REPLBUFF, ST+WC),
+  PROC(rename,	 rename,	rename,		fhandle2, RC_REPLBUFF, ST+WC+WC),
+  PROC(link,	 link,		link,		fhandle2, RC_REPLBUFF, ST+pAT+WC),
+  PROC(readdir,	 readdir,	readdir,	fhandle,  RC_NOCACHE, 0),
+  PROC(readdirplus,readdirplus,	readdir,	fhandle,  RC_NOCACHE, 0),
+  PROC(fsstat,	 fhandle,	fsstat,		void,     RC_NOCACHE, ST+pAT+2*6+1),
+  PROC(fsinfo,   fhandle,	fsinfo,		void,     RC_NOCACHE, ST+pAT+12),
+  PROC(pathconf, fhandle,	pathconf,	void,     RC_NOCACHE, ST+pAT+6),
+  PROC(commit,	 commit,	commit,		fhandle,  RC_NOCACHE, ST+WC+2),
+  PROC(getxattr, getxattr,	getxattr,	fhandle,  RC_NOCACHE, ST+1+EA),
+  PROC(setxattr, setxattr,	setxattr,	fhandle,  RC_NOCACHE, ST),
+};
+
+struct svc_version	selinux_nfsd_version3 = {
+		.vs_vers	= 3,
+		.vs_nproc	= 24,
+		.vs_proc	= selinux_nfsd_procedures3,
+		.vs_dispatch	= nfsd_dispatch,
+		.vs_xdrsize	= NFS3EA_SVC_XDRSIZE,
+};
+#endif /* CONFIG_NFSD_SELINUX */
+
+#ifdef CONFIG_NFSD_XATTR
+static struct svc_procedure		xattr_nfsd_procedures3[24] = {
+  PROC(null,	 void,		void,		void,	  RC_NOCACHE, ST),
+  PROC(getattr,	 fhandle,	attrstat,	fhandle,  RC_NOCACHE, ST+AT),
+  PROC(setattr,  sattr,		wccstat,	fhandle,  RC_REPLBUFF, ST+WC),
+  PROC(lookup,	 dirop,		dirop,		fhandle2, RC_NOCACHE, ST+FH+pAT+pAT),
+  PROC(access,	 access,	access,		fhandle,  RC_NOCACHE, ST+pAT+1),
+  PROC(readlink, readlink,	readlink,	fhandle,  RC_NOCACHE, ST+pAT+1+NFS3_MAXPATHLEN/4),
+  PROC(read,	 read,		read,		fhandle,  RC_NOCACHE, ST+pAT+4+NFSSVC_MAXBLKSIZE),
+  PROC(write,	 write,		write,		fhandle,  RC_REPLBUFF, ST+WC+4),
+  PROC(create,	 create,	create,		fhandle2, RC_REPLBUFF, ST+(1+FH+pAT)+WC),
+  PROC(mkdir,	 mkdir,		create,		fhandle2, RC_REPLBUFF, ST+(1+FH+pAT)+WC),
+  PROC(symlink,	 symlink,	create,		fhandle2, RC_REPLBUFF, ST+(1+FH+pAT)+WC),
+  PROC(mknod,	 mknod,		create,		fhandle2, RC_REPLBUFF, ST+(1+FH+pAT)+WC),
+  PROC(remove,	 dirop,		wccstat,	fhandle,  RC_REPLBUFF, ST+WC),
+  PROC(rmdir,	 dirop,		wccstat,	fhandle,  RC_REPLBUFF, ST+WC),
+  PROC(rename,	 rename,	rename,		fhandle2, RC_REPLBUFF, ST+WC+WC),
+  PROC(link,	 link,		link,		fhandle2, RC_REPLBUFF, ST+pAT+WC),
+  PROC(readdir,	 readdir,	readdir,	fhandle,  RC_NOCACHE, 0),
+  PROC(readdirplus,readdirplus,	readdir,	fhandle,  RC_NOCACHE, 0),
+  PROC(fsstat,	 fhandle,	fsstat,		void,     RC_NOCACHE, ST+pAT+2*6+1),
+  PROC(fsinfo,   fhandle,	fsinfo,		void,     RC_NOCACHE, ST+pAT+12),
+  PROC(pathconf, fhandle,	pathconf,	void,     RC_NOCACHE, ST+pAT+6),
+  PROC(commit,	 commit,	commit,		fhandle,  RC_NOCACHE, ST+WC+2),
+  PROC(getxattr, getxattr,	getxattr,	fhandle,  RC_NOCACHE, ST+1+EA),
+  PROC(setxattr, setxattr,	setxattr,	fhandle,  RC_NOCACHE, ST),
+};
+
+struct svc_version	xattr_nfsd_version3 = {
+		.vs_vers	= 3,
+		.vs_nproc	= 24,
+		.vs_proc	= xattr_nfsd_procedures3,
+		.vs_dispatch	= nfsd_dispatch,
+		.vs_xdrsize	= NFS3EA_SVC_XDRSIZE,
+};
+#endif /* CONFIG_NFSD_XATTR */
Index: nfs-2.6/fs/nfsd/nfs3xdr.c
diff -u nfs-2.6/fs/nfsd/nfs3xdr.c:1.1.1.4 nfs-2.6/fs/nfsd/nfs3xdr.c:1.15
--- nfs-2.6/fs/nfsd/nfs3xdr.c:1.1.1.4	Mon Apr  5 11:55:36 2004
+++ nfs-2.6/fs/nfsd/nfs3xdr.c	Mon Apr  5 16:31:52 2004
@@ -21,6 +21,7 @@
 #include <linux/sunrpc/svc.h>
 #include <linux/nfsd/nfsd.h>
 #include <linux/nfsd/xdr3.h>
+#include <linux/security.h>
 
 #define NFSDDBG_FACILITY		NFSDDBG_XDR
 
@@ -241,7 +242,8 @@
 encode_post_op_attr(struct svc_rqst *rqstp, u32 *p, struct svc_fh *fhp)
 {
 	struct dentry *dentry = fhp->fh_dentry;
-	if (dentry && dentry->d_inode != NULL) {
+	if (dentry && dentry->d_inode != NULL && 
+	    !security_inode_getattr(fhp->fh_export->ex_mnt, dentry)) {
 		*p++ = xdr_one;		/* attributes follow */
 		return encode_fattr3(rqstp, p, fhp);
 	}
@@ -596,6 +598,34 @@
 	return xdr_argsize_check(rqstp, p);
 }
 
+#ifdef CONFIG_NFSD_XATTR
+int
+nfs3svc_decode_getxattrargs(struct svc_rqst *rqstp, u32 *p, 
+			    struct nfsd3_getxattrargs *args)
+{
+	if (!(p = decode_fh(p, &args->fh)) ||
+	      !(p = xdr_decode_string_inplace(p, &args->name, &args->namelen,
+					      XATTR_NAME_MAX)))
+		return 0;
+	args->size = ntohl(*p++);
+	return xdr_argsize_check(rqstp, p);
+}    
+
+int
+nfs3svc_decode_setxattrargs(struct svc_rqst *rqstp, u32 *p, 
+			    struct nfsd3_setxattrargs *args)
+{
+	if (!(p = decode_fh(p, &args->fh)) ||
+	      !(p = xdr_decode_string_inplace(p, &args->name, &args->namelen,
+					      XATTR_NAME_MAX)) ||
+		!(p = xdr_decode_string_inplace(p, &args->value, &args->size,
+						NFS_XATTR_SIZE_MAX)))
+		return 0;
+	args->flags = ntohl(*p++);
+	return xdr_argsize_check(rqstp, p);
+}    
+#endif /* CONFIG_NFSD_XATTR */
+
 /*
  * XDR encode functions
  */
@@ -1045,6 +1075,32 @@
 	}
 	return xdr_ressize_check(rqstp, p);
 }
+
+#ifdef CONFIG_NFSD_XATTR
+/* GETXATTR */
+int
+nfs3svc_encode_getxattrres(struct svc_rqst *rqstp, u32 *p,
+			   struct nfsd3_getxattrres *res)
+{
+	struct iovec *resv = &rqstp->rq_res.head[0];
+	char *cp;
+	int len;
+
+	*p++ = htonl(res->xattrlen);
+	p = xdr_encode_array(p, res->buffer, res->buflen);
+	cp = (char *)p;
+	len = cp - (char *)resv->iov_base;
+	return xdr_ressize_check(rqstp, p);
+}
+
+/* SETXATTR */
+int
+nfs3svc_encode_setxattrres(struct svc_rqst *rqstp, u32 *p,
+			   struct nfsd3_setxattrres *res)
+{
+	return xdr_ressize_check(rqstp, p);
+}
+#endif /* CONFIG_NFSD_XATTR */
 
 /*
  * XDR release functions
Index: nfs-2.6/fs/nfsd/nfsfh.c
diff -u nfs-2.6/fs/nfsd/nfsfh.c:1.1.1.4 nfs-2.6/fs/nfsd/nfsfh.c:1.9
--- nfs-2.6/fs/nfsd/nfsfh.c:1.1.1.4	Fri Apr  2 14:13:50 2004
+++ nfs-2.6/fs/nfsd/nfsfh.c	Mon Apr  5 09:35:36 2004
@@ -149,8 +149,14 @@
 		if (!exp || IS_ERR(exp))
 			goto out;
 
-		/* Check if the request originated from a secure port. */
 		error = nfserr_perm;
+#ifdef CONFIG_NFSD_SELINUX
+                if (EX_SELINUX(exp) && !rqstp->rq_selinux) {
+			printk(KERN_WARNING "nfsd: request not using SELinux\n");
+			goto out;
+		}
+#endif
+		/* Check if the request originated from a secure port. */
 		if (!rqstp->rq_secure && EX_SECURE(exp)) {
 			printk(KERN_WARNING
 			       "nfsd: request from insecure port (%08x:%d)!\n",
Index: nfs-2.6/fs/nfsd/nfsproc.c
diff -u nfs-2.6/fs/nfsd/nfsproc.c:1.1.1.3 nfs-2.6/fs/nfsd/nfsproc.c:1.3
--- nfs-2.6/fs/nfsd/nfsproc.c:1.1.1.3	Fri Apr  2 14:13:50 2004
+++ nfs-2.6/fs/nfsd/nfsproc.c	Mon Apr  5 09:35:36 2004
@@ -579,6 +579,10 @@
 		{ nfserr_nospc, -ENOSPC },
 		{ nfserr_rofs, -EROFS },
 		{ nfserr_mlink, -EMLINK },
+#ifdef CONFIG_NFSD_XATTR
+		{ nfserr_erange, -ERANGE },
+		{ nfserr_eopnotsupp, -EOPNOTSUPP },
+#endif
 		{ nfserr_nametoolong, -ENAMETOOLONG },
 		{ nfserr_notempty, -ENOTEMPTY },
 #ifdef EDQUOT
Index: nfs-2.6/fs/nfsd/nfssvc.c
diff -u nfs-2.6/fs/nfsd/nfssvc.c:1.1.1.1 nfs-2.6/fs/nfsd/nfssvc.c:1.11
--- nfs-2.6/fs/nfsd/nfssvc.c:1.1.1.1	Mon Aug 18 12:48:27 2003
+++ nfs-2.6/fs/nfsd/nfssvc.c	Tue Mar  9 08:58:38 2004
@@ -31,6 +31,8 @@
 #include <linux/nfsd/stats.h>
 #include <linux/nfsd/cache.h>
 #include <linux/lockd/bind.h>
+#include <linux/sunrpc/xattr.h>
+#include <linux/sunrpc/selinux.h>
 
 #define NFSDDBG_FACILITY	NFSDDBG_SVC
 
@@ -48,7 +50,8 @@
  */
 #define	SIG_NOCLEAN	SIGHUP
 
-extern struct svc_program	nfsd_program;
+extern struct svc_program	nfsd_program, selinux_nfsd_program, 
+	                        xattr_nfsd_program;
 static void			nfsd(struct svc_rqst *rqstp);
 struct timeval			nfssvc_boot;
 static struct svc_serv 		*nfsd_serv;
@@ -371,6 +374,11 @@
 
 #define NFSD_NRVERS		(sizeof(nfsd_version)/sizeof(nfsd_version[0]))
 struct svc_program		nfsd_program = {
+#ifdef CONFIG_NFSD_XATTR
+	.pg_next		= &xattr_nfsd_program,	/* next registered program */
+#else
+	.pg_next		= NULL,
+#endif
 	.pg_prog		= NFS_PROGRAM,		/* program number */
 	.pg_nvers		= NFSD_NRVERS,		/* nr of entries in nfsd_version */
 	.pg_vers		= nfsd_version,		/* version table */
@@ -378,3 +386,43 @@
 	.pg_class		= "nfsd",		/* authentication class */
 	.pg_stats		= &nfsd_svcstats,	/* version table */
 };
+
+#ifdef CONFIG_NFSD_XATTR
+extern struct svc_version	xattr_nfsd_version3;
+static struct svc_version *	xattr_nfsd_version[] = {	
+	[3] = &xattr_nfsd_version3,
+};
+#define XATTR_NFSD_NRVERS	(sizeof(xattr_nfsd_version)/sizeof(xattr_nfsd_version[0]))
+
+struct svc_program		xattr_nfsd_program = {
+#ifdef CONFIG_NFSD_SELINUX
+	.pg_next		= &selinux_nfsd_program, /* next registered program */
+#else
+	.pg_next		= NULL,
+#endif
+	.pg_prog		= XATTR_NFS_PROGRAM,	/* program number */
+	.pg_nvers		= XATTR_NFSD_NRVERS,	/* nr of entries in nfsd_version */
+	.pg_vers		= xattr_nfsd_version,	/* version table */
+	.pg_name		= "xattr_nfsd",		/* program name */
+	.pg_class		= "xattr_nfsd",		/* authentication class */
+	.pg_stats		= &xattr_nfsd_svcstats,	
+};
+#endif /* CONFIG_NFSD_XATTR */
+
+#ifdef CONFIG_NFSD_SELINUX
+extern struct svc_version	selinux_nfsd_version3;
+static struct svc_version *	selinux_nfsd_version[] = {	
+	[3] = &selinux_nfsd_version3,
+};
+#define SELINUX_NFSD_NRVERS	(sizeof(selinux_nfsd_version)/sizeof(selinux_nfsd_version[0]))
+
+struct svc_program		selinux_nfsd_program = {
+	.pg_next		= NULL,			/* next registered program */
+	.pg_prog		= SELINUX_NFS_PROGRAM,	/* program number */
+	.pg_nvers		= SELINUX_NFSD_NRVERS,	/* nr of entries in nfsd_version */
+	.pg_vers		= selinux_nfsd_version,	/* version table */
+	.pg_name		= "selinux_nfsd",	/* program name */
+	.pg_class		= "selinux_nfsd",       /* authentication class */
+	.pg_stats		= &selinux_nfsd_svcstats,	
+};
+#endif /* CONFIG_NFSD_SELINUX */
Index: nfs-2.6/fs/nfsd/stats.c
diff -u nfs-2.6/fs/nfsd/stats.c:1.1.1.2 nfs-2.6/fs/nfsd/stats.c:1.4
--- nfs-2.6/fs/nfsd/stats.c:1.1.1.2	Fri Apr  2 14:13:50 2004
+++ nfs-2.6/fs/nfsd/stats.c	Mon Apr  5 09:35:36 2004
@@ -39,6 +39,17 @@
 struct svc_stat		nfsd_svcstats = {
 	.program	= &nfsd_program,
 };
+#ifdef CONFIG_NFSD_SELINUX
+struct svc_stat		selinux_nfsd_svcstats = {
+	.program	= &selinux_nfsd_program,
+};
+#endif
+
+#ifdef CONFIG_NFSD_XATTR
+struct svc_stat		xattr_nfsd_svcstats = {
+	.program	= &xattr_nfsd_program,
+};
+#endif
 
 static int nfsd_proc_show(struct seq_file *seq, void *v)
 {
@@ -71,7 +82,14 @@
 	
 	/* show my rpc info */
 	svc_seq_show(seq, &nfsd_svcstats);
-
+#ifdef CONFIG_NFSD_XATTR
+	seq_printf(seq,"xattr:\n");
+	svc_seq_show(seq, &xattr_nfsd_svcstats);
+#endif
+#ifdef CONFIG_NFSD_SELINUX
+	seq_printf(seq,"selinux:\n");
+	svc_seq_show(seq, &selinux_nfsd_svcstats);
+#endif
 	return 0;
 }
 
Index: nfs-2.6/fs/nfsd/vfs.c
diff -u nfs-2.6/fs/nfsd/vfs.c:1.1.1.5 nfs-2.6/fs/nfsd/vfs.c:1.13
--- nfs-2.6/fs/nfsd/vfs.c:1.1.1.5	Mon Apr  5 11:55:37 2004
+++ nfs-2.6/fs/nfsd/vfs.c	Mon Apr  5 16:31:52 2004
@@ -44,6 +44,7 @@
 #include <linux/nfsd/nfsfh.h>
 #include <linux/quotaops.h>
 #include <linux/dnotify.h>
+#include <linux/security.h>
 
 #include <asm/uaccess.h>
 
@@ -1584,7 +1585,7 @@
 	 */
 	if ((acc & MAY_OWNER_OVERRIDE) &&
 	    inode->i_uid == current->fsuid)
-		return 0;
+		return security_inode_permission(inode, acc & (MAY_READ|MAY_WRITE|MAY_EXEC), NULL);
 
 	err = permission(inode, acc & (MAY_READ|MAY_WRITE|MAY_EXEC), NULL);
 
@@ -1633,3 +1634,74 @@
 	nfsdstats.ra_size = cache_size;
 	return 0;
 }
+
+#ifdef CONFIG_NFSD_XATTR
+/*
+ * Get file extended attribute
+ * After this call fhp needs an fh_put
+ */
+int
+nfsd_getxattr(struct svc_rqst *rqstp, struct svc_fh *fhp, char *name, 
+	      u32 size, char *buffer, int *xattrlen)
+{
+	struct dentry *dentry;
+	int error;
+
+	error = fh_verify(rqstp, fhp, 0, MAY_NOP);
+	if (error)
+		return error;
+	dentry = fhp->fh_dentry;
+	error = -EOPNOTSUPP;
+	if (dentry->d_inode->i_op && dentry->d_inode->i_op->getxattr) {
+		error = security_inode_getxattr(dentry, name);
+		if (error)
+			return  nfserrno(error);
+		if (size == 0)
+			buffer = NULL;
+		if (size > NFS_XATTR_SIZE_MAX) {
+			*xattrlen = dentry->d_inode->i_op->getxattr(dentry, name, 
+							    NULL, size);
+			if (*xattrlen > NFS_XATTR_SIZE_MAX){
+				*xattrlen = 0;
+				return  nfserrno(-EOPNOTSUPP);
+			}
+		}
+		*xattrlen = dentry->d_inode->i_op->getxattr(dentry, name, 
+							    buffer, size);
+		if (*xattrlen < 0) {
+			error = *xattrlen;
+			*xattrlen = 0;
+		}
+	}
+	return  nfserrno(error);
+}
+
+/*
+ * Set file extended attribute
+ * After this call fhp needs an fh_put
+ */
+int
+nfsd_setxattr(struct svc_rqst *rqstp, struct svc_fh *fhp, char *name, 
+	      char *value, u32 size, int flags)
+{
+	struct dentry *dentry;
+	int error;
+
+	error = fh_verify(rqstp, fhp, 0, MAY_NOP);
+	if (error)
+		return error;
+	dentry = fhp->fh_dentry;
+	error = nfserr_opnotsupp;
+	if (dentry->d_inode->i_op && dentry->d_inode->i_op->setxattr) {
+		error = security_inode_setxattr(dentry, name, value, size, flags);
+		if (error)
+			return nfserrno(error);
+		error = dentry->d_inode->i_op->setxattr(dentry, name, value, 
+							size, flags);
+		if (!error)
+			security_inode_post_setxattr(dentry, name, value, size, 
+						     flags);
+	}
+	return error;
+}
+#endif /* CONFIG_NFSD_XATTR */
Index: nfs-2.6/include/linux/nfs.h
diff -u nfs-2.6/include/linux/nfs.h:1.1.1.2 nfs-2.6/include/linux/nfs.h:1.3
--- nfs-2.6/include/linux/nfs.h:1.1.1.2	Fri Apr  2 14:17:06 2004
+++ nfs-2.6/include/linux/nfs.h	Mon Apr  5 09:35:40 2004
@@ -56,12 +56,14 @@
 	NFSERR_NOSPC = 28,		/* v2 v3 v4 */
 	NFSERR_ROFS = 30,		/* v2 v3 v4 */
 	NFSERR_MLINK = 31,		/*    v3 v4 */
+	NFSERR_ERANGE = 34,		/*    v3ea  */
 	NFSERR_OPNOTSUPP = 45,		/* v2 v3 */
 	NFSERR_NAMETOOLONG = 63,	/* v2 v3 v4 */
 	NFSERR_NOTEMPTY = 66,		/* v2 v3 v4 */
 	NFSERR_DQUOT = 69,		/* v2 v3 v4 */
 	NFSERR_STALE = 70,		/* v2 v3 v4 */
 	NFSERR_REMOTE = 71,		/* v2 v3 */
+	NFSERR_EOPNOTSUPP = 95,		/*    v3ea  */
 	NFSERR_WFLUSH = 99,		/* v2    */
 	NFSERR_BADHANDLE = 10001,	/*    v3 v4 */
 	NFSERR_NOT_SYNC = 10002,	/*    v3 */
Index: nfs-2.6/include/linux/nfs3.h
diff -u nfs-2.6/include/linux/nfs3.h:1.1.1.1 nfs-2.6/include/linux/nfs3.h:1.2
--- nfs-2.6/include/linux/nfs3.h:1.1.1.1	Mon Aug 18 12:47:53 2003
+++ nfs-2.6/include/linux/nfs3.h	Wed Nov 26 14:19:57 2003
@@ -87,6 +87,8 @@
 #define NFS3PROC_FSINFO		19
 #define NFS3PROC_PATHCONF	20
 #define NFS3PROC_COMMIT		21
+#define NFS3PROC_GETXATTR	22
+#define NFS3PROC_SETXATTR	23
 
 #define NFS_MNT3_PROGRAM	100005
 #define NFS_MNT3_VERSION	3
Index: nfs-2.6/include/linux/nfs_mount.h
diff -u nfs-2.6/include/linux/nfs_mount.h:1.1.1.2 nfs-2.6/include/linux/nfs_mount.h:1.4
--- nfs-2.6/include/linux/nfs_mount.h:1.1.1.2	Thu Feb 19 14:08:42 2004
+++ nfs-2.6/include/linux/nfs_mount.h	Fri Feb 20 08:32:59 2004
@@ -60,6 +60,8 @@
 #define NFS_MOUNT_BROKEN_SUID	0x0400	/* 4 */
 #define NFS_MOUNT_STRICTLOCK	0x1000	/* reserved for NFSv4 */
 #define NFS_MOUNT_SECFLAVOUR	0x2000	/* 5 */
-#define NFS_MOUNT_FLAGMASK	0xFFFF
+#define NFS_MOUNT_XATTR		0x100000 /* xattr */
+#define NFS_MOUNT_SELINUX	0x200000 /* SELinux */
+#define NFS_MOUNT_FLAGMASK	0x20FFFF
 
 #endif
Index: nfs-2.6/include/linux/nfs_xdr.h
diff -u nfs-2.6/include/linux/nfs_xdr.h:1.1.1.4 nfs-2.6/include/linux/nfs_xdr.h:1.8
--- nfs-2.6/include/linux/nfs_xdr.h:1.1.1.4	Mon Apr  5 11:58:16 2004
+++ nfs-2.6/include/linux/nfs_xdr.h	Mon Apr  5 16:32:14 2004
@@ -439,6 +439,24 @@
 	struct page **		pages;
 };
 
+#ifdef CONFIG_NFS_XATTR
+struct nfs3_getxattrargs {
+	struct nfs_fh *		fh;
+	const char *		name;
+	unsigned int		namelen;
+	unsigned int		bufsize;
+};
+
+struct nfs3_setxattrargs {
+	struct nfs_fh *		fh;
+	const char *		name;
+	unsigned int		namelen;
+	const char *		value;
+	int			size;
+	int			flags;
+};
+#endif /* CONFIG_NFS_XATTR */
+
 struct nfs3_diropres {
 	struct nfs_fattr *	dir_attr;
 	struct nfs_fh *		fh;
@@ -472,6 +490,17 @@
 	int			plus;
 };
 
+#ifdef CONFIG_NFS_XATTR
+struct nfs3_getxattrres {	
+	char *			buffer;
+	unsigned int		size;
+	unsigned int *		xattrlen;
+};
+
+struct nfs3_setxattrres {
+};
+#endif /* CONFIG_NFS_XATTR */
+
 #ifdef CONFIG_NFS_V4
 
 typedef u64 clientid4;
@@ -740,6 +769,12 @@
 	void	(*read_setup)   (struct nfs_read_data *, unsigned int count);
 	void	(*write_setup)  (struct nfs_write_data *, unsigned int count, int how);
 	void	(*commit_setup) (struct nfs_write_data *, u64 start, u32 len, int how);
+#ifdef CONFIG_NFS_XATTR
+	int	(*getxattr) (struct dentry *, const char *, void *, unsigned int, 
+			     unsigned int *);
+	int	(*setxattr) (struct dentry *, const char *, const void *, 
+			     unsigned int, int);
+#endif
 	int	(*file_open)   (struct inode *, struct file *);
 	int	(*file_release) (struct inode *, struct file *);
 	void	(*request_init)(struct nfs_page *, struct file *);
@@ -765,5 +800,14 @@
 extern struct rpc_version	nfs_version4;
 extern struct rpc_program	nfs_program;
 extern struct rpc_stat		nfs_rpcstat;
+#ifdef CONFIG_NFS_XATTR
+extern struct nfs_rpc_ops	nfs_v3ea_clientops;
+extern struct rpc_version	xattr_nfs_version3;
+extern struct rpc_version	selinux_nfs_version3;
+extern struct rpc_program	selinux_nfs_program;
+extern struct rpc_program	xattr_nfs_program;
+extern struct rpc_stat		selinux_rpcstat;
+extern struct rpc_stat		xattr_rpcstat;
+#endif /* CONFIG_NFS_XATTR */
 
 #endif
Index: nfs-2.6/include/linux/nfsd/export.h
diff -u nfs-2.6/include/linux/nfsd/export.h:1.1.1.2 nfs-2.6/include/linux/nfsd/export.h:1.4
--- nfs-2.6/include/linux/nfsd/export.h:1.1.1.2	Mon Jan 12 13:54:35 2004
+++ nfs-2.6/include/linux/nfsd/export.h	Tue Feb 24 15:43:56 2004
@@ -40,7 +40,8 @@
 #define NFSEXP_FSID		0x2000
 #define	NFSEXP_CROSSMOUNT	0x4000
 #define	NFSEXP_NOACL		0x8000	/* reserved for possible ACL related use */
-#define NFSEXP_ALLFLAGS		0xFE3F
+#define NFSEXP_SELINUX		0x100000
+#define NFSEXP_ALLFLAGS		0x10FE3F
 
 
 #ifdef __KERNEL__
@@ -75,6 +76,7 @@
 #define EX_RDONLY(exp)		((exp)->ex_flags & NFSEXP_READONLY)
 #define EX_NOHIDE(exp)		((exp)->ex_flags & NFSEXP_NOHIDE)
 #define EX_WGATHER(exp)		((exp)->ex_flags & NFSEXP_GATHERED_WRITES)
+#define EX_SELINUX(exp)		((exp)->ex_flags & NFSEXP_SELINUX)
 
 
 /*
Index: nfs-2.6/include/linux/nfsd/nfsd.h
diff -u nfs-2.6/include/linux/nfsd/nfsd.h:1.1.1.3 nfs-2.6/include/linux/nfsd/nfsd.h:1.8
--- nfs-2.6/include/linux/nfsd/nfsd.h:1.1.1.3	Fri Apr  2 14:17:21 2004
+++ nfs-2.6/include/linux/nfsd/nfsd.h	Mon Apr  5 09:35:40 2004
@@ -57,9 +57,18 @@
 typedef int (*nfsd_dirop_t)(struct inode *, struct dentry *, int, int);
 
 extern struct svc_program	nfsd_program;
-extern struct svc_version	nfsd_version2, nfsd_version3,
+extern struct svc_version	nfsd_version2, nfsd_version3, 
 				nfsd_version4;
 
+#ifdef CONFIG_NFSD_XATTR
+extern struct svc_program	xattr_nfsd_program;
+extern struct svc_version	xattr_nfsd_version3;
+#endif
+#ifdef CONFIG_NFSD_SELINUX
+extern struct svc_program	selinux_nfsd_program;
+extern struct svc_version	selinux_nfsd_version3;
+#endif 
+
 /*
  * Function prototypes.
  */
@@ -118,6 +127,12 @@
 
 int		nfsd_notify_change(struct inode *, struct iattr *);
 int		nfsd_permission(struct svc_export *, struct dentry *, int);
+#ifdef CONFIG_NFSD_XATTR
+int		nfsd_getxattr(struct svc_rqst *, struct svc_fh *, char *, 
+			      u32, char *, int *);
+int 		nfsd_setxattr(struct svc_rqst *, struct svc_fh *, char *, 
+			      char *, u32, int);
+#endif
 
 
 /* 
@@ -158,12 +173,14 @@
 #define	nfserr_nospc		__constant_htonl(NFSERR_NOSPC)
 #define	nfserr_rofs		__constant_htonl(NFSERR_ROFS)
 #define	nfserr_mlink		__constant_htonl(NFSERR_MLINK)
+#define nfserr_erange		__constant_htonl(NFSERR_ERANGE)
 #define	nfserr_opnotsupp	__constant_htonl(NFSERR_OPNOTSUPP)
 #define	nfserr_nametoolong	__constant_htonl(NFSERR_NAMETOOLONG)
 #define	nfserr_notempty		__constant_htonl(NFSERR_NOTEMPTY)
 #define	nfserr_dquot		__constant_htonl(NFSERR_DQUOT)
 #define	nfserr_stale		__constant_htonl(NFSERR_STALE)
 #define	nfserr_remote		__constant_htonl(NFSERR_REMOTE)
+#define nfserr_eopnotsupp	__constant_htonl(NFSERR_EOPNOTSUPP)
 #define	nfserr_wflush		__constant_htonl(NFSERR_WFLUSH)
 #define	nfserr_badhandle	__constant_htonl(NFSERR_BADHANDLE)
 #define	nfserr_notsync		__constant_htonl(NFSERR_NOT_SYNC)
Index: nfs-2.6/include/linux/nfsd/stats.h
diff -u nfs-2.6/include/linux/nfsd/stats.h:1.1.1.1 nfs-2.6/include/linux/nfsd/stats.h:1.3
--- nfs-2.6/include/linux/nfsd/stats.h:1.1.1.1	Mon Aug 18 12:47:57 2003
+++ nfs-2.6/include/linux/nfsd/stats.h	Wed Mar  3 15:53:35 2004
@@ -36,6 +36,13 @@
 
 extern struct nfsd_stats	nfsdstats;
 extern struct svc_stat		nfsd_svcstats;
+#ifdef CONFIG_NFSD_XATTR
+extern struct svc_stat		xattr_nfsd_svcstats;
+#endif
+#ifdef CONFIG_NFSD_SELINUX
+extern struct svc_stat		selinux_nfsd_svcstats;
+#endif 
+	                        
 
 void	nfsd_stat_init(void);
 void	nfsd_stat_shutdown(void);
Index: nfs-2.6/include/linux/nfsd/xdr3.h
diff -u nfs-2.6/include/linux/nfsd/xdr3.h:1.1.1.1 nfs-2.6/include/linux/nfsd/xdr3.h:1.6
--- nfs-2.6/include/linux/nfsd/xdr3.h:1.1.1.1	Mon Aug 18 12:47:57 2003
+++ nfs-2.6/include/linux/nfsd/xdr3.h	Tue Mar  9 08:59:12 2004
@@ -10,6 +10,8 @@
 #define _LINUX_NFSD_XDR3_H
 
 #include <linux/nfsd/xdr.h>
+#include <linux/sunrpc/xattr.h>
+#include <linux/sunrpc/selinux.h>
 
 struct nfsd3_sattrargs {
 	struct svc_fh		fh;
@@ -110,6 +112,24 @@
 	__u32			count;
 };
 
+#ifdef CONFIG_NFSD_XATTR
+struct nfsd3_getxattrargs {
+	struct svc_fh		fh;
+	char *			name;
+	__u32			namelen;
+	__u32			size;
+};
+
+struct nfsd3_setxattrargs {
+	struct svc_fh		fh;
+	char *			name;
+	__u32			namelen;
+	char *			value;			
+	__u32			size;
+	int			flags;
+};
+#endif /* CONFIG_NFSD_XATTR */
+
 struct nfsd3_attrstat {
 	__u32			status;
 	struct svc_fh		fh;
@@ -208,6 +228,22 @@
 	struct svc_fh		fh;
 };
 
+#ifdef CONFIG_NFSD_XATTR
+struct nfsd3_getxattrres {
+	__u32			status;
+	struct svc_fh		fh;
+	int			xattrlen;
+	char 			buffer[NFS_XATTR_SIZE_MAX];
+	__u32			buflen;
+	
+};
+
+struct nfsd3_setxattrres {
+	__u32			status;
+	struct svc_fh		fh;
+};
+#endif /* CONFIG_NFSD_XATTR */
+
 /* dummy type for release */
 struct nfsd3_fhandle_pair {
 	__u32			dummy;
@@ -244,6 +280,38 @@
 
 #define NFS3_SVC_XDRSIZE		sizeof(union nfsd3_xdrstore)
 
+#ifdef CONFIG_NFSD_XATTR
+union nfsd3ea_xdrstore {
+	struct nfsd3_sattrargs		sattrargs;
+	struct nfsd3_diropargs		diropargs;
+	struct nfsd3_readargs		readargs;
+	struct nfsd3_writeargs		writeargs;
+	struct nfsd3_createargs		createargs;
+	struct nfsd3_renameargs		renameargs;
+	struct nfsd3_linkargs		linkargs;
+	struct nfsd3_symlinkargs	symlinkargs;
+	struct nfsd3_readdirargs	readdirargs;
+	struct nfsd3_getxattrargs	getxattrargs;
+	struct nfsd3_setxattrargs	setxattrargs;
+	struct nfsd3_diropres 		diropres;
+	struct nfsd3_accessres		accessres;
+	struct nfsd3_readlinkres	readlinkres;
+	struct nfsd3_readres		readres;
+	struct nfsd3_writeres		writeres;
+	struct nfsd3_renameres		renameres;
+	struct nfsd3_linkres		linkres;
+	struct nfsd3_readdirres		readdirres;
+	struct nfsd3_fsstatres		fsstatres;
+	struct nfsd3_fsinfores		fsinfores;
+	struct nfsd3_pathconfres	pathconfres;
+	struct nfsd3_commitres		commitres;
+	struct nfsd3_getxattrres	getxattrres;
+	struct nfsd3_setxattrres	setxattrres;
+};
+
+#define NFS3EA_SVC_XDRSIZE		sizeof(union nfsd3ea_xdrstore)
+#endif /* CONFIG_NFSD_XATTR */
+
 int nfs3svc_decode_fhandle(struct svc_rqst *, u32 *, struct nfsd_fhandle *);
 int nfs3svc_decode_sattrargs(struct svc_rqst *, u32 *,
 				struct nfsd3_sattrargs *);
@@ -275,6 +343,12 @@
 				struct nfsd3_readdirargs *);
 int nfs3svc_decode_commitargs(struct svc_rqst *, u32 *,
 				struct nfsd3_commitargs *);
+#ifdef CONFIG_NFSD_XATTR
+int nfs3svc_decode_getxattrargs(struct svc_rqst *, u32 *, 
+				struct nfsd3_getxattrargs *);
+int nfs3svc_decode_setxattrargs(struct svc_rqst *, u32 *, 
+				struct nfsd3_setxattrargs *);
+#endif
 int nfs3svc_encode_voidres(struct svc_rqst *, u32 *, void *);
 int nfs3svc_encode_attrstat(struct svc_rqst *, u32 *,
 				struct nfsd3_attrstat *);
@@ -304,7 +378,12 @@
 				struct nfsd3_pathconfres *);
 int nfs3svc_encode_commitres(struct svc_rqst *, u32 *,
 				struct nfsd3_commitres *);
-
+#ifdef CONFIG_NFSD_XATTR
+int nfs3svc_encode_getxattrres(struct svc_rqst *, u32 *,
+			       struct nfsd3_getxattrres *);
+int nfs3svc_encode_setxattrres(struct svc_rqst *, u32 *,
+			       struct nfsd3_setxattrres *);
+#endif
 int nfs3svc_release_fhandle(struct svc_rqst *, u32 *,
 				struct nfsd3_attrstat *);
 int nfs3svc_release_fhandle2(struct svc_rqst *, u32 *,
Index: nfs-2.6/include/linux/sunrpc/clnt.h
diff -u nfs-2.6/include/linux/sunrpc/clnt.h:1.1.1.2 nfs-2.6/include/linux/sunrpc/clnt.h:1.3
--- nfs-2.6/include/linux/sunrpc/clnt.h:1.1.1.2	Thu Feb 19 14:09:01 2004
+++ nfs-2.6/include/linux/sunrpc/clnt.h	Fri Feb 20 08:33:00 2004
@@ -53,7 +53,8 @@
 				cl_autobind : 1,/* use getport() */
 				cl_droppriv : 1,/* enable NFS suid hack */
 				cl_oneshot  : 1,/* dispose after use */
-				cl_dead     : 1;/* abandoned */
+				cl_dead     : 1,/* abandoned */
+				cl_selinux  : 1;/* Use SELinux */
 
 	struct rpc_rtt *	cl_rtt;		/* RTO estimator data */
 	struct rpc_portmap *	cl_pmap;	/* port mapping */
Index: nfs-2.6/include/linux/sunrpc/selinux.h
diff -u /dev/null nfs-2.6/include/linux/sunrpc/selinux.h:1.7
--- /dev/null	Wed Apr  7 15:16:10 2004
+++ nfs-2.6/include/linux/sunrpc/selinux.h	Thu Mar  4 16:43:35 2004
@@ -0,0 +1,48 @@
+/* -*- linux-c -*- */
+
+#ifndef _SELINUX_RPC_H_
+#define _SELINUX_RPC_H_
+
+
+#include <linux/types.h>
+
+#define SELINUX_NFS_PROGRAM 	100006
+
+#define SELINUX_RPC_MAXCONTEXTLEN 255
+/* Size is 4 for length + Max length of context + 1 for \0*/
+#define SELINUX_RPC_SZ	(SELINUX_RPC_MAXCONTEXTLEN + 5)
+#define SELINUX_RPC_WORDS ((SELINUX_RPC_SZ+3)>>2)
+
+
+#ifdef CONFIG_SECURITY_SELINUX
+extern int security_sid_to_network_context(u32 sid, char *context,
+					   u32 *len);
+extern int security_network_context_to_sid(char *context, u32 len, 
+					   u32 *sid);
+extern u32 security_get_sid(void);
+extern int security_set_fssid(u32 sid);
+extern void security_clear_fssid(void);
+
+#else /* CONFIG_SECURITY_SELINUX not defined */
+static inline int security_sid_to_network_context(u32 sid, char *context, u32 *len)
+{
+	*len = 0;
+	return -EOPNOTSUPP;
+}
+static inline int security_network_context_to_sid(char *context, u32 len, u32 *sid)
+{
+	*sid = 0;
+	return -EOPNOTSUPP;
+}
+static inline u32 security_get_sid(void)
+{
+	return 0;
+}
+static inline int security_set_fssid(u32 sid)
+{
+	return 0;
+}
+static inline void security_clear_fssid(void){}
+#endif /* CONFIG_SECURITY_SELINUX */
+
+#endif
Index: nfs-2.6/include/linux/sunrpc/svc.h
diff -u nfs-2.6/include/linux/sunrpc/svc.h:1.1.1.3 nfs-2.6/include/linux/sunrpc/svc.h:1.6
--- nfs-2.6/include/linux/sunrpc/svc.h:1.1.1.3	Fri Apr  2 14:17:23 2004
+++ nfs-2.6/include/linux/sunrpc/svc.h	Mon Apr  5 09:35:40 2004
@@ -128,7 +128,8 @@
 	u32			rq_proc;	/* procedure number */
 	u32			rq_prot;	/* IP protocol */
 	unsigned short
-				rq_secure  : 1;	/* secure port */
+				rq_secure  : 1,	/* secure port */
+				rq_selinux : 1; /* Use SELinux */
 
 
 	__u32			rq_daddr;	/* dest addr of request - reply from here */
@@ -151,6 +152,7 @@
 						 * determine what device number
 						 * to report (real or virtual)
 						 */
+	u32			rq_sid;		/* SELinux SID */
 
 	wait_queue_head_t	rq_wait;	/* synchronization */
 };
@@ -235,6 +237,7 @@
  * RPC program
  */
 struct svc_program {
+	struct svc_program *	pg_next;	/* other programs */
 	u32			pg_prog;	/* program number */
 	unsigned int		pg_lovers;	/* lowest version */
 	unsigned int		pg_hivers;	/* lowest version */
Index: nfs-2.6/include/linux/sunrpc/xattr.h
diff -u /dev/null nfs-2.6/include/linux/sunrpc/xattr.h:1.1
--- /dev/null	Wed Apr  7 15:16:10 2004
+++ nfs-2.6/include/linux/sunrpc/xattr.h	Thu Jan 15 16:14:59 2004
@@ -0,0 +1,10 @@
+/*
+  File: include/linux/sunrpc/xattr.h
+ 
+*/
+
+#define XATTR_NFS_PROGRAM 	100007
+#define NFS_XATTR_SIZE_MAX	1024
+
+extern int nfs_setxattr(struct dentry *, const char *, const void *, size_t, int);
+extern ssize_t nfs_getxattr(struct dentry *, const char *, void *, size_t);
Index: nfs-2.6/net/sunrpc/clnt.c
diff -u nfs-2.6/net/sunrpc/clnt.c:1.1.1.4 nfs-2.6/net/sunrpc/clnt.c:1.15
--- nfs-2.6/net/sunrpc/clnt.c:1.1.1.4	Mon Apr  5 11:59:26 2004
+++ nfs-2.6/net/sunrpc/clnt.c	Mon Apr  5 16:32:17 2004
@@ -34,6 +34,7 @@
 #include <linux/sunrpc/rpc_pipe_fs.h>
 
 #include <linux/nfs.h>
+#include <linux/sunrpc/selinux.h>
 
 
 #define RPC_SLACK_SPACE		(1024)	/* total overkill */
@@ -937,6 +938,9 @@
 	struct rpc_xprt *xprt = clnt->cl_xprt;
 	struct rpc_rqst	*req = task->tk_rqstp;
 	u32		*p = req->rq_svec[0].iov_base;
+	u32 sid, len, quadlen; /* for SELinux */
+	char *context; /* for SELinux */
+	int rc; /* for SELinux */
 
 	/* FIXME: check buffer size? */
 	if (xprt->stream)
@@ -947,7 +951,20 @@
 	*p++ = htonl(clnt->cl_prog);	/* program number */
 	*p++ = htonl(clnt->cl_vers);	/* program version */
 	*p++ = htonl(task->tk_msg.rpc_proc->p_proc);	/* procedure */
-	return rpcauth_marshcred(task, p);
+	p = rpcauth_marshcred(task, p);
+#ifdef CONFIG_NFS_SELINUX
+	if (clnt->cl_selinux) {
+		sid = security_get_sid();
+		context = (char *)(p+1);
+		rc = security_sid_to_network_context(sid, context, &len);
+		if (rc)
+			rpc_exit(task, rc);
+		quadlen = XDR_QUADLEN(len);
+		*p++ = htonl(len);
+		p += quadlen;
+	}
+#endif /* CONFIG_NFS_SELINUX */
+	return p;
 }
 
 /*
Index: nfs-2.6/net/sunrpc/rpc_pipe.c
diff -u nfs-2.6/net/sunrpc/rpc_pipe.c:1.1.1.3 nfs-2.6/net/sunrpc/rpc_pipe.c:1.9
--- nfs-2.6/net/sunrpc/rpc_pipe.c:1.1.1.3	Thu Feb 19 14:10:23 2004
+++ nfs-2.6/net/sunrpc/rpc_pipe.c	Wed Mar  3 15:54:14 2004
@@ -358,6 +358,12 @@
 	RPCAUTH_mount,
 	RPCAUTH_nfs,
 	RPCAUTH_portmap,
+#ifdef CONFIG_NFS_SELINUX
+	RPCAUTH_selinuxnfs,
+#endif
+#ifdef CONFIG_NFS_XATTR
+	RPCAUTH_xattrnfs,
+#endif
 	RPCAUTH_statd,
 	RPCAUTH_RootEOF
 };
@@ -388,6 +394,18 @@
 		.name = "portmap",
 		.mode = S_IFDIR | S_IRUGO | S_IXUGO,
 	},
+#ifdef CONFIG_NFS_SELINUX
+	[RPCAUTH_selinuxnfs] = {
+		.name = "selinuxnfs",
+		.mode = S_IFDIR | S_IRUGO | S_IXUGO,
+	},
+#endif
+#ifdef CONFIG_NFS_XATTR
+	[RPCAUTH_xattrnfs] = {
+		.name = "xattrnfs",
+		.mode = S_IFDIR | S_IRUGO | S_IXUGO,
+	},
+#endif
 	[RPCAUTH_statd] = {
 		.name = "statd",
 		.mode = S_IFDIR | S_IRUGO | S_IXUGO,
Index: nfs-2.6/net/sunrpc/svc.c
diff -u nfs-2.6/net/sunrpc/svc.c:1.1.1.2 nfs-2.6/net/sunrpc/svc.c:1.18
--- nfs-2.6/net/sunrpc/svc.c:1.1.1.2	Fri Apr  2 14:18:39 2004
+++ nfs-2.6/net/sunrpc/svc.c	Mon Apr  5 09:35:41 2004
@@ -18,6 +18,7 @@
 #include <linux/sunrpc/stats.h>
 #include <linux/sunrpc/svcsock.h>
 #include <linux/sunrpc/clnt.h>
+#include <linux/sunrpc/selinux.h>
 
 #define RPCDBG_FACILITY	RPCDBG_SVCDSP
 #define RPC_PARANOIA 1
@@ -29,6 +30,7 @@
 svc_create(struct svc_program *prog, unsigned int bufsize)
 {
 	struct svc_serv	*serv;
+	struct svc_program *progp;
 	int vers;
 	unsigned int xdrsize;
 
@@ -41,14 +43,16 @@
 	serv->sv_bufsz	   = bufsize? bufsize : 4096;
 	prog->pg_lovers = prog->pg_nvers-1;
 	xdrsize = 0;
-	for (vers=0; vers<prog->pg_nvers ; vers++)
-		if (prog->pg_vers[vers]) {
-			prog->pg_hivers = vers;
-			if (prog->pg_lovers > vers)
-				prog->pg_lovers = vers;
-			if (prog->pg_vers[vers]->vs_xdrsize > xdrsize)
-				xdrsize = prog->pg_vers[vers]->vs_xdrsize;
-		}
+	for (progp = prog; progp; progp = progp->pg_next) {
+		for (vers=0; vers<progp->pg_nvers ; vers++)
+			if (progp->pg_vers[vers]) {
+				progp->pg_hivers = vers;
+				if (progp->pg_lovers > vers)
+					prog->pg_lovers = vers;
+				if (progp->pg_vers[vers]->vs_xdrsize > xdrsize)
+					xdrsize = progp->pg_vers[vers]->vs_xdrsize;
+			}
+	}
 	serv->sv_xdrsize   = xdrsize;
 	INIT_LIST_HEAD(&serv->sv_threads);
 	INIT_LIST_HEAD(&serv->sv_sockets);
@@ -158,13 +162,14 @@
 	struct svc_rqst	*rqstp;
 	int		error = -ENOMEM;
 
+	struct svc_program *prog = serv->sv_program;
+
 	rqstp = kmalloc(sizeof(*rqstp), GFP_KERNEL);
 	if (!rqstp)
 		goto out;
 
 	memset(rqstp, 0, sizeof(*rqstp));
 	init_waitqueue_head(&rqstp->rq_wait);
-
 	if (!(rqstp->rq_argp = (u32 *) kmalloc(serv->sv_xdrsize, GFP_KERNEL))
 	 || !(rqstp->rq_resp = (u32 *) kmalloc(serv->sv_xdrsize, GFP_KERNEL))
 	 || !svc_init_buffer(rqstp, serv->sv_bufsz))
@@ -227,18 +232,19 @@
 	if (!port)
 		clear_thread_flag(TIF_SIGPENDING);
 
-	for (i = 0; i < progp->pg_nvers; i++) {
-		if (progp->pg_vers[i] == NULL)
-			continue;
-		error = rpc_register(progp->pg_prog, i, proto, port, &dummy);
-		if (error < 0)
-			break;
-		if (port && !dummy) {
-			error = -EACCES;
-			break;
+	for (progp = serv->sv_program; progp; progp = progp->pg_next) {		
+		for (i = 0; i < progp->pg_nvers; i++) {
+			if (progp->pg_vers[i] == NULL)
+				continue;
+			error = rpc_register(progp->pg_prog, i, proto, port, &dummy);
+			if (error < 0)
+				break;
+			if (port && !dummy) {
+				error = -EACCES;
+				break;
+			}
 		}
 	}
-
 	if (!port) {
 		spin_lock_irqsave(&current->sighand->siglock, flags);
 		recalc_sigpending();
@@ -263,6 +269,8 @@
 	u32			*statp;
 	u32			dir, prog, vers, proc,
 				auth_stat, rpc_stat;
+	u32			len, quadlen; /* for SELinux */
+	char *			context; /* for SELinux */
 
 	rpc_stat = rpc_success;
 
@@ -325,9 +333,27 @@
 	case SVC_COMPLETE:
 		goto sendit;
 	}
-		
-	progp = serv->sv_program;
-	if (prog != progp->pg_prog)
+
+#ifdef CONFIG_NFSD_SELINUX
+	if (prog == SELINUX_NFS_PROGRAM){
+		rqstp->rq_selinux = 1;
+		len = ntohl(svc_getu32(argv));
+		context = argv->iov_base;
+		quadlen = XDR_QUADLEN(len);
+		if (quadlen > argv->iov_len)
+			goto err_garbage;
+		argv->iov_base = (void*)((u32*)argv->iov_base + quadlen);
+		argv->iov_len -= quadlen;
+		if (security_network_context_to_sid(context, len, &rqstp->rq_sid))
+			goto err_garbage;
+	} else {
+		rqstp->rq_selinux = 0;
+	}
+#endif /* CONFIG_NFSD_SELINUX */
+	for (progp = serv->sv_program; progp; progp = progp->pg_next)
+		if (prog == progp->pg_prog)
+			break;
+	if (progp == NULL)
 		goto err_bad_prog;
 
 	if (vers >= progp->pg_nvers ||
@@ -360,6 +386,10 @@
 	if (procp->pc_xdrressize)
 		svc_reserve(rqstp, procp->pc_xdrressize<<2);
 
+#ifdef CONFIG_NFSD_SELINUX
+	if (rqstp->rq_selinux)
+		security_set_fssid(rqstp->rq_sid);
+#endif
 	/* Call the function that processes the request. */
 	if (!versp->vs_dispatch) {
 		/* Decode arguments */
@@ -385,7 +415,10 @@
 			goto dropit;
 		}
 	}
-
+#ifdef CONFIG_NFSD_SELINUX
+	if (rqstp->rq_selinux)
+		security_clear_fssid();
+#endif
 	/* Check RPC status result */
 	if (*statp != rpc_success)
 		resv->iov_len = ((void*)statp)  - resv->iov_base + 4;
@@ -403,6 +436,10 @@
 	return svc_send(rqstp);
 
  dropit:
+#ifdef CONFIG_NFSD_SELINUX
+	if (rqstp->rq_selinux)
+		security_clear_fssid();
+#endif
 	svc_authorise(rqstp);	/* doesn't hurt to call this twice */
 	dprintk("svc: svc_process dropit\n");
 	svc_drop(rqstp);
@@ -440,9 +477,9 @@
 
 err_bad_prog:
 #ifdef RPC_PARANOIA
-	if (prog != 100227 || progp->pg_prog != 100003)
-		printk("svc: unknown program %d (me %d)\n", prog, progp->pg_prog);
-	/* else it is just a Solaris client seeing if ACLs are supported */
+	if (prog != 100227 || serv->sv_program->pg_prog != 100003)
+		printk("svc: unknown program %d\n", prog);
+	/* else it is just a client seeing if ACLs are supported */
 #endif
 	serv->sv_stats->rpcbadfmt++;
 	svc_putu32(resv, rpc_prog_unavail);
@@ -469,6 +506,10 @@
 err_garbage:
 #ifdef RPC_PARANOIA
 	printk("svc: failed to decode args\n");
+#endif
+#ifdef CONFIG_NFSD_SELINUX
+	if (rqstp->rq_selinux)
+		security_clear_fssid();
 #endif
 	rpc_stat = rpc_garbage_args;
 err_bad:
Index: nfs-2.6/security/selinux/Makefile
diff -u nfs-2.6/security/selinux/Makefile:1.1.1.2 nfs-2.6/security/selinux/Makefile:1.4
--- nfs-2.6/security/selinux/Makefile:1.1.1.2	Thu Feb 19 14:10:33 2004
+++ nfs-2.6/security/selinux/Makefile	Thu Mar  4 14:36:53 2004
@@ -6,6 +6,8 @@
 
 selinux-y := avc.o hooks.o selinuxfs.o netlink.o
 
+selinux-$(CONFIG_NFS_SELINUX) += selinux_nfs.o
+
 selinux-$(CONFIG_SECURITY_NETWORK) += netif.o
 
 EXTRA_CFLAGS += -Isecurity/selinux/include
Index: nfs-2.6/security/selinux/hooks.c
diff -u nfs-2.6/security/selinux/hooks.c:1.1.1.6 nfs-2.6/security/selinux/hooks.c:1.15
--- nfs-2.6/security/selinux/hooks.c:1.1.1.6	Mon Apr  5 11:59:34 2004
+++ nfs-2.6/security/selinux/hooks.c	Mon Apr  5 16:32:25 2004
@@ -117,7 +117,7 @@
 	memset(tsec, 0, sizeof(struct task_security_struct));
 	tsec->magic = SELINUX_MAGIC;
 	tsec->task = task;
-	tsec->osid = tsec->sid = SECINITSID_UNLABELED;
+	tsec->fssid = tsec->osid = tsec->sid = SECINITSID_UNLABELED;
 	task->security = tsec;
 
 	return 0;
@@ -151,7 +151,8 @@
 	isec->sid = SECINITSID_UNLABELED;
 	isec->sclass = SECCLASS_FILE;
 	if (tsec && tsec->magic == SELINUX_MAGIC)
-		isec->task_sid = tsec->sid;
+		isec->task_sid = (tsec->fssid != SECINITSID_UNLABELED) ?
+			tsec->fssid : tsec->sid;
 	else
 		isec->task_sid = SECINITSID_UNLABELED;
 	inode->i_security = isec;
@@ -189,8 +190,13 @@
 	fsec->magic = SELINUX_MAGIC;
 	fsec->file = file;
 	if (tsec && tsec->magic == SELINUX_MAGIC) {
-		fsec->sid = tsec->sid;
-		fsec->fown_sid = tsec->sid;
+		if (tsec->fssid != SECINITSID_UNLABELED) {
+			fsec->sid = tsec->fssid;
+			fsec->fown_sid = tsec->fssid;
+		} else {
+			fsec->sid = tsec->sid;
+			fsec->fown_sid = tsec->sid;
+		}
 	} else {
 		fsec->sid = SECINITSID_UNLABELED;
 		fsec->fown_sid = SECINITSID_UNLABELED;
@@ -335,7 +341,7 @@
 	if (sb->s_type->fs_flags & FS_BINARY_MOUNTDATA) {
 
 		/* NFS we understand. */
-		if (!strcmp(name, "nfs")) {
+		if (!strcmp(name, "nfs") || !strcmp(name, "selinuxnfs")) {
 			struct nfs_mount_data *d = data;
 
 			if (d->version <  NFS_MOUNT_VERSION)
@@ -947,6 +953,7 @@
 	struct task_security_struct *tsec;
 	struct inode_security_struct *isec;
 	struct avc_audit_data ad;
+	u32 tsid;
 
 	tsec = tsk->security;
 	isec = inode->i_security;
@@ -957,7 +964,8 @@
 		ad.u.fs.inode = inode;
 	}
 
-	return avc_has_perm(tsec->sid, isec->sid, isec->sclass,
+	tsid = (tsec->fssid != SECINITSID_UNLABELED) ? tsec->fssid : tsec->sid;
+	return avc_has_perm(tsid, isec->sid, isec->sclass,
 			    perms, aeref ? aeref : &isec->avcr, adp);
 }
 
@@ -996,13 +1004,15 @@
 	struct inode *inode = dentry->d_inode;
 	struct avc_audit_data ad;
 	int rc;
+	u32 tsid;
 
 	AVC_AUDIT_DATA_INIT(&ad, FS);
 	ad.u.fs.mnt = mnt;
 	ad.u.fs.dentry = dentry;
 
-	if (tsec->sid != fsec->sid) {
-		rc = avc_has_perm(tsec->sid, fsec->sid,
+	tsid = (tsec->fssid != SECINITSID_UNLABELED) ? tsec->fssid : tsec->sid;
+	if (tsid != fsec->sid) {
+		rc = avc_has_perm(tsid, fsec->sid,
 				  SECCLASS_FD,
 				  FD__USE,
 				  &fsec->avcr, &ad);
@@ -1028,6 +1038,7 @@
 	u32 newsid;
 	struct avc_audit_data ad;
 	int rc;
+	u32 tsid;
 
 	tsec = current->security;
 	dsec = dir->i_security;
@@ -1036,7 +1047,8 @@
 	AVC_AUDIT_DATA_INIT(&ad, FS);
 	ad.u.fs.dentry = dentry;
 
-	rc = avc_has_perm(tsec->sid, dsec->sid, SECCLASS_DIR,
+	tsid = (tsec->fssid != SECINITSID_UNLABELED) ? tsec->fssid : tsec->sid;
+	rc = avc_has_perm(tsid, dsec->sid, SECCLASS_DIR,
 			  DIR__ADD_NAME | DIR__SEARCH,
 			  &dsec->avcr, &ad);
 	if (rc)
@@ -1045,13 +1057,13 @@
 	if (tsec->create_sid && sbsec->behavior != SECURITY_FS_USE_MNTPOINT) {
 		newsid = tsec->create_sid;
 	} else {
-		rc = security_transition_sid(tsec->sid, dsec->sid, tclass,
+		rc = security_transition_sid(tsid, dsec->sid, tclass,
 					     &newsid);
 		if (rc)
 			return rc;
 	}
 
-	rc = avc_has_perm(tsec->sid, newsid, tclass, FILE__CREATE, NULL, &ad);
+	rc = avc_has_perm(tsid, newsid, tclass, FILE__CREATE, NULL, &ad);
 	if (rc)
 		return rc;
 
@@ -1073,7 +1085,7 @@
 	struct task_security_struct *tsec;
 	struct inode_security_struct *dsec, *isec;
 	struct avc_audit_data ad;
-	u32 av;
+	u32 av, tsid;
 	int rc;
 
 	tsec = current->security;
@@ -1083,9 +1095,10 @@
 	AVC_AUDIT_DATA_INIT(&ad, FS);
 	ad.u.fs.dentry = dentry;
 
+	tsid = (tsec->fssid != SECINITSID_UNLABELED) ? tsec->fssid : tsec->sid;
 	av = DIR__SEARCH;
 	av |= (kind ? DIR__REMOVE_NAME : DIR__ADD_NAME);
-	rc = avc_has_perm(tsec->sid, dsec->sid, SECCLASS_DIR,
+	rc = avc_has_perm(tsid, dsec->sid, SECCLASS_DIR,
 			  av, &dsec->avcr, &ad);
 	if (rc)
 		return rc;
@@ -1105,7 +1118,7 @@
 		return 0;
 	}
 
-	rc = avc_has_perm(tsec->sid, isec->sid, isec->sclass,
+	rc = avc_has_perm(tsid, isec->sid, isec->sclass,
 			  av, &isec->avcr, &ad);
 	return rc;
 }
@@ -1118,7 +1131,7 @@
 	struct task_security_struct *tsec;
 	struct inode_security_struct *old_dsec, *new_dsec, *old_isec, *new_isec;
 	struct avc_audit_data ad;
-	u32 av;
+	u32 av, tsid;
 	int old_is_dir, new_is_dir;
 	int rc;
 
@@ -1130,20 +1143,21 @@
 
 	AVC_AUDIT_DATA_INIT(&ad, FS);
 
+	tsid = (tsec->fssid != SECINITSID_UNLABELED) ? tsec->fssid : tsec->sid;
 	ad.u.fs.dentry = old_dentry;
-	rc = avc_has_perm(tsec->sid, old_dsec->sid, SECCLASS_DIR,
+	rc = avc_has_perm(tsid, old_dsec->sid, SECCLASS_DIR,
 			  DIR__REMOVE_NAME | DIR__SEARCH,
 			  &old_dsec->avcr, &ad);
 	if (rc)
 		return rc;
-	rc = avc_has_perm(tsec->sid, old_isec->sid,
+	rc = avc_has_perm(tsid, old_isec->sid,
 			  old_isec->sclass,
 			  FILE__RENAME,
 			  &old_isec->avcr, &ad);
 	if (rc)
 		return rc;
 	if (old_is_dir && new_dir != old_dir) {
-		rc = avc_has_perm(tsec->sid, old_isec->sid,
+		rc = avc_has_perm(tsid, old_isec->sid,
 				  old_isec->sclass,
 				  DIR__REPARENT,
 				  &old_isec->avcr, &ad);
@@ -1155,14 +1169,14 @@
 	av = DIR__ADD_NAME | DIR__SEARCH;
 	if (new_dentry->d_inode)
 		av |= DIR__REMOVE_NAME;
-	rc = avc_has_perm(tsec->sid, new_dsec->sid, SECCLASS_DIR,
+	rc = avc_has_perm(tsid, new_dsec->sid, SECCLASS_DIR,
 			  av,&new_dsec->avcr, &ad);
 	if (rc)
 		return rc;
 	if (new_dentry->d_inode) {
 		new_isec = new_dentry->d_inode->i_security;
 		new_is_dir = S_ISDIR(new_dentry->d_inode->i_mode);
-		rc = avc_has_perm(tsec->sid, new_isec->sid,
+		rc = avc_has_perm(tsid, new_isec->sid,
 				  new_isec->sclass,
 				  (new_is_dir ? DIR__RMDIR : FILE__UNLINK),
 				  &new_isec->avcr, &ad);
@@ -1181,10 +1195,12 @@
 {
 	struct task_security_struct *tsec;
 	struct superblock_security_struct *sbsec;
+	u32 tsid;
 
 	tsec = tsk->security;
 	sbsec = sb->s_security;
-	return avc_has_perm(tsec->sid, sbsec->sid, SECCLASS_FILESYSTEM,
+	tsid = (tsec->fssid != SECINITSID_UNLABELED) ? tsec->fssid : tsec->sid;
+	return avc_has_perm(tsid, sbsec->sid, SECCLASS_FILESYSTEM,
 			    perms, NULL, ad);
 }
 
@@ -1255,7 +1271,7 @@
 	struct inode *inode;
 	struct inode_security_struct *dsec;
 	struct superblock_security_struct *sbsec;
-	u32 newsid;
+	u32 newsid, tsid;
 	char *context;
 	unsigned int len;
 	int rc;
@@ -1277,7 +1293,8 @@
 	if (tsec->create_sid && sbsec->behavior != SECURITY_FS_USE_MNTPOINT) {
 		newsid = tsec->create_sid;
 	} else {
-		rc = security_transition_sid(tsec->sid, dsec->sid,
+		tsid = (tsec->fssid != SECINITSID_UNLABELED) ? tsec->fssid : tsec->sid;
+		rc = security_transition_sid(tsid, dsec->sid,
 					     inode_mode_to_security_class(inode->i_mode),
 					     &newsid);
 		if (rc) {
@@ -2109,7 +2126,7 @@
 	struct inode_security_struct *isec = inode->i_security;
 	struct superblock_security_struct *sbsec;
 	struct avc_audit_data ad;
-	u32 newsid;
+	u32 newsid, tsid;
 	int rc = 0;
 
 	if (strcmp(name, XATTR_NAME_SELINUX)) {
@@ -2133,7 +2150,8 @@
 	AVC_AUDIT_DATA_INIT(&ad,FS);
 	ad.u.fs.dentry = dentry;
 
-	rc = avc_has_perm(tsec->sid, isec->sid, isec->sclass,
+	tsid = (tsec->fssid != SECINITSID_UNLABELED) ? tsec->fssid : tsec->sid;
+	rc = avc_has_perm(tsid, isec->sid, isec->sclass,
 			  FILE__RELABELFROM,
 			  &isec->avcr, &ad);
 	if (rc)
@@ -2143,7 +2161,7 @@
 	if (rc)
 		return rc;
 
-	rc = avc_has_perm(tsec->sid, newsid, isec->sclass,
+	rc = avc_has_perm(tsid, newsid, isec->sclass,
 			  FILE__RELABELTO, NULL, &ad);
 	if (rc)
 		return rc;
Index: nfs-2.6/security/selinux/selinux_nfs.c
diff -u /dev/null nfs-2.6/security/selinux/selinux_nfs.c:1.9
--- /dev/null	Wed Apr  7 15:16:29 2004
+++ nfs-2.6/security/selinux/selinux_nfs.c	Thu Mar  4 16:54:27 2004
@@ -0,0 +1,52 @@
+/*
+ *  NSA Security-Enhanced Linux (SELinux) NFS Related Funtions
+ *
+ *  This file contains the SELinux Related Functions for NFS.
+ *
+ */
+#include <linux/types.h>
+#include <linux/slab.h>
+#include <flask.h>
+#include <security.h>
+#include <objsec.h>
+
+int security_sid_to_network_context(u32 sid, char *context, u32 *len)
+{
+	int rc;
+	char *s = NULL;
+	*len = 0;
+	
+	rc = security_sid_to_context(sid, &s, len);
+	if (rc) 
+		return rc;
+	memcpy(context, s, *len);
+	kfree(s);
+	return rc;
+}
+
+int security_network_context_to_sid(char *context, u32 len, u32 *sid)
+{
+	return security_context_to_sid(context, len, sid);
+}
+
+__u32 security_get_sid(void)
+{
+	struct task_security_struct *tsec = current->security;
+
+	return tsec->sid;
+}
+	
+void security_set_fssid(u32 sid)
+{
+	struct task_security_struct *tsec = current->security;
+	
+	tsec->fssid = sid;
+}
+
+void security_clear_fssid(void)
+{
+	struct task_security_struct *tsec = current->security;
+	
+	tsec->fssid = SECINITSID_UNLABELED;
+}
+
Index: nfs-2.6/security/selinux/include/objsec.h
diff -u nfs-2.6/security/selinux/include/objsec.h:1.1.1.3 nfs-2.6/security/selinux/include/objsec.h:1.5
--- nfs-2.6/security/selinux/include/objsec.h:1.1.1.3	Thu Feb 19 14:10:35 2004
+++ nfs-2.6/security/selinux/include/objsec.h	Fri Feb 20 08:33:01 2004
@@ -33,6 +33,7 @@
 	u32 sid;             /* current SID */
 	u32 exec_sid;        /* exec SID */
 	u32 create_sid;      /* fscreate SID */
+	u32 fssid;           /* nfs SID */
         struct avc_entry_ref avcr;     /* reference to process permissions */
 };
 
