diff -up tightvnc-1.5.0-20081015svn3022/unix/vncviewer/vncviewer.man.manminor tightvnc-1.5.0-20081015svn3022/unix/vncviewer/vncviewer.man
--- tightvnc-1.5.0-20081015svn3022/unix/vncviewer/vncviewer.man.manminor	2008-10-23 12:58:02.000000000 +0200
+++ tightvnc-1.5.0-20081015svn3022/unix/vncviewer/vncviewer.man	2008-10-23 12:58:23.000000000 +0200
@@ -8,6 +8,10 @@ vncviewer \- VNC viewer for X
 .br
 .B vncviewer
 .RI [ options ] 
+.RI [ host ][:: port ]
+.br
+.B vncviewer
+.RI [ options ] 
 .B \-listen
 .RI [ port ]
 .SH DESCRIPTION
