diff -up tigervnc-0.0.90-20090407svn3761/unix/vncserver.rh494801 tigervnc-0.0.90-20090407svn3761/unix/vncserver
--- tigervnc-0.0.90-20090407svn3761/unix/vncserver.rh494801	2009-04-08 15:36:58.463812689 +0200
+++ tigervnc-0.0.90-20090407svn3761/unix/vncserver	2009-04-08 15:37:13.552448279 +0200
@@ -203,7 +203,6 @@ $cmd .= " -pixelformat $pixelformat" if 
 $cmd .= " -rfbwait 30000";
 $cmd .= " -rfbauth $vncUserDir/passwd";
 $cmd .= " -rfbport $vncPort";
-$cmd .= " -fp $fontPath" if ($fontPath);
 $cmd .= " -pn";
 
 # Add color database stuff here, e.g.:
