--- mozilla/configure.in.orig	2007-07-04 17:42:33.000000000 +0000
+++ mozilla/configure.in	2007-07-04 17:43:08.000000000 +0000
@@ -7563,6 +7563,7 @@
 AC_CONFIG_HEADER(
 gfx/gfx-config.h
 netwerk/necko-config.h
+intl/uconv/native/config.h
 xpcom/xpcom-config.h
 xpcom/xpcom-private.h
 )
diff -Nur mozilla/intl/uconv/native/config.h.in mozilla/intl/uconv/native/config.h.in
--- mozilla/intl/uconv/native/config.h.in 1970-01-01 01:00:00 +0100
+++ mozilla/intl/uconv/native/config.h.in 2007-07-03 14:53:53 +0100
@@ -0,0 +1,2 @@
+/* Define if iconv() supports const input */
+#undef HAVE_ICONV_WITH_CONST_INPUT
diff -Nur mozilla/intl/uconv/native/nsNativeUConvService.cpp mozilla/intl/uconv/native/nsNativeUConvService.cpp
--- mozilla/intl/uconv/native/nsNativeUConvService.cpp    2007-07-03 15:10:08 +0100
+++ mozilla/intl/uconv/native/nsNativeUConvService.cpp    2007-07-03 15:05:07 +0100
@@ -37,6 +37,7 @@


 #ifdef MOZ_USE_NATIVE_UCONV
+#include "config.h"
 #include "nsString.h"
 #include "nsIGenericFactory.h"

@@ -227,7 +228,11 @@

     do {
         if ( iconv(mConverter,
+#ifdef HAVE_ICONV_WITH_CONST_INPUT
+                   (const char **)&aSrc,
+#else
                    (char **)&aSrc,
+#endif
                    &inLeft,
                    (char **)&aDest,
                    &outLeft) == (size_t) -1 ) {
@@ -374,7 +379,11 @@

     do {
         if ( iconv(mConverter,
+#ifdef HAVE_ICONV_WITH_CONST_INPUT
+                   (const char **)&aSrc,
+#else
                    (char **)&aSrc,
+#endif
                    &inLeft,
                    (char **)&aDest,
                    &outLeft) == (size_t) -1 ) {