Index: vlc-0.8.5/Makefile.am
===================================================================
--- vlc-0.8.5.orig/Makefile.am
+++ vlc-0.8.5/Makefile.am
@@ -313,7 +313,7 @@ vlc-bundle: vlc
 endif
 
 # Install the symlinks
-install-exec-local:
+install-exec-local: install-binPROGRAMS
 	for i in "" $(ALIASES) ; do if test -n "$$i" ; then \
 	  rm -f "$(DESTDIR)$(bindir)/$$i" && \
 	  ln -sf vlc "$(DESTDIR)$(bindir)/$$i" ; \
