Index: vlc-0.8.5/mozilla/vlcshell.cpp
===================================================================
--- vlc-0.8.5.orig/mozilla/vlcshell.cpp
+++ vlc-0.8.5/mozilla/vlcshell.cpp
@@ -47,10 +47,8 @@
 #include <npapi.h>
 #include <npruntime.h>
 
-/* This is from mozilla java, do we really need it? */
-#if 0
+/* This is from mozilla java, needed for jref type */
 #include <jri.h>
-#endif
 
 #include "vlcplugin.h"
 #include "vlcruntime.h"
Index: vlc-0.8.5/mozilla/Makefile.am
===================================================================
--- vlc-0.8.5.orig/mozilla/Makefile.am
+++ vlc-0.8.5/mozilla/Makefile.am
@@ -158,7 +158,7 @@ libvlcplugin_la_SOURCES = $(SOURCES_mozi
 libvlcplugin_la_CFLAGS = `$(VLC_CONFIG) --cflags mozilla`
 libvlcplugin_la_CXXFLAGS = `$(VLC_CONFIG) --cxxflags mozilla`
 libvlcplugin_la_LDFLAGS = `$(VLC_CONFIG) --libs mozilla` -module -avoid-version -shrext $(LIBEXT)
-libvlcplugin_la_LIBADD = ../src/libvlc.la
+libvlcplugin_la_LIBADD = ../src/libvlc.la -lXt
 
 libnpvlc_a_SOURCES = $(SOURCES_mozilla_common) $(SOURCES_support)
 libnpvlc_a_CFLAGS = `$(VLC_CONFIG) --cflags mozilla $(pic)` \
