Index: luatex-beta-0.46.0/source/texk/web2c/luatexdir/image/pdftoepdf.cc
===================================================================
--- luatex-beta-0.46.0.orig/source/texk/web2c/luatexdir/image/pdftoepdf.cc
+++ luatex-beta-0.46.0/source/texk/web2c/luatexdir/image/pdftoepdf.cc
@@ -672,7 +672,7 @@ static void writeEncodings(PDF pdf)
     }
     for (r = encodingList; r != NULL; r = n) {
         n = r->next;
-        delete r->font;
+	r->font->decRefCnt();
         delete r;
     }
 }
