Index: xine-ui-0.99.4/configure.ac
===================================================================
--- xine-ui-0.99.4.orig/configure.ac
+++ xine-ui-0.99.4/configure.ac
@@ -571,7 +571,7 @@ dnl
 AC_CHECK_LIB(posix4, nanosleep)
 AC_CHECK_FUNCS(vsnprintf snprintf strpbrk strsep nanosleep)
 AC_CHECK_HEADER(stdarg.h, AC_DEFINE(HAVE_STDARGS,,[Define this is you have stdarg.h header file installed]))
-AC_CHECK_HEADERS(string.h strings.h alloca.h linux/kd.h)
+AC_CHECK_HEADERS(string.h strings.h alloca.h linux/kd.hb execinfo.h)
 
 
 dnl
Index: xine-ui-0.99.4/src/xitk/xine-toolkit/xitk.c
===================================================================
--- xine-ui-0.99.4.orig/src/xitk/xine-toolkit/xitk.c
+++ xine-ui-0.99.4/src/xitk/xine-toolkit/xitk.c
@@ -46,7 +46,7 @@
 
 #include <locale.h>
 
-#ifdef __linux__
+#ifdef HAVE_EXECINFO_H
 #include <execinfo.h>
 #endif
 
@@ -322,8 +322,7 @@ static void xitk_signal_handler(int sig)
       fprintf(stderr, "xiTK received SIGSEGV signal, RIP.\n");
 #ifndef DEBUG
       abort();
-#else
-#ifdef __linux__
+#elif defined(HAVE_EXECINFO_H)
       void    *backtrace_array[255];
       char   **backtrace_strings;
       int      entries, i;
@@ -341,7 +340,6 @@ static void xitk_signal_handler(int sig)
 	}
       }
 #endif
-#endif
     }
     break;
 
