Index: xine-ui-0.99.4/src/xitk/help.c
===================================================================
--- xine-ui-0.99.4.orig/src/xitk/help.c
+++ xine-ui-0.99.4/src/xitk/help.c
@@ -392,7 +392,7 @@ void help_panel(void) {
   xitk_list_append_content ((XITK_WIDGET_LIST_LIST(help->widget_list)),
     (help->tabs = 
      xitk_noskin_tabs_create(help->widget_list, 
-			     &tab, 20, 24, WINDOW_WIDTH - 40, tabsfontname)));
+			     &tab, 20, 24, WINDOW_WIDTH - 40, 28, tabsfontname)));
   
   xitk_enable_and_show_widget(help->tabs);
 
Index: xine-ui-0.99.4/src/xitk/setup.c
===================================================================
--- xine-ui-0.99.4.orig/src/xitk/setup.c
+++ xine-ui-0.99.4/src/xitk/setup.c
@@ -56,7 +56,7 @@ static char                     *tabsfon
     int                  lbearing, rbearing, width, ascent, descent;                            \
     xitk_font_t         *fs;                                                                    \
                                                                                                 \
-    image = xitk_image_create_image(gGui->imlib_data, FRAME_WIDTH + 1, FRAME_HEIGHT + 1);       \
+    image = xitk_image_create_image(gGui->imlib_data, FRAME_WIDTH + 1, FRAME_HEIGHT + 2);       \
                                                                                                 \
     fs = xitk_font_load_font(gGui->display, boldfontname);                                      \
     xitk_font_set_font(fs, (XITK_WIDGET_LIST_GC(setup->widget_list)));                          \
@@ -1010,7 +1010,7 @@ static void setup_sections (void) {
   tab.userdata          = NULL;
   xitk_list_append_content ((XITK_WIDGET_LIST_LIST(setup->widget_list)),
     (setup->tabs = 
-     xitk_noskin_tabs_create(setup->widget_list, &tab, 20, 24, WINDOW_WIDTH - 40, tabsfontname)));
+     xitk_noskin_tabs_create(setup->widget_list, &tab, 20, 24, WINDOW_WIDTH - 40, 28, tabsfontname)));
   
   xitk_enable_and_show_widget(setup->tabs);
 
@@ -1051,7 +1051,6 @@ static void setup_nextprev_wg(xitk_widge
 
   if(rpos != setup->first_displayed) {
     setup->first_displayed = rpos;
-    setup_clear_tab();
     setup_paint_widgets();
   }
 }
Index: xine-ui-0.99.4/src/xitk/viewlog.c
===================================================================
--- xine-ui-0.99.4.orig/src/xitk/viewlog.c
+++ xine-ui-0.99.4/src/xitk/viewlog.c
@@ -319,7 +319,7 @@ static void viewlog_create_tabs(void) {
   tab.userdata          = NULL;
   xitk_list_append_content ((XITK_WIDGET_LIST_LIST(viewlog->widget_list)),
     (viewlog->tabs = 
-     xitk_noskin_tabs_create(viewlog->widget_list, &tab, 20, 24, WINDOW_WIDTH - 40, tabsfontname)));
+     xitk_noskin_tabs_create(viewlog->widget_list, &tab, 20, 24, WINDOW_WIDTH - 40, 28, tabsfontname)));
   xitk_enable_and_show_widget(viewlog->tabs);
 
   bg = xitk_image_create_xitk_pixmap(gGui->imlib_data, WINDOW_WIDTH, WINDOW_HEIGHT);
Index: xine-ui-0.99.4/src/xitk/xine-toolkit/tabs.c
===================================================================
--- xine-ui-0.99.4.orig/src/xitk/xine-toolkit/tabs.c
+++ xine-ui-0.99.4/src/xitk/xine-toolkit/tabs.c
@@ -323,7 +323,8 @@ char *xitk_tabs_get_current_tab_selected
  */
 xitk_widget_t *xitk_noskin_tabs_create(xitk_widget_list_t *wl,
 				       xitk_tabs_widget_t *t, 
-				       int x, int y, int width, char *fontname) {
+				       int x, int y, int width, int height,
+				       char *fontname) {
   xitk_widget_t         *mywidget;
   tabs_private_data_t   *private_data;
   
@@ -367,7 +368,10 @@ xitk_widget_t *xitk_noskin_tabs_create(x
     XITK_WIDGET_INIT(&lb, t->imlibdata);
     XITK_WIDGET_INIT(&b, t->imlibdata);
 
-    private_data->bheight = fheight * 2;
+    if (height == 0)
+      private_data->bheight = fheight * 2;
+    else
+      private_data->bheight = height;
 
     for(i = 0; i < t->num_entries; i++) {
 
Index: xine-ui-0.99.4/src/xitk/xine-toolkit/test.c
===================================================================
--- xine-ui-0.99.4.orig/src/xitk/xine-toolkit/test.c
+++ xine-ui-0.99.4/src/xitk/xine-toolkit/test.c
@@ -510,7 +510,7 @@ static void create_tabs(void) {
   t.userdata          = NULL;
   xitk_list_append_content (XITK_WIDGET_LIST_LIST(test->widget_list),
 		    (test->tabs = 
-		     xitk_noskin_tabs_create(test->widget_list, &t, x, y, w, fontname)));
+		     xitk_noskin_tabs_create(test->widget_list, &t, x, y, w, 0, fontname)));
   xitk_enable_and_show_widget(test->tabs);
 
 }
Index: xine-ui-0.99.4/src/xitk/xine-toolkit/xitk.h
===================================================================
--- xine-ui-0.99.4.orig/src/xitk/xine-toolkit/xitk.h
+++ xine-ui-0.99.4/src/xitk/xine-toolkit/xitk.h
@@ -2351,7 +2351,8 @@ void xitk_window_dialog_set_modal(xitk_w
 
 xitk_widget_t *xitk_noskin_tabs_create(xitk_widget_list_t *wl,
 				       xitk_tabs_widget_t *t, 
-				       int x, int y, int width, char *fontname);
+				       int x, int y, int width, int height,
+				       char *fontname);
 int xitk_tabs_get_current_selected(xitk_widget_t *w);
 char *xitk_tabs_get_current_tab_selected(xitk_widget_t *w);
 void xitk_tabs_set_current_selected(xitk_widget_t *w, int select);
