diff -ur -x '*~' xine-ui-0.99.4/src/common/utils.c xine-ui-0.99.4-alloca/src/common/utils.c
--- xine-ui-0.99.4/src/common/utils.c	2004-04-15 22:55:36.000000000 +0200
+++ xine-ui-0.99.4-alloca/src/common/utils.c	2005-07-30 11:59:33.778826216 +0200
@@ -34,7 +34,10 @@
 #include <sys/stat.h>
 #include <sys/wait.h>
 #include <sys/utsname.h>
+
+#ifdef HAVE_ALLOCA_H
 #include <alloca.h>
+#endif
 
 #ifdef HAVE_X11
 #include <X11/Xlib.h>
