diff -ur xine-ui-0.99.4/src/xitk/network.c xine-ui-0.99.4-select/src/xitk/network.c
--- xine-ui-0.99.4/src/xitk/network.c	2005-07-16 21:05:32.000000000 +0200
+++ xine-ui-0.99.4-select/src/xitk/network.c	2005-07-30 12:03:42.489016520 +0200
@@ -57,6 +57,7 @@
 #include <time.h>
 #include <sys/ioctl.h>
 #include <sys/types.h>
+#include <sys/select.h>
 #include <sys/stat.h>
 #include <sys/signal.h>
 #include <sys/time.h>
diff -ur xine-ui-0.99.4/src/xitk/session.c xine-ui-0.99.4-select/src/xitk/session.c
--- xine-ui-0.99.4/src/xitk/session.c	2004-09-08 23:13:27.000000000 +0200
+++ xine-ui-0.99.4-select/src/xitk/session.c	2005-07-30 12:04:57.221655432 +0200
@@ -41,6 +41,7 @@
 #include <sys/stat.h>
 #include <sys/socket.h>
 #include <sys/un.h>
+#include <sys/select.h>
 #include <errno.h>
 
 #include "common.h"
