--- pub/diag/Makefile
+++ pub/diag/Makefile
@@ -12,7 +12,7 @@
 prefix = $(PREFIX)/usr
 exec_prefix = $(prefix)
 instdir=$(exec_prefix)/sbin
-MANDIR = $(prefix)/man
+MANDIR = $(prefix)/share/man
 MANPAGES = etherwake.8 mii-diag.8 netdiag.8
 
 # Diagnostics that can use libmii.
@@ -44,9 +44,9 @@
 mii-diag: $(MII_DEPEND)
 
 install:  $(DIAGS) $(UTILS)
-	install -m 755 $^ $(instdir)
-	mkdir -p $(MANDIR)/man8/
-	install -m 444 $(MANPAGES) $(MANDIR)/man8/
+	mkdir -p $(DESTDIR)$(instdir) $(DESTDIR)$(MANDIR)/man8/
+	install -m 755 $^ $(DESTDIR)$(instdir)
+	install -m 444 $(MANPAGES) $(DESTDIR)$(MANDIR)/man8/
 
 netdiag.tgz: netdiag.spec $(MANPAGES) Makefile $(LIBS) $(addsuffix .c, $(DIAGS) $(UTILS))
 	tar cfvz $@ $^
