--- pub/diag/alta-diag.c
+++ pub/diag/alta-diag.c
@@ -84,10 +84,6 @@ static const char long_usage_msg[] =
 #include <strings.h>
 #include <errno.h>
 
-/* The following are required only with unaligned field accesses. */
-#include <asm/types.h>
-#include <asm/unaligned.h>
-
 #if defined(__linux__)  &&  __GNU_LIBRARY__ == 1
 #include <asm/io.h>			/* Newer libraries use <sys/io.h> instead. */
 #else
--- pub/diag/diag-example.c
+++ pub/diag/diag-example.c
@@ -90,10 +90,6 @@ static const char long_usage_msg[] =
 #include <sys/types.h>
 #include <ctype.h>
 
-/* The following are required only with unaligned field accesses. */
-#include <asm/types.h>
-#include <asm/unaligned.h>
-
 #if defined(__linux__)  &&  __GNU_LIBRARY__ == 1
 #include <asm/io.h>			/* Newer libraries use <sys/io.h> instead. */
 #else
--- pub/diag/myson-diag.c
+++ pub/diag/myson-diag.c
@@ -81,10 +81,6 @@ static const char long_usage_msg[] =
 #include <string.h>
 #include <errno.h>
 
-/* The following are required only with unaligned field accesses. */
-#include <asm/types.h>
-#include <asm/unaligned.h>
-
 #if defined(__linux__)  &&  __GNU_LIBRARY__ == 1
 #include <asm/io.h>			/* Newer libraries use <sys/io.h> instead. */
 #else
--- pub/diag/natsemi-diag.c
+++ pub/diag/natsemi-diag.c
@@ -90,10 +90,6 @@ static const char long_usage_msg[] =
 #include <sys/types.h>
 #include <ctype.h>
 
-/* The following are required only with unaligned field accesses. */
-#include <asm/types.h>
-#include <asm/unaligned.h>
-
 #if defined(__linux__)  &&  __GNU_LIBRARY__ == 1
 #include <asm/io.h>			/* Newer libraries use <sys/io.h> instead. */
 #else
--- pub/diag/ns820-diag.c
+++ pub/diag/ns820-diag.c
@@ -83,10 +83,6 @@ static const char long_usage_msg[] =
 #include <errno.h>
 #include <ctype.h>
 
-/* The following are required only with unaligned field accesses. */
-#include <asm/types.h>
-#include <asm/unaligned.h>
-
 #if defined(__linux__)  &&  __GNU_LIBRARY__ == 1
 #include <asm/io.h>			/* Newer libraries use <sys/io.h> instead. */
 #else
--- pub/diag/pcnet-diag.c
+++ pub/diag/pcnet-diag.c
@@ -87,9 +87,6 @@ static const char long_usage_msg[] =
 #include <string.h>
 #include <errno.h>
 
-#include <asm/types.h>
-#include <asm/unaligned.h>
-
 #if defined(__linux__)  &&  __GNU_LIBRARY__ == 1
 #include <asm/io.h>
 #else
--- pub/diag/tulip-diag.c
+++ pub/diag/tulip-diag.c
@@ -90,9 +90,6 @@ static const char long_usage_msg[] =
 #include <sys/types.h>
 #include <ctype.h>
 
-#include <asm/types.h>
-#include <asm/unaligned.h>
-
 #if defined(__linux__)  &&  __GNU_LIBRARY__ == 1
 #include <asm/io.h>			/* Newer libraries use <sys/io.h> instead. */
 #else
@@ -249,6 +246,13 @@ static int emergency_rewrite = 0;
 static int scan_proc_pci(int card_num);
 static int parse_media_type(const char *capabilities);
 static int get_media_index(const char *name);
+
+u16 inline get_unaligned(void *ptr)
+{
+	u16 ret;
+	memcpy(&ret, ptr, sizeof(ret));
+	return ret;
+}
 
 int
 main(int argc, char **argv)
--- pub/diag/via-diag.c
+++ pub/diag/via-diag.c
@@ -90,10 +90,6 @@ static const char long_usage_msg[] =
 #include <sys/types.h>
 #include <ctype.h>
 
-/* The following are required only with unaligned field accesses. */
-#include <asm/types.h>
-#include <asm/unaligned.h>
-
 #if defined(__linux__)  &&  __GNU_LIBRARY__ == 1
 #include <asm/io.h>			/* Newer libraries use <sys/io.h> instead. */
 #else
