--- pub/diag/Makefile
+++ pub/diag/Makefile
@@ -15,6 +15,9 @@ instdir=$(exec_prefix)/sbin
 MANDIR = $(prefix)/man
 MANPAGES = etherwake.8 mii-diag.8 netdiag.8
 
+# a lot of these files require sys/io.h
+ifeq ($(shell echo | $(CC) -E -include sys/io.h - >& /dev/null && echo yes),yes)
+
 # Diagnostics that can use libmii.
 MDIAGS=alta-diag eepro100-diag epic-diag myson-diag\
    natsemi-diag ns820-diag pcnet-diag rtl8139-diag starfire-diag tulip-diag \
@@ -23,20 +26,28 @@ MDIAGS=alta-diag eepro100-diag epic-diag
 # PCI memory regions into user-level diagnostics, and thus are not distributed.
 
 DIAGS=ne2k-pci-diag $(MDIAGS)
-UTILS=pci-config mii-diag ether-wake
+UTILS=pci-config mii-diag ether-wake diag-example
 LIBS = libflash.c libmii.c
 
 ifeq ($(wildcard libflash.c), libflash.c)
   FLASH_DEPEND = libflash.o
   override CFLAGS += -DLIBFLASH
 endif
+
+else
+MANPAGES := etherwake.8 mii-diag.8
+DIAGS :=
+UTILS := mii-diag ether-wake
+LIBS := libmii.c
+endif
+
 ifeq ($(wildcard libmii.c), libmii.c)
   MII_DEPEND += libmii.o
   override CFLAGS += -DLIBMII
 endif
 
 
-most:  $(DIAGS) $(UTILS) diag-example
+most:  $(DIAGS) $(UTILS)
 all:  most test-configmap config-map.o
 
 diag-example $(MDIAGS): $(FLASH_DEPEND) $(MII_DEPEND)
