diff -Nru resin-4.0.6.vanilla/modules/resin/src/boot-manifest resin-4.0.6/modules/resin/src/boot-manifest
--- resin-4.0.6.vanilla/modules/resin/src/boot-manifest	2010-04-19 21:47:32.837392736 +0100
+++ resin-4.0.6/modules/resin/src/boot-manifest	2010-04-19 21:49:59.417465510 +0100
@@ -1,5 +1,5 @@
 Main-Class: com.caucho.server.bootstrap.ResinBoot
-Class-Path: resin.jar jta-101.jar isorelax.jar jsdk-24.jar jmx-12.jar jca-15.jar jstl-11.jar
+Class-Path: resin.jar jta-101.jar isorelax.jar resin-servlet-api.jar jmx-12.jar jca-15.jar jstl-11.jar
 Extension-Name: com.caucho.server.bootstrap
 Implementation-Vendor: Caucho Technology, Inc.
 Implementation-Version: 3.0.1
diff -Nru resin-4.0.6.vanilla/modules/resin/src/manifest resin-4.0.6/modules/resin/src/manifest
--- resin-4.0.6.vanilla/modules/resin/src/manifest	2010-04-19 21:47:33.421393155 +0100
+++ resin-4.0.6/modules/resin/src/manifest	2010-04-19 21:49:59.417465510 +0100
@@ -1,4 +1,4 @@
-Class-Path: javaee-16.jar javaee-15.jar pro.jar license.jar hessian.jar quercus.jar resin-kernel.jar jcache-16.jar webservices-api.jar
+Class-Path: javaee-16.jar javaee-15.jar pro.jar license.jar hessian.jar quercus.jar resin-kernel.jar jcache-16.jar webservices-api.jar resin-servlet-api.jar
 Implementation-Vendor: Caucho Technology, Inc.
 Implementation-Version: 4.0.6
 J2EE-DeploymentFactory-Implementation-Class: com.caucho.j2ee.deployclient.DeploymentFactoryImpl
diff -Nru resin-4.0.6.vanilla/modules/resin/src/manifest.in resin-4.0.6/modules/resin/src/manifest.in
--- resin-4.0.6.vanilla/modules/resin/src/manifest.in	2010-04-19 21:47:33.421393155 +0100
+++ resin-4.0.6/modules/resin/src/manifest.in	2010-04-19 21:49:59.417465510 +0100
@@ -1,4 +1,4 @@
-Class-Path: javaee-16.jar javaee-15.jar pro.jar license.jar hessian.jar quercus.jar resin-kernel.jar jcache-16.jar webservices-api.jar
+Class-Path: javaee-16.jar javaee-15.jar pro.jar license.jar hessian.jar quercus.jar resin-kernel.jar jcache-16.jar webservices-api.jar resin-servlet-api.jar
 Implementation-Vendor: Caucho Technology, Inc.
 Implementation-Version: @VERSION@
 J2EE-DeploymentFactory-Implementation-Class: com.caucho.j2ee.deployclient.DeploymentFactoryImpl
