Index: vlc-0.9.10/projects/mozilla/vlcshell.cpp
===================================================================
--- vlc-0.9.10.orig/projects/mozilla/vlcshell.cpp
+++ vlc-0.9.10/projects/mozilla/vlcshell.cpp
@@ -36,10 +36,8 @@
 #   include <mozilla-config.h>
 #endif
 
-/* This is from mozilla java, do we really need it? */
-#if 0
+/* This is from mozilla java, needed for jref type */
 #include <jri.h>
-#endif
 
 #include "vlcplugin.h"
 #include "vlcshell.h"
