Index: luatex-beta-0.29.0/src/texk/web2c/luatexdir/image/pdftoepdf.cc
===================================================================
--- luatex-beta-0.29.0.orig/src/texk/web2c/luatexdir/image/pdftoepdf.cc
+++ luatex-beta-0.29.0/src/texk/web2c/luatexdir/image/pdftoepdf.cc
@@ -24,10 +24,11 @@
 #include <stdio.h>
 #include <string.h>
 #include <ctype.h>
-#include <aconf.h>
-#include <GString.h>
-#include <gmem.h>
-#include <gfile.h>
+#include <dirent.h>
+#include <poppler-config.h>
+#include <goo/GooString.h>
+#include <goo/gmem.h>
+#include <goo/gfile.h>
 #include <config.h>
 #include <assert.h>
 #include "Object.h"
@@ -163,7 +164,7 @@ static PdfDocument *refPdfDocument(char 
         pdf_doc->file_path = xstrdup(file_path);
         void **aa = avl_probe(PdfDocumentTree, pdf_doc);
         assert(aa != NULL);
-        GString *docName = new GString(pdf_doc->file_path);
+        GooString *docName = new GooString(pdf_doc->file_path);
         pdf_doc->doc = new PDFDoc(docName);     // takes ownership of docName
         if (!pdf_doc->doc->isOk() || !pdf_doc->doc->okToPrint())
             pdftex_fail("xpdf: reading PDF image failed");
@@ -534,7 +535,7 @@ static void copyObject(Object * obj)
     int i, l, c;
     Ref ref;
     char *p;
-    GString *s;
+    GooString *s;
     if (obj->isBool()) {
         pdf_printf("%s", obj->getBool()? "true" : "false");
     } else if (obj->isInt()) {
@@ -726,7 +727,7 @@ read_pdf_info(image_dict * idict, intege
     img_totalpages(idict) = pdf_doc->doc->getCatalog()->getNumPages();
     if (img_pagename(idict)) {
         // get page by name
-        GString name(img_pagename(idict));
+        GooString name(img_pagename(idict));
         LinkDest *link = pdf_doc->doc->findDest(&name);
         if (link == NULL || !link->isOk())
             pdftex_fail("PDF inclusion: invalid destination <%s>",
Index: luatex-beta-0.29.0/src/texk/web2c/luatexdir/utils.c
===================================================================
--- luatex-beta-0.29.0.orig/src/texk/web2c/luatexdir/utils.c
+++ luatex-beta-0.29.0/src/texk/web2c/luatexdir/utils.c
@@ -41,7 +41,7 @@
 #include "ptexlib.h"
 
 #include "png.h"
-#include "xpdf/config.h"        /* just to get the xpdf version */
+#include "poppler-config.h"        /* just to get the poppler version */
 
 static const char __svn_version[] =
     "$Id: 010_all_poppler.patch,v 1.1 2008/08/30 07:34:19 aballier Exp $ $URL: http://scm.foundry.supelec.fr/svn/luatex/tags/beta-0.29.0/src/texk/web2c/luatexdir/utils.c $";
@@ -1131,7 +1131,7 @@ void initversionstring(char **versions)
     (void) asprintf(versions,
                     "Compiled with libpng %s; using libpng %s\n"
                     "Compiled with zlib %s; using zlib %s\n"
-                    "Compiled with xpdf version %s\n",
+                    "Compiled with poppler version %s\n",
                     PNG_LIBPNG_VER_STRING, png_libpng_ver,
                     ZLIB_VERSION, zlib_version, xpdfVersion);
 }
