Index: luatex-beta-0.29.0/src/libs/lua51/Makefile
===================================================================
--- luatex-beta-0.29.0.orig/src/libs/lua51/Makefile
+++ luatex-beta-0.29.0/src/libs/lua51/Makefile
@@ -7,10 +7,7 @@
 # Your platform. See PLATS for possible values.
 PLAT= none
 
-CC= gcc
-CFLAGS= -g -O2 -Wall $(XCFLAGS) $(MYCFLAGS) $(COCOFLAGS)
-AR= ar rcu
-RANLIB= ranlib
+CFLAGS+= -Wall $(XCFLAGS) $(MYCFLAGS) $(COCOFLAGS)
 RM= rm -f
 LIBS= $(XLDFLAGS) -lm $(MYLIBS)
 
@@ -64,7 +61,7 @@ o:	$(ALL_O)
 a:	$(ALL_A)
 
 $(LUA_A): $(CORE_O) $(LIB_O)
-	$(AR) $@ $?
+	$(AR) rcu $@ $?
 	$(RANLIB) $@
 
 $(LUA_T): $(LUA_O) $(LUA_A)
