Index: luatex-beta-0.29.0/src/libs/luafontforge/Makefile
===================================================================
--- luatex-beta-0.29.0.orig/src/libs/luafontforge/Makefile
+++ luatex-beta-0.29.0/src/libs/luafontforge/Makefile
@@ -5,7 +5,6 @@ srcdir=../../../src/libs/luafontforge
 FF_LIB=ffdummies.o luafflib.o
 
 _CFLAGS=-Wall -DFONTFORGE_CONFIG_NO_WINDOWING_UI=1 -DLUA_FF_LIB=1 $(XCFLAGS)
-CFLAGS = -g -O2
 
 .PHONY: newfile
 
@@ -17,7 +16,7 @@ fontforge: $(FF_LIB) newfile
 	ranlib libff.a
 
 %.o: $(srcdir)/src/%.c
-	gcc $(_CFLAGS) $(CFLAGS) -I$(srcdir)/../lua51 -I$(srcdir)/fontforge/inc -I$(srcdir)/fontforge/fontforge -c $< -o $@
+	$(CC) $(_CFLAGS) $(CFLAGS) -I$(srcdir)/../lua51 -I$(srcdir)/fontforge/inc -I$(srcdir)/fontforge/fontforge -c $< -o $@
 
 clean:
 	rm  -f libff.a $(FF_LIB)
Index: luatex-beta-0.29.0/src/libs/luasocket/config
===================================================================
--- luatex-beta-0.29.0.orig/src/libs/luasocket/config
+++ luatex-beta-0.29.0/src/libs/luasocket/config
@@ -49,11 +49,10 @@ INSTALL_EXEC=cp
 #------
 # Compiler and linker settings
 # for Linux
-CC=gcc
 DEF=-DLUASOCKET_DEBUG 
-CFLAGS= $(LUAINC) $(DEF) -pedantic -Wall -O2 -fpic
-LDFLAGS=-O -shared -fpic
-LD=gcc 
+CFLAGS+= $(LUAINC) $(DEF) -pedantic -Wall -fpic
+LDFLAGS+=-O -shared -fpic
+LD=$(CC)
 
 #------
 # End of makefile configuration
Index: luatex-beta-0.29.0/src/libs/luafontforge/fontforge/Unicode/Makefile
===================================================================
--- luatex-beta-0.29.0.orig/src/libs/luafontforge/fontforge/Unicode/Makefile
+++ luatex-beta-0.29.0/src/libs/luafontforge/fontforge/Unicode/Makefile
@@ -2,12 +2,9 @@
 top_srcdir = ../../../../src/libs/luafontforge/fontforge
 srcdir = $(top_srcdir)/Unicode
 
-CC = gcc
-
 OBJECTS = alphabet.o char.o memory.o cjk.o ustring.o gwwiconv.o
 
 _CFLAGS = -I$(top_srcdir)/inc -Wall -DLUA_FF_LIB=1 $(XCFLAGS)
-CFLAGS = -g -O2
 
 all: $(OBJECTS)
 
Index: luatex-beta-0.29.0/src/libs/luafontforge/fontforge/fontforge/Makefile
===================================================================
--- luatex-beta-0.29.0.orig/src/libs/luafontforge/fontforge/fontforge/Makefile
+++ luatex-beta-0.29.0/src/libs/luafontforge/fontforge/fontforge/Makefile
@@ -2,10 +2,6 @@
 top_srcdir = ../../../../src/libs/luafontforge/fontforge
 srcdir = $(top_srcdir)/fontforge
 
-CC = gcc
-AR = ar cru
-RANLIB = ranlib
-
 OBJECTS =autohint.o charview.o fontinfo.o \
  fontview.o fvfonts.o parsepfa.o parsettf.o prefs.o psread.o namelist.o \
  splinefill.o splineoverlap.o splinestroke.o splineutil.o splineutil2.o stamp.o \
@@ -22,7 +18,7 @@ _CFLAGS = -I$(top_srcdir)/inc -I$(srcdir
  -DNODYNAMIC=1 -D_NO_FREETYPE=1 -D_NO_LIBUNINAMESLIST=1 \
  -DFONTFORGE_CONFIG_NO_WINDOWING_UI=1 -DLUA_FF_LIB=1 -UHAVE_LIBINTL_H
 
-CFLAGS = -Wall -g -O2
+CFLAGS += -Wall
 
 all: $(OBJECTS)
 
@@ -30,7 +26,7 @@ all: $(OBJECTS)
 # disasterous effects. This file may not be compiled with the optimizer.
 # On the other hand we do need CPPFLAGS to get the right includes...
 splinerefigure.o: $(srcdir)/splinerefigure.c $(srcdir)/splinefont.h
-	$(CC) -g -c -o splinerefigure.o $(_CFLAGS) $(srcdir)/splinerefigure.c
+	$(CC) -c -o splinerefigure.o $(_CFLAGS) $(CFLAGS) $(srcdir)/splinerefigure.c
 
 %.o: $(srcdir)/%.c $(srcdir)/splinefont.h $(srcdir)/views.h $(srcdir)/stemdb.h $(srcdir)/ttf.h $(srcdir)/unicoderange.h $(srcdir)/psfont.h $(srcdir)/groups.h $(srcdir)/edgelist.h $(srcdir)/edgelist2.h
 	$(CC)  $(_CFLAGS) $(CFLAGS) -c $< -o $@
Index: luatex-beta-0.29.0/src/libs/luasocket/src/Makefile
===================================================================
--- luatex-beta-0.29.0.orig/src/libs/luasocket/src/Makefile
+++ luatex-beta-0.29.0/src/libs/luasocket/src/Makefile
@@ -5,11 +5,9 @@ INSTALL_TOP_SHARE=/usr/local/share/lua/5
 INSTALL_TOP_LIB=/usr/local/lib/lua/5.1
 
 LUAINC = -I../../lua51
-CC = gcc
 
 DEF = -DLUASOCKET_DEBUG 
-CFLAGS = $(LUAINC) $(DEF) -pedantic -Wall -O2 $(XCFLAGS)
-AR=ar
+CFLAGS += $(LUAINC) $(DEF) -pedantic -Wall $(XCFLAGS)
 ARFLAGS=rc
 
 SOCKET_OBJS:= \
