Index: luatex-beta-0.29.0/src/texk/web2c/luatexdir/luatexlib.mk
===================================================================
--- luatex-beta-0.29.0.orig/src/texk/web2c/luatexdir/luatexlib.mk
+++ luatex-beta-0.29.0/src/texk/web2c/luatexdir/luatexlib.mk
@@ -108,7 +108,7 @@ LUASOCKETINC=-I../../lua51
 
 $(LUASOCKETDEP): $(LUASOCKETDIR)/src/luasocket.c $(LUASOCKETDIR)/src/luasocket.h
 	mkdir -p $(LUASOCKETDIR) && cd $(LUASOCKETDIR) && cp -R $(LUASOCKETSRCDIR)/* . && \
-    cd src && make
+    cd src && $(MAKE)
 
 
 # luapeg
