--- config/autoconf.mk.in.old	2007-01-20 21:19:33.000000000 +0100
+++ config/autoconf.mk.in	2007-01-20 21:20:16.000000000 +0100
@@ -479,6 +479,7 @@
 MOZ_ENABLE_PANGO        = @MOZ_ENABLE_PANGO@
 MOZ_PANGO_CFLAGS        = @MOZ_PANGO_CFLAGS@
 MOZ_PANGO_LIBS          = @MOZ_PANGO_LIBS@
+MOZ_PANGOCAIRO          = @MOZ_PANGOCAIRO@
 
 MOZ_EXTRA_X11CONVERTERS	= @MOZ_EXTRA_X11CONVERTERS@
 
--- gfx/src/gtk/mozilla-decoder.cpp.old	2007-01-20 21:21:30.000000000 +0100
+++ gfx/src/gtk/mozilla-decoder.cpp	2007-01-20 21:24:19.000000000 +0100
@@ -40,7 +40,11 @@
 #define PANGO_ENABLE_ENGINE
 
 #include "mozilla-decoder.h"
+#ifdef MOZ_PANGOCAIRO
+#include <pango/pangocairo.h>
+#else
 #include <pango/pangoxft.h>
+#endif
 #include <pango/pangofc-fontmap.h>
 #include <pango/pangofc-font.h>
 #include <gdk/gdkx.h>
@@ -207,9 +211,12 @@
             printf("unknown suffix used for mapping\n");
         }
     }
-
+#ifdef MOZ_PANGOCAIRO
+    pango_fc_font_map_add_decoder_find_func(PANGO_FC_FONT_MAP(pango_cairo_font_map_get_default()),
+#else
     pango_fc_font_map_add_decoder_find_func(PANGO_FC_FONT_MAP(pango_xft_get_font_map(GDK_DISPLAY(),gdk_x11_get_default_screen())),
-                                            mozilla_find_decoder,
+#endif
+                                         mozilla_find_decoder,
                                             NULL,
                                             NULL);
 
--- configure.in.old	2007-01-20 21:25:20.000000000 +0100
+++ configure.in	2007-01-20 21:32:21.000000000 +0100
@@ -3923,8 +3923,8 @@
     MOZ_ENABLE_GTK2=1
     MOZ_ENABLE_XREMOTE=1
     MOZ_ENABLE_COREXFONTS=${MOZ_ENABLE_COREXFONTS-}
-    TK_CFLAGS='$(MOZ_GTK2_CFLAGS)'
-    TK_LIBS='$(MOZ_GTK2_LIBS)'
+    TK_CFLAGS='$(MOZ_GTK2_CFLAGS) $(MOZ_PANGO_CFLAGS)'
+    TK_LIBS='$(MOZ_GTK2_LIBS) $(MOD_PANGO_LIBS)'
     AC_DEFINE(MOZ_WIDGET_GTK2)
     ;;
 
@@ -4581,12 +4581,14 @@
 then
     AC_DEFINE(MOZ_ENABLE_XFT)
     PKG_CHECK_MODULES(MOZ_XFT, xft)
-    PKG_CHECK_MODULES(_PANGOCHK, pango >= 1.1.0)
+    PKG_CHECK_MODULES(MOZ_PANGO, pangoxft >= 1.1.0 pangox)
 fi
 
 AC_SUBST(MOZ_ENABLE_XFT)
 AC_SUBST(MOZ_XFT_CFLAGS)
 AC_SUBST(MOZ_XFT_LIBS)
+AC_SUBST(MOZ_PANGO_CFLAGS)
+AC_SUBST(MOZ_PANGO_LIBS)
 
 dnl ========================================================
 dnl = pango font rendering
@@ -4600,8 +4602,17 @@
 then
     AC_DEFINE(MOZ_ENABLE_PANGO)
     PKG_CHECK_MODULES(MOZ_PANGO, pangoxft >= 1.6.0)
+    PKG_CHECK_MODULES(PANGOCAIRO_CHK, pangocairo >= 1.10.0,
+    [
+      MOZ_PANGOCAIRO=1
+      AC_DEFINE(MOZ_PANGOCAIRO)
+      MOZ_PANGO_CFLAGS="$PANGOCAIRO_CHK_CFLAGS $MOZ_PANGO_CFLAGS"
+      MOZ_PANGO_LIBS="$PANGOCAIRO_CHK_LIBS $MOZ_PANGO_LIBS"
+    ],
+    [:])
 
     AC_SUBST(MOZ_ENABLE_PANGO)
+    AC_SUBST(MOZ_PANGOCAIRO)
     AC_SUBST(MOZ_PANGO_CFLAGS)
     AC_SUBST(MOZ_PANGO_LIBS)
 fi
