--- xpinstall/packager/packages-unix.old	2007-01-20 21:14:14.000000000 +0100
+++ xpinstall/packager/packages-unix	2007-01-20 21:17:57.000000000 +0100
@@ -45,9 +45,6 @@
 
 [xpcom]
 bin/libmozjs.so
-bin/libnspr4.so
-bin/libplc4.so
-bin/libplds4.so
 bin/cpu/sparcv8plus/libnspr_flt4.so
 bin/libxpcom.so
 bin/libxpcom_core.so
@@ -469,17 +466,13 @@
 bin/components/talkback/*
 
 [psm]
-bin/libnssckbi.so
 bin/components/libpipboot.so
 bin/components/pipboot.xpt
 bin/components/libpipnss.so
 bin/components/pipnss.xpt
 bin/components/libpippki.so
 bin/components/pippki.xpt
-bin/libnss3.so
-bin/libsmime3.so
 bin/libsoftokn3.chk
-bin/libsoftokn3.so
 bin/libfreebl3.chk
 bin/libfreebl3.so
 bin/libfreebl_32fpu_3.chk
@@ -492,7 +485,6 @@
 bin/libfreebl_64fpu_3.so
 bin/libfreebl_64int_3.chk
 bin/libfreebl_64int_3.so
-bin/libssl3.so
 bin/chrome/pipnss.jar
 bin/chrome/pippki.jar
 
@@ -510,5 +502,17 @@
 bin/components/spellchecker.xpt
 bin/dictionaries/*
 
+[nspr]
+bin/libnspr4.so
+bin/libplc4.so
+bin/libplds4.so
+
+[nss]
+bin/libnssckbi.so
+bin/libnss3.so
+bin/libsmime3.so
+bin/libsoftokn3.so
+bin/libssl3.so
+
 [reporter]
 bin/chrome/reporter.jar
