diff -Nru resin-3.1.8.vanilla/configure.ac resin-3.1.8/configure.ac
--- resin-3.1.8.vanilla/configure.ac	2009-01-10 15:32:19.000000000 +0100
+++ resin-3.1.8/configure.ac	2009-01-10 15:32:49.000000000 +0100
@@ -1164,10 +1164,3 @@
 modules/c/src/resin_os/Makefile
 contrib/init.resin
 ], [])
-
-if test -n "${resin_pro_plugin}"; then
-  AC_OUTPUT([
-    modules/c/src/resin/Makefile
-    modules/c/src/resinssl/Makefile
-  ])
-fi
