diff -urN linux-2.4.27-hardened-0/Documentation/Configure.help linux-2.4.27-hardened-1/Documentation/Configure.help
--- linux-2.4.27-hardened-0/Documentation/Configure.help	2004-08-08 14:06:38.000000000 -0500
+++ linux-2.4.27-hardened-1/Documentation/Configure.help	2004-08-08 14:06:49.000000000 -0500
@@ -17273,6 +17273,19 @@
 
   If unsure, say N.
 
+kernel compilation parameters in /proc support
+CONFIG_PROC_CONFIG
+  This provides a listing in /proc/config of the configuration of
+  the compiled kernel. It's useful because people forget or lose
+  their notes on how they compiled which kernel image. You can
+  then recreate the .config file by
+       cat /proc/config > /usr/src/linux/.config
+
+  You can compile the code as a module.  If compiled in it
+  will enlarge your kernel by between 3-16KB. 
+
+  If unsure, say M.
+
 NFS file system support
 CONFIG_NFS_FS
   If you are connected to some other (usually local) Unix computer
diff -urN linux-2.4.27-hardened-0/Makefile linux-2.4.27-hardened-1/Makefile
--- linux-2.4.27-hardened-0/Makefile	2004-08-07 18:26:07.000000000 -0500
+++ linux-2.4.27-hardened-1/Makefile	2004-08-08 14:06:49.000000000 -0500
@@ -494,7 +494,7 @@
 sums:
 	find . -type f -print | sort | xargs sum > .SUMS
 
-dep-files: scripts/mkdep archdep include/linux/version.h
+dep-files: scripts/mkdep archdep include/linux/version.h scripts/kconfig.tk scripts/makeproconfigdata
 	rm -f .depend .hdepend
 	$(MAKE) $(patsubst %,_sfdep_%,$(SUBDIRS)) _FASTDEP_ALL_SUB_DIRS="$(SUBDIRS)"
 ifdef CONFIG_MODVERSIONS
@@ -553,6 +553,12 @@
 scripts/split-include: scripts/split-include.c
 	$(HOSTCC) $(HOSTCFLAGS) -o scripts/split-include scripts/split-include.c
 
+scripts/kconfig.tk:
+	$(MAKE) -C scripts kconfig.tk
+
+scripts/makeproconfigdata: scripts/makeproconfigdata.c
+	$(HOSTCC) $(HOSTCFLAGS) -o $@ $^
+
 #
 # RPM target
 #
diff -urN linux-2.4.27-hardened-0/fs/Config.in linux-2.4.27-hardened-1/fs/Config.in
--- linux-2.4.27-hardened-0/fs/Config.in	2004-08-08 14:06:38.000000000 -0500
+++ linux-2.4.27-hardened-1/fs/Config.in	2004-08-08 14:06:49.000000000 -0500
@@ -82,6 +82,9 @@
 tristate 'OS/2 HPFS file system support' CONFIG_HPFS_FS
 
 bool '/proc file system support' CONFIG_PROC_FS
+if [ "$CONFIG_PROC_FS" = "y" ]; then
+   tristate 'Kernel configuration visible in /proc/config' CONFIG_PROC_CONFIG
+fi
 
 # For some reason devfs corrupts memory badly on x86-64. Disable it 
 # for now.
diff -urN linux-2.4.27-hardened-0/fs/proc/proc_misc.c linux-2.4.27-hardened-1/fs/proc/proc_misc.c
--- linux-2.4.27-hardened-0/fs/proc/proc_misc.c	2004-08-07 18:26:06.000000000 -0500
+++ linux-2.4.27-hardened-1/fs/proc/proc_misc.c	2004-08-08 14:06:49.000000000 -0500
@@ -670,4 +670,7 @@
 			entry->proc_fops = &ppc_htab_operations;
 	}
 #endif
+#if defined(CONFIG_PROC_CONFIG) && ! defined(CONFIG_PROC_CONFIG_MODULE)
+	proc_config_init();
+#endif
 }
diff -urN linux-2.4.27-hardened-0/kernel/Makefile linux-2.4.27-hardened-1/kernel/Makefile
--- linux-2.4.27-hardened-0/kernel/Makefile	2004-08-08 14:06:38.000000000 -0500
+++ linux-2.4.27-hardened-1/kernel/Makefile	2004-08-08 14:06:49.000000000 -0500
@@ -9,7 +9,7 @@
 
 O_TARGET := kernel.o
 
-export-objs = signal.o sys.o kmod.o context.o ksyms.o pm.o exec_domain.o printk.o fork.o
+export-objs = signal.o sys.o kmod.o context.o ksyms.o pm.o exec_domain.o printk.o fork.o config.o
 
 obj-y     = sched.o dma.o fork.o exec_domain.o panic.o printk.o \
 	    module.o exit.o itimer.o info.o time.o softirq.o resource.o \
@@ -19,6 +19,7 @@
 obj-$(CONFIG_UID16) += uid16.o
 obj-$(CONFIG_MODULES) += ksyms.o
 obj-$(CONFIG_PM) += pm.o
+obj-$(CONFIG_PROC_CONFIG) += config.o
 
 ifneq ($(CONFIG_IA64),y)
 # According to Alan Modra <alan@linuxcare.com.au>, the -fno-omit-frame-pointer is
@@ -30,3 +31,11 @@
 endif
 
 include $(TOPDIR)/Rules.make
+
+config.o: config.c proconfig.data
+
+proconfig.data: $(TOPDIR)/.config $(TOPDIR)/scripts/makeproconfigdata
+	$(TOPDIR)/scripts/makeproconfigdata < $< > $@
+$(TOPDIR)/scripts/makeproconfigdata:
+	$(MAKE) -C $(TOPDIR)/scripts makeproconfigdata
+
diff -urN linux-2.4.27-hardened-0/kernel/config.c linux-2.4.27-hardened-1/kernel/config.c
--- linux-2.4.27-hardened-0/kernel/config.c	1969-12-31 18:00:00.000000000 -0600
+++ linux-2.4.27-hardened-1/kernel/config.c	2004-08-08 14:06:49.000000000 -0500
@@ -0,0 +1,450 @@
+#include <linux/fs.h>
+#include <linux/stat.h>
+#include <linux/config.h>
+#include <linux/version.h>
+#include <linux/slab.h>
+#include <linux/unistd.h>
+#include <linux/module.h>
+#include <linux/errno.h>
+#include <linux/proc_fs.h>
+#include <asm/uaccess.h>
+#include <linux/kmod.h>
+
+/*
+ * This is the printing engine and CRC database. 
+ *
+ * The /proc/config entry can be set to off or on by writing
+ * respectively to it the one-character string "0", "1". E.g., for
+ * example
+ *
+ *    echo -n "0" > /proc/config
+ *
+ */
+
+#ifdef MODULE
+#define proc_config_init init_module
+
+MODULE_AUTHOR ("Peter T. Breuer");
+MODULE_DESCRIPTION ("Proc Config v0.98 March 2003");
+MODULE_LICENSE ("GPL");
+#endif
+
+ /*
+  * typedefs
+  *   Handle - unique identifier for list of values (up to 256 of)
+  *   Offset - well, printing offsets for each line of output
+  *   Index  - offset of name string within buffer area
+  *   Value  - value of kernel parameter
+  */
+typedef short Offset;
+typedef short Index;
+typedef char Handle;
+typedef char *Value;
+
+ /*
+  * global struct
+  */
+struct ucStruct {
+    int      len;         // global counter for number of stored data items
+ /*
+  * These three simple arrays will be embedded in the kernel permamently
+  *
+  *  handle   - list of single char identifiers
+  *  value    - list of possible strings as config values
+  *  key      - list of prinary keys (shorts)
+  */
+    Handle * handle;
+    Value  * value;
+    Offset * offset;     // array of print string offsets
+    Index  * index;      // array of name string offsets
+    char   * buffer;     // name storge area
+};
+
+static struct ucStruct uc;
+
+ /*
+  * internal forward prototypes
+  */
+static int get_info_config (char *, char **, off_t, int, int *, void *);
+static int write_proc_config (struct file *file, const char *buffer,
+			      unsigned long count, void *data);
+static int proc_switch (struct ucStruct *uc, int x);
+static int proc_config_setup (struct ucStruct *uc);
+
+/*
+ * Free whichever of the tables uxOffset, ucName and ucScratch
+ * has been kmalloced.
+ */
+static void
+free_memory (struct ucStruct *uc)
+{
+        if (!uc)
+                return;
+
+	if (uc->offset)
+		kfree (uc->offset);
+	uc->offset = NULL;
+
+	if (uc->index)
+		kfree (uc->index);
+	uc->index = NULL;
+}
+
+static int
+alloc_memory(struct ucStruct *uc, int len)
+{
+        // table of name string offsets within buffer
+	if (!uc->index)
+                uc->index = kmalloc(sizeof(Index) * (len + 1), GFP_KERNEL);
+	if (!uc->index)
+		goto fail;
+        // the table of printing offsets
+	if (!uc->offset)
+	        uc->offset = kmalloc (sizeof(Offset) * (len + 1), GFP_KERNEL);
+	if (!uc->offset)
+		goto fail;
+        return 0;
+fail:
+        return -ENOMEM;
+}
+
+
+/*
+ * initialisation. Don't declare as __init - we want its data to hang
+ * around
+ */
+int __init
+proc_config_init (void)
+{
+    /*
+     * this routine lives in the kernel rump - it should be minimal in
+     * size.  It contains the basic lists (as statics) that the module
+     * requires in order to print a table of names and values. 
+     */
+
+	// we store names directly
+//#define PROCONF_KEY(x,y) #x ,
+#define PROCONF_KEY(x,y) #x "\000"
+#define PROCONF_KEY_BEGIN static char ucBuffer[] = 
+#define PROCONF_KEY_END ;
+
+	// actual distinct parameter values (very few)
+#define PROCONF_XDATA(y) #y ,
+#define PROCONF_XDATA_BEGIN static Value ucValue[] = {
+#define PROCONF_XDATA_END };
+
+	// indexes to the parameter values
+#define PROCONF_DATA(y) y
+#define PROCONF_DATA_BEGIN static Handle ucHandle[] =
+#define PROCONF_DATA_END ;
+
+	int ucLen;
+
+    /*
+     * this makes the local static name-key pair table ucName
+     * and the local static list ucValue of special valeues taken by
+     * the config params, and the local static list of ucHandle charvals
+     * and sets ucLen.
+     */
+
+#include "proconfig.data"
+
+        // pick up the length and tables
+
+	uc.len = ucLen;
+	uc.handle = ucHandle;
+	uc.value = ucValue;
+	uc.buffer = ucBuffer;
+            
+        
+        if (alloc_memory(&uc, ucLen) < 0)
+                goto fail;
+
+	printk (KERN_INFO "Proc Config support by ptb@it.uc3m.es\n");
+
+        
+
+	if (proc_config_setup (&uc) < 0)
+                goto fail;
+
+	do {
+		struct proc_dir_entry *res =
+		 create_proc_read_entry ("config", 0440, NULL, &get_info_config, NULL);
+		if (!res)
+			goto fail;
+		res->write_proc = &write_proc_config;
+	} while (0);
+
+	return 0;
+
+      fail:
+	printk (KERN_INFO "memory alloc failed in proc_config_init\n");
+	free_memory (&uc);
+	return -ENOMEM;
+}
+
+#ifdef MODULE
+void __exit
+cleanup_module (void)
+{
+	remove_proc_entry ("config", &proc_root);
+	free_memory (&uc);
+	printk (KERN_INFO "Proc Config unloaded\n");
+}
+#endif
+
+/*
+ * A name may be compressed. That means it may contain refs to
+ * other names within it. A refernce is a character with the high bit
+ * set. If the 64 bit is also set, then the reference is continued to
+ * the next character too.  The reference is a base 64 integer.
+ *
+ * This routine expands the name into the buffer supplied.
+ */
+static int
+expandname(struct ucStruct *uc, int index, char *buf) {
+        unsigned char * name = & uc->buffer[uc->index[index]];
+        int i, len = strlen(name);
+        int count = 0;
+        for (i = 0; i < len; i++) {
+            if (! (name[i] & 128)) {
+                buf[count++] = name[i];
+            } else {
+                int index = 0;
+                do {
+                        index *= 64;
+                        index += name[i] & 0x3f;
+                        if (! (name[i] & 64)) 
+                            // no more in this number
+                            break;
+                        i++;
+                } while (name[i] & 128);
+                // recursion! It'll only happen very few times. 
+                count += expandname(uc, index, &buf[count]);
+                buf[count++] = '_';
+            }
+        }
+        buf[count] = 0;
+        return count;
+}
+
+static int
+print_entries (struct ucStruct *uc, char *buf, int index, int len)
+{
+	int count = 0;
+
+	// start printing out entries till nearly len bytes output
+	for (; index < uc->len && count < len; index++) {
+		int n = uc->offset[index + 1] - uc->offset[index];
+                char nambuf[64];
+                // ignore 0-values (= 'n')  entries
+                if (uc->handle[index] == 0)
+                    continue;
+		// stop before we overrun
+		if (count + n > len)
+			break;
+                expandname(uc, index, nambuf);
+		n = sprintf (buf, "CONFIG_%s=%s\n", nambuf,
+			  uc->value[(int) uc->handle[index]]);
+		buf += n;
+		count += n;
+	}
+
+	// update callers notion of where we got to this time
+	return count;
+}
+
+static int
+proc_config_setup (struct ucStruct *uc)
+{
+
+	// This routine might be in the data module. It's called from there.
+	// It just so happens that we work on data here.
+
+	int offset = 0;
+	int count = 0;
+        int tot = 0;
+
+        // construct the name index map
+        for (count = 0; count < uc->len; count++) {
+            int len = strlen (& uc->buffer[tot]);
+            uc->index[count] = tot;
+            tot += len + 1;
+        }
+
+	if (!uc->offset)
+		goto fail;
+
+	// construct print offset map
+	for (count = 0; count < uc->len; count++) {
+
+                char nambuf[64];
+		int m = strlen (uc->value[(int) uc->handle[count]]);
+		int n = expandname (uc, count, nambuf);
+
+		uc->offset[count] = offset;
+                // ignore 0-valued entries (= 'n')
+                if (uc->handle[count] == 0)
+                    continue;
+		// CONFIG_...=...\n
+		offset += 7 + n + 1 + m + 1;
+	}
+	uc->offset[uc->len] = offset;
+
+        printk (KERN_INFO "proc config counted %d bytes in names\n", tot);
+
+        printk (KERN_INFO "proc config counted %d bytes in value handles\n", count * sizeof(Handle));
+
+	return 0;
+
+      fail:
+	printk (KERN_ALERT "out of memory in proc_config_setup\n");
+	return -ENOMEM;
+
+}
+
+/*
+ * given desired character offset in printout list, find the next
+ * table entry to print
+ */
+static int
+proc_config_find (struct ucStruct *uc, int offset)
+{
+
+#ifdef PROCONF_KISS
+
+        // linear search
+	int i = 0;
+	for (i = 0; i < uc->len; i++) {
+		if (ucOffset[i + 1] > offset)
+			break;
+	}
+
+	return i;
+
+#else
+
+        // binary search
+	int lower = 0;
+	int upper = uc->len - 1;
+
+    #if DEBUG > 0        
+	printk (KERN_INFO
+		"proc_config_find entered with offset %d, ucLen %d\n",
+		offset, uc->len);
+    #endif
+	if (upper < lower)
+		return -EINVAL;
+
+	if (uc->offset[upper + 1] <= offset) {
+		return upper + 1;
+	}
+	if (uc->offset[lower] > offset) {
+		return lower - 1;
+	}
+
+	while (upper > lower + 1) {
+		int i = (lower + upper) / 2;
+		if (uc->offset[i + 1] > offset)
+			upper = i;
+		else
+			lower = i;
+	}
+
+	return upper;
+
+#endif
+
+}
+
+/* 
+ * print proc data
+ */
+static int
+get_info_config (char *page, char **start, off_t offset, int len, int *eof,
+		 void *data)
+{
+	int count = 0;		// chars output
+
+	// bsearch internal table until we locate the internal offset
+	int i = proc_config_find (&uc, offset);
+	if (i >= 0)
+		// now print as many entries as we can from there on
+		count = print_entries (&uc, page, i, len);
+
+	*start = (char *) count;
+	if (count <= 0)
+		*eof = 1;
+	return count;
+}
+
+
+/*
+ * turn proc listing on or off or half on. Half-on is when it prints
+ * CRCs only.
+ */
+static int
+proc_switch (struct ucStruct *uc, int x)
+{
+
+	static int ucLenForReal;
+
+        void
+        proc_switch_off (void)
+        {
+	        if (ucLenForReal <= 0) {
+		        // we were on
+		        free_memory (uc);
+		        ucLenForReal = uc->len;
+		        uc->len = 0;
+	        }
+        }
+
+        void
+        proc_switch_on (void)
+        {
+	        if (ucLenForReal > 0) {
+		        // we were off
+                        if (alloc_memory(uc, ucLenForReal) < 0)
+                                return;
+		        uc->len = ucLenForReal;
+		        ucLenForReal = 0;
+	                proc_config_setup (uc);
+	        }
+        }
+
+	switch (x) {
+	  case 0:
+		proc_switch_off ();
+		break;
+	  case 1:
+		proc_switch_on ();
+		break;
+	}
+	return ucLenForReal;
+}
+
+
+int
+write_proc_config (struct file *file, const char *buffer,
+		   unsigned long count, void *data)
+{
+        int intval;
+#if DEBUG > 0
+	printk (KERN_INFO
+		"write_proc_config file %x, buff %x, count %lu, data %x\n",
+		(unsigned) file, (unsigned) buffer, count,
+		(unsigned) data);
+#endif
+        if (sscanf(buffer, "%d", &intval) >= 1) {
+		proc_switch (&uc, intval);
+		return count;
+	}
+	return -EINVAL;
+}
+
+
+/*
+ *  compile line
+ *  gcc -O2 -c -D__KERNEL__ -DMODULE -I "kernel-source-dir"/include -c config.c
+ */
diff -urN linux-2.4.27-hardened-0/scripts/Makefile linux-2.4.27-hardened-1/scripts/Makefile
--- linux-2.4.27-hardened-0/scripts/Makefile	2000-03-16 13:27:17.000000000 -0600
+++ linux-2.4.27-hardened-1/scripts/Makefile	2004-08-08 14:06:49.000000000 -0500
@@ -39,7 +39,10 @@
 docproc: docproc.o
 	${HOSTCC} -o docproc docproc.o
 
+makeproconfigdata: makeproconfigdata.c
+	$(HOSTCC) -o $@ $^
+
 clean:
-	rm -f *~ kconfig.tk *.o tkparse mkdep split-include docproc
+	rm -f *~ kconfig.tk *.o tkparse mkdep split-include docproc makeproconfigdata
 
 include $(TOPDIR)/Rules.make
diff -urN linux-2.4.27-hardened-0/scripts/makeproconfigdata.c linux-2.4.27-hardened-1/scripts/makeproconfigdata.c
--- linux-2.4.27-hardened-0/scripts/makeproconfigdata.c	1969-12-31 18:00:00.000000000 -0600
+++ linux-2.4.27-hardened-1/scripts/makeproconfigdata.c	2004-08-08 14:06:49.000000000 -0500
@@ -0,0 +1,231 @@
+
+# include <stdio.h>
+# include <stdlib.h>
+# include <string.h>
+# include <ctype.h>
+
+#define MIN(x,y) ((x)<(y) ? (x) : (y) )
+
+/* makeproconfigdata.c : make a suitable kernel/proconfig.data file
+   Author and Copyright Peter T. Breuer (ptb@it.uc3m.es) 27/2/00 
+   With contributions from cksum.c in the GNU textutils package
+   as reproduced below.
+
+   This code is under the GNU General Public License as published by
+   the Free Software Foundation; either version 2, or (at your option)
+   any later version.
+ */
+
+      // do inplace compression on the name
+static void
+compress (char *name, char *names[], int ucLen)
+{
+    int j;
+    int maxlen = -1;
+    int maxj = -1;
+    char *realname = strdup (name);
+
+    for (j = 0; j < ucLen; j++) {
+
+	char *possible = names[j];
+	int len = strlen (possible);
+
+	if (strncmp (name, possible, len) == 0 && name[len] == '_') {
+	    if (maxlen < len) {
+		maxlen = len;
+		maxj = j;
+	    }
+	}
+    }				// efor
+
+    // we are a '_' continuation of something
+
+    if (maxj >= 0 && maxlen >= 2) {
+
+	int n, offset = 0;
+	int len = maxlen;
+	j = maxj;
+
+	if (j < 64) {
+	    sprintf (name + offset, "\\%.3o%n%c", j | 128, &n, 0);
+	    offset += n;
+	    sprintf (name + offset, "%s", realname + len + 1);
+	}
+	else {
+	    int k = j / 64;
+	    j %= 64;
+	    sprintf (name + offset, "\\%.3o%n%c", k | 128 | 64, &n, 0);
+	    offset += n;
+	    sprintf (name + offset, "\\%.3o%n%c", j | 128, &n, 0);
+	    offset += n;
+	    sprintf (name + offset, "%s", realname + len + 1);
+	}
+    }
+
+}
+
+int
+main (int argc, char *args[])
+{
+
+# define MAXLINELENGTH 1024
+    static char s[MAXLINELENGTH];
+    int count = 3;
+#define MAXPARM 4096
+    // "n" is in 0 position so we can never choose it
+    char *extras[MAXPARM] = { "n", "y", "m", NULL, };
+    short tokens[MAXPARM];
+    char *names[MAXPARM];
+    int ucLen = 0;
+    int i;
+
+    printf ("\n");
+    printf ("/* generated by $0 from $TOPDIR/.config, do not edit */\n");
+    printf ("\n");
+
+    printf ("PROCONF_KEY_BEGIN\n");
+
+    // seed the array with some invisible stuff that helps compression
+    do {
+	int i = 0;
+	char *name;
+	char *array[] = {
+	    "BLK_DEV",
+	    "MTDRAM",
+	    "IP",
+	    "IP_ROUTE",
+	    "IP_PIMSM",
+	    "IP_NF",
+	    "IP_NF_MATCH",
+	    "IP_NF_TARGET",
+	    "IP6",
+	    "IP6_NF",
+	    "IP6_NF_MATCH",
+	    "IP6_NF_TARGET",
+	    "AIC7XXX",
+	    "HISAX",
+            "ISDN_DRV",
+	    "RADIO",
+	    "VIDEO",
+	    "REISERFS",
+	    "NCPFS",
+	    "FBCON",
+	    NULL,
+	};
+	while ((name = array[i++]) != NULL) {
+	    char *realname = strdup (name);
+	    strcpy (s, name);
+	    name = s;
+	    names[ucLen] = realname;
+	    compress (name, names, ucLen);
+	    printf ("PROCONF_KEY( %s , %lu )\n", name, ucLen);
+            // the zero means it will be ignored by the module
+	    tokens[ucLen++] = 0;
+	}
+    } while (0);
+
+    do {
+	int n, m;
+	char *err = fgets (s, MAXLINELENGTH, stdin);
+	char *t;
+	char *name, *data, *realname;
+	int ndata;
+	if (!err)
+	    break;
+	n = strlen (s);
+	// FIXME. Skip white space here
+
+	// sanity check. String begins with ·CONFIG_"
+	if (strncmp (s, "CONFIG_", 7) != 0)
+	    continue;
+	// remove the "CONFIG_" from the buffer
+	memmove (s, s + 7, n - 7 + 1);
+	// sanity check. There is an "=" sign
+	t = strchr (s, '=');
+	if (!t)
+	    continue;
+	// replace the "=" sign by space
+	*t = ' ';
+	// split the buffer by white space and thus get the name part (before "=")
+	name = strtok (s, " ");
+	if (!name)
+	    continue;
+	// get the data part (after "=" sign)
+	data = strtok (NULL, " ");
+	// sanity check. There has to be a data part!
+	if (!data)
+	    continue;
+	m = strlen (data);
+	// remove trailing spaces instead of running strtok again (why?)
+	do {
+	    if (m <= 0)
+		break;
+	    if (!isspace (data[m - 1]))
+		break;
+	    data[--m] = 0;
+	} while (1);
+
+	// check for common data values
+	if (1) {
+	    int j;
+	    // see if we already have the data item
+	    for (j = 0; j < count; j++) {
+		if (strcmp (data, extras[j]) == 0)
+		    break;
+	    }
+	    if (j >= count) {
+		char *buf = strdup (data);
+		if (!buf)
+		    break;
+		// we didn't have it
+		ndata = count;
+		extras[count++] = buf;
+	    }
+	    else {
+		// we did have it
+		ndata = j;
+	    }
+
+	}
+	// save name
+	realname = strdup (name);
+	names[ucLen] = realname;
+	compress (name, names, ucLen);
+
+	printf ("PROCONF_KEY( %s , %lu )\n", name, ucLen);
+	tokens[ucLen++] = ndata;
+
+    } while (1);
+
+    printf ("PROCONF_KEY_END\n");
+    printf ("\n");
+
+    printf ("PROCONF_XDATA_BEGIN\n");
+    printf ("PROCONF_XDATA( n )\n");
+    printf ("PROCONF_XDATA( y )\n");
+    printf ("PROCONF_XDATA( m )\n");
+    for (i = 3; i < count; i++)
+	printf ("PROCONF_XDATA( %s )\n", extras[i]);
+    printf ("PROCONF_XDATA_END\n");
+    printf ("\n");
+
+    printf ("PROCONF_DATA_BEGIN\n");
+    for (i = 0; i < ucLen; i++) {
+	int i0, i1, i2;
+	int x = tokens[i];
+	i0 = x % 8;
+	x /= 8;
+	i1 = x % 8;
+	x /= 8;
+	i2 = x % 8;
+	/* yes, right, I don't want to take chances with formatting */
+	printf ("PROCONF_DATA( \"\\%c%c%c\" )\n", i2 + '0',
+		i1 + '0', i0 + '0');
+    }
+    printf ("PROCONF_DATA_END\n");
+    printf ("\n");
+
+    printf ("ucLen = %d;\n", ucLen);
+    return 0;
+
+}
