? build.log
? strict-aliasing.patch
? po/stamp-po
Index: xine-lib-1.1.2cvs20060606/m4/optimizations.m4
===================================================================
--- xine-lib-1.1.2cvs20060606.orig/m4/optimizations.m4
+++ xine-lib-1.1.2cvs20060606/m4/optimizations.m4
@@ -22,7 +22,6 @@ AC_DEFUN([AC_OPTIMIZATIONS], [
         dnl
         AC_TRY_CFLAGS("-mpreferred-stack-boundary=2",
             m_psb="-mpreferred-stack-boundary=2", m_psb="")
-        AC_TRY_CFLAGS("-fno-strict-aliasing", f_nsa="-fno-strict-aliasing", f_nsa="")
         AC_TRY_CFLAGS("-fschedule-insns2", f_si="-fschedule-insns2", f_si="")
         AC_TRY_CFLAGS("-mwide-multiply", m_wm="-mwide-multiply", m_wm="")
         dnl
@@ -104,7 +103,7 @@ AC_DEFUN([AC_OPTIMIZATIONS], [
             AC_TRY_CFLAGS("$sarchopt=athlon", k7cpu="athlon", k7cpu="i686")
 
             dnl add x86 specific gcc CFLAGS
-            CFLAGS="-O3 -pipe -fomit-frame-pointer $f_af $f_al $f_aj $m_wm $m_psb -fexpensive-optimizations $f_si $f_nsa -ffast-math $INLINE_FUNCTIONS $CFLAGS"
+            CFLAGS="-O3 -pipe -fomit-frame-pointer $f_af $f_al $f_aj $m_wm $m_psb -fexpensive-optimizations $f_si -ffast-math $INLINE_FUNCTIONS $CFLAGS"
 
             DEBUG_CFLAGS="-O $DEBUG_CFLAGS"
 
@@ -188,11 +187,11 @@ AC_DEFUN([AC_OPTIMIZATIONS], [
         DEBUG_CFLAGS="-O3 -mieee $DEBUG_CFLAGS"
         ;;
       *darwin*)
-        CFLAGS="-O3 -pipe -fomit-frame-pointer $m_wm $m_psb -fexpensive-optimizations $f_si $f_nsa -ffast-math $INLINE_FUNCTIONS -no-cpp-precomp -D_INTL_REDIRECT_MACROS $CFLAGS"
+        CFLAGS="-O3 -pipe -fomit-frame-pointer $m_wm $m_psb -fexpensive-optimizations $f_si -ffast-math $INLINE_FUNCTIONS -no-cpp-precomp -D_INTL_REDIRECT_MACROS $CFLAGS"
         DEBUG_CFLAGS="-O3 $DEBUG_CFLAGS"
         ;;
       ppc-*-linux* | powerpc-*)
-        CFLAGS="-O3 -pipe -fomit-frame-pointer $m_wm $m_psb -fexpensive-optimizations $f_si $f_nsa -ffast-math $INLINE_FUNCTIONS $CFLAGS"
+        CFLAGS="-O3 -pipe -fomit-frame-pointer $m_wm $m_psb -fexpensive-optimizations $f_si -ffast-math $INLINE_FUNCTIONS $CFLAGS"
         DEBUG_CFLAGS="-O3 $DEBUG_CFLAGS"
         ;;
       sparc*-*-linux*)
@@ -243,7 +242,7 @@ AC_DEFUN([AC_OPTIMIZATIONS], [
         DEBUG_CFLAGS="$cc_debug_cflags $DEBUG_CFLAGS"
         ;;
       x86_64-*)
-        CFLAGS="-O3 -fomit-frame-pointer $m_wm $m_psb -fexpensive-optimizations $f_si $f_nsa -ffast-math $INLINE_FUNCTIONS $CFLAGS"
+        CFLAGS="-O3 -fomit-frame-pointer $m_wm $m_psb -fexpensive-optimizations $f_si -ffast-math $INLINE_FUNCTIONS $CFLAGS"
         DEBUG_CFLAGS="-g $DEBUG_CFLAGS"
         ;;
       armv4l-*-linux*)
Index: xine-lib-1.1.2cvs20060606/src/demuxers/demux_mpeg_block.c
===================================================================
--- xine-lib-1.1.2cvs20060606.orig/src/demuxers/demux_mpeg_block.c
+++ xine-lib-1.1.2cvs20060606/src/demuxers/demux_mpeg_block.c
@@ -1387,7 +1387,7 @@ static demux_plugin_t *open_plugin (demu
   this->demux_plugin.get_optional_data = demux_mpeg_block_get_optional_data;
   this->demux_plugin.demux_class       = class_gen;
 
-  this->scratch    = xine_xmalloc_aligned (512, 4096, (void**) &this->scratch_base);
+  this->scratch    = xine_xmalloc_aligned (512, 4096, &this->scratch_base);
   this->status     = DEMUX_FINISHED;
 
   lprintf ("open_plugin:detection_method=%d\n",
Index: xine-lib-1.1.2cvs20060606/src/demuxers/demux_mpeg_pes.c
===================================================================
--- xine-lib-1.1.2cvs20060606.orig/src/demuxers/demux_mpeg_pes.c
+++ xine-lib-1.1.2cvs20060606/src/demuxers/demux_mpeg_pes.c
@@ -1495,7 +1495,7 @@ static demux_plugin_t *open_plugin (demu
   this->demux_plugin.get_optional_data = demux_mpeg_pes_get_optional_data;
   this->demux_plugin.demux_class       = class_gen;
 
-  this->scratch    = xine_xmalloc_aligned (512, 4096, (void**) &this->scratch_base);
+  this->scratch    = xine_xmalloc_aligned (512, 4096, &this->scratch_base);
   this->status     = DEMUX_FINISHED;
   /* Don't start demuxing stream until we see a program_stream_pack_header */
   /* We need to system header in order to identify is the stream is mpeg1 or mpeg2. */
Index: xine-lib-1.1.2cvs20060606/src/demuxers/Makefile.am
===================================================================
--- xine-lib-1.1.2cvs20060606.orig/src/demuxers/Makefile.am
+++ xine-lib-1.1.2cvs20060606/src/demuxers/Makefile.am
@@ -84,6 +84,7 @@ xineplug_dmx_qt_la_LDFLAGS = -avoid-vers
 
 xineplug_dmx_asf_la_SOURCES = demux_asf.c
 xineplug_dmx_asf_la_LIBADD = $(XINE_LIB)
+xineplug_dmx_asf_la_CFLAGS = $(AM_CFLAGS) -fno-strict-aliasing
 xineplug_dmx_asf_la_LDFLAGS = -avoid-version -module @XINE_PLUGIN_MIN_SYMS@
 
 xineplug_dmx_fli_la_SOURCES = demux_fli.c
@@ -144,6 +145,7 @@ xineplug_dmx_nsv_la_LIBADD = $(XINE_LIB)
 xineplug_dmx_nsv_la_LDFLAGS = -avoid-version -module @XINE_PLUGIN_MIN_SYMS@
 
 xineplug_dmx_matroska_la_SOURCES = demux_matroska.c ebml.c
+xineplug_dmx_matroska_la_CFLAGS = $(AM_CFLAGS) -fno-strict-aliasing
 xineplug_dmx_matroska_la_LIBADD = $(XINE_LIB) $(ZLIB_LIBS)
 xineplug_dmx_matroska_la_LDFLAGS = -avoid-version -module @XINE_PLUGIN_MIN_SYMS@
 
Index: xine-lib-1.1.2cvs20060606/src/dxr3/video_out_dxr3.c
===================================================================
--- xine-lib-1.1.2cvs20060606.orig/src/dxr3/video_out_dxr3.c
+++ xine-lib-1.1.2cvs20060606/src/dxr3/video_out_dxr3.c
@@ -734,7 +734,7 @@ static void dxr3_update_frame_format(vo_
       /* planar format, only base[0] */
       /* add one extra line for field swap stuff */
       frame->real_base[0] = xine_xmalloc_aligned(16, image_size + frame->vo_frame.pitches[0],
-        (void**)&frame->mem);
+        &frame->mem);
 
       /* don't use first line */
       frame->real_base[0] += frame->vo_frame.pitches[0];
@@ -762,7 +762,7 @@ static void dxr3_update_frame_format(vo_
 
       /* add one extra line for field swap stuff */
       frame->real_base[0] = xine_xmalloc_aligned(16, image_size_y + frame->vo_frame.pitches[0] +
-        image_size_u + image_size_v, (void**)&frame->mem);
+        image_size_u + image_size_v, &frame->mem);
 
       /* don't use first line */
       frame->real_base[0] += frame->vo_frame.pitches[0];
Index: xine-lib-1.1.2cvs20060606/src/input/libdvdnav/nav_print.c
===================================================================
--- xine-lib-1.1.2cvs20060606.orig/src/input/libdvdnav/nav_print.c
+++ xine-lib-1.1.2cvs20060606/src/input/libdvdnav/nav_print.c
@@ -65,7 +65,8 @@ static void navPrint_PCI_GI(pci_gi_t *pc
   printf("pci_gi:\n");
   printf("nv_pck_lbn    0x%08x\n", pci_gi->nv_pck_lbn);
   printf("vobu_cat      0x%04x\n", pci_gi->vobu_cat);
-  printf("vobu_uop_ctl  0x%08x\n", *(uint32_t*)&pci_gi->vobu_uop_ctl);
+/* This will break strict aliasing, better avoid as this seems to be useless
+  printf("vobu_uop_ctl  0x%08x\n", *(uint32_t*)&pci_gi->vobu_uop_ctl); */
   printf("vobu_s_ptm    0x%08x\n", pci_gi->vobu_s_ptm);
   printf("vobu_e_ptm    0x%08x\n", pci_gi->vobu_e_ptm);
   printf("vobu_se_e_ptm 0x%08x\n", pci_gi->vobu_se_e_ptm);
Index: xine-lib-1.1.2cvs20060606/src/libdts/Makefile.am
===================================================================
--- xine-lib-1.1.2cvs20060606.orig/src/libdts/Makefile.am
+++ xine-lib-1.1.2cvs20060606/src/libdts/Makefile.am
@@ -22,6 +22,7 @@ xineplug_decode_dts_la_LIBADD = $(XINE_L
 xineplug_decode_dts_la_CFLAGS = $(LIBDTS_CFLAGS)
 else
 xineplug_decode_dts_la_LIBADD = $(XINE_LIB)
+xineplug_decode_dts_la_CFLAGS = -fno-strict-aliasing
 endif
 
 noinst_HEADERS = bitstream.h dts.h dts_internal.h tables.h tables_adpcm.h \
Index: xine-lib-1.1.2cvs20060606/src/libfaad/Makefile.am
===================================================================
--- xine-lib-1.1.2cvs20060606.orig/src/libfaad/Makefile.am
+++ xine-lib-1.1.2cvs20060606/src/libfaad/Makefile.am
@@ -12,6 +12,7 @@ lib_LTLIBRARIES = $(faad_module)
 
 VPATH = @srcdir@:@srcdir@/codebook:
 
+xineplug_decode_faad_la_CFLAGS = $(AM_CFLAGS) -fno-strict-aliasing
 xineplug_decode_faad_la_SOURCES = \
 	bits.c \
 	cfft.c \
Index: xine-lib-1.1.2cvs20060606/src/libffmpeg/libavcodec/alpha/Makefile.am
===================================================================
--- xine-lib-1.1.2cvs20060606.orig/src/libffmpeg/libavcodec/alpha/Makefile.am
+++ xine-lib-1.1.2cvs20060606/src/libffmpeg/libavcodec/alpha/Makefile.am
@@ -1,6 +1,7 @@
 include $(top_srcdir)/misc/Makefile.common
 
 AM_CPPFLAGS = $(LIBFFMPEG_CPPFLAGS)
+AM_CFLAGS = -fno-strict-aliasing
 
 EXTRA_DIST = asm.h \
 	dsputil_alpha.c \
Index: xine-lib-1.1.2cvs20060606/src/libffmpeg/libavcodec/armv4l/Makefile.am
===================================================================
--- xine-lib-1.1.2cvs20060606.orig/src/libffmpeg/libavcodec/armv4l/Makefile.am
+++ xine-lib-1.1.2cvs20060606/src/libffmpeg/libavcodec/armv4l/Makefile.am
@@ -1,6 +1,6 @@
 include $(top_srcdir)/misc/Makefile.common
 
-AM_CFLAGS = -O2
+AM_CFLAGS = -O2 -fno-strict-aliasing
 AM_CPPFLAGS = $(LIBFFMPEG_CPPFLAGS)
 ASFLAGS =
 
Index: xine-lib-1.1.2cvs20060606/src/libffmpeg/libavcodec/i386/Makefile.am
===================================================================
--- xine-lib-1.1.2cvs20060606.orig/src/libffmpeg/libavcodec/i386/Makefile.am
+++ xine-lib-1.1.2cvs20060606/src/libffmpeg/libavcodec/i386/Makefile.am
@@ -2,7 +2,7 @@ include $(top_srcdir)/misc/Makefile.comm
 
 # -fomit-frame-pointer is always needed. it might cause debug to not
 # work, but at least it compiles.
-AM_CFLAGS = -fomit-frame-pointer
+AM_CFLAGS = -fomit-frame-pointer -fno-strict-aliasing
 # CFLAGS is here to filter out -funroll-loops because it causes bad
 # behavior of libavcodec
 CFLAGS := `echo @CFLAGS@ | sed -e 's/-funroll-loops//g'`
Index: xine-lib-1.1.2cvs20060606/src/libffmpeg/libavcodec/libpostproc/Makefile.am
===================================================================
--- xine-lib-1.1.2cvs20060606.orig/src/libffmpeg/libavcodec/libpostproc/Makefile.am
+++ xine-lib-1.1.2cvs20060606/src/libffmpeg/libavcodec/libpostproc/Makefile.am
@@ -2,7 +2,7 @@ include $(top_srcdir)/misc/Makefile.comm
 
 # -fomit-frame-pointer is always needed. it might cause debug to not
 # work, but at least it compiles.
-AM_CFLAGS = -fomit-frame-pointer
+AM_CFLAGS = -fomit-frame-pointer -fno-strict-aliasing
 AM_CPPFLAGS = $(LIBFFMPEG_CPPFLAGS) -I$(top_srcdir)/src/libffmpeg/libavcodec  
 ASFLAGS =
 
Index: xine-lib-1.1.2cvs20060606/src/libffmpeg/libavcodec/Makefile.am
===================================================================
--- xine-lib-1.1.2cvs20060606.orig/src/libffmpeg/libavcodec/Makefile.am
+++ xine-lib-1.1.2cvs20060606/src/libffmpeg/libavcodec/Makefile.am
@@ -9,7 +9,7 @@ EXTRA_DIST = motion_est_template.c \
 # we need to compile everything in debug mode, including the encoders,
 # otherwise we get unresolved symbols, because some unsatisfied function calls
 # are not optimized away with debug optimization
-AM_CFLAGS = `test "$(CFLAGS)" = "$(DEBUG_CFLAGS)" && echo -DCONFIG_ENCODERS`
+AM_CFLAGS = `test "$(CFLAGS)" = "$(DEBUG_CFLAGS)" && echo -DCONFIG_ENCODERS` -fno-strict-aliasing
 AM_CPPFLAGS = $(ZLIB_CPPFLAGS) $(LIBFFMPEG_CPPFLAGS) \
 	-I$(top_srcdir)/src/libffmpeg/libavutil
 ASFLAGS =
Index: xine-lib-1.1.2cvs20060606/src/libffmpeg/libavcodec/mlib/Makefile.am
===================================================================
--- xine-lib-1.1.2cvs20060606.orig/src/libffmpeg/libavcodec/mlib/Makefile.am
+++ xine-lib-1.1.2cvs20060606/src/libffmpeg/libavcodec/mlib/Makefile.am
@@ -1,6 +1,7 @@
 include $(top_srcdir)/misc/Makefile.common
 
 AM_CPPFLAGS = $(LIBFFMPEG_CPPFLAGS)
+AM_CFLAGS = -fno-strict-aliasing
 ASFLAGS =
 
 noinst_LTLIBRARIES = libavcodec_mlib.la
Index: xine-lib-1.1.2cvs20060606/src/libffmpeg/libavcodec/ppc/Makefile.am
===================================================================
--- xine-lib-1.1.2cvs20060606.orig/src/libffmpeg/libavcodec/ppc/Makefile.am
+++ xine-lib-1.1.2cvs20060606/src/libffmpeg/libavcodec/ppc/Makefile.am
@@ -1,6 +1,7 @@
 include $(top_srcdir)/misc/Makefile.common
 
 AM_CPPFLAGS = $(LIBFFMPEG_CPPFLAGS)
+AM_CFLAGS = -fno-strict-aliasing
 # CFLAGS is here to filter out -funroll-loops because it causes bad
 # behavior of libavcodec
 CFLAGS = `echo @CFLAGS@ | sed -e 's/-funroll-loops//g'`
Index: xine-lib-1.1.2cvs20060606/src/libffmpeg/libavcodec/sparc/Makefile.am
===================================================================
--- xine-lib-1.1.2cvs20060606.orig/src/libffmpeg/libavcodec/sparc/Makefile.am
+++ xine-lib-1.1.2cvs20060606/src/libffmpeg/libavcodec/sparc/Makefile.am
@@ -1,6 +1,7 @@
 include $(top_srcdir)/misc/Makefile.common
 
 AM_CPPFLAGS = $(LIBFFMPEG_CPPFLAGS) -I$(top_srcdir)/src/libffmpeg/libavutil
+AM_CFLAGS = -fno-strict-aliasing
 ASFLAGS =
 
 noinst_LTLIBRARIES = libavcodec_sparc.la
Index: xine-lib-1.1.2cvs20060606/src/libffmpeg/libavutil/Makefile.am
===================================================================
--- xine-lib-1.1.2cvs20060606.orig/src/libffmpeg/libavutil/Makefile.am
+++ xine-lib-1.1.2cvs20060606/src/libffmpeg/libavutil/Makefile.am
@@ -1,6 +1,7 @@
 include $(top_srcdir)/misc/Makefile.common
 
 AM_CPPFLAGS = $(LIBFFMPEG_CPPFLAGS)
+AM_CFLAGS = -fno-strict-aliasing
 ASFLAGS =
 
 noinst_LTLIBRARIES = libavutil.la
Index: xine-lib-1.1.2cvs20060606/src/libmpeg2/decode.c
===================================================================
--- xine-lib-1.1.2cvs20060606.orig/src/libmpeg2/decode.c
+++ xine-lib-1.1.2cvs20060606/src/libmpeg2/decode.c
@@ -69,10 +69,10 @@ void mpeg2_init (mpeg2dec_t * mpeg2dec, 
 
     if( !mpeg2dec->chunk_buffer )
       mpeg2dec->chunk_buffer = xine_xmalloc_aligned (16, BUFFER_SIZE + 4, 
-						     (void**)&mpeg2dec->chunk_base);
+						     &mpeg2dec->chunk_base);
     if( !mpeg2dec->picture )
       mpeg2dec->picture = xine_xmalloc_aligned (16, sizeof (picture_t),
-						(void**)&mpeg2dec->picture_base);
+						&mpeg2dec->picture_base);
 
     mpeg2dec->shift = 0xffffff00;
     mpeg2dec->new_sequence = 0;
Index: xine-lib-1.1.2cvs20060606/src/libspucc/Makefile.am
===================================================================
--- xine-lib-1.1.2cvs20060606.orig/src/libspucc/Makefile.am
+++ xine-lib-1.1.2cvs20060606/src/libspucc/Makefile.am
@@ -5,6 +5,7 @@ libdir = $(XINE_PLUGINDIR)
 lib_LTLIBRARIES = xineplug_decode_spucc.la
 
 xineplug_decode_spucc_la_SOURCES =  cc_decoder.c xine_decoder.c
+xineplug_decode_spucc_la_CFLAGS = $(AM_CFLAGS) -fno-strict-aliasing
 xineplug_decode_spucc_la_LIBADD = $(XINE_LIB)
 xineplug_decode_spucc_la_LDFLAGS = -avoid-version -module @XINE_PLUGIN_MIN_SYMS@
 
Index: xine-lib-1.1.2cvs20060606/src/libxineadec/Makefile.am
===================================================================
--- xine-lib-1.1.2cvs20060606.orig/src/libxineadec/Makefile.am
+++ xine-lib-1.1.2cvs20060606/src/libxineadec/Makefile.am
@@ -19,7 +19,7 @@ xineplug_decode_gsm610_la_LIBADD = \
 	$(top_builddir)/src/libxineadec/gsm610/libgsm610.la
 
 xineplug_decode_nsf_la_SOURCES = nsf.c
-#xineplug_decode_nsf_la_CFLAGS = -DNSF_PLAYER
+xineplug_decode_nsf_la_CFLAGS = -DNSF_PLAYER -fno-strict-aliasing
 xineplug_decode_nsf_la_LDFLAGS = -avoid-version -module @XINE_PLUGIN_MIN_SYMS@
 xineplug_decode_nsf_la_LIBADD = \
 	$(XINE_LIB) \
Index: xine-lib-1.1.2cvs20060606/src/libxineadec/nosefart/Makefile.am
===================================================================
--- xine-lib-1.1.2cvs20060606.orig/src/libxineadec/nosefart/Makefile.am
+++ xine-lib-1.1.2cvs20060606/src/libxineadec/nosefart/Makefile.am
@@ -1,6 +1,6 @@
 include $(top_srcdir)/misc/Makefile.common
 
-AM_CFLAGS = -DNSF_PLAYER
+AM_CFLAGS = -DNSF_PLAYER -fno-strict-aliasing
 
 noinst_LTLIBRARIES = libnosefart.la
 
Index: xine-lib-1.1.2cvs20060606/src/video_out/Makefile.am
===================================================================
--- xine-lib-1.1.2cvs20060606.orig/src/video_out/Makefile.am
+++ xine-lib-1.1.2cvs20060606/src/video_out/Makefile.am
@@ -11,12 +11,13 @@ VIDIX_CFLAGS = -I$(top_builddir)/src/vid
 AM_CFLAGS = $(X_CFLAGS) -DXINE_COMPILE \
         $(SUNDGA_CFLAGS) $(SDL_CFLAGS) $(VIDIX_CFLAGS) \
 	$(AALIB_CFLAGS) $(MLIB_CFLAGS) $(LIBSTK_CFLAGS) \
-	$(DIRECTFB_CFLAGS) $(CACA_CFLAGS)
+	$(DIRECTFB_CFLAGS) $(CACA_CFLAGS) $(X11OSD_STRICT_ALIASING)
 
 libdir = $(XINE_PLUGINDIR)
 
 if HAVE_X11
 X11OSD = x11osd.c
+X11OSD_STRICT_ALIASING = -fno-strict-aliasing
 xshm_module = xineplug_vo_out_xshm.la
 if HAVE_XV
 xv_module = xineplug_vo_out_xv.la
Index: xine-lib-1.1.2cvs20060606/src/xine-engine/Makefile.am
===================================================================
--- xine-lib-1.1.2cvs20060606.orig/src/xine-engine/Makefile.am
+++ xine-lib-1.1.2cvs20060606/src/xine-engine/Makefile.am
@@ -1,7 +1,7 @@
 include $(top_srcdir)/misc/Makefile.common
 include $(top_srcdir)/lib/Makefile.common
 
-AM_CFLAGS = $(X_CFLAGS) $(FT2_CFLAGS)
+AM_CFLAGS = $(X_CFLAGS) $(FT2_CFLAGS) -fno-strict-aliasing
 AM_CPPFLAGS = $(ZLIB_CPPFLAGS) -DXINE_LIBRARY_COMPILE
 
 LIBTOOL = $(SHELL) $(top_builddir)/libtool
