Index: xine-lib-1.1.2cvs20060606/src/input/input_http.c
===================================================================
--- xine-lib-1.1.2cvs20060606.orig/src/input/input_http.c
+++ xine-lib-1.1.2cvs20060606/src/input/input_http.c
@@ -900,10 +900,10 @@ static int http_plugin_open (input_plugi
 	len = 0;
     } else
       len ++;
-    if ( len >= buflen ) {
+    if ( len >= BUFSIZE ) {
        _x_message(this->stream, XINE_MSG_PERMISSION_ERROR, this->mrl, NULL);
        xine_log (this->stream->xine, XINE_LOG_MSG,
-         _("input_http: buffer exhausted after %d bytes."), buflen);
+         _("input_http: buffer exhausted after %d bytes."), BUFSIZE);
        return 0;
     }
   }
