Index: xine-lib-1.1.2cvs20060606/configure.ac
===================================================================
--- xine-lib-1.1.2cvs20060606.orig/configure.ac
+++ xine-lib-1.1.2cvs20060606/configure.ac
@@ -163,6 +163,8 @@ AM_ICONV
 AM_GNU_GETTEXT([external])
 AC_PROG_GMSGFMT_PLURAL
 
+AC_CHECK_FUNCS([nl_langinfo])
+
 dnl ---------------------------------------------
 dnl Checks for typedefs, structures, and compiler characteristics.
 dnl ---------------------------------------------
Index: xine-lib-1.1.2cvs20060606/src/xine-utils/utils.c
===================================================================
--- xine-lib-1.1.2cvs20060606.orig/src/xine-utils/utils.c
+++ xine-lib-1.1.2cvs20060606/src/xine-utils/utils.c
@@ -50,7 +50,7 @@
 #include <ucontext.h>
 #endif
 
-#ifdef HAVE_LANGINFO_CODESET
+#ifdef HAVE_NL_LANGINFO
 #include <langinfo.h>
 #endif
 
@@ -531,7 +531,7 @@ static char *_get_lang(void) {
 char *xine_get_system_encoding(void) {
   char *codeset = NULL;
   
-#ifdef HAVE_LANGINFO_CODESET
+#ifdef HAVE_NL_LANGINFO
   codeset = nl_langinfo(CODESET);
 #endif
   /*
Index: xine-lib-1.1.2cvs20060606/src/input/mms.c
===================================================================
--- xine-lib-1.1.2cvs20060606.orig/src/input/mms.c
+++ xine-lib-1.1.2cvs20060606/src/input/mms.c
@@ -45,7 +45,7 @@
 #include <stdlib.h>
 #include <time.h>
 
-#if defined(HAVE_ICONV) && defined(HAVE_LANGINFO_CODESET)
+#if defined(HAVE_ICONV) && defined(HAVE_NL_LANGINFO)
 #define USE_ICONV
 #include <iconv.h>
 #include <locale.h>
Index: xine-lib-1.1.2cvs20060606/src/input/vcd/libcdio/iso9660_fs.c
===================================================================
--- xine-lib-1.1.2cvs20060606.orig/src/input/vcd/libcdio/iso9660_fs.c
+++ xine-lib-1.1.2cvs20060606/src/input/vcd/libcdio/iso9660_fs.c
@@ -35,7 +35,7 @@
 # include <iconv.h>
 #endif
 
-#ifdef HAVE_LANGINFO_CODESET
+#ifdef HAVE_NL_LANGINFO
 #include <langinfo.h>
 #endif
 
@@ -148,7 +148,7 @@ ucs2be_to_locale(ICONV_CONST char *psz_u
 {
 
   iconv_t ic = 
-#if defined(HAVE_LANGINFO_CODESET)
+#if defined(HAVE_NL_LANGINFO)
     iconv_open(nl_langinfo(CODESET), "UCS-2BE");
 #else 
     iconv_open("ASCII", "UCS-2BE");
@@ -161,7 +161,7 @@ ucs2be_to_locale(ICONV_CONST char *psz_u
   int i_outlen_actual;
 
   if (-1 == (size_t) ic) {
-#if defined(HAVE_LANGINFO_CODESET)
+#if defined(HAVE_NL_LANGINFO)
     cdio_info("Failed to get conversion table for locale, trying ASCII");
     ic = iconv_open("ASCII", "UCS-2BE");
     if (-1 == (size_t) ic) {
