Index: xine-lib-1.1.2cvs20060606/configure.ac
===================================================================
--- xine-lib-1.1.2cvs20060606.orig/configure.ac
+++ xine-lib-1.1.2cvs20060606/configure.ac
@@ -85,7 +85,7 @@ AC_SUBST(SPEC_VERSION)
 LIBNAME=libxine$XINE_MAJOR
 AC_SUBST(LIBNAME)
 
-AC_DEFINE_UNQUOTED(XINE_TEXTDOMAIN, "$LIBNAME", [catalog message text domain])
+AC_DEFINE_UNQUOTED(XINE_TEXTDOMAIN, "xine-lib", [catalog message text domain])
 
 dnl
 AC_CANONICAL_TARGET
Index: xine-lib-1.1.2cvs20060606/src/input/vcd/vcdplayer.h
===================================================================
--- xine-lib-1.1.2cvs20060606.orig/src/input/vcd/vcdplayer.h
+++ xine-lib-1.1.2cvs20060606/src/input/vcd/vcdplayer.h
@@ -33,7 +33,7 @@
 #ifdef ENABLE_NLS
 #include <locale.h>
 #    include <libintl.h>
-#    define _(String) dgettext ("libxine1", String)
+#    define _(String) dgettext ("xine-lib", String)
 #else
 /* Stubs that do something close enough.  */
 #    define _(String) (String)
