? po/remove-potcdate.sed
? po/stamp-po
Index: src/input/libdvdnav/dvd_reader.c
===================================================================
RCS file: /cvsroot/xine/xine-lib/src/input/libdvdnav/dvd_reader.c,v
retrieving revision 1.12
diff -u -B -r1.12 dvd_reader.c
--- src/input/libdvdnav/dvd_reader.c	27 May 2006 16:13:17 -0000	1.12
+++ src/input/libdvdnav/dvd_reader.c	15 Jun 2006 14:24:00 -0000
@@ -482,7 +482,7 @@
                     auth_drive = DVDOpenImageFile( me->mnt_fsname, have_css );
 		    /* If the device is not encrypted, don't access the device
 		     * directly as it would fail for non-UDF DVDs */
-		    if ( dvdinput_is_encrypted( auth_drive->dev ) == 0) {
+		    if ( auth_drive && dvdinput_is_encrypted( auth_drive->dev ) == 0) {
 		      DVDClose( auth_drive );
 		      auth_drive = NULL;
 		      break;
