? po/remove-potcdate.sed
? po/stamp-po
Index: xine-lib-1.1.2cvs20060606/src/input/libdvdnav/dvd_reader.c
===================================================================
--- xine-lib-1.1.2cvs20060606.orig/src/input/libdvdnav/dvd_reader.c
+++ xine-lib-1.1.2cvs20060606/src/input/libdvdnav/dvd_reader.c
@@ -482,7 +482,7 @@ dvd_reader_t *DVDOpen( const char *ppath
                     auth_drive = DVDOpenImageFile( me->mnt_fsname, have_css );
 		    /* If the device is not encrypted, don't access the device
 		     * directly as it would fail for non-UDF DVDs */
-		    if ( dvdinput_is_encrypted( auth_drive->dev ) == 0) {
+		    if ( auth_drive && dvdinput_is_encrypted( auth_drive->dev ) == 0) {
 		      DVDClose( auth_drive );
 		      auth_drive = NULL;
 		      break;
