--- libao-0.8.6/configure.in	2005-01-11 15:37:15 -0600
+++ libao-0.8.6-new/configure.in	2005-08-01 18:19:21 -0500
@@ -72,9 +72,10 @@
                 PROFILE="-pg -g -O20 -D__NO_MATH_INLINES -fsigned-char -mv8" ;;
         *-darwin*)
                 PLUGIN_LDFLAGS="-module -avoid-version"
-                DEBUG="-g -Wall -D__NO_MATH_INLINES -fsigned-char -Ddlsym=dlsym_auto_underscore"
-                CFLAGS="-D__NO_MATH_INLINES -fsigned-char -Ddlsym=dlsym_auto_underscore"
-                PROFILE="-g -pg -D__NO_MATH_INLINES -fsigned-char -Ddlsym=dlsym_auto_underscore" ;;
+                DEBUG="-g -Wall -D__NO_MATH_INLINES -fsigned-char"
+                CFLAGS="-D__NO_MATH_INLINES -fsigned-char "
+                PROFILE="-g -pg -D__NO_MATH_INLINES -fsigned-char"
+	        LIBS="-Wl,-framework -Wl,CoreAudio" ;;
         *)
                 PLUGIN_LDFLAGS="-export-dynamic -avoid-version"
                 DEBUG="-g -Wall -D__NO_MATH_INLINES -fsigned-char"
@@ -111,7 +112,7 @@
 	DLOPEN_FLAG='(RTLD_LAZY)'
 	SHARED_LIB_EXT='.sl'
 	;;    
-    *openbsd* | *netbsd* | *solaris2.7)
+    *openbsd* | *netbsd* | *solaris2.7 | *darwin* | *freebsd*)
 	DLOPEN_FLAG='(RTLD_LAZY)'
 	SHARED_LIB_EXT='.so'
 	;;
@@ -365,5 +369,6 @@
 dnl Plugins get special LDFLAGS
 AC_SUBST(PLUGIN_LDFLAGS)
 
+AM_DL()
 
 AC_OUTPUT(Makefile src/Makefile doc/Makefile include/Makefile include/ao/Makefile include/ao/os_types.h src/plugins/Makefile src/plugins/esd/Makefile src/plugins/oss/Makefile src/plugins/alsa/Makefile src/plugins/alsa09/Makefile src/plugins/sun/Makefile src/plugins/irix/Makefile src/plugins/arts/Makefile src/plugins/macosx/Makefile src/plugins/nas/Makefile src/plugins/polyp/Makefile debian/Makefile ao.pc)
Only in libao-0.8.6: m4
diff -x '*~' -ur libao-0.8.6/src/Makefile.am libao-0.8.6-fbsd/src/Makefile.am
--- libao-0.8.6/src/Makefile.am	2004-11-09 09:20:26.000000000 +0100
+++ libao-0.8.6-fbsd/src/Makefile.am	2005-04-29 17:11:08.581865400 +0200
@@ -8,7 +8,7 @@
 lib_LTLIBRARIES = libao.la
 
 libao_la_SOURCES = audio_out.c config.c ao_null.c ao_wav.c ao_au.c ao_raw.c ao_aixs.c ao_private.h
-libao_la_LDFLAGS = -version-info @LIB_CURRENT@:@LIB_REVISION@:@LIB_AGE@ -ldl
+libao_la_LDFLAGS = -version-info @LIB_CURRENT@:@LIB_REVISION@:@LIB_AGE@ @DYNAMIC_LD_LIBS@
 
 
 debug:
