make bundled libtool code bearable

--- jpeg/configure
+++ jpeg/configure
@@ -1559,7 +1559,7 @@
   if test "x$LTSTATIC" = xno; then
     disable_static="--disable-static"
   fi
-  $srcdir/ltconfig $disable_shared $disable_static $srcdir/ltmain.sh
+  $srcdir/ltconfig $disable_shared $disable_static $srcdir/ltmain.sh $CHOST
 fi
 
 # Select memory manager depending on user input.
--- jpeg/ltconfig
+++ jpeg/ltconfig
@@ -553,7 +553,9 @@
     # On HP-UX, both CC and GCC only warn that PIC is supported... then they
     # create non-PIC objects.  So, if there were any warnings, we assume that
     # PIC is not supported.
+    # Make sure we only test warnings on HP-UX (pic_flag == +Z) or we can
+    # easily break Linux builds http://bugs.gentoo.org/70947
-    if test -s conftest.err; then
+    if test -s conftest.err -a "$pic_flag" = "+Z"; then
       echo "$ac_t"no 1>&6
       can_build_shared=no
       pic_flag=
@@ -716,7 +716,7 @@
 
   # See if GNU ld supports shared libraries.
   if $LD --help 2>&1 | egrep ': supported targets:.* elf' > /dev/null; then
-    archive_cmds='$CC -shared ${wl}-soname $wl$soname -o $lib$libobjs'
+    archive_cmds='$CC $LDFLAGS -shared ${wl}-soname $wl$soname -o $lib$libobjs'
     runpath_var=LD_RUN_PATH
     ld_shlibs=yes
   else
@@ -1210,7 +1210,6 @@
   else
     # Only the GNU ld.so supports shared libraries on MkLinux.
     case "$host_cpu" in
-    powerpc*) dynamic_linker=no ;;
     *) dynamic_linker='Linux ld.so' ;;
     esac
   fi
