# https://bugs.gentoo.org/show_bug.cgi?id=165777

--- mozilla.orig/xpcom/reflect/xptcall/src/md/unix/Makefile.in.orig	2006-12-02 21:30:12.000000000 +0000
+++ mozilla/xpcom/reflect/xptcall/src/md/unix/Makefile.in	2006-12-02 21:46:11.000000000 +0000
@@ -281,9 +281,15 @@
 AS		:= $(CC) -c -x assembler-with-cpp
 endif
 ifeq ($(OS_ARCH)$(OS_TEST),Linuxppc64)
+ifeq ($(USE_64),1)
 CPPSRCS		:= xptcinvoke_ppc64_linux.cpp xptcstubs_ppc64_linux.cpp
 ASFILES		:= xptcinvoke_asm_ppc64_linux.s xptcstubs_asm_ppc64_linux.s
 AS		:= $(CC) -c -x assembler-with-cpp
+else
+CPPSRCS         := xptcinvoke_ppc_linux.cpp xptcstubs_ppc_linux.cpp
+ASFILES         := xptcinvoke_asm_ppc_linux.s xptcstubs_asm_ppc_linux.s
+AS              := $(CC) -c -x assembler-with-cpp
+endif
 endif
 
 #
