# Upstream https://bugzilla.mozilla.org/show_bug.cgi?id=312998

--- mozilla/embedding/browser/gtk/src/EmbedWindow.cpp.visible	2006-09-20 14:35:58.000000000 -0400
+++ mozilla/embedding/browser/gtk/src/EmbedWindow.cpp	2006-09-20 14:36:14.000000000 -0400
@@ -359,7 +359,13 @@
 NS_IMETHODIMP
 EmbedWindow::GetVisibility(PRBool *aVisibility)
 {
-  *aVisibility = mVisibility;
+  // Work around the problem that sometimes the window
+  // is already visible even though mVisibility isn't true
+  // yet.
+  *aVisibility = mVisibility ||
+                 (!mOwner->mIsChrome &&
+                  mOwner->mOwningWidget &&
+                  GTK_WIDGET_MAPPED(mOwner->mOwningWidget));
   return NS_OK;
 }
