--- tools/regression/src/compiler_status.cpp.orig	2007-11-15 17:31:06.000000000 +0100
+++ tools/regression/src/compiler_status.cpp	2008-08-20 15:22:17.000000000 +0200
@@ -789,7 +789,7 @@
     // - Boost.Build V2 location with top-lelve "build-dir" 
     // - Boost.Build V1 location without ALL_LOCATE_TARGET
     string relative( fs::initial_path().string() );
-    relative.erase( 0, boost_root.string().size()+1 );    
+    relative.erase( 0, boost_root.string().size() );    
     fs::path bin_path = find_bin_path(relative);
 
     report << "<table border=\"1\" cellspacing=\"0\" cellpadding=\"5\">\n";
