--- configure.ac	2008-11-09 21:10:40.000000000 +0100
+++ configure.ac.new	2008-11-09 21:12:18.000000000 +0100
@@ -2,7 +2,7 @@
 # Process this file with autoconf to produce a configure script.
 
 AC_PREREQ(2.57)
-AC_INIT(tslib, 0.0.2, kergoth@handhelds.org)
+AC_INIT(tslib, 1.0, kergoth@handhelds.org)
 # AC_CONFIG_AUX_DIR(config)
 AM_INIT_AUTOMAKE(dist-bzip2)
 AC_CONFIG_SRCDIR([src/ts_close.c])
@@ -43,7 +43,6 @@
 
 # Checks for library functions.
 AC_PROG_GCC_TRADITIONAL
-AC_FUNC_MALLOC
 AC_FUNC_MMAP
 AC_FUNC_SELECT_ARGTYPES
 AC_TYPE_SIGNAL
@@ -193,7 +192,7 @@
 AC_SUBST(TS_CONF)
 
 # Library versioning
-LT_RELEASE=0.0
+LT_RELEASE=1.0
 LT_CURRENT=1
 LT_REVISION=1
 LT_AGE=1
