--- configure.ac.orig	2010-02-08 10:41:57.000000000 -0800
+++ configure.ac	2010-02-08 11:20:52.000000000 -0800
@@ -206,5 +206,6 @@
                  plugins/Makefile
                  src/Makefile
                  tests/Makefile
+                 tslib.pc:tslib.pc.in
                  tslib-$LT_RELEASE.pc:tslib.pc.in])
 AC_OUTPUT
--- Makefile.am.orig	2006-08-24 14:02:55.000000000 -0700
+++ Makefile.am	2010-02-08 11:51:46.000000000 -0800
@@ -16,4 +16,4 @@
 AUTOMAKE_OPTIONS = 1.6
 
 pkgconfigdir = $(libdir)/pkgconfig
-pkgconfig_DATA = $(PACKAGE)-$(LT_RELEASE).pc
+pkgconfig_DATA = $(PACKAGE)-$(LT_RELEASE).pc $(PACKAGE).pc
