From 2d9cc62265dd1ac18a8e619e3a637dec05fc2e4d Mon Sep 17 00:00:00 2001
From: Michael Hennerich <michael.hennerich@analog.com>
Date: Sun, 14 Nov 2010 15:03:44 -0500
Subject: [PATCH] fbutils: add support for RGB24

Signed-off-by: Michael Hennerich <michael.hennerich@analog.com>
Signed-off-by: Mike Frysinger <vapier@gentoo.org>
---
 tests/fbutils.c |   12 ++++++++++++
 1 files changed, 12 insertions(+), 0 deletions(-)

diff --git a/tests/fbutils.c b/tests/fbutils.c
index ec267f7..fd7a0ed 100644
--- a/tests/fbutils.c
+++ b/tests/fbutils.c
@@ -249,6 +249,7 @@ void setcolor(unsigned colidx, unsigned value)
         	        perror("ioctl FBIOPUTCMAP");
 		break;
 	case 2:
+	case 3:
 	case 4:
 		red = (value >> 16) & 0xff;
 		green = (value >> 8) & 0xff;
@@ -276,6 +277,17 @@ static inline void __setpixel (union multiptr loc, unsigned xormode, unsigned co
 		else
 			*loc.p16 = color;
 		break;
+	case 3:
+		if (xormode) {
+			*loc.p8++ ^= (color >> 16) & 0xff;
+			*loc.p8++ ^= (color >> 8) & 0xff;
+			*loc.p8 ^= color & 0xff;
+		} else {
+			*loc.p8++ = (color >> 16) & 0xff;
+			*loc.p8++ = (color >> 8) & 0xff;
+			*loc.p8 = color & 0xff;
+		}
+		break;
 	case 4:
 		if (xormode)
 			*loc.p32 ^= color;
-- 
1.7.8.4

