Index: vlc-0.8.6-test2/mozilla/vlcshell.cpp
===================================================================
--- vlc-0.8.6-test2.orig/mozilla/vlcshell.cpp
+++ vlc-0.8.6-test2/mozilla/vlcshell.cpp
@@ -35,10 +35,8 @@
 #   include <mozilla-config.h>
 #endif
 
-/* This is from mozilla java, do we really need it? */
-#if 0
+/* This is from mozilla java, needed for jref type */
 #include <jri.h>
-#endif
 
 #include "vlcplugin.h"
 
