https://bugzilla.mozilla.org/show_bug.cgi?id=356609


--- mozilla/xpcom/reflect/xptcall/src/md/unix/xptcstubs_arm.cpp~	2006-06-20 11:10:37.000000000 +0100
+++ mozilla/xpcom/reflect/xptcall/src/md/unix/xptcstubs_arm.cpp	2006-06-20 11:10:37.000000000 +0100
@@ -45,7 +45,7 @@
 #endif
 
 /* Specify explicitly a symbol for this function, don't try to guess the c++ mangled symbol.  */
-static nsresult PrepareAndDispatch(nsXPTCStubBase* self, uint32 methodIndex, PRUint32* args) asm("_PrepareAndDispatch");
+static nsresult PrepareAndDispatch(nsXPTCStubBase* self, uint32 methodIndex, PRUint32* args) asm("_PrepareAndDispatch") __attribute__((used));
 
 static nsresult
 PrepareAndDispatch(nsXPTCStubBase* self, uint32 methodIndex, PRUint32* args)
