--- resin-4.0.26.vanilla/modules/c/src/apache2/Makefile.in	2012-03-04 11:33:18.455432143 +0100
+++ resin-4.0.26/modules/c/src/apache2/Makefile.in	2012-03-04 11:43:23.381046283 +0100
@@ -27,7 +27,7 @@
 #	mv mod_caucho.la mod_caucho.so
 
 mod_caucho.la	: $(OBJ_PICS)
-	$(LIBTOOL) --mode=link $(LD_SHLIB) $(LDFLAGS_SHLIB) -module -o $@ -avoid-version -rpath $(apache_libexec) $(OBJ_PICS) $(LIBS_SHLIB)
+	$(LIBTOOL) --mode=link $(LD_SHLIB) $(LDFLAGS) $(LDFLAGS_SHLIB) -module -o $@ -avoid-version -rpath $(apache_libexec) $(OBJ_PICS) $(LIBS_SHLIB)
 
 .c.lo	:
 	$(LIBTOOL) --mode=compile $(CC) -o $@ -c $(INCLUDES) $(CFLAGS) $<
