--- resin-4.0.26.vanilla/configure.ac	2012-03-04 11:33:20.347437179 +0100
+++ resin-4.0.26/configure.ac	2012-03-04 11:46:18.913514661 +0100
@@ -783,16 +783,16 @@
     linux*)
       LD_SHLIB=${CC}
 
-      CFLAGS="$CFLAGS -pthread -fPIC -fno-omit-frame-pointer -O2"
+      CFLAGS="$CFLAGS -pthread -fPIC -fno-omit-frame-pointer"
       CFLAGS_JNI="$CFLAGS_JNI -fPIC"
       LDFLAGS_SHLIB="-shared -fPIC"
 
       if test -n "$b64_jni"; then
-        CFLAGS_JNI="$CFLAGS_JNI -m64 -fPIC"
-        LDFLAGS_SHLIB="$LDFLAGS_SHLIB -m64"
+        CFLAGS_JNI="$CFLAGS_JNI -fPIC"
+        LDFLAGS_SHLIB="$LDFLAGS_SHLIB "
       else
-        CFLAGS_JNI="$CFLAGS_JNI -m32"
-        LDFLAGS_SHLIB="$LDFLAGS_SHLIB -m32"
+        CFLAGS_JNI="$CFLAGS_JNI "
+        LDFLAGS_SHLIB="$LDFLAGS_SHLIB "
       fi
       ;;
 
@@ -805,12 +805,12 @@
         CFLAGS_JNI="$CFLAGS_JNI -fPIC -fno-omit-frame-pointer"
 
         if test -n "$b64"; then
-          CFLAGS="$CFLAGS -m64"
-          LDFLAGS_SHLIB="-m64 -R /usr/sfw/lib/64 -R /usr/local/lib/sparcv9"
+          CFLAGS="$CFLAGS "
+          LDFLAGS_SHLIB=" -R /usr/sfw/lib/64 -R /usr/local/lib/sparcv9"
         fi
         if test -n "${b64_jni}"; then
-          CFLAGS_JNI="$CFLAGS_JNI -m64"
-          LDFLAGS_SHLIB="-m64 -R /usr/sfw/lib/64 -R /usr/local/lib/sparcv9"
+          CFLAGS_JNI="$CFLAGS_JNI "
+          LDFLAGS_SHLIB=" -R /usr/sfw/lib/64 -R /usr/local/lib/sparcv9"
         fi
 
         LDFLAGS_SHLIB="$LDFLAGS_SHLIB -shared"
@@ -819,11 +819,11 @@
         CFLAGS_SHLIB="-mt -KPIC -xO2"
 
         if test -n "$b64"; then
-          CFLAGS="$CFLAGS -m64"
+          CFLAGS="$CFLAGS "
         fi
         if test -n "$b64_jni"; then
-          CFLAGS_JNI="$CFLAGS_JNI -m64"
-          LDFLAGS_SHLIB="$LDFLAGS_SHLIB -m64 -G -mt -z defs -z text"
+          CFLAGS_JNI="$CFLAGS_JNI "
+          LDFLAGS_SHLIB="$LDFLAGS_SHLIB  -G -mt -z defs -z text"
           LDFLAGS_SHLIB="$LDFLAGS_SHLIB -R/usr/lib/64:/usr/sfw/lib/64:/usr/local/lib/sparcv9 -YP,/usr/lib/64:/usr/sfw/lib/64:/usr/local/lib/sparcv9"
         fi
 
@@ -840,7 +840,7 @@
         LDFLAGS_SHLIB="-arch x86_64 $LDFLAGS_SHLIB"
       fi
       if test -n "$b64_jni"; then
-        CFLAGS_JNI="$CFLAGS_JNI -m64"
+        CFLAGS_JNI="$CFLAGS_JNI "
         LDFLAGS_SHLIB="-arch x86_64 $LDFLAGS_SHLIB"
       fi
       ;;
@@ -871,8 +871,8 @@
 #  case "$target_cpu" in
 #    x86_64)
 #           if test -z "$b64_jni"; then
-#  	     CFLAGS_JNI="$CFLAGS_JNI -m32"
-#  	     LDFLAGS_SHLIB="$LDFLAGS_SHLIB -m32"
+#  	     CFLAGS_JNI="$CFLAGS_JNI "
+#  	     LDFLAGS_SHLIB="$LDFLAGS_SHLIB "
 #   	   fi
 #	;;
 #  esac	
@@ -1114,11 +1114,6 @@
     echo "checking for JNI in $JAVA_HOME/include ... found"
   else
     JNI=""
-    AC_MSG_ERROR([
-
-    *** Can't find JNI directory in JAVA_HOME=${JAVA_HOME}
-    *** JNI is expected in ${JAVA_HOME}/include/$jni_os
-  ])
   fi
 
   HAS_JVMTI=""
@@ -1133,8 +1128,6 @@
     echo "Using JVMTI for class reloading"
     
     CFLAGS="$CFLAGS -DHAS_JVMTI"
-  else
-    AC_MSG_ERROR([Can't find JVMTI directory ${JNI_INCLUDE_DIR}/jvmti.h])
   fi
 fi
 
