diff -Nru resin-4.0.26.vanilla/modules/c/src/resin_os/Makefile.in resin-4.0.26/modules/c/src/resin_os/Makefile.in
--- resin-4.0.26.vanilla/modules/c/src/resin_os/Makefile.in	2012-03-04 11:33:18.455432143 +0100
+++ resin-4.0.26/modules/c/src/resin_os/Makefile.in	2012-03-04 12:02:44.379794892 +0100
@@ -20,10 +20,10 @@
 all	: libresin_os.$(SO)
 
 libresin_os.so	: $(OBJ)
-	$(CC) $(LDFLAGS_SHLIB) -o $@ $(OBJ) $(PROXY_LIBS)
+	$(CC) $(LDFLAGS_SHLIB) $(LDFLAGS) -o $@ $(OBJ) $(PROXY_LIBS)
 
 libresin_os.jnilib	: $(OBJ)
-	$(CC) $(LDFLAGS_SHLIB) -framework JavaVM -o $@ $(OBJ) $(PROXY_LIBS)
+	$(CC) $(LDFLAGS_SHLIB) -framework JavaVM $(LDFLAGS) -o $@ $(OBJ) $(PROXY_LIBS)
 
 boot.o		: Makefile resin.h
 java.o		: Makefile resin.h
diff -Nru resin-4.0.26.vanilla/modules/c/src/resinssl/Makefile.in resin-4.0.26/modules/c/src/resinssl/Makefile.in
--- resin-4.0.26.vanilla/modules/c/src/resinssl/Makefile.in	2012-03-04 11:33:18.475432198 +0100
+++ resin-4.0.26/modules/c/src/resinssl/Makefile.in	2012-03-04 12:05:36.972254697 +0100
@@ -21,10 +21,10 @@
 all	: libresinssl.$(SO)
 
 libresinssl.so	: $(OBJ)
-	$(CC) $(LDFLAGS_SHLIB) -o $@ $(OBJ) $(PROXY_LIBS)
+	$(CC) $(LDFLAGS_SHLIB) $(LDFLAGS) -o $@ $(OBJ) $(PROXY_LIBS)
 
 libresinssl.jnilib	: $(OBJ)
-	$(CC) $(LDFLAGS_SHLIB) -framework JavaVM -o $@ $(OBJ) $(PROXY_LIBS)
+	$(CC) $(LDFLAGS_SHLIB) -framework JavaVM $(LDFLAGS) -o $@ $(OBJ) $(PROXY_LIBS)
 
 $(OBJ)	: ../resin/resin.h Makefile
 
