--- xpcom/sample/Makefile.in.orig	2007-04-18 05:24:25.000000000 +0200
+++ xpcom/sample/Makefile.in	2007-04-18 05:25:16.000000000 +0200
@@ -89,7 +89,7 @@
 # from this Makefile. We link against the "dependent glue" and against the
 # frozen XPCOM shared library.
 EXTRA_DSO_LDOPTS = \
-		$(DIST)/lib/$(LIB_PREFIX)xpcomglue_s.$(LIB_SUFFIX) \
+		$(XPCOM_STANDALONE_GLUE_LDOPTS) \
 		$(MOZ_COMPONENT_LIBS) \
 		$(NULL)
 
