# https://bugzilla.mozilla.org/show_bug.cgi?id=360493

Index: browser/app/profile/firefox.js
===================================================================
RCS file: /cvsroot/mozilla/browser/app/profile/firefox.js,v
retrieving revision 1.71.2.77
diff -u -8 -p -r1.71.2.77 firefox.js
--- browser/app/profile/firefox.js	4 Dec 2006 20:04:19 -0000	1.71.2.77
+++ browser/app/profile/firefox.js	8 Dec 2006 00:21:00 -0000
@@ -378,17 +378,17 @@ pref("alerts.slideIncrementTime", 10);
 pref("alerts.totalOpenTime", 4000);
 pref("alerts.height", 50);
 
 pref("browser.xul.error_pages.enabled", true);
 
 pref("signon.rememberSignons",              true);
 pref("signon.expireMasterPassword",         false);
 pref("signon.SignonFileName",               "signons.txt");
-pref("signon.prefillForms",                 true);
+pref("signon.prefillForms",                 false);
 
 // We want to make sure mail URLs are handled externally...
 pref("network.protocol-handler.external.mailto", true); // for mail
 pref("network.protocol-handler.external.news", true);   // for news
 pref("network.protocol-handler.external.snews", true);  // for secure news
 pref("network.protocol-handler.external.nntp", true);   // also news
 // ...without warning dialogs
 pref("network.protocol-handler.warn-external.mailto", false);
