Index: vlc-0.8.6h/configure.ac
===================================================================
--- vlc-0.8.6h.orig/configure.ac
+++ vlc-0.8.6h/configure.ac
@@ -3107,9 +3107,10 @@ if test "${enable_dts}" != "no"; then
         VLC_ADD_PLUGINS([dtstofloat32])
         VLC_ADD_LDFLAGS([dtstofloat32],[-ldts_pic])
       ],[
-        AC_CHECK_LIB(dts, dts_free, [
+	PKG_CHECK_MODULES([DCA], [libdts >= 0.0.2], [
           VLC_ADD_PLUGINS([dtstofloat32])
-          VLC_ADD_LDFLAGS([dtstofloat32],[-ldts])
+	  VLC_ADD_CPPFLAGS([dtstofloat32], [${DCA_CFLAGS}])
+          VLC_ADD_LDFLAGS([dtstofloat32],[${DCA_LIBS}])
         ],[
           if test "${enable_dts}" = "yes"; then
             AC_MSG_ERROR([Could not find libdts on your system: you may get it from http://www.videolan.org/dtsdec.html])
