https://bugs.gentoo.org/384553

--- emacs-22.3-orig/src/image.c
+++ emacs-22.3/src/image.c
@@ -6427,7 +6427,7 @@
 {
   xassert (png_ptr != NULL);
   image_error ("PNG error: %s", build_string (msg), Qnil);
-  longjmp (png_ptr->jmpbuf, 1);
+  longjmp (png_jmpbuf (png_ptr), 1);
 }
 
 
@@ -6603,7 +6603,7 @@
 
   /* Set error jump-back.  We come back here when the PNG library
      detects an error.  */
-  if (setjmp (png_ptr->jmpbuf))
+  if (setjmp (png_jmpbuf (png_ptr)))
     {
     error:
       if (png_ptr)
