--- mozilla/extensions/java/xpcom/interfaces/org/mozilla/xpcom/Mozilla.java
+++ mozilla/extensions/java/xpcom/interfaces/org/mozilla/xpcom/Mozilla.java
@@ -465,18 +465,18 @@ IXPCOMError {
       grePath = getPathFromConfigFile(env, aVersions, aProperties);
       if (grePath != null) {
         return grePath;
       }
     }
 
     final String greUserConfFile = ".gre.config";
     final String greUserConfDir = ".gre.d";
-    final String greConfPath = "/etc/gre.conf";
-    final String greConfDir = "/etc/gre.d";
+    final String greConfPath = "@GENTOO_PORTAGE_EPREFIX@/etc/gre.conf";
+    final String greConfDir = "@GENTOO_PORTAGE_EPREFIX@/etc/gre.d";
 
     env = System.getProperty("user.home");
     if (env != null) {
       // Look in ~/.gre.config
       grePath = getPathFromConfigFile(env + File.separator + greUserConfFile,
                                       aVersions, aProperties);
       if (grePath != null) {
         return grePath;
--- mozilla/xpcom/build/nsXPCOMPrivate.h
+++ mozilla/xpcom/build/nsXPCOMPrivate.h
@@ -274,18 +274,18 @@ void LogTerm();
 #define GRE_FRAMEWORK_NAME "XUL.framework"
 #define XUL_DLL            "XUL"
 #else
 #define XPCOM_SEARCH_KEY  "LD_LIBRARY_PATH"
 #define XUL_DLL   "libxul"MOZ_DLL_SUFFIX
 #endif
 
 #define GRE_CONF_NAME ".gre.config"
-#define GRE_CONF_PATH "/etc/gre.conf"
-#define GRE_CONF_DIR  "/etc/gre.d"
+#define GRE_CONF_PATH "@GENTOO_PORTAGE_EPREFIX@/etc/gre.conf"
+#define GRE_CONF_DIR  "@GENTOO_PORTAGE_EPREFIX@/etc/gre.d"
 #define GRE_USER_CONF_DIR ".gre.d"
 #endif
 
 #if defined(XP_WIN) || defined(XP_OS2)
   #define XPCOM_FILE_PATH_SEPARATOR       "\\"
   #define XPCOM_ENV_PATH_SEPARATOR        ";"
 #elif defined(XP_UNIX) || defined(XP_BEOS)
   #define XPCOM_FILE_PATH_SEPARATOR       "/"
--- mozilla/xulrunner/app/nsRegisterGREUnix.cpp
+++ mozilla/xulrunner/app/nsRegisterGREUnix.cpp
@@ -107,17 +107,17 @@ RegisterXULRunner(PRBool aRegisterGlobal
                   const GREProperty *aProperties, PRUint32 aPropertiesLen,
                   const char *aGREMilestone)
 {
   // Register ourself in /etc/gre.d or ~/.gre.d/ and record what key we created
   // for future unregistration.
 
   nsresult rv;
 
-  char root[MAXPATHLEN] = "/etc/gre.d";
+  char root[MAXPATHLEN] = "@GENTOO_PORTAGE_EPREFIX@/etc/gre.d";
 
   if (!aRegisterGlobally) {
     char *home = PR_GetEnv("HOME");
     if (!home || !*home)
       return PR_FALSE;
 
     PR_snprintf(root, MAXPATHLEN, "%s/.gre.d", home);
   }
@@ -199,17 +199,17 @@ RegisterXULRunner(PRBool aRegisterGlobal
 }
 
 void
 UnregisterXULRunner(PRBool aRegisterGlobally, nsIFile* aLocation,
                     const char *aGREMilestone)
 {
   nsresult rv;
 
-  char root[MAXPATHLEN] = "/etc/gre.d";
+  char root[MAXPATHLEN] = "@GENTOO_PORTAGE_EPREFIX@/etc/gre.d";
 
   if (!aRegisterGlobally) {
     char *home = PR_GetEnv("HOME");
     if (!home || !*home)
       return;
 
     PR_snprintf(root, MAXPATHLEN, "%s/.gre.d", home);
   }
--- mozilla/xulrunner/installer/Makefile.in
+++ mozilla/xulrunner/installer/Makefile.in
@@ -68,17 +68,17 @@ INSTALL_SDK = 1
 include $(topsrcdir)/toolkit/mozapps/installer/packager.mk
 
 $(MOZILLA_VERSION).system.conf: $(topsrcdir)/config/milestone.txt Makefile
 	printf "[%s]\nGRE_PATH=%s\nxulrunner=true\nabi=%s" \
 	  $(MOZILLA_VERSION) $(installdir) $(TARGET_XPCOM_ABI)> $@
 
 ifndef SKIP_GRE_REGISTRATION
 # to register xulrunner per-user, override this with $HOME/.gre.d
-regdir = /etc/gre.d
+regdir = @sysconfdir@/gre.d
 
 install:: $(MOZILLA_VERSION).system.conf
 	$(NSINSTALL) -D $(DESTDIR)$(regdir)
 	$(SYSINSTALL) $(IFLAGS1) $^ $(DESTDIR)$(regdir)
 endif
 
 # Add pkg-config files to the install:: target
 
