Index: TiMidity++-2.13.2/interface/x_sherry.c
===================================================================
--- TiMidity++-2.13.2.orig/interface/x_sherry.c
+++ TiMidity++-2.13.2/interface/x_sherry.c
@@ -782,7 +782,7 @@ static void sry_pal_v2r(uint8 *data)
 static void png_read_func(png_structp png_ptr, char *buff, size_t n)
 {
     struct timidity_file *tf;
-    tf = (struct timidity_file *)png_ptr->io_ptr;
+    tf = (struct timidity_file *)png_get_io_ptr(png_ptr);
     tf_read(buff, 1, n, tf);
 }
 
@@ -938,9 +938,7 @@ static void sry_load_png(uint8 *data)
 		       6*7*6, MAX_SCREEN_COLORS,
 		       NULL, 1);
 	/*???*/
-	png_set_PLTE(pngPtr, infoPtr, pngPtr->palette, pngPtr->num_palette);
-	palette = pngPtr->palette;
-	numPalette = pngPtr->num_palette;
+        png_get_PLTE(pngPtr, infoPtr, &palette, &numPalette);
     }
 
     if(png_get_valid(pngPtr, infoPtr, PNG_INFO_tRNS))
