Index: luatex-beta-0.30.3/src/texk/web2c/luatexdir/luatexlib.mk
===================================================================
--- luatex-beta-0.30.3.orig/src/texk/web2c/luatexdir/luatexlib.mk
+++ luatex-beta-0.30.3/src/texk/web2c/luatexdir/luatexlib.mk
@@ -69,6 +69,7 @@ endif
 
 ZZIPLIBDIR=../../libs/zziplib
 ZZIPLIBSRCDIR=$(srcdir)/$(ZZIPLIBDIR)
+ZZIPLIBINC=-I../$(ZZIPLIBSRCDIR) -I../$(ZZIPLIBDIR)
 ZZIPLIBDEP = $(ZZIPLIBDIR)/zzip/libzzip.a
 
 $(ZZIPLIBDEP): $(ZZIPLIBSRCDIR)
@@ -80,7 +81,7 @@ $(ZZIPLIBDEP): $(ZZIPLIBSRCDIR)
 LUAZIPDIR=../../libs/luazip
 LUAZIPSRCDIR=$(srcdir)/$(LUAZIPDIR)
 LUAZIPDEP=$(LUAZIPDIR)/src/luazip.o
-LUAZIPINC=-I../../lua51 -I../$(ZZIPLIBSRCDIR) -I../$(ZZIPLIBDIR)
+LUAZIPINC=-I../../lua51 $(ZZIPLIBINC)
 
 $(LUAZIPDEP): $(LUAZIPDIR)/src/luazip.c
 	mkdir -p $(LUAZIPDIR) && cd $(LUAZIPDIR) && cp -R $(LUAZIPSRCDIR)/* . && \
@@ -154,11 +155,11 @@ LIBMPLIBDEP=$(LIBMPDIR)/lmplib.o $(LIBMP
 # Convenience variables.
 
 luatexlibs = $(luapdflib) $(LDLIBPNG) $(LDZLIB) $(LDLIBXPDF) $(LIBMD5DEP) $(LDLIBOBSD) \
-             $(LIBLUADEP) $(SLNUNICODEDEP)  $(LUAZIPDEP) $(ZZIPLIBDEP) $(LUAFSDEP) $(LUASOCKETDEP) \
+             $(LIBLUADEP) $(SLNUNICODEDEP)  $(LUAZIPDEP) $(LDZZIPLIB)  $(LUAFSDEP) $(LUASOCKETDEP) \
              $(LUAPEGDEP) $(LUAMDVDEP)  $(LUAZLIBDEP) $(LUAFFDEP) $(LIBMPLIBDEP)
 
 luatexlibsdep = $(luapdflib) $(LIBPNGDEP) $(ZLIBDEP) $(LIBXPDFDEP) $(LIBMD5DEP) $(LIBOBSDDEP) \
-                $(LIBLUADEP) $(SLNUNICODEDEP) $(ZZIPLIBDEP) $(LUAZIPDEP)  $(LUAFSDEP) $(LUASOCKETDEP) \
+                $(LIBLUADEP) $(SLNUNICODEDEP) $(LUAZIPDEP)  $(LUAFSDEP) $(LUASOCKETDEP) \
                 $(LUAPEGDEP) $(LUAMDVDEP)  $(LUAZLIBDEP) $(LUAFFDEP) $(makecpool)
 
 ## end of luatexlib.mk - Makefile fragment for libraries used by pdf[ex]tex.
