Index: luatex-beta-0.30.3/src/texk/web2c/luatexdir/Makefile.in
===================================================================
--- luatex-beta-0.30.3.orig/src/texk/web2c/luatexdir/Makefile.in
+++ luatex-beta-0.30.3/src/texk/web2c/luatexdir/Makefile.in
@@ -17,7 +17,7 @@ ALL_CXXFLAGS = @CXXFLAGS@ @DEFS@ $(XXCFL
   $(LIBXPDFCPPFLAGS) \
   -I$(LIBOBSDCOMPATDIR) -I$(LIBOBSDCOMPATDIR)/.. \
   -I$(LIBOBSDCOMPATFSRCDIR) -I$(LIBOBSDCOMPATFSRCDIR)/.. \
-  -I$(LIBPNGSRCDIR) -I$(ZLIBSRCDIR) \
+  $(LIBPNGINCLUDES) -I$(ZLIBSRCDIR) \
   -DPDF_PARSER_ONLY -DDISABLE_OUTLINE
 CXX = @CXX@
 
@@ -31,6 +31,7 @@ CTANGLE = ../ctangle
 
 LIBPNGDIR=../../../libs/libpng
 LIBPNGSRCDIR=$(srcdir)/$(LIBPNGDIR)
+LIBPNGINCLUDES=-I$(LIBPNGSRCDIR)
 
 ZLIBDIR=../../../libs/zlib
 ZLIBSRCDIR=$(srcdir)/$(ZLIBDIR)
Index: luatex-beta-0.30.3/src/texk/web2c/luatexdir/luatex.mk
===================================================================
--- luatex-beta-0.30.3.orig/src/texk/web2c/luatexdir/luatex.mk
+++ luatex-beta-0.30.3/src/texk/web2c/luatexdir/luatex.mk
@@ -13,6 +13,7 @@ ZLIBSRCDIR=$(srcdir)/$(ZLIBDIR)
 # libpng
 LIBPNGDIR=../../libs/libpng
 LIBPNGSRCDIR=$(srcdir)/$(LIBPNGDIR)
+LIBPNGINCLUDES=-I$(LIBPNGSRCDIR)
 # libxpdf
 LIBXPDFDIR=../../libs/xpdf
 LIBXPDFSRCDIR=$(srcdir)/$(LIBXPDFDIR)
@@ -23,7 +24,7 @@ LIBOBSDCOMPATFSRCDIR=$(srcdir)/$(LIBOBSD
 
 SVN_REV := $(shell $(srcdir)/$(luatexdir)/get_svnversion.sh $(srcdir))
 
-XCPPFLAGS=-I$(LIBOBSDCOMPATDIR) -I$(LIBOBSDCOMPATDIR)/.. -I$(LIBOBSDCOMPATFSRCDIR) -I$(LIBOBSDCOMPATFSRCDIR)/..  -I$(ZLIBSRCDIR) -I$(LIBPNGSRCDIR) $(LIBXPDFINCLUDE) -Dextra_version_info=`date +-%Y%m%d%H` -DSVN_REV=\"$(SVN_REV)\"
+XCPPFLAGS=-I$(LIBOBSDCOMPATDIR) -I$(LIBOBSDCOMPATDIR)/.. -I$(LIBOBSDCOMPATFSRCDIR) -I$(LIBOBSDCOMPATFSRCDIR)/..  -I$(ZLIBSRCDIR) $(LIBPNGINCLUDES) $(LIBXPDFINCLUDE) -Dextra_version_info=`date +-%Y%m%d%H` -DSVN_REV=\"$(SVN_REV)\"
 
 Makefile: $(srcdir)/$(luatexdir)/luatex.mk
 
