Index: luatex-beta-0.30.3/src/texk/web2c/luatexdir/luatex.mk
===================================================================
--- luatex-beta-0.30.3.orig/src/texk/web2c/luatexdir/luatex.mk
+++ luatex-beta-0.30.3/src/texk/web2c/luatexdir/luatex.mk
@@ -129,6 +129,8 @@ install-luatex: install-luatex-exec
 install-programs: @LTEX@ install-luatex-exec
 install-luatex-exec: $(luatex) $(bindir)
 	for p in luatex; do $(INSTALL_LIBTOOL_PROG) $$p $(bindir); done
+	cd $(DESTDIR)$(bindir) && rm -f texlua && $(LN) luatex texlua
+	cd $(DESTDIR)$(bindir) && rm -f texluac && $(LN) luatex texluac
 
 # 
 # luatex binaries archive
