Index: luatex-beta-0.30.3/src/texk/web2c/luatexdir/image/pdftoepdf.cc
===================================================================
--- luatex-beta-0.30.3.orig/src/texk/web2c/luatexdir/image/pdftoepdf.cc
+++ luatex-beta-0.30.3/src/texk/web2c/luatexdir/image/pdftoepdf.cc
@@ -658,7 +658,7 @@ static void writeEncodings()
     }
     for (r = encodingList; r != NULL; r = n) {
         n = r->next;
-        delete r->font;
+        r->font->decRefCnt();
         delete r;
     }
 }
