diff -Nru resin-4.0.9.vanilla/modules/resin/src/boot-manifest resin-4.0.9/modules/resin/src/boot-manifest
--- resin-4.0.9.vanilla/modules/resin/src/boot-manifest	2010-08-19 17:44:18.810963762 +0200
+++ resin-4.0.9/modules/resin/src/boot-manifest	2010-08-19 17:50:04.030963761 +0200
@@ -1,5 +1,5 @@
 Main-Class: com.caucho.server.bootstrap.ResinBoot
-Class-Path: resin.jar jta-101.jar isorelax.jar jsdk-24.jar jmx-12.jar jca-15.jar jstl-11.jar
+Class-Path: resin.jar jta-101.jar isorelax.jar resin-servlet-api.jar jmx-12.jar jca-15.jar jstl-11.jar
 Extension-Name: com.caucho.server.bootstrap
 Implementation-Vendor: Caucho Technology, Inc.
 Implementation-Version: 3.0.1
diff -Nru resin-4.0.9.vanilla/modules/resin/src/manifest resin-4.0.9/modules/resin/src/manifest
--- resin-4.0.9.vanilla/modules/resin/src/manifest	2010-08-19 17:44:19.522963762 +0200
+++ resin-4.0.9/modules/resin/src/manifest	2010-08-19 17:50:04.030963761 +0200
@@ -1,4 +1,4 @@
-Class-Path: javaee-16.jar javaee-15.jar pro.jar license.jar hessian.jar quercus.jar resin-kernel.jar jcache-16.jar webservices-api.jar
+Class-Path: javaee-16.jar javaee-15.jar pro.jar license.jar hessian.jar quercus.jar resin-kernel.jar jcache-16.jar webservices-api.jar resin-servlet-api.jar
 Implementation-Vendor: Caucho Technology, Inc.
 Implementation-Version: 4.0.9
 J2EE-DeploymentFactory-Implementation-Class: com.caucho.j2ee.deployclient.DeploymentFactoryImpl
diff -Nru resin-4.0.9.vanilla/modules/resin/src/manifest.in resin-4.0.9/modules/resin/src/manifest.in
--- resin-4.0.9.vanilla/modules/resin/src/manifest.in	2010-08-19 17:44:19.522963762 +0200
+++ resin-4.0.9/modules/resin/src/manifest.in	2010-08-19 17:50:04.030963762 +0200
@@ -1,4 +1,4 @@
-Class-Path: javaee-16.jar javaee-15.jar pro.jar license.jar hessian.jar quercus.jar resin-kernel.jar jcache-16.jar webservices-api.jar
+Class-Path: javaee-16.jar javaee-15.jar pro.jar license.jar hessian.jar quercus.jar resin-kernel.jar jcache-16.jar webservices-api.jar resin-servlet-api.jar
 Implementation-Vendor: Caucho Technology, Inc.
 Implementation-Version: @VERSION@
 J2EE-DeploymentFactory-Implementation-Class: com.caucho.j2ee.deployclient.DeploymentFactoryImpl
