diff -Nru resin-4.0.9.vanilla/Makefile.am resin-4.0.9/Makefile.am
--- resin-4.0.9.vanilla/Makefile.am	2010-08-19 17:44:20.150963763 +0200
+++ resin-4.0.9/Makefile.am	2010-08-19 17:47:36.938963766 +0200
@@ -20,34 +20,31 @@
 	- rm -r $(LIBEXEC)
 
 install	:
-	if test "$(resin_home)" != $(abs_builddir); then \
-	  mkdir -p $(resin_home)/bin; \
-	  mkdir -p $(resin_home)/keys; \
-	  mkdir -p $(resin_home)/licenses; \
-	  mkdir -p $(resin_home)/lib; \
-	  mkdir -p $(resin_home)/$(LIBEXEC); \
-	  cp bin/* $(resin_home)/bin; \
-	  cp lib/*.jar $(resin_home)/lib; \
+	if test "$(DESTDIR)/$(resin_home)" != $(abs_builddir); then \
+	  mkdir -p $(DESTDIR)/$(resin_home)/bin; \
+	  mkdir -p $(DESTDIR)/$(resin_home)/keys; \
+	  mkdir -p $(DESTDIR)/$(resin_home)/licenses; \
+	  mkdir -p $(DESTDIR)/$(resin_home)/lib; \
+	  mkdir -p $(DESTDIR)/$(resin_home)/$(LIBEXEC); \
+	  cp bin/* $(DESTDIR)/$(resin_home)/bin; \
+	  cp lib/*.jar $(DESTDIR)/$(resin_home)/lib; \
 	fi
-	if test "$(resin_root)" != $(abs_builddir); then \
-	  mkdir -p $(resin_root)/webapps; \
-	  cp -r $(CP_U_FLAG) webapps/* $(resin_root)/webapps; \
-	  mkdir -p $(resin_root)/doc; \
-	  cp -r $(CP_U_FLAG) doc/* $(resin_root)/doc; \
-	  mkdir -p $(resin_root)/project-jars; \
-	  cp -r $(CP_U_FLAG) project-jars/* $(resin_root)/project-jars; \
-	  mkdir -p $(resin_log); \
+	if test "$(DESTDIR)/$(resin_root)" != $(abs_builddir); then \
+	  mkdir -p $(DESTDIR)/$(resin_root)/webapps; \
+	  cp -r $(CP_U_FLAG) webapps/* $(DESTDIR)/$(resin_root)/webapps; \
+	  mkdir -p $(DESTDIR)/$(resin_root)/doc; \
+	  cp -r $(CP_U_FLAG) doc/* $(DESTDIR)/$(resin_root)/doc; \
+	  mkdir -p $(DESTDIR)/$(resin_root)/project-jars; \
+	  cp -r $(CP_U_FLAG) project-jars/* $(DESTDIR)/$(resin_root)/project-jars; \
+	  mkdir -p $(DESTDIR)/$(resin_log); \
 	fi
-	if test "$(resin_conf)" != $(abs_builddir)/conf; then \
-	  mkdir -p $(resin_conf); \
-	  cp $(CP_U_FLAG) conf/app-default.xml $(resin_conf)/app-default.xml; \
-	  cp $(CP_U_FLAG) conf/resin.xml $(resin_conf)/resin.xml; \
+	if test "$(DESTDIR)/$(resin_conf)" != $(abs_builddir)/conf; then \
+	  mkdir -p $(DESTDIR)/$(resin_conf); \
+	  cp $(CP_U_FLAG) conf/app-default.xml $(DESTDIR)/$(resin_conf)/app-default.xml; \
+	  cp $(CP_U_FLAG) conf/resin.xml $(DESTDIR)/$(resin_conf)/resin.xml; \
 	fi
 	(cd modules/c/src; $(MAKE) install)
-	if test -r $(resin_pro)/modules/c/src/Makefile; then \
-		(cd $(resin_pro)/modules/c/src; $(MAKE) install) \
-	fi
-	@ if test -n "$(resin_init_d)" -a "$(resin_home)" != $(abs_builddir); then \
+	@ if false && test -n "$(resin_init_d)" -a "$(resin_home)" != $(abs_builddir); then \
 		if test -f "${resin_init_d}"; then\
 			if test \( -w "${resin_init_d}" \) -o \( -w `dirname ${resin_init_d}` \); then \
 				cp $(CP_U_FLAG) $(abs_builddir)/init.d/resin $(resin_init_d); \
diff -Nru resin-4.0.9.vanilla/modules/c/src/resin_os/Makefile.in resin-4.0.9/modules/c/src/resin_os/Makefile.in
--- resin-4.0.9.vanilla/modules/c/src/resin_os/Makefile.in	2010-08-19 17:44:19.566963762 +0200
+++ resin-4.0.9/modules/c/src/resin_os/Makefile.in	2010-08-19 17:45:06.686963763 +0200
@@ -39,8 +39,8 @@
 
 install	: libresin_os.$(SO)
 #	cp resin $(resin_home)/bin
-	- mkdir -p $(resin_home)/$(LIBEXEC)
-	cp libresin_os.$(SO) $(resin_home)/$(LIBEXEC)
+	- mkdir -p $(DESTDIR)/$(resin_home)/$(LIBEXEC)
+	cp libresin_os.$(SO) $(DESTDIR)/$(resin_home)/$(LIBEXEC)
 
 
 
