diff -Nru ppp-2.4.5.orig/pppd/ipv6cp.c ppp-2.4.5/pppd/ipv6cp.c
--- ppp-2.4.5.orig/pppd/ipv6cp.c	2009-11-16 23:26:07.000000000 +0100
+++ ppp-2.4.5/pppd/ipv6cp.c	2010-08-08 09:59:19.000000000 +0200
@@ -153,6 +153,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <unistd.h>
+#include <stdlib.h>
 #include <netdb.h>
 #include <sys/param.h>
 #include <sys/types.h>
diff -Nru ppp-2.4.5.orig//pppdump/pppdump.c ppp-2.4.5/pppdump/pppdump.c
--- ppp-2.4.5.orig//pppdump/pppdump.c	2009-11-16 23:26:07.000000000 +0100
+++ ppp-2.4.5/pppdump/pppdump.c	2011-11-11 01:47:39.000000000 +0100
@@ -191,7 +191,7 @@
 	    show_time(f, c);
 	    break;
 	default:
-	    printf("?%.2x\n");
+	    printf("?\n");
 	}
     }
 }
@@ -421,7 +421,7 @@
 	    show_time(f, c);
 	    break;
 	default:
-	    printf("?%.2x\n");
+	    printf("?\n");
 	}
     }
 }
