diff -Nru ppp-2.4.5.orig/contrib/pppgetpass/Makefile.linux ppp-2.4.5/contrib/pppgetpass/Makefile.linux
--- ppp-2.4.5.orig/contrib/pppgetpass/Makefile.linux	2009-11-16 23:26:07.000000000 +0100
+++ ppp-2.4.5/contrib/pppgetpass/Makefile.linux	2010-08-08 10:38:06.000000000 +0200
@@ -3,9 +3,9 @@
 pppgetpass.vt: pppgetpass.vt.o
 
 pppgetpass.gtk: pppgetpass.gtk.o
-	$(CC) $(LDFLAGS) pppgetpass.gtk.o `gtk-config --libs` -o pppgetpass.gtk
+	$(CC) $(LDFLAGS) pppgetpass.gtk.o `pkg-config --libs gtk+-2.0` -o pppgetpass.gtk
 pppgetpass.gtk.o: pppgetpass.gtk.c
-	$(CC) $(CFLAGS) -c pppgetpass.gtk.c `gtk-config --cflags`
+	$(CC) $(CFLAGS) -c pppgetpass.gtk.c `pkg-config --cflags gtk+-2.0`
 
 install: all
 	install -m 755 pppgetpass.sh /usr/bin/pppgetpass
