diff -Nru ppp-2.4.3.orig/pppd/plugins/dhcp/Makefile ppp-2.4.3/pppd/plugins/dhcp/Makefile
--- ppp-2.4.3.orig/pppd/plugins/dhcp/Makefile	2002-09-30 13:21:58.000000000 +0300
+++ ppp-2.4.3/pppd/plugins/dhcp/Makefile	2006-05-07 19:40:49.603186500 +0300
@@ -1,6 +1,6 @@
 
 PLUGIN=dhcpc.so
-CFLAGS=-I../.. -I../../../include -O2
+CFLAGS=$(COPTS) -I../.. -I../../../include -fPIC
 
 all: $(PLUGIN)
 
@@ -9,7 +9,7 @@
 	$(INSTALL) -s -c -m 755 dhcpc.so $(LIBDIR)
 
 dhcpc.so: dhcpc.o clientpacket.o packet.o socket.o options.o 
-	gcc -o dhcpc.so -shared dhcpc.o clientpacket.o packet.o socket.o options.o
+	$(CC) $(LDFLAGS) $(CFLAGS) -o $@ $^
 
 
 clean:
diff -Nru ppp-2.4.3.orig/pppd/plugins/dhcp/Makefile.linux ppp-2.4.3/pppd/plugins/dhcp/Makefile.linux
--- ppp-2.4.3.orig/pppd/plugins/dhcp/Makefile.linux	2002-09-30 13:21:58.000000000 +0300
+++ ppp-2.4.3/pppd/plugins/dhcp/Makefile.linux	2006-05-07 19:40:49.603186500 +0300
@@ -1,6 +1,6 @@
 
 PLUGIN=dhcpc.so
-CFLAGS=-I../.. -I../../../include -O2
+CFLAGS=$(COPTS) -I../.. -I../../../include -fPIC
 
 all: $(PLUGIN)
 
@@ -9,7 +9,7 @@
 	$(INSTALL) -s -c -m 755 dhcpc.so $(LIBDIR)
 
 dhcpc.so: dhcpc.o clientpacket.o packet.o socket.o options.o 
-	gcc -o dhcpc.so -shared dhcpc.o clientpacket.o packet.o socket.o options.o
+	$(CC) $(LDFLAGS) $(CFLAGS) -o $@ $^
 
 
 clean:
