diff -Nru resin-3.0.25.vanilla/modules/resin/src/com/caucho/server/resin/Resin.java resin-3.0.24/modules/resin/src/com/caucho/server/resin/Resin.java
--- resin-3.0.25.vanilla/modules/resin/src/com/caucho/server/resin/Resin.java	2006-08-01 12:43:54.000000000 +0200
+++ resin-3.0.25/modules/resin/src/com/caucho/server/resin/Resin.java	2007-08-17 09:12:32.000000000 +0200
@@ -700,10 +700,10 @@
       throw new ConfigException(L().l("The following system property must be set:\n  -Djava.util.logging.manager=com.caucho.log.LogManagerImpl\nThe JDK 1.4 Logging manager must be set to Resin's log manager."));
     }
 
-    validatePackage("javax.servlet.Servlet", new String[] {"2.4", "1.4"});
+    validatePackage("javax.servlet.Servlet", new String[] {"2.4", "1.5"});
     validatePackage("javax.servlet.jsp.jstl.core.Config", new String[] {"1.1"});
     validatePackage("javax.management.MBeanServer", new String[] { "1.2", "1.5" });
-    validatePackage("javax.resource.spi.ResourceAdapter", new String[] {"1.5", "1.4"});
+    validatePackage("javax.resource.spi.ResourceAdapter", new String[] {"1.5", "1.5"});
   }
 
   /**

