diff -Nru resin-3.0.25.vanilla/bin/wrapper.pl resin-3.0.24/bin/wrapper.pl
--- resin-3.0.25.vanilla/bin/wrapper.pl	2007-08-14 08:56:36.000000000 +0200
+++ resin-3.0.25/bin/wrapper.pl	2007-08-17 09:12:32.000000000 +0200
@@ -447,9 +447,6 @@
 	$pid=`cat $pid_file`;
 	chop($pid);
 
-	if ($cmd eq "stop") {
-	    print("Stopping ${name}\n");
-	}
 	$signal = 15;
 	$count = 10;
 	while (kill($signal, $pid) && $count-- > 0) {
@@ -565,8 +562,6 @@
 	$SIG{CHLD} = 'IGNORE';
     }
 
-    print "\n\nResin wrapper.pl closing.\n\n";
-    
     close(S);
     close(C); # closing the accepted socket should start orderly shutdown
     # unlink needs to happen relatively soon so restart's pid won't
@@ -743,8 +738,6 @@
       $date = `date`;
       chop($date);
 
-      print "Resin $name $cmd at $date\n";
-
       # create a keepalive socket
       # when the wrapper dies, the httpd class will detect that and
       # close gracefully
diff -Nru resin-3.0.25.vanilla/bin/wrapper.pl.in resin-3.0.24/bin/wrapper.pl.in
--- resin-3.0.25.vanilla/bin/wrapper.pl.in	2007-01-01 11:02:08.000000000 +0100
+++ resin-3.0.25/bin/wrapper.pl.in	2007-08-17 09:12:32.000000000 +0200
@@ -447,9 +447,6 @@
 	$pid=`cat $pid_file`;
 	chop($pid);
 
-	if ($cmd eq "stop") {
-	    print("Stopping ${name}\n");
-	}
 	$signal = 15;
 	$count = 10;
 	while (kill($signal, $pid) && $count-- > 0) {
@@ -565,8 +562,6 @@
 	$SIG{CHLD} = 'IGNORE';
     }
 
-    print "\n\nResin wrapper.pl closing.\n\n";
-    
     close(S);
     close(C); # closing the accepted socket should start orderly shutdown
     # unlink needs to happen relatively soon so restart's pid won't
@@ -743,8 +738,6 @@
       $date = `date`;
       chop($date);
 
-      print "Resin $name $cmd at $date\n";
-
       # create a keepalive socket
       # when the wrapper dies, the httpd class will detect that and
       # close gracefully

