Index: vlc-0.8.6-rc1/configure.ac
===================================================================
--- vlc-0.8.6-rc1.orig/configure.ac
+++ vlc-0.8.6-rc1/configure.ac
@@ -3058,7 +3058,7 @@ if test "${enable_dts}" != "no"; then
         VLC_ADD_LDFLAGS([dtstofloat32],[-ldts_pic])
       ],[
         AC_CHECK_LIB(dts, dts_free, [
-          VLC_ADD_BUILTINS([dtstofloat32])
+          VLC_ADD_PLUGINS([dtstofloat32])
           VLC_ADD_LDFLAGS([dtstofloat32],[-ldts])
         ],[
           if test "${enable_dts}" = "yes"; then
