diff -Naur vnc-4_1_2-unixsrc.orig/common/network/TcpSocket.cxx vnc-4_1_2-unixsrc/common/network/TcpSocket.cxx
--- vnc-4_1_2-unixsrc.orig/common/network/TcpSocket.cxx	2006-05-15 18:56:20.000000000 +0200
+++ vnc-4_1_2-unixsrc/common/network/TcpSocket.cxx	2008-03-10 12:27:22.000000000 +0100
@@ -35,6 +35,7 @@
 #include <string.h>
 #include <signal.h>
 #include <fcntl.h>
+#include <stdlib.h>
 #endif
 
 #include <network/TcpSocket.h>
diff -Naur vnc-4_1_2-unixsrc.orig/unix/tx/TXImage.cxx vnc-4_1_2-unixsrc/unix/tx/TXImage.cxx
--- vnc-4_1_2-unixsrc.orig/unix/tx/TXImage.cxx	2006-05-15 18:56:20.000000000 +0200
+++ vnc-4_1_2-unixsrc/unix/tx/TXImage.cxx	2008-03-10 12:36:06.000000000 +0100
@@ -22,6 +22,8 @@
 
 #include <stdio.h>
 #include <strings.h>
+#include <malloc.h>
+#include <stdlib.h>
 #include <sys/types.h>
 #include <sys/ipc.h>
 #include <sys/shm.h>
@@ -34,6 +36,7 @@
 #include "TXWindow.h"
 #include "TXImage.h"
 
+
 using namespace rfb;
 
 static rfb::LogWriter vlog("TXImage");
diff -Naur vnc-4_1_2-unixsrc.orig/unix/x0vncserver/Image.cxx vnc-4_1_2-unixsrc/unix/x0vncserver/Image.cxx
--- vnc-4_1_2-unixsrc.orig/unix/x0vncserver/Image.cxx	2006-05-15 18:56:20.000000000 +0200
+++ vnc-4_1_2-unixsrc/unix/x0vncserver/Image.cxx	2008-03-10 12:42:17.000000000 +0100
@@ -21,6 +21,8 @@
 
 
 #include <stdio.h>
+#include <malloc.h>
+#include <stdlib.h>
 #include <sys/types.h>
 #include <sys/ipc.h>
 #include <sys/shm.h>
