--- unix/xorg-x11-server-source/configure.ac.old	2008-06-10 14:19:49.000000000 +0000
+++ unix/xorg-x11-server-source/configure.ac	2008-06-10 14:20:57.000000000 +0000
@@ -1674,7 +1674,9 @@
 DIX_CFLAGS="-DHAVE_DIX_CONFIG_H"
 AC_SUBST([DIX_CFLAGS])
 
-AC_SUBST([libdir exec_prefix prefix])
+AC_SUBST([libdir])
+AC_SUBST([exec_prefix])
+AC_SUBST([prefix])
 
 # Man page sections - used in config utils & generating man pages
 XORG_MANPAGE_SECTIONS
