--- vnc-4_1_2-unixsrc/unix/xorg-x11-server-source/hw/xfree86/int10/Makefile.am.s390	2007-06-06 15:00:23.000000000 +0200
+++ vnc-4_1_2-unixsrc/unix/xorg-x11-server-source/hw/xfree86/int10/Makefile.am	2007-06-06 15:00:26.000000000 +0200
@@ -38,7 +38,8 @@ libint10_la_SOURCES = \
 endif
 
 if INT10_STUB
-AM_CFLAGS = $(I386_VIDEO_CFLAGS) -D_VM86_LINUX $(XORG_CFLAGS) $(EXTRA_CFLAGS)
+AM_CFLAGS = $(I386_VIDEO_CFLAGS) -D_VM86_LINUX $(XORG_CFLAGS) $(EXTRA_CFLAGS) \
+	-I../common -I../os-support -I../os-support/bus
 libint10_la_SOURCES = stub.c xf86int10module.c
 endif
 
--- vnc-4_1_2-unixsrc/unix/xorg-x11-server-source/hw/xfree86/common/compiler.h.s390	2007-06-06 15:00:24.000000000 +0200
+++ vnc-4_1_2-unixsrc/unix/xorg-x11-server-source/hw/xfree86/common/compiler.h	2007-06-06 15:00:26.000000000 +0200
@@ -1353,6 +1353,15 @@ do {									\
     : "0" (_beg), "r" (_end), "r" (_flg));				\
 } while (0)
 
+#   elif defined(__s390__) || defined(__s390x__)
+
+#define inb(x) (0)
+#define inw(x) (0)
+#define inl(x) (0)
+#define outb(x,y) (0)
+#define outw(x,y) (0)
+#define outl(x,y) (0)
+
 #   else /* ix86 */
 
 #    define ldq_u(p)	(*((unsigned long  *)(p)))
--- vnc-4_1_2-unixsrc/unix/xorg-x11-server-source/hw/xfree86/common/xf86Helper.c.s390	2007-06-06 15:00:24.000000000 +0200
+++ vnc-4_1_2-unixsrc/unix/xorg-x11-server-source/hw/xfree86/common/xf86Helper.c	2007-06-06 15:00:26.000000000 +0200
@@ -2089,6 +2089,7 @@ xf86GetClocks(ScrnInfoPtr pScrn, int num
 	      void (*BlankScreen)(ScrnInfoPtr, Bool), IOADDRESS vertsyncreg,
 	      int maskval, int knownclkindex, int knownclkvalue)
 {
+#if !defined(__s390__) && !defined(__s390x__)
     register int status = vertsyncreg;
     unsigned long i, cnt, rcnt, sync;
 
@@ -2182,6 +2183,7 @@ finish:
 
     /* Restore registers that were written on */
     (*ClockFunc)(pScrn, CLK_REG_RESTORE);
+#endif /* !defined(__s390__) && !defined(__s390x__) */
 }
 
 _X_EXPORT void
--- vnc-4_1_2-unixsrc/unix/xorg-x11-server-source/hw/xfree86/os-support/linux/lnx_video.c.s390	2007-06-06 15:00:24.000000000 +0200
+++ vnc-4_1_2-unixsrc/unix/xorg-x11-server-source/hw/xfree86/os-support/linux/lnx_video.c	2007-06-06 15:03:10.000000000 +0200
@@ -62,7 +62,8 @@
 #elif !defined(__powerpc__) && \
       !defined(__mc68000__) && \
       !defined(__sparc__) && \
-      !defined(__mips__)
+      !defined(__mips__) && \
+      !defined(__s390)
 
 /*
  * Due to conflicts with "compiler.h", don't rely on <sys/io.h> to declare

