Index: faad2-2.0/plugins/xmms/src/libmp4.c
===================================================================
--- faad2-2.0.orig/plugins/xmms/src/libmp4.c
+++ faad2-2.0/plugins/xmms/src/libmp4.c
@@ -34,6 +34,9 @@ static void	mp4_getSongInfo(char *);
 static int	mp4_isFile(char *);
 static void*	mp4Decode(void *);
 
+void getMP4info(char* file);
+int getAACTrack(MP4FileHandle file);
+
 InputPlugin mp4_ip =
   {
     0,	// handle
@@ -112,11 +115,13 @@ static int	mp4_isFile(char *filename)
     gchar*	extention;
 
     extention = strrchr(filename, '.');
+
     if (extention &&
-	!strcasecmp(extention, ".mp4") ||	// official extention
-	!strcasecmp(extention, ".m4a") ||	// Apple mp4 extention
-	!strcasecmp(extention, ".aac")		// old MPEG2/4-AAC extention
-	){
+        (!strcasecmp(extention, ".mp4") ||	// official extention
+         !strcasecmp(extention, ".m4a") ||	// Apple mp4 extention
+         !strcasecmp(extention, ".aac")		  // old MPEG2/4-AAC extention
+        ))
+    {
       return (1);
     }
   }
