Index: faad2-2.0/configure.in
===================================================================
--- faad2-2.0.orig/configure.in
+++ faad2-2.0/configure.in
@@ -80,11 +80,7 @@ if test x$external_mp4v2 = xyes; then
   AC_SUBST(MPEG4IP_PLAYER_PLUGIN_DIR)
   AC_MSG_NOTICE([*** Building with external mp4v2 ***])
 else
-  AC_MSG_NOTICE([*** Building with internal mp4v2 ***])
-  AM_CONDITIONAL(HAVE_MPEG4IP, false)
-  if test x$WITHMP4V2 = xyes; then
-     AM_CONDITIONAL(WITH_MP4V2, true)
-  fi
+  AC_MSG_ERROR([Internal mp4v2 is going to hurt, you need mpeg4ip!])
 fi
 
 if test x$WITHXMMS = xyes; then
