Index: faad2-2.0/frontend/main.c
===================================================================
--- faad2-2.0.orig/frontend/main.c
+++ faad2-2.0/frontend/main.c
@@ -1053,7 +1053,8 @@ int main(int argc, char *argv[])
             { "info",       0, 0, 'i' },
             { "stdio",      0, 0, 'w' },
             { "stdio",      0, 0, 'g' },
-            { "help",       0, 0, 'h' }
+            { "help",       0, 0, 'h' },
+            { 0,            0, 0,  0  }
         };
 
         c = getopt_long(argc, argv, "o:a:s:f:b:l:wgdhit",
