--- gnusound-0.7.4/doc/Makefile.in.old	2007-02-24 13:55:32.000000000 +0100
+++ gnusound-0.7.4/doc/Makefile.in	2007-02-24 13:55:45.000000000 +0100
@@ -19,7 +19,7 @@
 LANGUAGES=C
 name=@PACKAGE@
 prefix=@prefix@
-docrootdir=@GNOME_HELP_DIR@
+datadir=@GNOME_HELP_DIR@
 docname=$(name)
 xslstylesheet=@DB2XHTML_STYLESHEET@
 omfdir=@datadir@/omf/$(name)
@@ -47,15 +47,15 @@
 	rm -f Makefile
 
 uninstall:
-	rm -rf $(docrootdir)
+	rm -rf $(datadir)
 
 install:
 	mkdir -p $(omfdir)
 	for lang in $(LANGUAGES); do \
-		mkdir -p $(docrootdir)/$$lang; \
+		mkdir -p $(datadir)/$$lang; \
 		install -m 644 -o $(owner_user) -g $(owner_group) \
 			$$lang/*.xml $$lang/*.html $$lang/topic.dat \
-			$(docrootdir)/$$lang; \
+			$(datadir)/$$lang; \
                 install -m 644 -o $(owner_user) -g $(owner_group) \
                         $$lang/$(name).omf \
                         $(omfdir)/$(name)-$$lang.omf; \
