Index: gnusound-0.7.4/configure.ac
===================================================================
--- gnusound-0.7.4.orig/configure.ac
+++ gnusound-0.7.4/configure.ac
@@ -301,7 +301,7 @@ AC_ARG_WITH(gnome2,
 if test "x$use_gnome2" = "xyes"; then
   PKG_CHECK_MODULES(GNOME2,libgnomeui-2.0 >= 2.0.0 libglade-2.0 gthread-2.0,
         AC_DEFINE(HAVE_GNOME2,,[use GNOME2])
-        CFLAGS="$CFLAGS -DGNOME_DISABLE_DEPRECATED -DGTK_DISABLE_DEPRECATED -rdynamic $GNOME2_CFLAGS"
+        CFLAGS="$CFLAGS -rdynamic $GNOME2_CFLAGS"
         LIBS="$LIBS -rdynamic $GNOME2_LIBS",
         use_gnome2=no)
 fi
