Index: gnusound-0.7.4/configure.ac
===================================================================
--- gnusound-0.7.4.orig/configure.ac
+++ gnusound-0.7.4/configure.ac
@@ -184,10 +184,13 @@ dnl FFMPEG
 AC_ARG_ENABLE(ffmpeg, AC_HELP_STRING([--disable-ffmpeg], [disable support for ffmpeg (default=autodetect)]), [ enable_ffmpeg=$enableval ], [ enable_ffmpeg=yes ])
 
 if test "x$enable_ffmpeg" = "xyes"; then
-AC_CHECK_HEADER([ffmpeg/avformat.h],
+AC_CHECK_HEADERS([ffmpeg/avformat.h libavformat/avformat.h],
   [AC_CHECK_LIB([avformat], [av_open_input_file],
     [AC_DEFINE(HAVE_FFMPEG)
-     echo "-lavformat -lavcodec -lm $use" > ${module_path}/.file_ffmpeg.cflags
+     extrainc=""
+     AC_CHECK_HEADER([libavcodec/avcodec.h],extrainc="$extrainc -DLIBAVCODEC_AVCODEC_H=1",[])
+     AC_CHECK_HEADER([libavformat/avformat.h],extrainc="$extrainc -DLIBAVFORMAT_AVFORMAT_H=1",[])
+     echo "-lavformat -lavcodec -lm $use $extrainc" > ${module_path}/.file_ffmpeg.cflags
      file_drivers="ffmpeg $file_drivers"], 
     [echo > ${module_path}/.file_ffmpeg.cflags], [-lavcodec -lm])])
 fi
Index: gnusound-0.7.4/modules/file_ffmpeg.c
===================================================================
--- gnusound-0.7.4.orig/modules/file_ffmpeg.c
+++ gnusound-0.7.4/modules/file_ffmpeg.c
@@ -37,8 +37,16 @@
 #warning "Not building ffmpeg support."
 #else
 
+#if LIBAVCODEC_AVCODEC_H
+#include <libavcodec/avcodec.h>
+#else
 #include <ffmpeg/avformat.h>
+#endif
+#if LIBAVFORMAT_AVFORMAT_H
+#include <libavformat/avformat.h>
+#else
 #include <ffmpeg/avcodec.h>
+#endif
 
 struct ffmpeg_data {
     AVFormatContext *format_context;
