? build.log
? enterprise
? runconfigure
? po/stamp-po
Index: src/xine-engine/load_plugins.c
===================================================================
--- src/xine-engine/load_plugins.c.orig
+++ src/xine-engine/load_plugins.c
@@ -402,7 +402,7 @@ static void _insert_node (xine_t *this,
     
     if (demux_old) {
       entry->priority = demux_new->priority = demux_old->priority;
-      lprintf("demux: %s, priority: %d\n", info->id, priority);
+      lprintf("demux: %s, priority: %d\n", info->id, entry->priority);
     } else {
       xprintf(this, XINE_VERBOSITY_LOG, 
               _("load_plugins: demuxer plugin %s does not provide a priority,"
@@ -419,7 +419,7 @@ static void _insert_node (xine_t *this,
     
     if (input_old) {
       entry->priority = input_new->priority = input_old->priority;
-      lprintf("input: %s, priority: %d\n", info->id, priority);
+      lprintf("input: %s, priority: %d\n", info->id, entry->priority);
     } else {
       xprintf(this, XINE_VERBOSITY_LOG, 
               _("load_plugins: input plugin %s does not provide a priority,"
