Index: stlport5.2-5.2.1/build/Makefiles/gmake/gcc.mak
===================================================================
--- stlport5.2-5.2.1.orig/build/Makefiles/gmake/gcc.mak	2008-12-10 10:56:51.000000000 +0100
+++ stlport5.2-5.2.1/build/Makefiles/gmake/gcc.mak	2009-01-11 12:03:07.000000000 +0100
@@ -246,7 +246,7 @@
 
 # optimization and debug compiler flags
 release-static : OPT += -O2
-release-shared : OPT += -O2
+release-shared : OPT +=
 
 dbg-static : OPT += -g
 dbg-shared : OPT += -g
