Index: stlport5.2-5.2.1/build/Makefiles/gmake/lib/gcc.mak
===================================================================
--- stlport5.2-5.2.1.orig/build/Makefiles/gmake/lib/gcc.mak	2009-01-11 12:17:25.000000000 +0100
+++ stlport5.2-5.2.1/build/Makefiles/gmake/lib/gcc.mak	2009-01-11 12:23:04.000000000 +0100
@@ -158,6 +158,9 @@
 END_OBJ := $(shell for o in crtendS.o crtn.o; do ${CXX} ${CXXFLAGS} -print-file-name=$$o; done)
 STDLIBS = -Wl,--whole-archive -lsupc++ ${_LGCC_EH} -Wl,--no-whole-archive ${_LGCC_S} -lpthread -lc -lm
 endif
+ifneq (,$(findstring $(M_ARCH),hppa parisc64 parisc))
+STDLIBS += -lgcc
+endif
 ifeq ($(OSNAME),openbsd)
 START_OBJ := $(shell for o in crtbeginS.o; do ${CXX} ${CXXFLAGS} -print-file-name=$$o; done)
 END_OBJ := $(shell for o in crtendS.o; do ${CXX} ${CXXFLAGS} -print-file-name=$$o; done)
Index: stlport5.2-5.2.1/build/Makefiles/gmake/app/gcc.mak
===================================================================
--- stlport5.2-5.2.1.orig/build/Makefiles/gmake/app/gcc.mak	2009-01-11 12:17:25.000000000 +0100
+++ stlport5.2-5.2.1/build/Makefiles/gmake/app/gcc.mak	2009-01-11 12:21:04.000000000 +0100
@@ -140,6 +140,9 @@
 START_OBJ := $(shell for o in crt1.o crti.o crtbegin.o; do ${CXX} ${CXXFLAGS} -print-file-name=$$o; done)
 END_OBJ := $(shell for o in crtend.o crtn.o; do ${CXX} ${CXXFLAGS} -print-file-name=$$o; done)
 STDLIBS = ${STLPORT_LIB} ${_LGCC_S} -lpthread -lc -lm
+ifneq (,$(findstring $(M_ARCH),hppa parisc64 parisc))
+STDLIBS += -lgcc
+endif
 endif
 
 ifeq ($(OSNAME),openbsd)
