Do we still need this patch? 2009-01-11, Torsten Werner

Index: stlport5.2-5.2.1/stlport/stl/debug/_debug.h
===================================================================
--- stlport5.2-5.2.1.orig/stlport/stl/debug/_debug.h	2008-12-10 10:56:51.000000000 +0100
+++ stlport5.2-5.2.1/stlport/stl/debug/_debug.h	2009-01-11 12:25:26.000000000 +0100
@@ -176,6 +176,7 @@
           }
 #  endif
 
+#  define _STLP_VERBOSE_PARAM(param) param
 #  define _STLP_DEBUG_CHECK(expr) _STLP_ASSERT(expr)
 
 #  if (_STLP_DEBUG_LEVEL == _STLP_STANDARD_DBG_LEVEL)
