diff --git a/build/Makefiles/gmake/CC.mak b/build/Makefiles/gmake/CC.mak
index 8383bb1..6e0e56b 100644
--- a/build/Makefiles/gmake/CC.mak
+++ b/build/Makefiles/gmake/CC.mak
@@ -25,9 +25,9 @@ endif
 
 CXX_VERSION := $(shell ${CXX} -V 2>&1 | grep ${CXX} | awk '{ print $$4; }')
 
-CXX_VERSION_MAJOR := $(shell echo ${CXX_VERSION} | awk 'BEGIN { FS = "."; } { print $$1; }')
-CXX_VERSION_MINOR := $(shell echo ${CXX_VERSION} | awk 'BEGIN { FS = "."; } { print $$2; }')
-CXX_VERSION_PATCH := $(shell echo ${CXX_VERSION} | awk 'BEGIN { FS = "."; } { if (NF > 2) {print $$3;}else{print "0"} }')
+CXX_VERSION_MAJOR := $(shell echo "${CXX_VERSION}" | awk 'BEGIN { FS = "."; } { print $$1; }')
+CXX_VERSION_MINOR := $(shell echo "${CXX_VERSION}" | awk 'BEGIN { FS = "."; } { print $$2; }')
+CXX_VERSION_PATCH := $(shell echo "${CXX_VERSION}" | awk 'BEGIN { FS = "."; } { if (NF > 2) {print $$3;}else{print "0"} }')
 
 DEFS ?=
 OPT ?=
diff --git a/build/Makefiles/gmake/aCC.mak b/build/Makefiles/gmake/aCC.mak
index 1ecbd14..c8853dd 100644
--- a/build/Makefiles/gmake/aCC.mak
+++ b/build/Makefiles/gmake/aCC.mak
@@ -25,8 +25,8 @@ CXX_VERSION := $(shell ${CXX} --version 2>&1 | grep ${CXX} | awk '{ print $$6; }
 ifeq ($(CXX_VERSION),)
 CXX_VERSION := $(shell ${CXX} --version)
 endif
-CXX_VERSION_MAJOR := $(shell echo ${CXX_VERSION} | awk 'BEGIN { FS = "."; } { print $$2; }')
-CXX_VERSION_MINOR := $(shell echo ${CXX_VERSION} | awk 'BEGIN { FS = "."; } { print $$3; }')
+CXX_VERSION_MAJOR := $(shell echo "${CXX_VERSION}" | awk 'BEGIN { FS = "."; } { print $$2; }')
+CXX_VERSION_MINOR := $(shell echo "${CXX_VERSION}" | awk 'BEGIN { FS = "."; } { print $$3; }')
 
 DEFS ?=
 OPT ?=
diff --git a/build/Makefiles/gmake/gcc.mak b/build/Makefiles/gmake/gcc.mak
index 16ec435..8afa377 100644
--- a/build/Makefiles/gmake/gcc.mak
+++ b/build/Makefiles/gmake/gcc.mak
@@ -32,9 +32,9 @@ AS := ${TARGET_OS}-${AS}
 endif
 
 CXX_VERSION := $(shell ${CXX} -dumpversion)
-CXX_VERSION_MAJOR := $(shell echo ${CXX_VERSION} | awk 'BEGIN { FS = "."; } { print $$1; }')
-CXX_VERSION_MINOR := $(shell echo ${CXX_VERSION} | awk 'BEGIN { FS = "."; } { print $$2; }')
-CXX_VERSION_PATCH := $(shell echo ${CXX_VERSION} | awk 'BEGIN { FS = "."; } { print $$3; }')
+CXX_VERSION_MAJOR := $(shell echo "${CXX_VERSION}" | awk 'BEGIN { FS = "."; } { print $$1; }')
+CXX_VERSION_MINOR := $(shell echo "${CXX_VERSION}" | awk 'BEGIN { FS = "."; } { print $$2; }')
+CXX_VERSION_PATCH := $(shell echo "${CXX_VERSION}" | awk 'BEGIN { FS = "."; } { print $$3; }')
 
 # Check that we need option -fuse-cxa-atexit for compiler
 _CXA_ATEXIT := $(shell ${CXX} -v 2>&1 | grep -q -e "--enable-__cxa_atexit" || echo "-fuse-cxa-atexit")
diff --git a/build/Makefiles/gmake/icc.mak b/build/Makefiles/gmake/icc.mak
index 8b0c47c..25b6322 100644
--- a/build/Makefiles/gmake/icc.mak
+++ b/build/Makefiles/gmake/icc.mak
@@ -25,9 +25,9 @@ ifeq (${CXX_VERSION},)
 CXX_VERSION := $(shell ${CXX} --version)
 endif
 
-CXX_VERSION_MAJOR := $(shell echo ${CXX_VERSION} | awk 'BEGIN { FS = "."; } { print $$1; }')
-CXX_VERSION_MINOR := $(shell echo ${CXX_VERSION} | awk 'BEGIN { FS = "."; } { print $$2; }')
-# CXX_VERSION_PATCH := $(shell echo ${CXX_VERSION} | awk 'BEGIN { FS = "."; } { print $$3; }')
+CXX_VERSION_MAJOR := $(shell echo "${CXX_VERSION}" | awk 'BEGIN { FS = "."; } { print $$1; }')
+CXX_VERSION_MINOR := $(shell echo "${CXX_VERSION}" | awk 'BEGIN { FS = "."; } { print $$2; }')
+# CXX_VERSION_PATCH := $(shell echo "${CXX_VERSION}" | awk 'BEGIN { FS = "."; } { print $$3; }')
 
 SYSINCLUDES := -I$(shell which icpc | xargs dirname | xargs dirname)/include/c++
 
