diff --git a/configure b/configure
index 3f4f286..71f3eca 100755
--- a/configure
+++ b/configure
@@ -255,6 +255,8 @@ do
       ;;
     --libdir=*)
       write_option "$option" INSTALL_LIB_DIR '${DESTDIR}'
+      write_option "$option" INSTALL_LIB_DIR_DBG '${DESTDIR}'
+      write_option "$option" INSTALL_LIB_DIR_STLDBG '${DESTDIR}'
       ;;
     --includedir=*)
       write_option "$option" INSTALL_HDR_DIR '${DESTDIR}'
