--- configure.old	2005-07-23 00:10:40 +0200
+++ configure	2005-07-23 00:11:21 +0200
@@ -2887,7 +2887,7 @@
   echo "$ac_t""$ac_cv_linux_config_modversions" 1>&6
   echo $ac_n "checking which kernel modules to build""... $ac_c" 1>&6
 echo "configure:2890: checking which kernel modules to build" >&5
-  if test "x$ac_linux_rhconfig" = "xyes" -o "x$ac_cv_linux_config_modversions" = "xno"; then
+  if test "x$ac_linux_rhconfig" = "xyes"; then
       MPS="MP SP"
   else
   if eval "test \"`echo '$''{'ac_cv_linux_config_smp'+set}'`\" = set"; then
