diff -Nru resin-3.1.8.vanilla/modules/c/src/apache2/mod_caucho.c resin-3.1.8/modules/c/src/apache2/mod_caucho.c
--- resin-3.1.8.vanilla/modules/c/src/apache2/mod_caucho.c	2009-01-10 15:28:46.000000000 +0100
+++ resin-3.1.8/modules/c/src/apache2/mod_caucho.c	2009-01-10 15:30:27.000000000 +0100
@@ -624,6 +624,16 @@
     }
   }
 #endif  
+  /* Add SSL Client certificate: (requires SSLOptions +StdEnvVars) */
+  {
+    int i;
+    char **env = (char **)ap_create_environment(r->pool, r->subprocess_env);
+    for (i = 0; env[i]; ++i) {
+      char *name = strsep(&(env[i]), "=");
+	  if (strncmp(name, "SSL_CLIENT_CERT", 16) == 0)
+		  cse_write_string(s, CSE_CLIENT_CERT, env[i]);
+    }
+  }
 }
 
 /**
