--- src/cf/osconf.m4.old	2006-09-09 14:10:43.000000000 +0200
+++ src/cf/osconf.m4	2006-09-09 14:11:41.000000000 +0200
@@ -3,11 +3,11 @@
 
 dnl defaults, override in case below as needed
 CFLAGS=
-XCFLAGS='${DBG} ${OPTMZ}'
+dnl XCFLAGS='${DBG} ${OPTMZ}'
 SHLIB_SUFFIX="so"
-CC="cc"
-CCOBJ="cc"
-MT_CC="cc"
+dnl CC="cc"
+CCOBJ="${CC}"
+MT_CC="${CC}"
 XLIBS="${LIB_AFSDB}"
 
 dnl debugging and optimization flag defaults
