Index: gpac-0.4.4/applications/osmo4_wx/Makefile
===================================================================
--- gpac-0.4.4.orig/applications/osmo4_wx/Makefile
+++ gpac-0.4.4/applications/osmo4_wx/Makefile
@@ -65,7 +65,7 @@ ifeq ($(CONFIG_DARWIN),yes)
 else
 	rm -f wxOsmo4.o
 	make -override BUILD_INSTALL=yes all
-	install -c -m 755 ../../bin/gcc/Osmo4 "$(prefix)/bin"
+	install -c -m 755 ../../bin/gcc/Osmo4 "$(DESTDIR)$(prefix)/bin"
 endif
 
 uninstall:
