Index: gpac-0.4.4/applications/mp4client/Makefile
===================================================================
--- gpac-0.4.4.orig/applications/mp4client/Makefile
+++ gpac-0.4.4/applications/mp4client/Makefile
@@ -48,8 +48,8 @@ MP4Client$(EXE): $(OBJS)
 clean: 
 	rm -f $(OBJS) ../../bin/gcc/$(PROG)
 
-install: clean
-	make -override BUILD_INSTALL=yes all
+install:
+	$(MAKE) -override BUILD_INSTALL=yes all
 	install -c -m 755 ../../bin/gcc/MP4Client "$(DESTDIR)$(prefix)/bin"
 
 uninstall:
Index: gpac-0.4.4/applications/osmo4_wx/Makefile
===================================================================
--- gpac-0.4.4.orig/applications/osmo4_wx/Makefile
+++ gpac-0.4.4/applications/osmo4_wx/Makefile
@@ -64,7 +64,7 @@ ifeq ($(CONFIG_DARWIN),yes)
 	echo -n 'APPLOsm4' > $(DESTDIR)$(mac_apps)/Osmo4.app/Contents/PkgInfo
 else
 	rm -f wxOsmo4.o
-	make -override BUILD_INSTALL=yes all
+	$(MAKE) -override BUILD_INSTALL=yes all
 	install -c -m 755 ../../bin/gcc/Osmo4 "$(DESTDIR)$(prefix)/bin"
 endif
 
