Index: gpac-0.4.4/applications/mp4box/Makefile
===================================================================
--- gpac-0.4.4.orig/applications/mp4box/Makefile
+++ gpac-0.4.4/applications/mp4box/Makefile
@@ -45,7 +45,7 @@ SRCS := $(OBJS:.o=.c) 
 all: $(PROG)
 
 $(PROG): $(OBJS)
-	$(CC) $(LDFLAGS) -o ../../bin/gcc/$@ $(OBJS) $(LINKFLAGS)
+	$(CC) $(LDFLAGS) -pthread -o ../../bin/gcc/$@ $(OBJS) $(LINKFLAGS)
 
 
 %.o: %.c
Index: gpac-0.4.4/applications/mp4client/Makefile
===================================================================
--- gpac-0.4.4.orig/applications/mp4client/Makefile
+++ gpac-0.4.4/applications/mp4client/Makefile
@@ -40,7 +40,7 @@ SRCS := $(OBJS:.o=.c) 
 all: $(PROG)
 
 MP4Client$(EXE): $(OBJS)
-	$(CC) $(LDFLAGS) -o ../../bin/gcc/$@ $(OBJS) -L../../bin/gcc -lgpac -lz
+	$(CC) $(LDFLAGS) -pthread -o ../../bin/gcc/$@ $(OBJS) -L../../bin/gcc -lgpac -lz
 
 %.o: %.c
 	$(CC) $(CFLAGS) $(INSTALL_FLAGS) -c -o $@ $< 
