Index: gpac-0.4.4/src/Makefile
===================================================================
--- gpac-0.4.4.orig/src/Makefile
+++ gpac-0.4.4/src/Makefile
@@ -91,6 +91,14 @@ CFLAGS+=-DGPAC_BIG_ENDIAN
 endif
 
 
+#4- flags used in utils/os_divers.c
+ifeq ($(CONFIG_LINUX), yes)
+CFLAGS+=-DCONFIG_LINUX
+endif
+ifeq ($(CONFIG_FREEBSD), yes)
+CFLAGS+=-DCONFIG_FREEBSD
+endif
+
 ## libgpac scenegraph compilation and linking options
 SCENEGRAPH_CFLAGS=
 ifeq ($(CONFIG_JS),no)
