Index: gpac-0.4.4/modules/ffmpeg_in/ffmpeg_decode.c
===================================================================
--- gpac-0.4.4.orig/modules/ffmpeg_in/ffmpeg_decode.c
+++ gpac-0.4.4/modules/ffmpeg_in/ffmpeg_decode.c
@@ -122,9 +122,17 @@ static GF_Err FFDEC_AttachStream(GF_Base
 			ffd->ctx->codec_type = CODEC_TYPE_AUDIO;
 			ffd->ctx->sample_rate = gf_bs_read_u32(bs);
 			ffd->ctx->channels = gf_bs_read_u16(bs);
+#if LIBAVCODEC_VERSION_INT < ((52<<16)+(0<<8)+0)
 			ffd->ctx->bits_per_sample = gf_bs_read_u16(bs);
+#else
+			ffd->ctx->bits_per_coded_sample = gf_bs_read_u16(bs);
+#endif
 			/*force 16 bits output*/
+#if LIBAVCODEC_VERSION_INT < ((52<<16)+(0<<8)+0)
 			ffd->ctx->bits_per_sample = 16;
+#else
+			ffd->ctx->bits_per_coded_sample = 16;
+#endif
 			ffd->ctx->frame_size = gf_bs_read_u16(bs);
 			ffd->ctx->block_align = gf_bs_read_u16(bs);
 		} else if (ffd->st==GF_STREAM_VISUAL) {
@@ -148,13 +156,21 @@ static GF_Err FFDEC_AttachStream(GF_Base
 			ffd->ctx->sample_rate = gf_bs_read_u16(bs);
 			ffd->ctx->frame_size = gf_bs_read_u16(bs);
 			ffd->ctx->channels = gf_bs_read_u8(bs);
+#if LIBAVCODEC_VERSION_INT < ((52<<16)+(0<<8)+0)
 			ffd->ctx->bits_per_sample = gf_bs_read_u8(bs);
+#else
+			ffd->ctx->bits_per_coded_sample = gf_bs_read_u8(bs);
+#endif
 			/*packed mode*/ gf_bs_read_u8(bs);
 			/*just in case...*/
 			if (codec_id == GF_4CC('a', 'm', 'r', ' ')) {
 			  ffd->ctx->sample_rate = 8000;
 			  ffd->ctx->channels = 1;
+#if LIBAVCODEC_VERSION_INT < ((52<<16)+(0<<8)+0)
 			  ffd->ctx->bits_per_sample = 16;
+#else
+			  ffd->ctx->bits_per_coded_sample = 16;
+#endif
 			  ffd->ctx->frame_size = 160;
 			}
 		} else if (ffd->st==GF_STREAM_VISUAL) {
@@ -245,12 +261,17 @@ static GF_Err FFDEC_AttachStream(GF_Base
 
 	/*setup audio streams*/
 	if (ffd->st==GF_STREAM_AUDIO) {
-		if ((ffd->codec->type == CODEC_ID_MP3LAME) || (ffd->codec->type == CODEC_ID_MP2)) {
+		if ((ffd->codec->type == CODEC_ID_MP3) || (ffd->codec->type == CODEC_ID_MP2)) {
 			ffd->ctx->frame_size = (ffd->ctx->sample_rate > 24000) ? 1152 : 576;
 		}
 		/*may be 0 (cfg not known yet)*/
+#if LIBAVCODEC_VERSION_INT < ((52<<16)+(0<<8)+0)
 		ffd->out_size = ffd->ctx->bits_per_sample * ffd->ctx->channels * ffd->ctx->frame_size / 8;
 		if (!ffd->ctx->bits_per_sample) ffd->ctx->bits_per_sample = 16;
+#else
+		ffd->out_size = ffd->ctx->bits_per_coded_sample * ffd->ctx->channels * ffd->ctx->frame_size / 8;
+		if (!ffd->ctx->bits_per_coded_sample) ffd->ctx->bits_per_coded_sample = 16;
+#endif
 		if (!ffd->ctx->sample_rate) ffd->ctx->sample_rate = 44100;
 		if (!ffd->ctx->channels) ffd->ctx->channels = 2;
 	} else {
@@ -330,7 +351,11 @@ static GF_Err FFDEC_GetCapabilities(GF_B
 		capability->cap.valueInt = ffd->ctx->channels;
 		break;
 	case GF_CODEC_BITS_PER_SAMPLE:
+#if LIBAVCODEC_VERSION_INT < ((52<<16)+(0<<8)+0)
 		capability->cap.valueInt = ffd->ctx->bits_per_sample;
+#else
+		capability->cap.valueInt = ffd->ctx->bits_per_coded_sample;
+#endif
 		break;
 	case GF_CODEC_BUFFER_MIN:
 		capability->cap.valueInt = (ffd->st==GF_STREAM_AUDIO) ? 4 : 1;
@@ -457,11 +482,20 @@ redecode:
 
 		/*first config*/
 		if (!ffd->out_size) {
+#if LIBAVCODEC_VERSION_INT < ((52<<16)+(0<<8)+0)
 			ffd->ctx->bits_per_sample = 16;
 			ffd->out_size = ffd->ctx->bits_per_sample * ffd->ctx->channels * ffd->ctx->frame_size / 8;
+#else
+			ffd->ctx->bits_per_coded_sample = 16;
+			ffd->out_size = ffd->ctx->bits_per_coded_sample * ffd->ctx->channels * ffd->ctx->frame_size / 8;
+#endif
 		}
 		if (ffd->out_size < (u32) gotpic) {
+#if LIBAVCODEC_VERSION_INT < ((52<<16)+(0<<8)+0)
 			ffd->ctx->bits_per_sample = 16;
+#else
+			ffd->ctx->bits_per_coded_sample = 16;
+#endif
 			/*looks like relying on frame_size is not a good idea for all codecs, so we use gotpic*/
 			(*outBufferLength) = ffd->out_size = gotpic;
 			return GF_BUFFER_TOO_SMALL;
Index: gpac-0.4.4/modules/ffmpeg_in/ffmpeg_demux.c
===================================================================
--- gpac-0.4.4.orig/modules/ffmpeg_in/ffmpeg_demux.c
+++ gpac-0.4.4/modules/ffmpeg_in/ffmpeg_demux.c
@@ -307,7 +307,11 @@ opaque_audio:
 			gf_bs_write_u32(bs, dec->codec_id);
 			gf_bs_write_u32(bs, dec->sample_rate);
 			gf_bs_write_u16(bs, dec->channels);
+#if LIBAVCODEC_VERSION_INT < ((52<<16)+(0<<8)+0)
 			gf_bs_write_u16(bs, dec->bits_per_sample);
+#else
+			gf_bs_write_u16(bs, dec->bits_per_coded_sample);
+#endif
 			gf_bs_write_u16(bs, dec->frame_size);
 			gf_bs_write_u16(bs, dec->block_align);
 
