--- comm-1.9.1/configure.in
+++ comm-1.9.1/configure.in
@@ -1487,6 +1487,9 @@
 x86_64 | ia64)
     CPU_ARCH="$OS_TEST"
     ;;
+arm*)
+    CPU_ARCH="arm"
+    ;;
 esac
 
 if test -z "$OS_TARGET"; then
--- comm-1.9.1/mozilla/js/src/configure.in
+++ comm-1.9.1/mozilla/js/src/configure.in
@@ -1221,6 +1221,9 @@
 x86_64 | ia64)
     CPU_ARCH="$OS_TEST"
     ;;
+arm*)
+    CPU_ARCH="arm"
+    ;;
 esac
 
 if test -z "$OS_TARGET"; then