--- boost/detail/endian.hpp.orig	2008-10-28 22:31:45.000000000 +0100
+++ boost/detail/endian.hpp	2008-10-29 00:21:05.000000000 +0100
@@ -52,7 +52,7 @@
    || defined(_POWER) || defined(__powerpc__) \
    || defined(__ppc__) || defined(__hpux) \
    || defined(_MIPSEB) || defined(_POWER) \
-   || defined(__s390__)
+   || defined(__s390__) || defined(__hppa)
 # define BOOST_BIG_ENDIAN
 # define BOOST_BYTE_ORDER 4321
 #elif defined(__i386__) || defined(__alpha__) \
