--- /usr/include/boost/crc.hpp~	2008-11-03 13:20:03.000000000 +0100
+++ /usr/include/boost/crc.hpp	2008-11-03 13:20:03.000000000 +0100
@@ -465,7 +465,7 @@
         // chokes over this unless we do the calculation for it:
         typedef value_type                   table_type[ 0x100 ];
 #else
-        typedef value_type                   table_type[ byte_combos ];
+        typedef value_type                   table_type[ 1ul << CHAR_BIT ];
 #endif
 
         static  void  init_table();
