--- src/cf/osconf.m4
+++ src/cf/osconf.m4
@@ -2,11 +2,11 @@
 AC_DEFUN([OPENAFS_OSCONF], [
 
 dnl defaults, override in case below as needed
-XCFLAGS='${DBG} ${OPTMZ}'
+dnl XCFLAGS='${DBG} ${OPTMZ}'
 SHLIB_SUFFIX="so"
-CC="cc"
-CCOBJ="cc"
-MT_CC="cc"
+dnl CC="cc"
+CCOBJ="${CC}"
+MT_CC="${CC}"
 XLIBS="${LIB_AFSDB} ${XBSA_XLIBS}"
 
 dnl debugging and optimization flag defaults
@@ -23,13 +23,13 @@
 
 dnl standard programs
 AC_PROG_RANLIB
-AC_CHECK_PROGS(AS, as, [${am_missing_run}as])
-AC_CHECK_PROGS(AR, ar, [${am_missing_run}ar])
+AC_CHECK_TOOLS(AS, as, [${am_missing_run}as])
+AC_CHECK_TOOLS(AR, ar, [${am_missing_run}ar])
 AC_CHECK_PROGS(MV, mv, [${am_missing_run}mv])
 AC_CHECK_PROGS(RM, rm, [${am_missing_run}rm])
-AC_CHECK_PROGS(LD, ld, [${am_missing_run}ld])
+AC_CHECK_TOOLS(LD, ld, [${am_missing_run}ld])
 AC_CHECK_PROGS(CP, cp, [${am_missing_run}cp])
-AC_CHECK_PROGS(STRIP, strip, [${am_missing_run}strip])
+AC_CHECK_TOOLS(STRIP, strip, [${am_missing_run}strip])
 AC_CHECK_PROGS(LORDER, lorder, [${am_missing_run}lorder])
 
 dnl TODO - need to disable STRIP if we are doing debugging in any user space code
