support separate linux kernel source and build dirs

diff -ur 2/openafs-1.4.8pre2/acinclude.m4 1/openafs-1.4.8pre2/acinclude.m4
--- 2/openafs-1.4.8pre2/acinclude.m4	2008-10-07 20:51:45.000000000 +0200
+++ 1/openafs-1.4.8pre2/acinclude.m4	2008-10-08 17:44:28.000000000 +0200
@@ -121,6 +121,9 @@
 AC_ARG_WITH(linux-kernel-headers,
 [  --with-linux-kernel-headers=path    	use the kernel headers found at path(optional, defaults to /usr/src/linux-2.4, then /usr/src/linux)]
 )
+AC_ARG_WITH(linux-kernel-build,
+[  --with-linux-kernel-build=path    	use the kernel build found at path(optional, defaults to /usr/src/linux-2.4, then /usr/src/linux)]
+)
 AC_ARG_WITH(bsd-kernel-headers,
 [  --with-bsd-kernel-headers=path    	use the kernel headers found at path(optional, defaults to /usr/src/sys)]
 )
@@ -210,15 +213,20 @@
 		     LINUX_KERNEL_PATH="/usr/src/linux"
 		   fi
 		 fi
-               if test -f "$LINUX_KERNEL_PATH/include/linux/utsrelease.h"; then
-		 linux_kvers=`fgrep UTS_RELEASE $LINUX_KERNEL_PATH/include/linux/utsrelease.h |awk 'BEGIN { FS="\"" } { print $[]2 }'|tail -n 1`
+		 if test "$with_linux_kernel_build" != "x"; then
+			 LINUX_KERNEL_BUILD="$with_linux_kernel_build"
+		 else
+		   LINUX_KERNEL_BUILD=$LINUX_KERNEL_PATH
+		 fi
+               if test -f "$LINUX_KERNEL_BUILD/include/linux/utsrelease.h"; then
+		 linux_kvers=`fgrep UTS_RELEASE $LINUX_KERNEL_BUILD/include/linux/utsrelease.h |awk 'BEGIN { FS="\"" } { print $[]2 }'|tail -n 1`
 		 LINUX_VERSION="$linux_kvers"
                else
-		 if test -f "$LINUX_KERNEL_PATH/include/linux/version.h"; then
-		  linux_kvers=`fgrep UTS_RELEASE $LINUX_KERNEL_PATH/include/linux/version.h |awk 'BEGIN { FS="\"" } { print $[]2 }'|tail -n 1`
+		 if test -f "$LINUX_KERNEL_BUILD/include/linux/version.h"; then
+		  linux_kvers=`fgrep UTS_RELEASE $LINUX_KERNEL_BUILD/include/linux/version.h |awk 'BEGIN { FS="\"" } { print $[]2 }'|tail -n 1`
 		  if test "x$linux_kvers" = "x"; then
-		    if test -f "$LINUX_KERNEL_PATH/include/linux/version-up.h"; then
-		      linux_kvers=`fgrep UTS_RELEASE $LINUX_KERNEL_PATH/include/linux/version-up.h |awk 'BEGIN { FS="\"" } { print $[]2 }'|tail -n 1`
+		    if test -f "$LINUX_KERNEL_BUILD/include/linux/version-up.h"; then
+		      linux_kvers=`fgrep UTS_RELEASE $LINUX_KERNEL_BUILD/include/linux/version-up.h |awk 'BEGIN { FS="\"" } { print $[]2 }'|tail -n 1`
 		      if test "x$linux_kvers" = "x"; then
 
 		        AC_MSG_ERROR(Linux headers lack version definition [2])
@@ -237,7 +245,7 @@
                     enable_kernel_module="no"
                  fi
                fi
-		 if test ! -f "$LINUX_KERNEL_PATH/include/linux/autoconf.h"; then
+		 if test ! -f "$LINUX_KERNEL_BUILD/include/linux/autoconf.h"; then
 		     enable_kernel_module="no"
 		 fi
 		 if test "x$enable_kernel_module" = "xno"; then
@@ -1395,6 +1403,7 @@
 AC_SUBST(ENABLE_KERNEL_MODULE)
 AC_SUBST(LIB_AFSDB)
 AC_SUBST(LINUX_KERNEL_PATH)
+AC_SUBST(LINUX_KERNEL_BUILD)
 AC_SUBST(BSD_KERNEL_PATH)
 AC_SUBST(BSD_KERNEL_BUILD)
 AC_SUBST(LINUX_VERSION)
diff -ur 2/openafs-1.4.8pre2/src/cf/linux-test1.m4 1/openafs-1.4.8pre2/src/cf/linux-test1.m4
--- 2/openafs-1.4.8pre2/src/cf/linux-test1.m4	2008-01-07 19:34:59.000000000 +0100
+++ 1/openafs-1.4.8pre2/src/cf/linux-test1.m4	2008-10-08 17:32:10.000000000 +0200
@@ -27,8 +27,8 @@
 
 MODULE_LICENSE("http://www.openafs.org/dl/license10.html");
 _ACEOF
-    echo make -C $LINUX_KERNEL_PATH M=$SRCDIR_PARENT/conftest.dir modules KBUILD_VERBOSE=1 >&AS_MESSAGE_LOG_FD &&
-    make -C $LINUX_KERNEL_PATH M=$SRCDIR_PARENT/conftest.dir modules KBUILD_VERBOSE=1 >&AS_MESSAGE_LOG_FD 2>conftest.err &&
+    echo make -C $LINUX_KERNEL_BUILD M=$SRCDIR_PARENT/conftest.dir modules KBUILD_VERBOSE=1 >&AS_MESSAGE_LOG_FD &&
+    make -C $LINUX_KERNEL_BUILD M=$SRCDIR_PARENT/conftest.dir modules KBUILD_VERBOSE=1 >&AS_MESSAGE_LOG_FD 2>conftest.err &&
     ! grep -i "^WARNING: .* undefined!$" conftest.err >/dev/null 2>&1
     then [$3]
     else
diff -ur 2/openafs-1.4.8pre2/src/config/Makefile.config.in 1/openafs-1.4.8pre2/src/config/Makefile.config.in
--- 2/openafs-1.4.8pre2/src/config/Makefile.config.in	2008-07-01 01:57:45.000000000 +0200
+++ 1/openafs-1.4.8pre2/src/config/Makefile.config.in	2008-10-08 17:42:41.000000000 +0200
@@ -48,6 +48,7 @@
 LIB_AFSDB = @LIB_AFSDB@
 LIBS = @LIBS@
 LINUX_KERNEL_PATH = @LINUX_KERNEL_PATH@
+LINUX_KERNEL_BUILD = @LINUX_KERNEL_BUILD@
 LINUX_VERSION = @LINUX_VERSION@
 LINUX_VERS = @LINUX_VERSION@
 LWP_OPTMZ = @LWP_OPTMZ@
diff -ur 2/openafs-1.4.8pre2/src/libafs/MakefileProto.LINUX.in 1/openafs-1.4.8pre2/src/libafs/MakefileProto.LINUX.in
--- 2/openafs-1.4.8pre2/src/libafs/MakefileProto.LINUX.in	2008-09-03 18:59:29.000000000 +0200
+++ 1/openafs-1.4.8pre2/src/libafs/MakefileProto.LINUX.in	2008-10-08 17:41:18.000000000 +0200
@@ -38,6 +38,7 @@
 COMMON_DEFINES=-D__KERNEL__ -DKERNEL -D_KERNEL -DMODULE ${SMP_DEF}
 
 LINUX_KERNEL_PATH=@LINUX_KERNEL_PATH@
+LINUX_KERNEL_BUILD=@LINUX_KERNEL_BUILD@
 
 # System specific build commands and flags
 <linux26 linux_26>
@@ -257,7 +258,7 @@
 .FORCE:
 libafs.ko: .FORCE
 	env EXTRA_CFLAGS="${EXTRA_CFLAGS}" @TOP_SRCDIR@/libafs/make_kbuild_makefile.pl ${KDIR} $@ @TOP_OBJDIR@/src/config/Makefile.config Makefile.afs Makefile.common
-	env EXTRA_CFLAGS="${EXTRA_CFLAGS}" $(MAKE) -C ${LINUX_KERNEL_PATH} M=@TOP_OBJDIR@/src/libafs/${KDIR} modules
+	env EXTRA_CFLAGS="${EXTRA_CFLAGS}" $(MAKE) -C ${LINUX_KERNEL_BUILD} M=@TOP_OBJDIR@/src/libafs/${KDIR} modules
         
 <all -linux26 -linux_26 -umlinux26>
 ${LIBAFS}: $(AFSAOBJS) $(AFSNONFSOBJS)
