--- a/src/bu_utils/fms.c
+++ b/src/bu_utils/fms.c
@@ -13,6 +13,8 @@
 
 #undef	IN
 #include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
 #include <errno.h>
 #include <sys/types.h>		/* for mtio.h */
 #include <afs/cmd.h>
--- a/src/config/config.c
+++ b/src/config/config.c
@@ -14,6 +14,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <errno.h>
+#include <stdlib.h>
 
 #include "AFS_component_version_number.c"
 
--- a/src/gtx/input.c
+++ b/src/gtx/input.c
@@ -14,6 +14,7 @@
 #ifdef AFS_HPUX_ENV
 #include <sys/types.h>
 #endif
+#include <stdlib.h>
 #include <lwp.h>
 
 #include "gtxobjects.h"
--- a/src/ptserver/map.c
+++ b/src/ptserver/map.c
@@ -33,6 +33,7 @@
 
 
 #ifdef SUPERGROUPS
+#include <string.h>
 #include <errno.h>
 #include "map.h"
 #ifdef STDLIB_HAS_MALLOC_PROTOS
--- a/src/venus/test/fulltest.c
+++ b/src/venus/test/fulltest.c
@@ -16,6 +16,7 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <stdio.h>
+#include <stdlib.h>
 #include <unistd.h>
 
 main(argc, argv)
--- a/src/venus/test/getinitparams.c
+++ b/src/venus/test/getinitparams.c
@@ -13,6 +13,7 @@
 
 #include <afs/param.h>
 #include <stdio.h>
+#include <stdlib.h>
 #include <netinet/in.h>
 #include <afs/vice.h>
 #include <afs/venus.h>
--- a/src/venus/test/idtest.c
+++ b/src/venus/test/idtest.c
@@ -10,7 +10,7 @@
 #include <afsconfig.h>
 #include <afs/param.h>
 #include <stdio.h>
-
+#include <stdlib.h>
 
 main(argc, argv)
 {
--- a/src/venus/up.c
+++ b/src/venus/up.c
@@ -41,6 +41,7 @@
 #include <netinet/in.h>
 #endif
 
+#include <stdlib.h>
 #include <string.h>
 
 #include <afs/venus.h>
--- a/src/vol/common.c
+++ b/src/vol/common.c
@@ -16,6 +16,7 @@
 
 #include <sys/types.h>
 #include <stdio.h>
+#include <stdlib.h>
 #include <afsconfig.h>
 #include <afs/param.h>
 
--- a/src/volser/common.c
+++ b/src/volser/common.c
@@ -12,6 +12,7 @@
 
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <afs/afsutil.h>
 #include <afs/com_err.h>
 
--- a/src/rx/xdr_arrayn.c
+++ b/src/rx/xdr_arrayn.c
@@ -60,6 +60,7 @@
 #endif
 #else
 #include <stdio.h>
+#include <string.h>
 #endif
 #include "xdr.h"
 
