http://bugs.gentoo.org/352356

--- openafs-1.4.14/src/ptserver/map.c
+++ openafs-1.4.14/src/ptserver/map.c
@@ -40,6 +40,9 @@
 #else
 #include "malloc.h"
 #endif
+#ifdef HAVE_STDINT_H
+# include <stdint.h>
+#endif
 
 #undef PRINT_MAP_ERROR
 /* #define MAP_DEBUG /**/
@@ -54,10 +57,10 @@
     int m_data[MDATA];
 };
 
-#define MAP(p)	((struct bitmap*)((int)(p)&~1))
-#define NEGMAP(p)	(((int)(p))&1)
+#define MAP(p)	((struct bitmap*)((intptr_t)(p)&~1))
+#define NEGMAP(p)	(((intptr_t)(p))&1)
 #define POSMAP(p)	(!NEGMAP(p))
-#define NOT_MAP(mp)	((struct map *) (((int)(mp)) ^ 1))
+#define NOT_MAP(mp)	((struct map *) (((intptr_t)(mp)) ^ 1))
 
 #define NUMBERTOBIT(n)	((n) & ((1<<LSHIFT)-1))
 #define NUMBERTOINDEX(n)	((n>>LSHIFT) & ((1<<MSHIFT)-1))
