diff -ur vlc-snapshot-20050607/modules/gui/wxwindows/wizard.cpp vlc-snapshot-20050607-cast/modules/gui/wxwindows/wizard.cpp
--- vlc-snapshot-20050607/modules/gui/wxwindows/wizard.cpp	2005-06-07 22:50:10.000000000 +0200
+++ vlc-snapshot-20050607-cast/modules/gui/wxwindows/wizard.cpp	2005-06-10 05:09:12.874656928 +0200
@@ -1085,7 +1085,7 @@
         }
     }
     p_parent->SetStream( methods_array[i_method].psz_access ,
-                         (char *)address_txtctrl->GetValue().mb_str() );
+                         const_cast<char*>((const char *)address_txtctrl->GetValue().mb_str()) );
 
     /* Set the action for the muxer page */
     ((wizEncapPage*)GetNext())->SetAction( p_parent->GetAction() );
