diff -ur -x '*~' vlc-0.8.2/modules/gui/wxwindows/preferences.cpp vlc-0.8.2-debug/modules/gui/wxwindows/preferences.cpp
--- vlc-0.8.2/modules/gui/wxwindows/preferences.cpp	2005-06-25 15:43:09.000000000 +0200
+++ vlc-0.8.2-debug/modules/gui/wxwindows/preferences.cpp	2005-07-13 12:15:01.659757328 +0200
@@ -336,7 +336,7 @@
     b_advanced = VLC_FALSE;
 
     root_item = AddRoot( wxT("") );
-    wxASSERT_MSG(root_item.IsOk(), "Could not add root item");
+    wxASSERT_MSG(root_item.IsOk(), wxT("Could not add root item"));
 
     wxImageList *p_images = new wxImageList( 16,16,TRUE );
     p_images->Add( wxIcon( audio_xpm ) );
