diff -Nru resin-4.0.25.vanilla/configure.ac resin-4.0.25/configure.ac
--- resin-4.0.25.vanilla/configure.ac	2012-02-11 19:37:27.161005340 +0100
+++ resin-4.0.25/configure.ac	2012-02-11 19:37:52.037162930 +0100
@@ -1122,11 +1122,6 @@
     echo "checking for JNI in $JAVA_HOME/include ... found"
   else
     JNI=""
-    AC_MSG_ERROR([
-
-    *** Can't find JNI directory in JAVA_HOME=${JAVA_HOME}
-    *** JNI is expected in ${JAVA_HOME}/include/$jni_os
-  ])
   fi
 
   HAS_JVMTI=""
@@ -1141,8 +1136,6 @@
     echo "Using JVMTI for class reloading"
     
     CFLAGS="$CFLAGS -DHAS_JVMTI"
-  else
-    AC_MSG_ERROR([Can't find JVMTI directory ${JNI_INCLUDE_DIR}/jvmti.h])
   fi
 fi
 
