diff -Nru resin-4.0.25.vanilla/configure.ac resin-4.0.25/configure.ac
--- resin-4.0.25.vanilla/configure.ac	2012-02-23 23:26:42.795297929 +0100
+++ resin-4.0.25/configure.ac	2012-02-23 23:30:57.796153924 +0100
@@ -788,11 +788,11 @@
       LDFLAGS_SHLIB="-shared -fPIC"
 
       if test -n "$b64_jni"; then
-        CFLAGS_JNI="$CFLAGS_JNI -m64 -fPIC"
-        LDFLAGS_SHLIB="$LDFLAGS_SHLIB -m64"
+        CFLAGS_JNI="$CFLAGS_JNI -fPIC"
+        LDFLAGS_SHLIB="$LDFLAGS_SHLIB"
       else
-        CFLAGS_JNI="$CFLAGS_JNI -m32"
-        LDFLAGS_SHLIB="$LDFLAGS_SHLIB -m32"
+        CFLAGS_JNI="$CFLAGS_JNI"
+        LDFLAGS_SHLIB="$LDFLAGS_SHLIB"
       fi
       ;;
 
@@ -805,12 +805,12 @@
         CFLAGS_JNI="$CFLAGS_JNI -fPIC -fno-omit-frame-pointer"
 
         if test -n "$b64"; then
-          CFLAGS="$CFLAGS -m64"
-          LDFLAGS_SHLIB="-m64 -R /usr/sfw/lib/64 -R /usr/local/lib/sparcv9"
+          CFLAGS="$CFLAGS"
+          LDFLAGS_SHLIB="-R /usr/sfw/lib/64 -R /usr/local/lib/sparcv9"
         fi
         if test -n "${b64_jni}"; then
-          CFLAGS_JNI="$CFLAGS_JNI -m64"
-          LDFLAGS_SHLIB="-m64 -R /usr/sfw/lib/64 -R /usr/local/lib/sparcv9"
+          CFLAGS_JNI="$CFLAGS_JNI"
+          LDFLAGS_SHLIB="-R /usr/sfw/lib/64 -R /usr/local/lib/sparcv9"
         fi
 
         LDFLAGS_SHLIB="$LDFLAGS_SHLIB -shared"
@@ -819,11 +819,11 @@
         CFLAGS_SHLIB="-mt -KPIC -xO2"
 
         if test -n "$b64"; then
-          CFLAGS="$CFLAGS -m64"
+          CFLAGS="$CFLAGS"
         fi
         if test -n "$b64_jni"; then
-          CFLAGS_JNI="$CFLAGS_JNI -m64"
-          LDFLAGS_SHLIB="$LDFLAGS_SHLIB -m64 -G -mt -z defs -z text"
+          CFLAGS_JNI="$CFLAGS_JNI"
+          LDFLAGS_SHLIB="$LDFLAGS_SHLIB -G -mt -z defs -z text"
           LDFLAGS_SHLIB="$LDFLAGS_SHLIB -R/usr/lib/64:/usr/sfw/lib/64:/usr/local/lib/sparcv9 -YP,/usr/lib/64:/usr/sfw/lib/64:/usr/local/lib/sparcv9"
         fi
 
@@ -840,7 +840,7 @@
         LDFLAGS_SHLIB="-arch x86_64 $LDFLAGS_SHLIB"
       fi
       if test -n "$b64_jni"; then
-        CFLAGS_JNI="$CFLAGS_JNI -m64"
+        CFLAGS_JNI="$CFLAGS_JNI"
         LDFLAGS_SHLIB="-arch x86_64 $LDFLAGS_SHLIB"
       fi
       ;;
@@ -871,8 +871,8 @@
 #  case "$target_cpu" in
 #    x86_64)
 #           if test -z "$b64_jni"; then
-#  	     CFLAGS_JNI="$CFLAGS_JNI -m32"
-#  	     LDFLAGS_SHLIB="$LDFLAGS_SHLIB -m32"
+#  	     CFLAGS_JNI="$CFLAGS_JNI"
+#  	     LDFLAGS_SHLIB="$LDFLAGS_SHLIB"
 #   	   fi
 #	;;
 #  esac	
diff -Nru resin-4.0.25.vanilla/modules/unix/src/configure.in resin-4.0.25/modules/unix/src/configure.in
--- resin-4.0.25.vanilla/modules/unix/src/configure.in	2012-02-23 23:26:41.279292834 +0100
+++ resin-4.0.25/modules/unix/src/configure.in	2012-02-23 23:30:42.076101182 +0100
@@ -434,7 +434,7 @@
 	LD_SHLIB=${CC}
 	
 	if test -n "$b64"; then
-  	  CFLAGS="$CFLAGS -m64"
+  	  CFLAGS="$CFLAGS"
 	  LDFLAGS_SHLIB="-shared -fPIC"
    	fi
 	;;
@@ -444,7 +444,7 @@
 	    CFLAGS_SHLIB="-fPIC"
 	    
 	   if test -n "$b64"; then
-  	     CFLAGS="$CFLAGS -m64"
+  	     CFLAGS="$CFLAGS"
    	   fi
 	else
 	    CFLAGS_SHLIB="-KPIC"
