Index: flac-1.1.2/src/libFLAC/libFLAC.m4
===================================================================
--- flac-1.1.2.orig/src/libFLAC/libFLAC.m4
+++ flac-1.1.2/src/libFLAC/libFLAC.m4
@@ -28,7 +28,7 @@ AC_ARG_ENABLE(libFLACtest, [  --disable-
   elif test "x$libFLAC_prefix" != "x" ; then
     LIBFLAC_CFLAGS="-I$libFLAC_prefix/include"
   elif test "$prefix" != "xNONE"; then
-    LIBFLAC_CFLAGS="-I$prefix/include"
+    LIBFLAC_CFLAGS="-I$libdir"
   fi
 
   AC_MSG_CHECKING(for libFLAC)
