Index: flac-1.1.2/include/FLAC/export.h
===================================================================
--- flac-1.1.2.orig/include/FLAC/export.h
+++ flac-1.1.2/include/FLAC/export.h
@@ -38,8 +38,8 @@
 
 #if defined(FLAC__NO_DLL) || !defined(_MSC_VER)
 
-#ifdef SUPPORT_ATTRIBUTE_VISIBILITY
-#define FLAC_API __attribute__((visibility("default")))
+#if defined(FLAC_API_EXPORTS) && SUPPORT_ATTRIBUTE_VISIBILITY
+#define FLAC_API __attribute__((visibility("protected")))
 #else
 #define FLAC_API
 #endif
Index: flac-1.1.2/include/FLAC++/export.h
===================================================================
--- flac-1.1.2.orig/include/FLAC++/export.h
+++ flac-1.1.2/include/FLAC++/export.h
@@ -38,7 +38,7 @@
 
 #if defined(FLAC__NO_DLL) || !defined(_MSC_VER)
 
-#ifdef SUPPORT_ATTRIBUTE_VISIBILITY
+#if defined(FLACPP_API_EXPORTS) && SUPPORT_ATTRIBUTE_VISIBILITY
 #define FLACPP_API __attribute__((visibility("default")))
 #else
 #define FLACPP_API
Index: flac-1.1.2/include/OggFLAC/export.h
===================================================================
--- flac-1.1.2.orig/include/OggFLAC/export.h
+++ flac-1.1.2/include/OggFLAC/export.h
@@ -38,8 +38,8 @@
 
 #if defined(FLAC__NO_DLL) || !defined(_MSC_VER)
 
-#ifdef SUPPORT_ATTRIBUTE_VISIBILITY
-#define OggFLAC_API __attribute__((visibility("default")))
+#if defined(OggFLAC_API_EXPORTS) && SUPPORT_ATTRIBUTE_VISIBILITY
+#define OggFLAC_API __attribute__((visibility("protected")))
 #else
 #define OggFLAC_API
 #endif
Index: flac-1.1.2/include/OggFLAC++/export.h
===================================================================
--- flac-1.1.2.orig/include/OggFLAC++/export.h
+++ flac-1.1.2/include/OggFLAC++/export.h
@@ -38,8 +38,7 @@
 
 #if defined(FLAC__NO_DLL) || !defined(_MSC_VER)
 
-#ifdef SUPPORT_ATTRIBUTE_VISIBILITY
-
+#if defined(OggFLACPP_API_EXPORTS) && SUPPORT_ATTRIBUTE_VISIBILITY
 #define OggFLACPP_API __attribute__((visibility("default")))
 #else
 #define OggFLACPP_API
Index: flac-1.1.2/src/libFLAC/Makefile.am
===================================================================
--- flac-1.1.2.orig/src/libFLAC/Makefile.am
+++ flac-1.1.2/src/libFLAC/Makefile.am
@@ -45,7 +45,7 @@ CPUCFLAGS = -maltivec -mabi=altivec -DFL
 endif
 endif
 
-AM_CFLAGS = $(DEBUGCFLAGS) $(CPUCFLAGS)
+AM_CFLAGS = $(DEBUGCFLAGS) $(CPUCFLAGS) -DFLAC_API_EXPORTS
 
 if FLaC__NO_ASM
 else
Index: flac-1.1.2/src/libFLAC++/Makefile.am
===================================================================
--- flac-1.1.2.orig/src/libFLAC++/Makefile.am
+++ flac-1.1.2/src/libFLAC++/Makefile.am
@@ -40,6 +40,7 @@ EXTRA_DIST = \
 	libFLAC++.m4
 
 # see 'http://www.gnu.org/software/libtool/manual.html#SEC35' for numbering convention
+libFLAC___la_CXXFLAGS = -DFLACPP_API_EXPORTS
 libFLAC___la_LDFLAGS = -version-info 5:0:0
 libFLAC___la_LIBADD = ../libFLAC/libFLAC.la
 
Index: flac-1.1.2/src/libOggFLAC/Makefile.am
===================================================================
--- flac-1.1.2.orig/src/libOggFLAC/Makefile.am
+++ flac-1.1.2/src/libOggFLAC/Makefile.am
@@ -33,7 +33,7 @@ if DEBUG
 DEBUGCFLAGS = 
 endif
 
-AM_CFLAGS = @OGG_CFLAGS@ $(DEBUGCFLAGS)
+AM_CFLAGS = @OGG_CFLAGS@ $(DEBUGCFLAGS) -DOggFLAC_API_EXPORTS
 
 SUBDIRS = include .
 
Index: flac-1.1.2/src/libOggFLAC++/Makefile.am
===================================================================
--- flac-1.1.2.orig/src/libOggFLAC++/Makefile.am
+++ flac-1.1.2/src/libOggFLAC++/Makefile.am
@@ -40,6 +40,7 @@ EXTRA_DIST = \
 	libOggFLAC++.m4
 
 # see 'http://www.gnu.org/software/libtool/manual.html#SEC35' for numbering convention
+libOggFLAC___la_CXXFLAGS = -DOggFLACPP_API_EXPORTS
 libOggFLAC___la_LDFLAGS = -version-info 2:0:0
 libOggFLAC___la_LIBADD = ../libOggFLAC/libOggFLAC.la
 
