--- samba-3.0.20/source/configure.in.orig	2005-09-02 09:43:16.000000000 +0200
+++ samba-3.0.20/source/configure.in	2005-09-02 09:46:24.000000000 +0200
@@ -231,6 +231,7 @@
 AC_SUBST(SMBWRAP_OBJS)
 AC_SUBST(SMBWRAP_INC)
 AC_SUBST(EXTRA_BIN_PROGS)
+AC_SUBST(EXTRA_SUBIN_PROGS)
 AC_SUBST(EXTRA_SBIN_PROGS)
 AC_SUBST(EXTRA_ALL_TARGETS)
 
@@ -3261,7 +3262,7 @@
 	*linux*)
 		AC_MSG_RESULT(yes)
 		AC_DEFINE(WITH_SMBMOUNT,1,[Whether to build smbmount])
-		EXTRA_BIN_PROGS="$EXTRA_BIN_PROGS bin/smbmount bin/smbmnt bin/smbumount"
+		EXTRA_SUBIN_PROGS="$EXTRA_SUBIN_PROGS client/mount.cifs$(EXEEXT) client/umount.cifs$(EXEEXT) bin/smbmount$(EXEEXT) bin/smbmnt$(EXEEXT) bin/smbumount$(EXEEXT)"
 		;;
 	*)
 		AC_MSG_ERROR(not on a linux system!)
--- samba-3.0.20/source/Makefile.in.orig	2005-09-08 15:26:39.000000000 +0200
+++ samba-3.0.20/source/Makefile.in	2005-09-08 15:27:46.000000000 +0200
@@ -25,6 +25,7 @@
 PYTHON=@PYTHON@
 PERL=@PERL@
 
+LAZYLDFLAGS=-Wl,-z,now
 TERMLDFLAGS=@TERMLDFLAGS@
 TERMLIBS=@TERMLIBS@
 PRINT_LIBS=@PRINT_LIBS@
@@ -71,6 +72,9 @@
 # The permissions to give the executables
 INSTALLPERMS = 0755
 
+# The permissions for setuid executables
+INSTALLSUPERMS = 4755
+
 # set these to where to find various files
 # These can be overridden by command line switches (see smbd(8))
 # or in smb.conf (see smb.conf(5))
@@ -128,6 +132,8 @@
 	bin/profiles@EXEEXT@ bin/ntlm_auth@EXEEXT@ \
 	bin/smbcquotas@EXEEXT@
 
+SUBIN_PROGS = @EXTRA_SUBIN_PROGS@
+
 TORTURE_PROGS = bin/smbtorture@EXEEXT@ bin/msgtest@EXEEXT@ \
 	bin/masktest@EXEEXT@ bin/locktest@EXEEXT@ \
 	bin/locktest2@EXEEXT@ bin/nsstest@EXEEXT@ bin/vfstest@EXEEXT@
@@ -553,6 +559,10 @@
 
 UMOUNT_OBJ = client/smbumount.o
 
+MCIFS_OBJ = client/mount.cifs.o
+
+UMCIFS_OBJ = client/umount.cifs.o
+
 NMBLOOKUP_OBJ = utils/nmblookup.o $(PARAM_OBJ) $(LIBNMB_OBJ) \
                $(LIB_NONSMBD_OBJ) $(POPT_LIB_OBJ) $(SECRETS_OBJ) $(LIBSAMBA_OBJ)
 
@@ -697,7 +707,7 @@
 ######################################################################
 # now the rules...
 ######################################################################
-all : SHOWFLAGS proto_exists $(SBIN_PROGS) $(BIN_PROGS) $(SHLIBS) \
+all : SHOWFLAGS proto_exists $(SBIN_PROGS) $(BIN_PROGS) $(SUBIN_PROGS) $(SHLIBS) \
 	$(MODULES) @EXTRA_ALL_TARGETS@
 
 pam_smbpass : SHOWFLAGS bin/pam_smbpass.@SHLIBEXT@
@@ -862,15 +872,23 @@
 
 bin/smbmount@EXEEXT@: $(MOUNT_OBJ) bin/.dummy
 	@echo Linking $@
-	@$(CC) $(FLAGS) @PIE_LDFLAGS@ -o $@ $(MOUNT_OBJ) $(DYNEXP) $(LDFLAGS) $(LIBS) $(KRB5LIBS) $(LDAP_LIBS)
+	@$(CC) $(FLAGS) @PIE_LDFLAGS@ -o $@ $(MOUNT_OBJ) $(DYNEXP) $(LDFLAGS) $(LIBS) $(KRB5LIBS) $(LDAP_LIBS) $(LAZYLDFLAGS)
 
 bin/smbmnt@EXEEXT@: $(MNT_OBJ) bin/.dummy
 	@echo Linking $@
-	@$(CC) $(FLAGS) @PIE_LDFLAGS@ -o $@ $(MNT_OBJ) $(DYNEXP) $(LDFLAGS) 
+	@$(CC) $(FLAGS) @PIE_LDFLAGS@ -o $@ $(MNT_OBJ) $(DYNEXP) $(LDFLAGS) $(LAZYLDFLAGS)
 
 bin/smbumount@EXEEXT@: $(UMOUNT_OBJ) bin/.dummy
 	@echo Linking $@
-	@$(CC) $(FLAGS) @PIE_LDFLAGS@ -o $@ $(UMOUNT_OBJ) $(DYNEXP) $(LDFLAGS)
+	@$(CC) $(FLAGS) @PIE_LDFLAGS@ -o $@ $(UMOUNT_OBJ) $(DYNEXP) $(LDFLAGS) $(LAZYLDFLAGS)
+
+client/mount.cifs@EXEEXT@: $(MCIFS_OBJ) bin/.dummy
+	@echo Linking $@
+	@$(CC) $(FLAGS) -o $@ $(MCIFS_OBJ) $(DYNEXP) $(LDFLAGS) $(LAZYLDFLAGS)
+
+client/umount.cifs@EXEEXT@: $(UMCIFS_OBJ) bin/.dummy
+	@echo Linking $@
+	@$(CC) $(FLAGS) -o $@ $(UMCIFS_OBJ) $(DYNEXP) $(LDFLAGS) $(LAZYLDFLAGS)
 
 bin/testparm@EXEEXT@: $(TESTPARM_OBJ) @BUILD_POPT@ bin/.dummy
 	@echo Linking $@
@@ -1295,7 +1313,7 @@
 
 bin/t_snprintf@EXEEXT@: lib/snprintf.c
 	$(CC) $(FLAGS) @PIE_LDFLAGS@ -o $@ $(DYNEXP) -DTEST_SNPRINTF lib/snprintf.c -lm
-install: installbin installman installscripts installdat installswat installmodules @INSTALLCLIENT@
+install: installbin installsubin installman installscripts installdat installswat installmodules @INSTALLCLIENT@
 
 install-everything: install installmodules
 
@@ -1317,6 +1335,8 @@
 	@$(SHELL) $(srcdir)/script/installbin.sh $(INSTALLPERMS) $(DESTDIR) $(DESTDIR)$(BASEDIR) $(DESTDIR)$(SBINDIR) $(DESTDIR)$(LIBDIR) $(DESTDIR)$(VARDIR) $(SBIN_PROGS)
 	@$(SHELL) $(srcdir)/script/installbin.sh $(INSTALLPERMS) $(DESTDIR) $(DESTDIR)$(BASEDIR) $(DESTDIR)$(BINDIR) $(DESTDIR)$(LIBDIR) $(DESTDIR)$(VARDIR) $(BIN_PROGS)
 
+installsubin: all installdirs
+	@$(SHELL) $(srcdir)/script/installbin.sh $(INSTALLSUPERMS) $(DESTDIR) $(DESTDIR)$(BASEDIR) $(DESTDIR)$(BINDIR) $(DESTDIR)$(LIBDIR) $(DESTDIR)$(VARDIR) $(SUBIN_PROGS)
 
 # Some symlinks are required for the 'probing' of modules. 
 # This mechanism should go at some point..
