--- samba-3.0.14a.orig/source/stf/smbcontrol.py	2005-02-25 18:59:28.000000000 +0100
+++ samba-3.0.14a/source/stf/smbcontrol.py	2005-07-21 11:40:45.000000000 +0200
@@ -227,10 +227,8 @@
         out = self.runcmd("smbcontrol --bogus self noop", expectedResult = 1)
 
 tests = [NoArgs, OneArg, SmbdDest, NmbdDest, WinbinddDest, PidDest,
-         SelfDest, BadDest, BadCmd, Debug, ForceElection, SamSync,
-         SamRepl, DmallocMark, DmallocChanged, Shutdown, DrvUpgrade,
-         CloseShare, Ping, Debuglevel, PrintNotify, Profile, ProfileLevel,
-         TimeoutArg, ConfigFileArg, BogusArg]
+         SelfDest, BadDest, BadCmd, PrintNotify, TimeoutArg, 
+         ConfigFileArg, BogusArg]
 
 # Handle execution of this file as a main program
 
