diff -u --recursive samba-3.0.14a/source/smbd/dir.c samba-3.0.14a.mod/source/smbd/dir.c
--- samba-3.0.14a/source/smbd/dir.c	2005-04-14 02:14:20.000000000 -0400
+++ samba-3.0.14a.mod/source/smbd/dir.c	2005-08-12 16:31:32.000000000 -0400
@@ -953,7 +953,7 @@
 			return False;
 		}
 		/* Honour _hide_special_ option */
-		if (hide_special && !file_is_special(conn, entry, pst)) {
+		if (hide_special && file_is_special(conn, entry, pst)) {
 			SAFE_FREE(entry);
 			return False;
 		}
