diff -uNr samba-3.0.22.orig/source/aclocal.m4 samba-3.0.22/source/aclocal.m4
--- samba-3.0.22.orig/source/aclocal.m4	2005-10-17 19:45:07.000000000 -0700
+++ samba-3.0.22/source/aclocal.m4	2006-06-29 14:58:57.000000000 -0700
@@ -748,23 +748,23 @@
 dnl		[ACTION-IF-CROSS-COMPILING = RUNTIME-ERROR])
 AC_DEFUN( [AC_TRY_RUN_STRICT],
 [
-	old_CFLAGS="$CFLAGS";
-	CFLAGS="$2";
-	export CFLAGS;
-	old_CPPFLAGS="$CPPFLAGS";
-	CPPFLAGS="$3";
-	export CPPFLAGS;
-	old_LDFLAGS="$LDFLAGS";
-	LDFLAGS="$4";
-	export LDFLAGS;
-	AC_TRY_RUN([$1],[$5],[$6],[$7]);
-	CFLAGS="$old_CFLAGS";
-	old_CFLAGS="";
-	export CFLAGS;
-	CPPFLAGS="$old_CPPFLAGS";
-	old_CPPFLAGS="";
-	export CPPFLAGS;
-	LDFLAGS="$old_LDFLAGS";
-	old_LDFLAGS="";
-	export LDFLAGS;
+	old_CFLAGS="$CFLAGS"
+	CFLAGS="$2"
+	export CFLAGS
+	old_CPPFLAGS="$CPPFLAGS"
+	CPPFLAGS="$3"
+	export CPPFLAGS
+	old_LDFLAGS="$LDFLAGS"
+	LDFLAGS="$4"
+	export LDFLAGS
+	AC_TRY_RUN([$1],[$5],[$6],[$7])
+	CFLAGS="$old_CFLAGS"
+	old_CFLAGS=""
+	export CFLAGS
+	CPPFLAGS="$old_CPPFLAGS"
+	old_CPPFLAGS=""
+	export CPPFLAGS
+	LDFLAGS="$old_LDFLAGS"
+	old_LDFLAGS=""
+	export LDFLAGS
 ])
