--- source/aclocal.m4.orig	2006-07-11 16:25:49.000000000 +0200
+++ source/aclocal.m4	2006-07-11 16:26:38.000000000 +0200
@@ -795,25 +795,25 @@
 dnl		[ACTION-IF-CROSS-COMPILING = RUNTIME-ERROR])
 AC_DEFUN( [AC_TRY_RUN_STRICT],
 [
-	old_CFLAGS="$CFLAGS";
-	CFLAGS="$2";
-	export CFLAGS;
-	old_CPPFLAGS="$CPPFLAGS";
-	CPPFLAGS="$3";
-	export CPPFLAGS;
-	old_LDFLAGS="$LDFLAGS";
-	LDFLAGS="$4";
-	export LDFLAGS;
+	old_CFLAGS="$CFLAGS"
+	CFLAGS="$2"
+	export CFLAGS
+	old_CPPFLAGS="$CPPFLAGS"
+	CPPFLAGS="$3"
+	export CPPFLAGS
+	old_LDFLAGS="$LDFLAGS"
+	LDFLAGS="$4"
+	export LDFLAGS
 	AC_TRY_RUN([$1],[$5],[$6],[$7])
-	CFLAGS="$old_CFLAGS";
-	old_CFLAGS="";
-	export CFLAGS;
-	CPPFLAGS="$old_CPPFLAGS";
-	old_CPPFLAGS="";
-	export CPPFLAGS;
-	LDFLAGS="$old_LDFLAGS";
-	old_LDFLAGS="";
-	export LDFLAGS;
+	CFLAGS="$old_CFLAGS"
+	old_CFLAGS=""
+	export CFLAGS
+	CPPFLAGS="$old_CPPFLAGS"
+	old_CPPFLAGS=""
+	export CPPFLAGS
+	LDFLAGS="$old_LDFLAGS"
+	old_LDFLAGS=""
+	export LDFLAGS
 ])
 
 dnl SMB_CHECK_SYSCONF(varname)
