diff -Nrc -x *~ gtk+-1.2.10.orig/gtk/gtkdnd.c gtk+-1.2.10/gtk/gtkdnd.c
*** gtk+-1.2.10.orig/gtk/gtkdnd.c	Fri Mar 23 18:24:31 2001
--- gtk+-1.2.10/gtk/gtkdnd.c	Sun Oct 28 12:06:34 2001
***************
*** 2673,2679 ****
    if (gdk_drag_motion (info->context, dest_window, protocol,
  		       x_root, y_root, action, 
  		       possible_actions,
! 		       time))
      {
        if (info->last_event)
  	gdk_event_free ((GdkEvent *)info->last_event);
--- 2673,2680 ----
    if (gdk_drag_motion (info->context, dest_window, protocol,
  		       x_root, y_root, action, 
  		       possible_actions,
! 		       time)
!       && info->last_event != event)
      {
        if (info->last_event)
  	gdk_event_free ((GdkEvent *)info->last_event);
