diff -Nrc -x *~ gtk+-1.2.9.orig/gdk/gdk.c gtk+-1.2.9/gdk/gdk.c
*** gtk+-1.2.9.orig/gdk/gdk.c	Thu Feb  1 20:35:44 2001
--- gtk+-1.2.9/gdk/gdk.c	Sat Mar 31 22:00:05 2001
***************
*** 704,719 ****
    
    if (return_val == Success)
      {
!       if (!window_private->destroyed)
! 	return_val = XGrabPointer (window_private->xdisplay,
! 				   xwindow,
! 				   owner_events,
! 				   xevent_mask,
! 				   GrabModeAsync, GrabModeAsync,
! 				   xconfine_to,
! 				   xcursor,
! 				   time);
!       else
  	return_val = AlreadyGrabbed;
      }
    
--- 704,721 ----
    
    if (return_val == Success)
      {
!       if (!window_private->destroyed) {
! 	if (getenv ("GTK_NO_GRAB") == NULL) {
! 	  return_val = XGrabPointer (window_private->xdisplay,
! 				     xwindow,
! 				     owner_events,
! 				     xevent_mask,
! 				     GrabModeAsync, GrabModeAsync,
! 				     xconfine_to,
! 				     xcursor,
! 				     time);
! 	}
!       } else
  	return_val = AlreadyGrabbed;
      }
    
***************
*** 802,814 ****
    window_private = (GdkWindowPrivate*) window;
    xwindow = window_private->xwindow;
    
!   if (!window_private->destroyed)
!     return XGrabKeyboard (window_private->xdisplay,
! 			  xwindow,
! 			  owner_events,
! 			  GrabModeAsync, GrabModeAsync,
! 			  time);
!   else
      return AlreadyGrabbed;
  }
  
--- 804,819 ----
    window_private = (GdkWindowPrivate*) window;
    xwindow = window_private->xwindow;
    
!   if (!window_private->destroyed) {
!     if (getenv ("GTK_NO_GRAB") == NULL) {
!       return XGrabKeyboard (window_private->xdisplay,
! 			    xwindow,
! 			    owner_events,
! 			    GrabModeAsync, GrabModeAsync,
! 			    time);
!     } else
!       return Success;
!   } else
      return AlreadyGrabbed;
  }
  
