diff -Nrc -x *~ gtk+-1.2.9.orig/gtk/gtkstyle.c gtk+-1.2.9/gtk/gtkstyle.c
*** gtk+-1.2.9.orig/gtk/gtkstyle.c	Tue Feb 20 11:46:58 2001
--- gtk+-1.2.9/gtk/gtkstyle.c	Mon Apr  2 00:33:55 2001
***************
*** 2255,2262 ****
    else if (height == -1)
      gdk_window_get_size (window, NULL, &height);
    
!   if (!style->bg_pixmap[state_type] || 
!       gdk_window_get_type (window) == GDK_WINDOW_PIXMAP)
      {
        if (area)
  	gdk_gc_set_clip_rectangle (style->bg_gc[state_type], area);
--- 2255,2261 ----
    else if (height == -1)
      gdk_window_get_size (window, NULL, &height);
    
!   if (!style->bg_pixmap[state_type])
      {
        if (area)
  	gdk_gc_set_clip_rectangle (style->bg_gc[state_type], area);
***************
*** 2327,2334 ****
    else
      gc1 = style->bg_gc[state_type];
    
!   if (!style->bg_pixmap[state_type] || gc1 != style->bg_gc[state_type] ||
!       gdk_window_get_type (window) == GDK_WINDOW_PIXMAP)
      {
        if (area)
  	gdk_gc_set_clip_rectangle (gc1, area);
--- 2326,2332 ----
    else
      gc1 = style->bg_gc[state_type];
    
!   if (!style->bg_pixmap[state_type] || gc1 != style->bg_gc[state_type])
      {
        if (area)
  	gdk_gc_set_clip_rectangle (gc1, area);
