--- gtk+-1.2.6/gtk/gtkrange.c.wide	Wed Feb  9 21:01:40 2000
+++ gtk+-1.2.6/gtk/gtkrange.c	Wed Feb  9 21:01:21 2000
@@ -160,8 +160,8 @@
   widget_class->focus_out_event = gtk_range_focus_out;
   widget_class->style_set = gtk_range_style_set;
 
-  class->slider_width = 11;
-  class->stepper_size = 11;
+  class->slider_width = 13;
+  class->stepper_size = 13;
   class->stepper_slider_spacing = 1;
   class->min_slider_size = 7;
   class->trough = 1;
