--- linux-2.6.3-openpax/security/selinux/avc.c	2004-02-17 21:58:52.000000000 -0600
+++ linux-2.6.3/security/selinux/avc.c	2004-03-07 18:24:57.000000000 -0600
@@ -143,6 +143,11 @@
 	char *scontext;
 	u32 scontext_len;
 
+#ifdef CONFIG_GRKERNSEC_PROC_IPADDR
+	if (current->curr_ip)
+		printk("ipaddr=%u.%u.%u.%u ", NIPQUAD(current->curr_ip));
+#endif /* CONFIG_GRKERNSEC_PROC_IPADDR */
+
  	rc = security_sid_to_context(ssid, &scontext, &scontext_len);
 	if (rc)
 		printk("ssid=%d", ssid);
