--- src/cf/osconf.m4	2011-02-14 18:18:29.000000000 +0000
+++ src/cf/osconf.m4	2011-03-19 08:30:06.349743558 +0000
@@ -2,8 +2,8 @@
 AC_DEFUN([OPENAFS_OSCONF], [
 
 dnl defaults, override in case below as needed
-CFLAGS=
-XCFLAGS='${DBG} ${OPTMZ}'
+dnl CFLAGS=
+dnl XCFLAGS='${DBG} ${OPTMZ}'
 RXDEBUG="-DRXDEBUG"
 SHLIB_SUFFIX="so"
 CCOBJ="$CC"
@@ -25,13 +25,13 @@
 
 dnl standard programs
 AC_PROG_RANLIB
-AC_CHECK_PROGS(AS, as, [${am_missing_run}as])
-AC_CHECK_PROGS(AR, ar, [${am_missing_run}ar])
+AC_CHECK_TOOL(AS, as, [${am_missing_run}as])
+AC_CHECK_TOOL(AR, ar, [${am_missing_run}ar])
 AC_CHECK_PROGS(MV, mv, [${am_missing_run}mv])
 AC_CHECK_PROGS(RM, rm, [${am_missing_run}rm])
-AC_CHECK_PROGS(LD, ld, [${am_missing_run}ld])
+AC_CHECK_TOOL(LD, ld, [${am_missing_run}ld])
 AC_CHECK_PROGS(CP, cp, [${am_missing_run}cp])
-AC_CHECK_PROGS(STRIP, strip, [${am_missing_run}strip])
+AC_CHECK_TOOL(STRIP, strip, [${am_missing_run}strip])
 AC_CHECK_PROGS(LORDER, lorder, [${am_missing_run}lorder])
 
 dnl TODO - need to disable STRIP if we are doing debugging in any user space code
