--- src/pam/Makefile.in	2011-02-14 18:18:29.000000000 +0000
+++ src/pam/Makefile.in	2011-03-19 08:42:19.239259375 +0000
@@ -29,7 +29,7 @@
 	  afs_pam_msg.o afs_message.o AFS_component_version_number.o
    OBJS = $(SHOBJS) test_pam.o
 INCLUDES=-I${TOP_OBJDIR}/src/config -I${TOP_INCDIR} 
-CFLAGS =  ${PAM_DBG} ${PAM_OPTMZ} ${INCLUDES} ${PAM_CFLAGS}
+CFLAGS =  ${PAM_DBG} ${PAM_OPTMZ} ${INCLUDES} ${PAM_CFLAGS} -DBINDIR='"${bindir}"'
 
 all: test_pam pam_afs.so.1 pam_afs.krb.so.1
 
--- src/pam/afs_util.h
+++ src/pam/afs_util.h
@@ -24,9 +24,9 @@
 		   const char *lifetime, const char *cell_name);
 extern afs_int32 getPAG(void);
 
-#define KLOG "/usr/afsws/bin/klog"
-#define KLOGKRB "/usr/afsws/bin/klog.krb"
-#define UNLOG "/usr/afsws/bin/unlog"
+#define KLOG BINDIR "/klog"
+#define KLOGKRB BINDIR "/klog.krb"
+#define UNLOG BINDIR "/unlog"
 #define IGNORE_MAX 1000
 
 #if	defined(AFS_HPUX_ENV)
