--- a/acinclude.m4
+++ b/acinclude.m4
@@ -406,7 +406,7 @@ case $system in
 		MKAFS_OSTYPE=DARWIN
                 AC_MSG_RESULT(x86_darwin)
                 ;;
-	i386-*-freebsd*)
+	i?86-*-freebsd*)
 		MKAFS_OSTYPE=FBSD
 		AC_MSG_RESULT(i386_fbsd)
 		;;
--- a/src/comerr/Makefile.in
+++ b/src/comerr/Makefile.in
@@ -36,7 +36,7 @@ compile_et:	compile_et.o error_table.o
 	*_linux* | *_umlinux* ) \
 	${CC} ${CFLAGS} -o compile_et compile_et.o error_table.o -L${TOP_LIBDIR} -lafsutil;; \
 	* ) \
-	${CC} ${CFLAGS} -o compile_et compile_et.o error_table.o -L${TOP_LIBDIR} -lafsutil -ll;; \
+	${CC} ${CFLAGS} -o compile_et compile_et.o error_table.o -L${TOP_LIBDIR} -lafsutil -lfl;; \
 	esac
 
 libafscom_err.a: error_msg.o et_name.o com_err.o AFS_component_version_number.o
--- a/src/libafs/MakefileProto.FBSD.in
+++ b/src/libafs/MakefileProto.FBSD.in
@@ -45,7 +45,7 @@ KDEFS=-Wall -nostdinc -I/usr/include -D_KERNEL -DKLD_MODULE \
 <i386_fbsd_60>
 	-mno-mmx -mno-3dnow -mno-sse -mno-sse2 \
 <all>
-	-mno-align-long-strings -fno-common -ffreestanding \
+	-fno-common -ffreestanding \
 	-I${KBLD} -include opt_global.h -fno-strict-aliasing
 
 DBUG = -O2
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -1162,7 +1162,14 @@ if test "x$with_bsd_kernel_build" != "x"; then
 else
 	case $AFS_SYSNAME in
 		*_fbsd_*)
-			BSD_KERNEL_BUILD="${BSD_KERNEL_PATH}/${HOST_CPU}/compile/GENERIC"
+			case $HOST_CPU in
+				i?86)
+					BSD_KERNEL_BUILD="${BSD_KERNEL_PATH}/i386/compile/GENERIC"
+					;;
+				*)
+					BSD_KERNEL_BUILD="${BSD_KERNEL_PATH}/${HOST_CPU}/compile/GENERIC"
+					;;
+			esac
 			;;
 		*_nbsd*)
 			BSD_KERNEL_BUILD="${BSD_KERNEL_PATH}/arch/${HOST_CPU}/compile/GENERIC"
