diff -uprN -X dontdiff linux-2.6.7/include/linux/sysctl.h linux-2.6.7-tcp-nmap-freak/include/linux/sysctl.h
--- linux-2.6.7/include/linux/sysctl.h	2004-06-24 11:46:04.972813656 +0200
+++ linux-2.6.7-tcp-nmap-freak/include/linux/sysctl.h	2004-06-24 11:45:38.577826304 +0200
@@ -342,6 +342,8 @@ enum
         NET_TCP_STACK_SYNFIN=108,
         NET_TCP_STACK_BOGUS=109,
         NET_TCP_STACK_ACK=110,
+        NET_IPV4_ICMP_RESTRICT=111,
+        NET_IPV4_TCP_RESTRICT=112,
 };
 
 enum {
diff -uprN -X dontdiff linux-2.6.7/net/ipv4/icmp.c linux-2.6.7-tcp-nmap-freak/net/ipv4/icmp.c
--- linux-2.6.7/net/ipv4/icmp.c	2004-06-16 07:20:03.000000000 +0200
+++ linux-2.6.7-tcp-nmap-freak/net/ipv4/icmp.c	2004-06-24 11:42:41.157798248 +0200
@@ -190,6 +190,10 @@ struct icmp_err icmp_err_convert[] = {
 int sysctl_icmp_echo_ignore_all;
 int sysctl_icmp_echo_ignore_broadcasts;
 
+#ifdef CONFIG_IP_NMAP_FREAK
+int sysctl_icmp_restrict = 0;
+#endif
+
 /* Control parameter - ignore bogus broadcast responses? */
 int sysctl_icmp_ignore_bogus_error_responses;
 
@@ -782,7 +786,12 @@ static void icmp_echo(struct sk_buff *sk
 		icmp_param.offset	   = 0;
 		icmp_param.data_len	   = skb->len;
 		icmp_param.head_len	   = sizeof(struct icmphdr);
+#ifdef CONFIG_IP_NMAP_FREAK
+		if (!sysctl_icmp_restrict)
+			icmp_reply(&icmp_param, skb);
+#else
 		icmp_reply(&icmp_param, skb);
+#endif
 	}
 }
 
diff -uprN -X dontdiff linux-2.6.7/net/ipv4/Kconfig linux-2.6.7-tcp-nmap-freak/net/ipv4/Kconfig
--- linux-2.6.7/net/ipv4/Kconfig	2004-06-24 11:46:05.000809400 +0200
+++ linux-2.6.7-tcp-nmap-freak/net/ipv4/Kconfig	2004-06-24 11:44:48.391455792 +0200
@@ -324,6 +324,21 @@ config SYN_COOKIES
 
 	  If unsure, say N.
 
+config IP_NMAP_FREAK
+	bool "IP: NMAP freak (disabled per default)"
+	depends on INET
+	default n
+	---help---
+	  This is a feature to prevent stealth,fin,rst scans and slows down
+	  the tcp connect scan, it also does not show the Operating System.
+
+	  You can turn this on(1) and off(0) using /proc
+
+	   echo 1 > /proc/sys/net/ipv4/tcp_restrict
+	   echo 1 > /proc/sys/net/ipv4/icmp_restrict
+
+	  If unsure, say N.
+
 config NET_STEALTH
 	bool "IP: TCP stealth options (enabled per default)"
 	depends on INET
diff -uprN -X dontdiff linux-2.6.7/net/ipv4/sysctl_net_ipv4.c linux-2.6.7-tcp-nmap-freak/net/ipv4/sysctl_net_ipv4.c
--- linux-2.6.7/net/ipv4/sysctl_net_ipv4.c	2004-06-24 11:46:05.015807120 +0200
+++ linux-2.6.7-tcp-nmap-freak/net/ipv4/sysctl_net_ipv4.c	2004-06-24 11:44:48.393455488 +0200
@@ -37,6 +37,11 @@ extern int sysctl_ip_dynaddr;
 extern int sysctl_icmp_ratelimit;
 extern int sysctl_icmp_ratemask;
 
+#ifdef CONFIG_IP_NMAP_FREAK
+extern int sysctl_icmp_restrict;
+extern int sysctl_tcp_restrict;
+#endif
+
 /* From igmp.c */
 extern int sysctl_igmp_max_memberships;
 extern int sysctl_igmp_max_msf;
@@ -325,6 +330,24 @@ ctl_table ipv4_table[] = {
 		.proc_handler	= &proc_dointvec
 	},
 #endif
+#ifdef CONFIG_IP_NMAP_FREAK
+	{
+		.ctl_name	= NET_IPV4_ICMP_RESTRICT,
+		.procname	= "icmp_restrict",
+		.data		= &sysctl_icmp_restrict,
+		.maxlen		= sizeof(int),
+		.mode		= 0644,
+		.proc_handler	= &proc_dointvec
+	},
+	{
+		.ctl_name	= NET_IPV4_TCP_RESTRICT,
+		.procname	= "tcp_restrict",
+		.data		= &sysctl_tcp_restrict,
+		.maxlen		= sizeof(int),
+		.mode		= 0644,
+		.proc_handler	= &proc_dointvec
+	},
+#endif
 #ifdef CONFIG_NET_STEALTH
 	{
 		.ctl_name	= NET_TCP_STACK_SYNFIN,
diff -uprN -X dontdiff linux-2.6.7/net/ipv4/tcp_ipv4.c linux-2.6.7-tcp-nmap-freak/net/ipv4/tcp_ipv4.c
--- linux-2.6.7/net/ipv4/tcp_ipv4.c	2004-06-24 11:46:05.115791920 +0200
+++ linux-2.6.7-tcp-nmap-freak/net/ipv4/tcp_ipv4.c	2004-06-24 11:44:48.427450320 +0200
@@ -79,6 +79,10 @@ extern int sysctl_ip_dynaddr;
 int sysctl_tcp_tw_reuse;
 int sysctl_tcp_low_latency;
 
+#ifdef CONFIG_IP_NMAP_FREAK
+int sysctl_tcp_restrict = 0;
+#endif
+
 #ifdef CONFIG_NET_STEALTH
 extern int sysctl_tcp_ignore_synfin;
 extern int sysctl_tcp_ignore_bogus;
@@ -1179,6 +1183,11 @@ static void tcp_v4_send_reset(struct sk_
 	struct tcphdr rth;
 	struct ip_reply_arg arg;
 
+#ifdef CONFIG_IP_NMAP_FREAK
+	if (sysctl_tcp_restrict)
+		return;
+#endif
+
 	/* Never send a reset in response to a reset. */
 	if (th->rst)
 		return;
