diff -uprN -X dontdiff /home/albeiro/src/rsbac/linux-2.6.7/drivers/net/Kconfig linux-2.6.7-netdev_random/drivers/net/Kconfig
--- /home/albeiro/src/rsbac/linux-2.6.7/drivers/net/Kconfig	2004-06-16 07:19:52.000000000 +0200
+++ linux-2.6.7-netdev_random/drivers/net/Kconfig	2004-06-23 23:16:46.124597472 +0200
@@ -104,6 +104,22 @@ config TUN
 
 	  If you don't know what to use this for, you don't need it.
 
+config NET_RANDOM
+	bool "Allow Net Devices to contribute to /dev/random"
+	depends on NETDEVICES && EXPERIMENTAL
+	---help---
+    	  If you say Y here, network device interrupts will contribute to the
+          kernel entropy pool. Normally, block devices and some other devices
+          feed the pool. Some systems, such as those that are headless or diskless,
+          need additional entropy sources.  Some people, however, feel that network
+          devices should not contribute to /dev/random because an external attacker
+          could observe incoming packets in an attempt to learn the entropy pool's
+          state.  If you say N, no network device will contribute entropy.
+
+          If you believe there is a chance of your network packets being observed
+          and you doubt the security of the entropy pool's one-way hash, do not
+	  enable this.  If unsure, say N.
+
 config ETHERTAP
 	tristate "Ethertap network tap"
 	depends on NETDEVICES && EXPERIMENTAL && NETLINK_DEV
diff -uprN -X dontdiff /home/albeiro/src/rsbac/linux-2.6.7/include/asm-alpha/signal.h linux-2.6.7-netdev_random/include/asm-alpha/signal.h
--- /home/albeiro/src/rsbac/linux-2.6.7/include/asm-alpha/signal.h	2004-06-16 07:19:23.000000000 +0200
+++ linux-2.6.7-netdev_random/include/asm-alpha/signal.h	2004-06-23 23:16:46.133596104 +0200
@@ -121,8 +121,15 @@ typedef unsigned long sigset_t;
 #define SA_PROBE		SA_ONESHOT
 #define SA_SAMPLE_RANDOM	SA_RESTART
 #define SA_SHIRQ		0x40000000
+
+#ifdef CONFIG_NET_RANDOM
+#define SA_NET_RANDOM	SA_SAMPLE_RANDOM
+#else
+#define SA_NET_RANDOM	0
 #endif
 
+#endif /* __KERNEL__ */
+
 #define SIG_BLOCK          1	/* for blocking signals */
 #define SIG_UNBLOCK        2	/* for unblocking signals */
 #define SIG_SETMASK        3	/* for setting the signal mask */
diff -uprN -X dontdiff /home/albeiro/src/rsbac/linux-2.6.7/include/asm-arm/signal.h linux-2.6.7-netdev_random/include/asm-arm/signal.h
--- /home/albeiro/src/rsbac/linux-2.6.7/include/asm-arm/signal.h	2004-06-16 07:19:22.000000000 +0200
+++ linux-2.6.7-netdev_random/include/asm-arm/signal.h	2004-06-23 23:16:46.141594888 +0200
@@ -126,8 +126,15 @@ typedef unsigned long sigset_t;
 #define SA_SAMPLE_RANDOM	0x10000000
 #define SA_IRQNOMASK		0x08000000
 #define SA_SHIRQ		0x04000000
+
+#ifdef CONFIG_NET_RANDOM
+#define SA_NET_RANDOM	SA_SAMPLE_RANDOM
+#else
+#define SA_NET_RANDOM	0
 #endif
 
+#endif /* __KERNEL__ */
+
 #define SIG_BLOCK          0	/* for blocking signals */
 #define SIG_UNBLOCK        1	/* for unblocking signals */
 #define SIG_SETMASK        2	/* for setting the signal mask */
diff -uprN -X dontdiff /home/albeiro/src/rsbac/linux-2.6.7/include/asm-cris/signal.h linux-2.6.7-netdev_random/include/asm-cris/signal.h
--- /home/albeiro/src/rsbac/linux-2.6.7/include/asm-cris/signal.h	2004-06-16 07:19:52.000000000 +0200
+++ linux-2.6.7-netdev_random/include/asm-cris/signal.h	2004-06-23 23:16:46.152593216 +0200
@@ -120,8 +120,15 @@ typedef unsigned long sigset_t;
 #define SA_PROBE		SA_ONESHOT
 #define SA_SAMPLE_RANDOM	SA_RESTART
 #define SA_SHIRQ		0x04000000
+
+#ifdef CONFIG_NET_RANDOM
+#define SA_NET_RANDOM	SA_SAMPLE_RANDOM
+#else
+#define SA_NET_RANDOM	0
 #endif
 
+#endif /* __KERNEL__ */
+
 #define SIG_BLOCK          0	/* for blocking signals */
 #define SIG_UNBLOCK        1	/* for unblocking signals */
 #define SIG_SETMASK        2	/* for setting the signal mask */
diff -uprN -X dontdiff /home/albeiro/src/rsbac/linux-2.6.7/include/asm-i386/signal.h linux-2.6.7-netdev_random/include/asm-i386/signal.h
--- /home/albeiro/src/rsbac/linux-2.6.7/include/asm-i386/signal.h	2004-06-16 07:18:59.000000000 +0200
+++ linux-2.6.7-netdev_random/include/asm-i386/signal.h	2004-06-23 23:16:46.161591848 +0200
@@ -121,8 +121,15 @@ typedef unsigned long sigset_t;
 #define SA_PROBE		SA_ONESHOT
 #define SA_SAMPLE_RANDOM	SA_RESTART
 #define SA_SHIRQ		0x04000000
+
+#ifdef CONFIG_NET_RANDOM
+#define SA_NET_RANDOM	SA_SAMPLE_RANDOM
+#else
+#define SA_NET_RANDOM	0
 #endif
 
+#endif /* __KERNEL__ */
+
 #define SIG_BLOCK          0	/* for blocking signals */
 #define SIG_UNBLOCK        1	/* for unblocking signals */
 #define SIG_SETMASK        2	/* for setting the signal mask */
diff -uprN -X dontdiff /home/albeiro/src/rsbac/linux-2.6.7/include/asm-ia64/signal.h linux-2.6.7-netdev_random/include/asm-ia64/signal.h
--- /home/albeiro/src/rsbac/linux-2.6.7/include/asm-ia64/signal.h	2004-06-16 07:18:58.000000000 +0200
+++ linux-2.6.7-netdev_random/include/asm-ia64/signal.h	2004-06-23 23:16:46.169590632 +0200
@@ -126,6 +126,12 @@
 #define SA_SHIRQ		0x04000000
 #define SA_PERCPU_IRQ		0x02000000
 
+#ifdef CONFIG_NET_RANDOM
+#define SA_NET_RANDOM	SA_SAMPLE_RANDOM
+#else
+#define SA_NET_RANDOM	0
+#endif
+
 #endif /* __KERNEL__ */
 
 #define SIG_BLOCK          0	/* for blocking signals */
diff -uprN -X dontdiff /home/albeiro/src/rsbac/linux-2.6.7/include/asm-m68k/signal.h linux-2.6.7-netdev_random/include/asm-m68k/signal.h
--- /home/albeiro/src/rsbac/linux-2.6.7/include/asm-m68k/signal.h	2004-06-16 07:19:01.000000000 +0200
+++ linux-2.6.7-netdev_random/include/asm-m68k/signal.h	2004-06-23 23:16:46.178589264 +0200
@@ -116,8 +116,15 @@ typedef unsigned long sigset_t;
 #define SA_PROBE		SA_ONESHOT
 #define SA_SAMPLE_RANDOM	SA_RESTART
 #define SA_SHIRQ		0x04000000
+
+#ifdef CONFIG_NET_RANDOM
+#define SA_NET_RANDOM	SA_SAMPLE_RANDOM
+#else
+#define SA_NET_RANDOM	0
 #endif
 
+#endif /* __KERNEL__ */
+
 #define SIG_BLOCK          0	/* for blocking signals */
 #define SIG_UNBLOCK        1	/* for unblocking signals */
 #define SIG_SETMASK        2	/* for setting the signal mask */
diff -uprN -X dontdiff /home/albeiro/src/rsbac/linux-2.6.7/include/asm-mips/signal.h linux-2.6.7-netdev_random/include/asm-mips/signal.h
--- /home/albeiro/src/rsbac/linux-2.6.7/include/asm-mips/signal.h	2004-06-16 07:19:10.000000000 +0200
+++ linux-2.6.7-netdev_random/include/asm-mips/signal.h	2004-06-23 23:16:46.188587744 +0200
@@ -111,6 +111,12 @@ typedef unsigned long old_sigset_t;		/* 
 #define SA_SAMPLE_RANDOM	SA_RESTART
 #define SA_SHIRQ		0x02000000
 
+#ifdef CONFIG_NET_RANDOM
+#define SA_NET_RANDOM	SA_SAMPLE_RANDOM
+#else
+#define SA_NET_RANDOM	0
+#endif
+
 #endif /* __KERNEL__ */
 
 #define SIG_BLOCK	1	/* for blocking signals */
diff -uprN -X dontdiff /home/albeiro/src/rsbac/linux-2.6.7/include/asm-parisc/signal.h linux-2.6.7-netdev_random/include/asm-parisc/signal.h
--- /home/albeiro/src/rsbac/linux-2.6.7/include/asm-parisc/signal.h	2004-06-16 07:20:04.000000000 +0200
+++ linux-2.6.7-netdev_random/include/asm-parisc/signal.h	2004-06-23 23:16:46.223582424 +0200
@@ -100,6 +100,12 @@
 #define SA_SAMPLE_RANDOM	SA_RESTART
 #define SA_SHIRQ		0x04000000
 
+#ifdef CONFIG_NET_RANDOM
+#define SA_NET_RANDOM	SA_SAMPLE_RANDOM
+#else
+#define SA_NET_RANDOM	0
+#endif
+
 #endif /* __KERNEL__ */
 
 #define SIG_BLOCK          0	/* for blocking signals */
diff -uprN -X dontdiff /home/albeiro/src/rsbac/linux-2.6.7/include/asm-ppc64/signal.h linux-2.6.7-netdev_random/include/asm-ppc64/signal.h
--- /home/albeiro/src/rsbac/linux-2.6.7/include/asm-ppc64/signal.h	2004-06-16 07:19:22.000000000 +0200
+++ linux-2.6.7-netdev_random/include/asm-ppc64/signal.h	2004-06-23 23:16:46.231581208 +0200
@@ -107,8 +107,15 @@ typedef struct {
 #define SA_PROBE		SA_ONESHOT
 #define SA_SAMPLE_RANDOM	SA_RESTART
 #define SA_SHIRQ		0x04000000
+
+#ifdef CONFIG_NET_RANDOM
+#define SA_NET_RANDOM	SA_SAMPLE_RANDOM
+#else
+#define SA_NET_RANDOM	0
 #endif
 
+#endif /* __KERNEL__ */
+
 #define SIG_BLOCK          0	/* for blocking signals */
 #define SIG_UNBLOCK        1	/* for unblocking signals */
 #define SIG_SETMASK        2	/* for setting the signal mask */
diff -uprN -X dontdiff /home/albeiro/src/rsbac/linux-2.6.7/include/asm-s390/signal.h linux-2.6.7-netdev_random/include/asm-s390/signal.h
--- /home/albeiro/src/rsbac/linux-2.6.7/include/asm-s390/signal.h	2004-06-16 07:19:43.000000000 +0200
+++ linux-2.6.7-netdev_random/include/asm-s390/signal.h	2004-06-23 23:16:46.240579840 +0200
@@ -129,8 +129,15 @@ typedef unsigned long sigset_t;
 #define SA_PROBE                SA_ONESHOT
 #define SA_SAMPLE_RANDOM        SA_RESTART
 #define SA_SHIRQ                0x04000000
+
+#ifdef CONFIG_NET_RANDOM
+#define SA_NET_RANDOM	SA_SAMPLE_RANDOM
+#else
+#define SA_NET_RANDOM	0
 #endif
 
+#endif /* __KERNEL__ */
+
 #define SIG_BLOCK          0    /* for blocking signals */
 #define SIG_UNBLOCK        1    /* for unblocking signals */
 #define SIG_SETMASK        2    /* for setting the signal mask */
diff -uprN -X dontdiff /home/albeiro/src/rsbac/linux-2.6.7/include/asm-sh/signal.h linux-2.6.7-netdev_random/include/asm-sh/signal.h
--- /home/albeiro/src/rsbac/linux-2.6.7/include/asm-sh/signal.h	2004-06-16 07:19:36.000000000 +0200
+++ linux-2.6.7-netdev_random/include/asm-sh/signal.h	2004-06-23 23:16:46.257577256 +0200
@@ -120,8 +120,15 @@ typedef unsigned long sigset_t;
 #define SA_PROBE		SA_ONESHOT
 #define SA_SAMPLE_RANDOM	SA_RESTART
 #define SA_SHIRQ		0x04000000
+
+#ifdef CONFIG_NET_RANDOM
+#define SA_NET_RANDOM	SA_SAMPLE_RANDOM
+#else
+#define SA_NET_RANDOM	0
 #endif
 
+#endif /* __KERNEL__ */
+
 #define SIG_BLOCK          0	/* for blocking signals */
 #define SIG_UNBLOCK        1	/* for unblocking signals */
 #define SIG_SETMASK        2	/* for setting the signal mask */
diff -uprN -X dontdiff /home/albeiro/src/rsbac/linux-2.6.7/include/asm-sparc/signal.h linux-2.6.7-netdev_random/include/asm-sparc/signal.h
--- /home/albeiro/src/rsbac/linux-2.6.7/include/asm-sparc/signal.h	2004-06-16 07:19:36.000000000 +0200
+++ linux-2.6.7-netdev_random/include/asm-sparc/signal.h	2004-06-23 23:16:46.268575584 +0200
@@ -176,8 +176,15 @@ struct sigstack {
 #define SA_PROBE SA_ONESHOT
 #define SA_SAMPLE_RANDOM SA_RESTART
 #define SA_STATIC_ALLOC		0x80
+
+#ifdef CONFIG_NET_RANDOM
+#define SA_NET_RANDOM	SA_SAMPLE_RANDOM
+#else
+#define SA_NET_RANDOM	0
 #endif
 
+#endif /* __KERNEL__ */
+
 /* Type of a signal handler.  */
 #ifdef __KERNEL__
 typedef void (*__sighandler_t)(int, int, struct sigcontext *, char *);
diff -uprN -X dontdiff /home/albeiro/src/rsbac/linux-2.6.7/include/asm-sparc64/signal.h linux-2.6.7-netdev_random/include/asm-sparc64/signal.h
--- /home/albeiro/src/rsbac/linux-2.6.7/include/asm-sparc64/signal.h	2004-06-16 07:19:02.000000000 +0200
+++ linux-2.6.7-netdev_random/include/asm-sparc64/signal.h	2004-06-23 23:16:46.278574064 +0200
@@ -182,8 +182,15 @@ struct sigstack {
 #define SA_PROBE SA_ONESHOT
 #define SA_SAMPLE_RANDOM SA_RESTART
 #define SA_STATIC_ALLOC		0x80
+
+#ifdef CONFIG_NET_RANDOM
+#define SA_NET_RANDOM	SA_SAMPLE_RANDOM
+#else
+#define SA_NET_RANDOM	0
 #endif
 
+#endif /* __KERNEL__ */
+
 /* Type of a signal handler.  */
 #ifdef __KERNEL__
 typedef void (*__sighandler_t)(int, struct sigcontext *);
diff -uprN -X dontdiff /home/albeiro/src/rsbac/linux-2.6.7/include/asm-x86_64/signal.h linux-2.6.7-netdev_random/include/asm-x86_64/signal.h
--- /home/albeiro/src/rsbac/linux-2.6.7/include/asm-x86_64/signal.h	2004-06-16 07:18:52.000000000 +0200
+++ linux-2.6.7-netdev_random/include/asm-x86_64/signal.h	2004-06-23 23:16:46.288572544 +0200
@@ -128,8 +128,15 @@ typedef unsigned long sigset_t;
 #define SA_PROBE		SA_ONESHOT
 #define SA_SAMPLE_RANDOM	SA_RESTART
 #define SA_SHIRQ		0x04000000
+
+#ifdef CONFIG_NET_RANDOM
+#define SA_NET_RANDOM	SA_SAMPLE_RANDOM
+#else
+#define SA_NET_RANDOM	0
 #endif
 
+#endif /* __KERNEL__ */
+
 #define SIG_BLOCK          0	/* for blocking signals */
 #define SIG_UNBLOCK        1	/* for unblocking signals */
 #define SIG_SETMASK        2	/* for setting the signal mask */
