--- xpdf/GlobalParams.cc
+++ xpdf/GlobalParams.cc
@@ -897,8 +897,12 @@
 				     xpdfKeyContextAny, "zoomPercent(125)"));
   keyBindings->append(new KeyBinding('+', xpdfKeyModNone,
 				     xpdfKeyContextAny, "zoomIn"));
+  keyBindings->append(new KeyBinding(xpdfKeyCodeAdd, xpdfKeyModNone,
+				     xpdfKeyContextAny, "zoomIn"));
   keyBindings->append(new KeyBinding('-', xpdfKeyModNone,
 				     xpdfKeyContextAny, "zoomOut"));
+  keyBindings->append(new KeyBinding(xpdfKeyCodeSubtract, xpdfKeyModNone,
+				     xpdfKeyContextAny, "zoomOut"));
   keyBindings->append(new KeyBinding('z', xpdfKeyModNone,
 				     xpdfKeyContextAny, "zoomFitPage"));
   keyBindings->append(new KeyBinding('w', xpdfKeyModNone,
--- xpdf/GlobalParams.h
+++ xpdf/GlobalParams.h
@@ -158,6 +158,8 @@
 #define xpdfKeyCodeDown           0x100d
 #define xpdfKeyCodeF1             0x1100
 #define xpdfKeyCodeF35            0x1122
+#define xpdfKeyCodeAdd            0x1200
+#define xpdfKeyCodeSubtract       0x1201
 #define xpdfKeyCodeMousePress1    0x2001
 #define xpdfKeyCodeMousePress2    0x2002
 #define xpdfKeyCodeMousePress3    0x2003
--- xpdf/XPDFViewer.cc
+++ xpdf/XPDFViewer.cc
@@ -110,6 +110,12 @@
 #ifndef XK_KP_Delete
 #define XK_KP_Delete            0xFF9F
 #endif
+#ifndef XK_KP_Add
+#define XK_KP_Add               0xFFAB
+#endif
+#ifndef XK_KP_Subtract
+#define XK_KP_Subtract          0xFFAD
+#endif
 
 //------------------------------------------------------------------------
 // GUI includes
@@ -576,6 +582,10 @@
     keyCode = xpdfKeyCodeDown;
   } else if (key >= XK_F1 && key <= XK_F35) {
     keyCode = xpdfKeyCodeF1 + (key - XK_F1);
+  } else if (key == XK_KP_Add) {
+    keyCode = xpdfKeyCodeAdd;
+  } else if (key == XK_KP_Subtract) {
+    keyCode = xpdfKeyCodeSubtract;
   } else {
     return;
   }
