diff -Naur evms-2.5.5.orig/tests/seq_dd.c evms-2.5.5/tests/seq_dd.c
--- evms-2.5.5.orig/tests/seq_dd.c	2009-02-20 08:53:31.000000000 +0100
+++ evms-2.5.5/tests/seq_dd.c	2009-02-20 08:54:18.000000000 +0100
@@ -108,7 +108,7 @@
 		return 1;
 	}
 
-	fd = open(filename, O_CREAT | O_WRONLY);
+	fd = open(filename, O_CREAT | O_WRONLY, S_IRUSR | S_IWUSR | S_IRGRP);
 	if (fd < 0) {
 		fprintf(stderr, "Error opening \"%s\": %s\n",
 			filename, strerror(errno));
