diff -Nru resin-3.1.5.vanilla/configure.ac resin-3.1.5/configure.ac
--- resin-3.1.5.vanilla/configure.ac	2008-02-27 22:36:36.000000000 +0100
+++ resin-3.1.5/configure.ac	2008-02-27 22:37:00.000000000 +0100
@@ -1146,10 +1146,3 @@
 modules/c/src/resin_os/Makefile
 contrib/init.resin
 ], [])
-
-if test -n "${resin_pro_plugin}"; then
-  AC_OUTPUT([
-    modules/c/src/resin/Makefile
-    modules/c/src/resinssl/Makefile
-  ])
-fi
