Index: blender-2.65/build_files/scons/tools/Blender.py
===================================================================
--- blender-2.65.orig/build_files/scons/tools/Blender.py
+++ blender-2.65/build_files/scons/tools/Blender.py
@@ -309,10 +309,7 @@ def setup_syslibs(lenv):
         syslibs += Split(lenv['BF_PTHREADS_LIB'])
     if lenv['WITH_BF_COLLADA'] and not lenv['WITH_BF_STATICOPENCOLLADA']:
         syslibs.append(lenv['BF_PCRE_LIB'])
-        if lenv['BF_DEBUG'] and (lenv['OURPLATFORM'] != 'linux'):
-            syslibs += [colladalib+'_d' for colladalib in Split(lenv['BF_OPENCOLLADA_LIB'])]
-        else:
-            syslibs += Split(lenv['BF_OPENCOLLADA_LIB'])
+        syslibs += Split(lenv['BF_OPENCOLLADA_LIB'])
         syslibs.append(lenv['BF_EXPAT_LIB'])
     syslibs += [
         'BulletSoftBody', 'BulletDynamics', 'BulletCollision', 'LinearMath',
