--- mozilla/config/rules.mk.orig	2007-07-04 17:34:35.000000000 +0000
+++ mozilla/config/rules.mk	2007-07-04 17:34:58.000000000 +0000
@@ -137,9 +137,9 @@
 
 ifdef SO_VERSION
 ifneq (,$(findstring $(SONAME),$(MKSHLIB)))
-SO_VERSION_MAJOR	:= $(shell echo $(SO_VERSION) | sed 's/^\([^.]*\)\(\.[^.]*\)\?\(\.[^.]*\)\?/\1/')
-SO_VERSION_MINOR	:= $(shell echo $(SO_VERSION) | sed 's/^\([^.]*\)\(\.[^.]*\)\?\(\.[^.]*\)\?/\2/')
-SO_VERSION_MICRO	:= $(shell echo $(SO_VERSION) | sed 's/^\([^.]*\)\(\.[^.]*\)\?\(\.[^.]*\)\?/\3/')
+SO_VERSION_MAJOR   := $(shell echo $(SO_VERSION) | sed -n -e 's/^\([^.]*\).*/\1/p')
+SO_VERSION_MINOR   := $(shell echo $(SO_VERSION) | sed -n -e 's/^\([^.]*\)\(\.[^.]*\).*/\2/p')
+SO_VERSION_MICRO   := $(shell echo $(SO_VERSION) | sed -n -e 's/^\([^.]*\)\(\.[^.]*\)\(\.[^.]*\).*/\3/p')
 
 SHARED_LIBRARY_LINKS	:= $(SONAME)
 ifdef SO_VERSION_MINOR