diff -up mozilla/xpcom/reflect/xptcall/src/md/unix/xptcinvoke_ipf64.cpp.broken mozilla/xpcom/reflect/xptcall/src/md/unix/xptcinvoke_ipf64.cpp
--- mozilla/xpcom/reflect/xptcall/src/md/unix/xptcinvoke_ipf64.cpp.broken	2004-04-18 10:18:18.000000000 -0400
+++ mozilla/xpcom/reflect/xptcall/src/md/unix/xptcinvoke_ipf64.cpp	2008-02-26 10:16:05.000000000 -0500
@@ -40,7 +40,6 @@
 #include "xptcprivate.h"
 
 #include <stdint.h>
-#include <iostream.h>
 
 // "This code is for IA64 only"
 
