Index: blender-2.63a/extern/libredcode/SConscript
===================================================================
--- blender-2.63a.orig/extern/libredcode/SConscript
+++ blender-2.63a/extern/libredcode/SConscript
@@ -7,7 +7,8 @@ import shutil
 Import('env')
 
 sources = env.Glob('*.c')
-incs = '. ../libopenjpeg'
+#incs = '. ../libopenjpeg'
+incs = '. /usr/include'
 
 env.BlenderLib ( libname='extern_redcode', 
                  sources=sources, includes=Split(incs),
Index: blender-2.63a/extern/SConscript
===================================================================
--- blender-2.63a.orig/extern/SConscript
+++ blender-2.63a/extern/SConscript
@@ -14,9 +14,6 @@ if env['WITH_BF_ELTOPO']:
 if env['WITH_BF_BULLET']:
     SConscript(['bullet2/src/SConscript'])
 
-if env['WITH_BF_OPENJPEG'] and env['BF_OPENJPEG_LIB'] == '':
-    SConscript(['libopenjpeg/SConscript'])
-
 if env['WITH_BF_REDCODE'] and env['BF_REDCODE_LIB'] == '':
     SConscript(['libredcode/SConscript'])
 
Index: blender-2.63a/build_files/scons/tools/Blender.py
===================================================================
--- blender-2.63a.orig/build_files/scons/tools/Blender.py
+++ blender-2.63a/build_files/scons/tools/Blender.py
@@ -258,6 +258,8 @@ def setup_syslibs(lenv):
         syslibs += Split(lenv['BF_FFMPEG_LIB'])
         if lenv['WITH_BF_OGG']:
             syslibs += Split(lenv['BF_OGG_LIB'])
+    if lenv['WITH_BF_OPENJPEG']:
+        syslibs += Split(lenv['BF_OPENJPEG_LIB'])
     if lenv['WITH_BF_JACK']:
         syslibs += Split(lenv['BF_JACK_LIB'])
     if lenv['WITH_BF_SNDFILE'] and not lenv['WITH_BF_STATICSNDFILE']:
