Index: blender-2.64a/extern/SConscript
===================================================================
--- blender-2.64a.orig/extern/SConscript
+++ blender-2.64a/extern/SConscript
@@ -2,7 +2,6 @@
 
 Import('env')
 
-SConscript(['glew/SConscript'])
 SConscript(['colamd/SConscript'])
 
 if env['WITH_BF_GAMEENGINE']:
Index: blender-2.64a/source/blender/editors/include/BIF_gl.h
===================================================================
--- blender-2.64a.orig/source/blender/editors/include/BIF_gl.h
+++ blender-2.64a/source/blender/editors/include/BIF_gl.h
@@ -33,7 +33,7 @@
 #ifndef __BIF_GL_H__
 #define __BIF_GL_H__
 
-#include "GL/glew.h"
+#include <GL/glew.h>
 
 /*
  * these should be phased out. cpack should be replaced in
