diff -Npur blender-2.64a.bullet/build_files/scons/tools/Blender.py blender-2.64a/build_files/scons/tools/Blender.py
--- blender-2.64a.bullet/build_files/scons/tools/Blender.py	2012-11-11 08:28:20.540415278 +0100
+++ blender-2.64a/build_files/scons/tools/Blender.py	2012-11-11 08:38:35.606732492 +0100
@@ -285,6 +285,8 @@ def setup_syslibs(lenv):
         syslibs += Split(lenv['BF_SDL_LIB'])
     if not lenv['WITH_BF_STATICOPENGL']:
         syslibs += Split(lenv['BF_OPENGL_LIB'])
+    if lenv['WITH_BF_BULLET']:
+	syslibs += Split(lenv['BF_BULLET_LIB'])
     if lenv['OURPLATFORM'] in ('win32-vc', 'win32-mingw','linuxcross', 'win64-vc', 'win64-mingw'):
         syslibs += Split(lenv['BF_PTHREADS_LIB'])
     if lenv['WITH_BF_COLLADA']:
diff -Npur blender-2.64a.bullet/doc/doxygen/Doxyfile blender-2.64a/doc/doxygen/Doxyfile
--- blender-2.64a.bullet/doc/doxygen/Doxyfile	2012-11-11 08:30:28.974502541 +0100
+++ blender-2.64a/doc/doxygen/Doxyfile	2012-11-11 08:38:35.608732352 +0100
@@ -617,8 +617,7 @@ INPUT                  = doxygen.main \
                          doxygen.intern \
                          doxygen.extern \
                          ../../source \
-                         ../../intern \
-                         ../../extern/bullet2
+                         ../../intern
 
 # This tag can be used to specify the character encoding of the source files 
 # that doxygen parses. Internally doxygen uses the UTF-8 encoding, which is 
diff -Npur blender-2.64a.bullet/extern/CMakeLists.txt blender-2.64a/extern/CMakeLists.txt
--- blender-2.64a.bullet/extern/CMakeLists.txt	2012-11-11 08:30:39.741755342 +0100
+++ blender-2.64a/extern/CMakeLists.txt	2012-11-11 08:38:35.608732352 +0100
@@ -28,10 +28,6 @@ remove_strict_flags()
 
 add_subdirectory(colamd)
 
-if(WITH_BULLET)
-	add_subdirectory(bullet2)
-endif()
-
 if(WITH_MOD_CLOTH_ELTOPO)
 	add_subdirectory(eltopo)
 endif()
diff -Npur blender-2.64a.bullet/intern/smoke/CMakeLists.txt blender-2.64a/intern/smoke/CMakeLists.txt
--- blender-2.64a.bullet/intern/smoke/CMakeLists.txt	2012-11-11 08:31:09.845666267 +0100
+++ blender-2.64a/intern/smoke/CMakeLists.txt	2012-11-11 08:38:35.608732352 +0100
@@ -26,10 +26,10 @@
 set(INC
 	intern
 	../memutil
-	../../extern/bullet2/src
 )
 
 set(INC_SYS
+	/usr/include/bullet
 	${PNG_INCLUDE_DIR}
 	${ZLIB_INCLUDE_DIRS}
 )
diff -Npur blender-2.64a.bullet/intern/smoke/SConscript blender-2.64a/intern/smoke/SConscript
--- blender-2.64a.bullet/intern/smoke/SConscript	2012-11-11 08:31:38.982644296 +0100
+++ blender-2.64a/intern/smoke/SConscript	2012-11-11 08:38:35.609732282 +0100
@@ -13,7 +13,7 @@ if env['WITH_BF_OPENMP']:
     defs += ' PARALLEL=1'
 
 incs += ' ' + env['BF_PNG_INC'] + ' ' + env['BF_ZLIB_INC'] 
-incs += ' intern ../../extern/bullet2/src ../memutil ../guardealloc '
+incs += ' intern /usr/include/bullet ../memutil ../guardealloc '
 
 if env['WITH_BF_FFTW3']:
     defs += ' WITH_FFTW3'
diff -Npur blender-2.64a.bullet/source/blender/blenkernel/CMakeLists.txt blender-2.64a/source/blender/blenkernel/CMakeLists.txt
--- blender-2.64a.bullet/source/blender/blenkernel/CMakeLists.txt	2012-11-11 08:31:49.775895293 +0100
+++ blender-2.64a/source/blender/blenkernel/CMakeLists.txt	2012-11-11 08:38:35.609732282 +0100
@@ -249,8 +249,8 @@ if(WITH_AUDASPACE)
 endif()
 
 if(WITH_BULLET)
-	list(APPEND INC
-		../../../extern/bullet2/src
+    list(APPEND INC_SYS
+		/usr/include/bullet
 	)
 	add_definitions(-DUSE_BULLET)
 endif()
diff -Npur blender-2.64a.bullet/source/blender/blenkernel/SConscript blender-2.64a/source/blender/blenkernel/SConscript
--- blender-2.64a.bullet/source/blender/blenkernel/SConscript	2012-11-11 08:32:03.512942005 +0100
+++ blender-2.64a/source/blender/blenkernel/SConscript	2012-11-11 08:38:35.610732213 +0100
@@ -13,7 +13,7 @@ incs += ' ../blenlib ../blenfont ../make
 incs += ' ../render/extern/include #/intern/decimation/extern ../makesrna'
 incs += ' ../imbuf ../ikplugin ../avi #/intern/elbeem/extern ../nodes ../modifiers'
 incs += ' #/intern/iksolver/extern ../blenloader'
-incs += ' #/extern/bullet2/src'
+incs += ' /usr/include/bullet'
 incs += ' #/intern/opennl/extern #/intern/bsp/extern'
 incs += ' ../gpu #/extern/glew/include'
 incs += ' ../bmesh'
diff -Npur blender-2.64a.bullet/source/gameengine/BlenderRoutines/CMakeLists.txt blender-2.64a/source/gameengine/BlenderRoutines/CMakeLists.txt
--- blender-2.64a.bullet/source/gameengine/BlenderRoutines/CMakeLists.txt	2012-11-11 08:32:28.151232221 +0100
+++ blender-2.64a/source/gameengine/BlenderRoutines/CMakeLists.txt	2012-11-11 08:41:06.026294063 +0100
@@ -23,7 +23,6 @@ set(INC
 	../../blender/makesdna
 	../../blender/makesrna
 	../../blender/windowmanager
-	../../../extern/bullet2/src
 	../../../intern/container
 	../../../intern/guardedalloc
 	../../../intern/moto/include
@@ -33,6 +32,7 @@ set(INC
 set(INC_SYS
 	${PTHREADS_INCLUDE_DIRS}
 	${GLEW_INCLUDE_PATH}
+	/usr/include/bullet
 )
 
 set(SRC
diff -Npur blender-2.64a.bullet/source/gameengine/Converter/CMakeLists.txt blender-2.64a/source/gameengine/Converter/CMakeLists.txt
--- blender-2.64a.bullet/source/gameengine/Converter/CMakeLists.txt	2012-11-11 08:32:46.903930869 +0100
+++ blender-2.64a/source/gameengine/Converter/CMakeLists.txt	2012-11-11 08:38:35.610732213 +0100
@@ -49,7 +49,6 @@ set(INC
 	../../blender/makesdna
 	../../blender/makesrna
 	../../blender/windowmanager
-	../../../extern/bullet2/src
 	../../../intern/container
 	../../../intern/guardedalloc
 	../../../intern/moto/include
@@ -60,6 +59,7 @@ set(INC
 set(INC_SYS
 	${PTHREADS_INCLUDE_DIRS}
 	/usr/include/eigen3
+	/usr/include/bullet
 
 )
 
diff -Npur blender-2.64a.bullet/source/gameengine/Ketsji/CMakeLists.txt blender-2.64a/source/gameengine/Ketsji/CMakeLists.txt
--- blender-2.64a.bullet/source/gameengine/Ketsji/CMakeLists.txt	2012-11-11 08:33:00.064017620 +0100
+++ blender-2.64a/source/gameengine/Ketsji/CMakeLists.txt	2012-11-11 08:38:35.611732144 +0100
@@ -248,9 +248,11 @@ endif()
 
 if(WITH_BULLET)
 	list(APPEND INC
-		../../../extern/bullet2/src
 		../Physics/Bullet
 	)
+	list(APPEND INC_SYS
+		/usr/include/bullet
+	)
 	add_definitions(-DUSE_BULLET)
 endif()
 
diff -Npur blender-2.64a.bullet/source/gameengine/Physics/Bullet/CcdPhysicsEnvironment.cpp blender-2.64a/source/gameengine/Physics/Bullet/CcdPhysicsEnvironment.cpp
--- blender-2.64a.bullet/source/gameengine/Physics/Bullet/CcdPhysicsEnvironment.cpp	2012-11-11 08:33:13.689072103 +0100
+++ blender-2.64a/source/gameengine/Physics/Bullet/CcdPhysicsEnvironment.cpp	2012-11-11 08:38:35.613732006 +0100
@@ -2813,7 +2813,8 @@ float		CcdPhysicsEnvironment::getApplied
 
 void	CcdPhysicsEnvironment::exportFile(const char* filename)
 {
-	btDefaultSerializer*	serializer = new btDefaultSerializer();
+	// Workaround bullet-2.77 api change.
+	btDefaultSerializer*	serializer = new btDefaultSerializer(0);
 	
 		
 	for (int i=0;i<m_dynamicsWorld->getNumCollisionObjects();i++)
diff -Npur blender-2.64a.bullet/source/gameengine/Physics/Bullet/CMakeLists.txt blender-2.64a/source/gameengine/Physics/Bullet/CMakeLists.txt
--- blender-2.64a.bullet/source/gameengine/Physics/Bullet/CMakeLists.txt	2012-11-11 08:33:26.599176200 +0100
+++ blender-2.64a/source/gameengine/Physics/Bullet/CMakeLists.txt	2012-11-11 08:38:35.613732006 +0100
@@ -37,7 +37,6 @@ set(INC
 	../../../blender/blenkernel
 	../../../blender/blenlib
 	../../../blender/makesdna
-	../../../../extern/bullet2/src
 	../../../../intern/container
 	../../../../intern/guardedalloc
 	../../../../intern/moto/include
@@ -45,6 +44,7 @@ set(INC
 )
 
 set(INC_SYS
+	/usr/include/bullet
 	${GLEW_INCLUDE_PATH}
 	${PYTHON_INCLUDE_DIRS}
 )
