diff -pur blender-2.64a.orig/build_files/scons/config/linux-config.py blender-2.64a.colamd/build_files/scons/config/linux-config.py
--- blender-2.64a.orig/build_files/scons/config/linux-config.py	2012-11-11 09:59:13.273986602 +0100
+++ blender-2.64a.colamd/build_files/scons/config/linux-config.py	2012-11-11 10:07:30.658160253 +0100
@@ -197,6 +197,11 @@ BF_EXPAT = '/usr'
 BF_EXPAT_LIB = 'expat'
 BF_EXPAT_LIBPATH = '/usr/lib'
 
+WITH_BF_COLAMD = True
+BF_COLAMD = '#extern/colamd'
+BF_COLAMD_INC = '${BF_COLAMD}/include'
+BF_COLAMD_LIB = 'extern_colamd'
+
 WITH_BF_JEMALLOC = False
 WITH_BF_STATICJEMALLOC = False
 BF_JEMALLOC = '/usr'
diff -pur blender-2.64a.orig/build_files/scons/tools/Blender.py blender-2.64a.colamd/build_files/scons/tools/Blender.py
--- blender-2.64a.orig/build_files/scons/tools/Blender.py	2012-11-11 09:59:34.361510075 +0100
+++ blender-2.64a.colamd/build_files/scons/tools/Blender.py	2012-11-11 10:07:30.659160183 +0100
@@ -296,6 +296,8 @@ def setup_syslibs(lenv):
         else:
             syslibs += Split(lenv['BF_OPENCOLLADA_LIB'])
         syslibs.append(lenv['BF_EXPAT_LIB'])
+    if lenv['WITH_BF_COLAMD']:
+        syslibs += Split(lenv['BF_COLAMD_LIB'])
 
     if lenv['WITH_BF_JEMALLOC']:
         if not lenv['WITH_BF_STATICJEMALLOC']:
diff -pur blender-2.64a.orig/build_files/scons/tools/btools.py blender-2.64a.colamd/build_files/scons/tools/btools.py
--- blender-2.64a.orig/build_files/scons/tools/btools.py	2012-11-11 09:59:52.129265995 +0100
+++ blender-2.64a.colamd/build_files/scons/tools/btools.py	2012-11-11 11:01:11.296482414 +0100
@@ -127,6 +127,7 @@ def validate_arguments(args, bc):
             'WITH_BF_FFTW3', 'BF_FFTW3', 'BF_FFTW3_INC', 'BF_FFTW3_LIB', 'BF_FFTW3_LIBPATH', 'WITH_BF_STATICFFTW3', 'BF_FFTW3_LIB_STATIC',
             'WITH_BF_STATICOPENGL', 'BF_OPENGL', 'BF_OPENGL_INC', 'BF_OPENGL_LIB', 'BF_OPENGL_LIBPATH', 'BF_OPENGL_LIB_STATIC',
             'WITH_BF_COLLADA', 'BF_COLLADA', 'BF_COLLADA_INC', 'BF_COLLADA_LIB', 'BF_OPENCOLLADA', 'BF_OPENCOLLADA_INC', 'BF_OPENCOLLADA_LIB', 'BF_OPENCOLLADA_LIBPATH', 'BF_PCRE', 'BF_PCRE_LIB', 'BF_PCRE_LIBPATH', 'BF_EXPAT', 'BF_EXPAT_LIB', 'BF_EXPAT_LIBPATH',
+            'WITH_BF_COLAMD', 'BF_COLAMD', 'BF_COLAMD_INC', 'BF_COLAMD_LIB',
             'WITH_BF_PLAYER',
             'WITH_BF_NOBLENDER',
             'WITH_BF_BINRELOC',
@@ -593,7 +594,12 @@ def read_opts(env, cfg, args):
         ('BF_BOOST_LIB_STATIC', 'Boost static library', ''),
 
         (BoolVariable('WITH_GHOST_XDND', 'Build with drag-n-drop support on Linux platforms using XDND protocol', True)),
-        (BoolVariable('WITH_BF_COMPOSITOR_LEGACY', 'Enable the legacy compositor', False))
+        (BoolVariable('WITH_BF_COMPOSITOR_LEGACY', 'Enable the legacy compositor', False)),
+        
+	(BoolVariable('WITH_BF_COLAMD', 'Use Colamd if true', True)),
+        ('BF_COLAMD', 'Colamd base dir', ''),
+        ('BF_COLAMD_INC', 'Colamd include path', ''),
+        ('BF_COLAMD_LIB', 'Colamd library', '')
     ) # end of opts.AddOptions()
 
     return localopts
diff -pur blender-2.64a.orig/extern/libmv/bundle.sh blender-2.64a.colamd/extern/libmv/bundle.sh
--- blender-2.64a.orig/extern/libmv/bundle.sh	2012-11-11 10:00:19.530347398 +0100
+++ blender-2.64a.colamd/extern/libmv/bundle.sh	2012-11-11 10:23:14.905045019 +0100
@@ -126,7 +126,6 @@ set(INC
 	.
 	third_party/ssba
 	third_party/ldl/Include
-	../colamd/Include
 	third_party/ceres/include
 )
 
@@ -134,6 +133,7 @@ set(INC_SYS
 	/usr/include/eigen3
 	\${PNG_INCLUDE_DIR}
 	\${ZLIB_INCLUDE_DIRS}
+	\${COLAMD_INCLUDE_DIRS}
 )
 
 
@@ -254,6 +254,7 @@ $src
 incs = '. /usr/include/eigen3 third_party/ceres/include'
 incs += ' ' + env['BF_PNG_INC']
 incs += ' ' + env['BF_ZLIB_INC']
+incs += ' ' + env['BF_COLAMD_INC']
 
 if env['OURPLATFORM'] in ('win32-vc', 'win32-mingw', 'linuxcross', 'win64-vc', 'win64-mingw'):
     incs += ' ./third_party/glog/src/windows ./third_party/glog/src/windows/glog'
Nur in blender-2.64a.colamd/extern/libmv: bundle.sh.rej.
diff -pur blender-2.64a.orig/extern/libmv/CMakeLists.txt blender-2.64a.colamd/extern/libmv/CMakeLists.txt
--- blender-2.64a.orig/extern/libmv/CMakeLists.txt	2012-11-11 10:00:34.289313990 +0100
+++ blender-2.64a.colamd/extern/libmv/CMakeLists.txt	2012-11-11 10:20:53.917916799 +0100
@@ -30,7 +30,6 @@ set(INC
 	.
 	third_party/ssba
 	third_party/ldl/Include
-	../colamd/Include
 	third_party/ceres/include
 )
 
@@ -38,6 +37,7 @@ set(INC_SYS
 	/usr/include/eigen3
 	${PNG_INCLUDE_DIR}
 	${ZLIB_INCLUDE_DIRS}
+	${COLAMD_INCLUDE_DIRS}
 )
 
 set(SRC
Nur in blender-2.64a.colamd/extern/libmv: CMakeLists.txt.rej.
diff -pur blender-2.64a.orig/extern/libmv/SConscript blender-2.64a.colamd/extern/libmv/SConscript
--- blender-2.64a.orig/extern/libmv/SConscript	2012-11-11 10:00:51.090137613 +0100
+++ blender-2.64a.colamd/extern/libmv/SConscript	2012-11-11 10:07:30.662159973 +0100
@@ -33,6 +33,7 @@ src += env.Glob('third_party/ssba/Math/*
 incs = '. /usr/include/eigen3 third_party/ceres/include'
 incs += ' ' + env['BF_PNG_INC']
 incs += ' ' + env['BF_ZLIB_INC']
+incs += ' ' + env['BF_COLAMD_INC']
 
 if env['OURPLATFORM'] in ('win32-vc', 'win32-mingw', 'linuxcross', 'win64-vc', 'win64-mingw'):
     incs += ' ./third_party/glog/src/windows ./third_party/glog/src/windows/glog'
@@ -62,7 +63,7 @@ else:
         ccflags_libmv += Split(env['REL_CCFLAGS'])
         cxxflags_libmv += Split(env['REL_CXXFLAGS'])
 
-incs += ' ./third_party/ssba ./third_party/ldl/Include ../colamd/Include'
+incs += ' ./third_party/ssba ./third_party/ldl/Include'
 
 env.BlenderLib ( libname = 'extern_libmv', sources=src, includes=Split(incs), defines=defs, libtype=['extern', 'player'], priority=[20,137], compileflags=cflags_libmv, cc_compileflags=ccflags_libmv, cxx_compileflags=cxxflags_libmv )
 
diff -pur blender-2.64a.orig/extern/libmv/third_party/ssba/Math/v3d_optimization.cpp blender-2.64a.colamd/extern/libmv/third_party/ssba/Math/v3d_optimization.cpp
--- blender-2.64a.orig/extern/libmv/third_party/ssba/Math/v3d_optimization.cpp	2012-11-11 10:01:05.715113586 +0100
+++ blender-2.64a.colamd/extern/libmv/third_party/ssba/Math/v3d_optimization.cpp	2012-11-11 10:07:30.663159903 +0100
@@ -21,7 +21,7 @@ with SSBA. If not, see <http://www.gnu.o
 
 #if defined(V3DLIB_ENABLE_SUITESPARSE)
 //# include "COLAMD/Include/colamd.h"
-# include "colamd.h"
+# include <colamd.h>
 extern "C"
 {
 //# include "LDL/Include/ldl.h"
diff -pur blender-2.64a.orig/intern/opennl/CMakeLists.txt blender-2.64a.colamd/intern/opennl/CMakeLists.txt
--- blender-2.64a.orig/intern/opennl/CMakeLists.txt	2012-11-11 10:01:30.562373806 +0100
+++ blender-2.64a.colamd/intern/opennl/CMakeLists.txt	2012-11-11 10:07:30.663159903 +0100
@@ -40,11 +40,10 @@ add_definitions(
 set(INC
 	extern
 	superlu
-	../../extern/colamd/Include
 )
 
 set(INC_SYS
-
+	/usr/include
 )
 
 set(SRC
diff -pur blender-2.64a.orig/intern/opennl/SConscript blender-2.64a.colamd/intern/opennl/SConscript
--- blender-2.64a.orig/intern/opennl/SConscript	2012-11-11 10:01:41.610600220 +0100
+++ blender-2.64a.colamd/intern/opennl/SConscript	2012-11-11 10:07:30.663159903 +0100
@@ -3,7 +3,7 @@ Import ('env')
 
 sources = env.Glob('intern/*.c') + env.Glob('superlu/*.c')
 
-incs = 'extern superlu ../../extern/colamd/Include'
+incs = 'extern superlu /usr/include'
 
 if (env['OURPLATFORM'] in ('win32-mingw', 'win64-mingw')):
     env.BlenderLib ('bf_intern_opennl', sources, Split(incs), [], libtype=['core','intern'], priority=[1,80] )
diff -pur blender-2.64a.orig/intern/opennl/superlu/get_perm_c.c blender-2.64a.colamd/intern/opennl/superlu/get_perm_c.c
--- blender-2.64a.orig/intern/opennl/superlu/get_perm_c.c	2012-11-11 10:01:53.211787917 +0100
+++ blender-2.64a.colamd/intern/opennl/superlu/get_perm_c.c	2012-11-11 10:07:30.664159833 +0100
@@ -10,7 +10,7 @@
  */
 
 #include "ssp_defs.h"
-#include "colamd.h"
+#include <colamd.h>
 
 extern int  genmmd_(int *, int *, int *, int *, int *, int *, int *, 
 		    int *, int *, int *, int *, int *);
