Index: blender-2.64a/source/blender/imbuf/intern/util.c
===================================================================
--- blender-2.64a.orig/source/blender/imbuf/intern/util.c
+++ blender-2.64a/source/blender/imbuf/intern/util.c
@@ -272,8 +272,8 @@ static int isffmpeg(const char *filename
 	AVFormatContext *pFormatCtx = NULL;
 	unsigned int i;
 	int videoStream;
-	AVCodec *pCodec;
-	AVCodecContext *pCodecCtx;
+	AVCodec *pCodec = NULL;
+	AVCodecContext *pCodecCtx = NULL;
 
 	if (BLI_testextensie(filename, ".swf") ||
 	    BLI_testextensie(filename, ".jpg") ||
Index: blender-2.64a/source/blender/imbuf/intern/anim_movie.c
===================================================================
--- blender-2.64a.orig/source/blender/imbuf/intern/anim_movie.c
+++ blender-2.64a/source/blender/imbuf/intern/anim_movie.c
@@ -464,9 +464,9 @@ static int startffmpeg(struct anim *anim
 {
 	int i, videoStream;
 
-	AVCodec *pCodec;
+	AVCodec *pCodec = NULL;
 	AVFormatContext *pFormatCtx = NULL;
-	AVCodecContext *pCodecCtx;
+	AVCodecContext *pCodecCtx = NULL;
 	int frs_num;
 	double frs_den;
 	int streamcount;
@@ -474,8 +474,8 @@ static int startffmpeg(struct anim *anim
 #ifdef FFMPEG_SWSCALE_COLOR_SPACE_SUPPORT
 	/* The following for color space determination */
 	int srcRange, dstRange, brightness, contrast, saturation;
-	int *table;
-	const int *inv_table;
+	int *table = NULL;
+	const int *inv_table = NULL;
 #endif
 
 	if (anim == 0) return(-1);
