Index: blender-2.64a/build_files/scons/tools/btools.py
===================================================================
--- blender-2.64a.orig/build_files/scons/tools/btools.py
+++ blender-2.64a/build_files/scons/tools/btools.py
@@ -484,7 +484,9 @@ def read_opts(env, cfg, args):
         ('BF_3DMOUSE_LIB', '3d mouse library', ''),
         ('BF_3DMOUSE_LIBPATH', '3d mouse library path', ''),
         ('BF_3DMOUSE_LIB_STATIC', '3d mouse static library', ''),
+    ) # end of opts.AddOptions()
 
+    localopts.AddVariables(
         ('CFLAGS', 'C only flags', []),
         ('CCFLAGS', 'Generic C and C++ flags', []),
         ('CXXFLAGS', 'C++ only flags', []),
