Index: blender-2.64a/build_files/scons/tools/Blender.py
===================================================================
--- blender-2.64a.orig/build_files/scons/tools/Blender.py
+++ blender-2.64a/build_files/scons/tools/Blender.py
@@ -295,7 +295,7 @@ def setup_syslibs(lenv):
         syslibs.append(lenv['BF_EXPAT_LIB'])
     if lenv['WITH_BF_COLAMD']:
         syslibs += Split(lenv['BF_COLAMD_LIB'])
-    syslibs += ['ldl']
+    syslibs += ['ldl', 'glog', 'gflags']
 
     if lenv['WITH_BF_JEMALLOC']:
         if not lenv['WITH_BF_STATICJEMALLOC']:
Index: blender-2.64a/extern/libmv/SConscript
===================================================================
--- blender-2.64a.orig/extern/libmv/SConscript
+++ blender-2.64a/extern/libmv/SConscript
@@ -25,7 +25,6 @@ src += env.Glob('libmv/numeric/*.cc')
 src += env.Glob('libmv/simple_pipeline/*.cc')
 src += env.Glob('libmv/tracking/*.cc')
 src += env.Glob('third_party/fast/*.c')
-src += env.Glob('third_party/gflags/*.cc')
 src += env.Glob('third_party/ssba/Geometry/*.cpp')
 src += env.Glob('third_party/ssba/Math/*.cpp')
 
@@ -55,8 +54,6 @@ if env['OURPLATFORM'] in ('win32-vc', 'w
             ccflags_libmv += Split(env['REL_CCFLAGS'])
             cxxflags_libmv += Split(env['REL_CXXFLAGS'])
 else:
-    src += env.Glob("third_party/glog/src/*.cc")
-    incs += ' ./third_party/glog/src'
     if not env['BF_DEBUG']:
         cflags_libmv += Split(env['REL_CFLAGS'])
         ccflags_libmv += Split(env['REL_CCFLAGS'])
Index: blender-2.64a/extern/libmv/libmv-capi.cpp
===================================================================
--- blender-2.64a.orig/extern/libmv/libmv-capi.cpp
+++ blender-2.64a/extern/libmv/libmv-capi.cpp
@@ -34,7 +34,7 @@
 
 #include "libmv-capi.h"
 
-#include "third_party/gflags/gflags/gflags.h"
+#include "gflags/gflags.h"
 #include "glog/logging.h"
 #include "libmv/logging/logging.h"
 
