Index: blender-2.63a/source/blender/imbuf/intern/util.c
===================================================================
--- blender-2.63a.orig/source/blender/imbuf/intern/util.c
+++ blender-2.63a/source/blender/imbuf/intern/util.c
@@ -243,11 +243,11 @@ void do_init_ffmpeg(void)
 
 static int isffmpeg (const char *filename)
 {
-	AVFormatContext *pFormatCtx;
+	AVFormatContext *pFormatCtx = NULL;
 	unsigned int i;
 	int videoStream;
-	AVCodec *pCodec;
-	AVCodecContext *pCodecCtx;
+	AVCodec *pCodec = NULL;
+	AVCodecContext *pCodecCtx = NULL;
 
 	do_init_ffmpeg();
 
