commit 3403c4274410115dbe5ce282aa8b045f8aedb002
Author: Pierre d'Herbemont <pdherbemont@free.fr>
Date:   Wed Aug 19 11:42:53 2009 +0200

    test: Disable test_media_list_player as it deadlocks.

Index: vlc-1.0.5/test/Makefile.am
===================================================================
--- vlc-1.0.5.orig/test/Makefile.am
+++ vlc-1.0.5/test/Makefile.am
@@ -11,7 +11,6 @@ check_PROGRAMS = \
 	test_libvlc_core \
 	test_libvlc_events \
 	test_libvlc_media_list \
-	test_libvlc_media_list_player \
 	$(NULL)
 EXTRA_PROGRAMS = \
 	test_libvlc_media_player \
