diff -urN mozillaorig/mail/base/content/mailCommands.js mozilla/mail/base/content/mailCommands.js
--- mozillaorig/mail/base/content/mailCommands.js	2007-02-07 15:24:40.000000000 +0100
+++ mozilla/mail/base/content/mailCommands.js	2007-02-07 16:18:59.000000000 +0100
@@ -271,8 +271,9 @@
       var messageIDScheme = messageID ? messageID.split(":")[0] : "";
       if (messageIDScheme && (messageIDScheme == 'http' || messageIDScheme == 'https') &&  "openComposeWindowForRSSArticle" in this)
         openComposeWindowForRSSArticle(messageID, hdr, type);
-      else if (type == msgComposeType.Reply ||
+       else if (type == msgComposeType.Reply ||
                type == msgComposeType.ReplyAll ||
+               type == msgComposeType.ReplyToList ||
                type == msgComposeType.ForwardInline ||
                type == msgComposeType.ReplyToGroup ||
                type == msgComposeType.ReplyToSender ||
diff -urN mozillaorig/mailnews/base/resources/content/mailCommands.js mozilla/mailnews/base/resources/content/mailCommands.js
--- mozillaorig/mailnews/base/resources/content/mailCommands.js	2007-02-07 15:26:44.000000000 +0100
+++ mozilla/mailnews/base/resources/content/mailCommands.js	2007-02-07 16:18:59.000000000 +0100
@@ -268,8 +268,9 @@
         }
       }
 
-      if (type == msgComposeType.Reply ||
+      if (type == msgComposeType.Reply || 
         type == msgComposeType.ReplyAll ||
+        type == msgComposeType.ReplyToList ||
         type == msgComposeType.ForwardInline ||
         type == msgComposeType.ReplyToGroup ||
         type == msgComposeType.ReplyToSender ||
diff -urN mozillaorig/mailnews/compose/public/nsIMsgComposeParams.idl mozilla/mailnews/compose/public/nsIMsgComposeParams.idl
--- mozillaorig/mailnews/compose/public/nsIMsgComposeParams.idl	2007-02-07 15:26:46.000000000 +0100
+++ mozilla/mailnews/compose/public/nsIMsgComposeParams.idl	2007-02-07 16:18:59.000000000 +0100
@@ -59,6 +59,7 @@
     const long Template                 = 10;
     const long MailToUrl                = 11;
     const long ReplyWithTemplate        = 12;
+    const long ReplyToList              = 13;
 };
 
 
diff -urN mozillaorig/mailnews/compose/src/nsMsgCompose.cpp mozilla/mailnews/compose/src/nsMsgCompose.cpp
--- mozillaorig/mailnews/compose/src/nsMsgCompose.cpp	2007-02-07 15:26:46.000000000 +0100
+++ mozilla/mailnews/compose/src/nsMsgCompose.cpp	2007-02-07 16:18:59.000000000 +0100
@@ -1751,6 +1751,7 @@
         default: break;
         case nsIMsgCompType::Reply :
         case nsIMsgCompType::ReplyAll:
+        case nsIMsgCompType::ReplyToList:
         case nsIMsgCompType::ReplyToGroup:
         case nsIMsgCompType::ReplyToSender:
         case nsIMsgCompType::ReplyToSenderAndGroup:
@@ -2195,8 +2196,13 @@
     if (!mCiteReference.IsEmpty())
       compose->SetCiteReference(mCiteReference);
 
-    if (mHeaders && (type == nsIMsgCompType::Reply || type == nsIMsgCompType::ReplyAll || type == nsIMsgCompType::ReplyToSender ||
-                     type == nsIMsgCompType::ReplyToGroup || type == nsIMsgCompType::ReplyToSenderAndGroup) && mQuoteOriginal)
+    if (mHeaders && (type == nsIMsgCompType::Reply ||
+                     type == nsIMsgCompType::ReplyAll ||
+                     type == nsIMsgCompType::ReplyToList ||
+                     type == nsIMsgCompType::ReplyToSender ||
+                     type == nsIMsgCompType::ReplyToGroup ||
+                     type == nsIMsgCompType::ReplyToSenderAndGroup) &&
+        mQuoteOriginal)
     {
       nsCOMPtr<nsIMsgCompFields> compFields;
       compose->GetCompFields(getter_AddRefs(compFields));
@@ -2213,6 +2219,7 @@
         nsAutoString followUpTo;
         nsAutoString messageId;
         nsAutoString references;
+        nsAutoString listPost;
         nsXPIDLCString outCString;
         PRBool needToRemoveDup = PR_FALSE;
         if (!mMimeConverter)
@@ -2266,6 +2273,29 @@
 
           needToRemoveDup = PR_TRUE;
         }
+
+        mHeaders->ExtractHeader(HEADER_LIST_POST, PR_TRUE, getter_Copies(outCString));
+        if (!outCString.IsEmpty())
+          mMimeConverter->DecodeMimeHeader(outCString, listPost, charset);
+
+        if (type == nsIMsgCompType::ReplyToList && ! listPost.IsEmpty())
+        {
+          nsString::const_iterator mailtoStart, mailtoEnd;
+          listPost.BeginReading(mailtoStart);
+          listPost.EndReading(mailtoEnd);
+          nsAutoString mailtoText(NS_LITERAL_STRING("<mailto:"));
+          PRBool mailtoFound = FindInReadable(mailtoText, mailtoStart, mailtoEnd);
+          
+          // Strip off the leading "<mailto:" and trailing ">"
+          if (mailtoFound && listPost.Equals(mailtoStart.get()) &&
+              listPost.RFindChar('>') == listPost.Length() - 1)
+          {
+            listPost.Cut(0, mailtoText.Length());
+            listPost.Cut(listPost.Length() - 1, 1);
+
+            compFields->SetTo(listPost);
+          } 
+        }
               
         mHeaders->ExtractHeader(HEADER_REPLY_TO, PR_FALSE, getter_Copies(outCString));
         if (outCString)
@@ -2303,7 +2333,8 @@
           mMimeConverter->DecodeMimeHeader(outCString, references, charset);
         }
 
-        if (! ((type == nsIMsgCompType::ReplyAll) && ! mailFollowupTo.IsEmpty()))
+        if (! ((type == nsIMsgCompType::ReplyAll) && ! mailFollowupTo.IsEmpty()) &&
+            ! ((type == nsIMsgCompType::ReplyToList) && ! listPost.IsEmpty()))
         {
           if (! mailReplyTo.IsEmpty())
           { // handle Mail-Reply-To (http://cr.yp.to/proto/replyto.html)
@@ -2808,6 +2839,7 @@
   // the header that we then look at when we actually send the message.
   if (mType == nsIMsgCompType::Reply || 
     mType == nsIMsgCompType::ReplyAll ||
+    mType == nsIMsgCompType::ReplyToList ||
     mType == nsIMsgCompType::ReplyToGroup ||
     mType == nsIMsgCompType::ReplyToSender ||
     mType == nsIMsgCompType::ReplyToSenderAndGroup ||
@@ -2846,6 +2878,7 @@
   // for this URI.
   if (mType == nsIMsgCompType::Reply || 
       mType == nsIMsgCompType::ReplyAll ||
+      mType == nsIMsgCompType::ReplyToList ||
       mType == nsIMsgCompType::ReplyToGroup ||
       mType == nsIMsgCompType::ReplyToSender ||
       mType == nsIMsgCompType::ReplyToSenderAndGroup ||
@@ -3778,6 +3811,7 @@
     case nsIMsgCompType::MailToUrl :    /* same as New */
     case nsIMsgCompType::Reply :        /* should not happen! but just in case */
     case nsIMsgCompType::ReplyAll :       /* should not happen! but just in case */
+    case nsIMsgCompType::ReplyToList :    /* should not happen! but just in case */
     case nsIMsgCompType::ForwardAsAttachment :  /* should not happen! but just in case */
     case nsIMsgCompType::ForwardInline :
     case nsIMsgCompType::NewsPost :
diff -urN mozillaorig/mailnews/mime/public/nsMailHeaders.h mozilla/mailnews/mime/public/nsMailHeaders.h
--- mozillaorig/mailnews/mime/public/nsMailHeaders.h	2007-02-07 15:26:58.000000000 +0100
+++ mozilla/mailnews/mime/public/nsMailHeaders.h	2007-02-07 16:18:59.000000000 +0100
@@ -65,6 +65,7 @@
 #define HEADER_FROM							            "From"
 #define HEADER_STATUS							            "Status"
 #define HEADER_LINES						            "Lines"
+#define HEADER_LIST_POST                    "List-Post"
 #define HEADER_MAIL_FOLLOWUP_TO				      "Mail-Followup-To"
 #define HEADER_MAIL_REPLY_TO				       "Mail-Reply-To"
 #define HEADER_MESSAGE_ID					          "Message-ID"
