diff -Nru hibernate-script-2.0.vanilla/hibernate.sh hibernate-script-2.0/hibernate.sh
--- hibernate-script-2.0.vanilla/hibernate.sh	2010-04-18 13:06:49.653466081 +0100
+++ hibernate-script-2.0/hibernate.sh	2010-04-18 13:27:09.566460136 +0100
@@ -713,11 +713,13 @@
 	fi
     fi
 
-    if [ "${conffiles_read#*$file_name}" != "$conffiles_read" ]; then
+    if [ -n "$(echo ${conffiles_read} | grep ${file_name})" ]; then
   	vecho 2 "$EXE: Skipping already included config file $file_name"
   	return 0
     fi
 
+    conffiles_read="${conffiles_read} ${file_name}"
+
     while true ; do
 	# Doing the read this way allows means we don't require a new-line
 	# at the end of the file.
