diff -ur StepMania-3.9-src.orig/src/arch/Sound/ALSA9Helpers.cpp StepMania-3.9-src/src/arch/Sound/ALSA9Helpers.cpp
--- StepMania-3.9-src.orig/src/arch/Sound/ALSA9Helpers.cpp	2004-07-18 00:27:29.000000000 +0300
+++ StepMania-3.9-src/src/arch/Sound/ALSA9Helpers.cpp	2007-08-07 16:53:38.000000000 +0300
@@ -256,16 +256,17 @@
 	{
 		CHECKPOINT;
 		dsnd_pcm_close(pcm);
+		pcm = 0;
 		CHECKPOINT;
-		RageException::ThrowNonfatal( "SetHWParams failed" );
 	}
-
-	SetSWParams();
+	else
+		SetSWParams();
 }
 
 Alsa9Buf::~Alsa9Buf()
 {
-	dsnd_pcm_close(pcm);
+	if(pcm)
+		dsnd_pcm_close(pcm);
 }
 
 
