diff -ur StepMania-3.9-src.orig/configure.ac StepMania-3.9-src/configure.ac
--- StepMania-3.9-src.orig/configure.ac	2007-08-06 07:35:56.000000000 +0300
+++ StepMania-3.9-src/configure.ac	2007-08-06 07:37:19.000000000 +0300
@@ -146,7 +146,9 @@
 	LIBS="$LIBS -ljpeg"
 else
 	AC_DEFINE(NO_JPEG_SUPPORT, 1, [JPEG support not available])
+	no_jpeg_support=yes
 fi
+AM_CONDITIONAL(NO_JPEG_SUPPORT, test "$no_jpeg_support" = "yes")
 
 AC_ARG_WITH(network, AC_HELP_STRING([--without-network], [Disable networking]), with_network=$withval, with_network=yes)
 if test "$with_network" = "no"; then
diff -ur StepMania-3.9-src.orig/src/Makefile.am StepMania-3.9-src/src/Makefile.am
--- StepMania-3.9-src.orig/src/Makefile.am	2007-08-06 07:35:55.000000000 +0300
+++ StepMania-3.9-src/src/Makefile.am	2007-08-06 07:36:05.000000000 +0300
@@ -211,10 +211,14 @@
 ScoreDisplayNormal.cpp ScoreDisplayNormal.h ScoreDisplayOni.cpp ScoreDisplayOni.h \
 ScoreDisplayPercentage.cpp ScoreDisplayPercentage.h ScoreDisplayRave.cpp ScoreDisplayRave.h
 
-Helpers = SDL_SaveJPEG.cpp SDL_SaveJPEG.h \
+Helpers = \
 	  SDL_dither.cpp SDL_dither.h \
 	  SDL_rotozoom.cpp SDL_rotozoom.h
 
+if !NO_JPEG_SUPPORT
+Helpers += SDL_SaveJPEG.cpp SDL_SaveJPEG.h
+endif
+
 PCRE = pcre/chartables.c pcre/get.c pcre/internal.h pcre/maketables.c pcre/pcre.c pcre/pcre.h pcre/study.c
 
 RageFile = \
@@ -235,7 +239,7 @@
 RageSurface.cpp RageSurface.h RageSurfaceUtils.cpp RageSurfaceUtils.h \
 RageSurfaceUtils_Palettize.cpp RageSurfaceUtils_Palettize.h \
 RageSurface_Load.cpp RageSurface_Load.h RageSurface_Load_PNG.cpp RageSurface_Load_PNG.h \
-RageSurface_Load_JPEG.cpp RageSurface_Load_JPEG.h RageSurface_Load_GIF.cpp RageSurface_Load_GIF.h \
+RageSurface_Load_GIF.cpp RageSurface_Load_GIF.h \
 RageSurface_Load_BMP.cpp RageSurface_Load_BMP.h \
 RageSurface_Load_XPM.cpp RageSurface_Load_XPM.h RageTexture.cpp RageTexture.h \
 RageSurface_Save_BMP.cpp RageSurface_Save_BMP.h \
@@ -243,6 +247,10 @@
 RageTimer.cpp RageTimer.h RageTypes.h RageUtil.cpp RageUtil.h RageUtil_CharConversions.cpp RageUtil_CharConversions.h \
 RageUtil_BackgroundLoader.cpp RageUtil_BackgroundLoader.h RageUtil_FileDB.cpp RageUtil_FileDB.h RageUtil_CircularBuffer.h
 
+if !NO_JPEG_SUPPORT
+Rage += RageSurface_Load_JPEG.cpp RageSurface_Load_JPEG.h 
+endif
+
 Actors = \
 Actor.cpp Actor.h ActorCollision.h ActorCommands.cpp ActorCommands.h ActorFrame.cpp ActorFrame.h \
 ActorScroller.cpp ActorScroller.h ActorUtil.cpp ActorUtil.h BitmapText.cpp BitmapText.h Model.cpp Model.h \
diff -ur StepMania-3.9-src.orig/src/RageDisplay.cpp StepMania-3.9-src/src/RageDisplay.cpp
--- StepMania-3.9-src.orig/src/RageDisplay.cpp	2007-08-06 07:35:55.000000000 +0300
+++ StepMania-3.9-src/src/RageDisplay.cpp	2007-08-06 07:36:20.000000000 +0300
@@ -674,6 +674,7 @@
 	}
 
 	bool bSuccess = false;
+#ifndef NO_JPEG_SUPPORT
 	switch( format )
 	{
 	case SAVE_LOSSLESS:
@@ -689,6 +690,9 @@
 		ASSERT(0);
 		return false;
 	}
+#else
+    bSuccess = RageSurface_Save_BMP( surface, out );
+#endif // NO_JPEG_SUPPORT
 
 	delete surface;
 	surface = NULL;
diff -ur StepMania-3.9-src.orig/src/RageSurface_Load.cpp StepMania-3.9-src/src/RageSurface_Load.cpp
--- StepMania-3.9-src.orig/src/RageSurface_Load.cpp	2007-08-06 07:35:55.000000000 +0300
+++ StepMania-3.9-src/src/RageSurface_Load.cpp	2007-08-06 07:36:40.000000000 +0300
@@ -18,8 +18,10 @@
 		result = RageSurface_Load_PNG( sPath, ret, bHeaderOnly, error );
 	else if( !format.CompareNoCase("gif") )
 		result = RageSurface_Load_GIF( sPath, ret, bHeaderOnly, error );
+#ifndef NO_JPEG_SUPPORT
 	else if( !format.CompareNoCase("jpg") )
 		result = RageSurface_Load_JPEG( sPath, ret, bHeaderOnly, error );
+#endif
 	else if( !format.CompareNoCase("bmp") )
 		result = RageSurface_Load_BMP( sPath, ret, bHeaderOnly, error );
 	else
