diff -Nru resin-4.0.13.vanilla/configure.ac resin-4.0.13/configure.ac
--- resin-4.0.13.vanilla/configure.ac	2010-12-12 12:18:03.216908371 +0100
+++ resin-4.0.13/configure.ac	2010-12-12 12:22:30.367266372 +0100
@@ -1072,11 +1072,6 @@
     echo "checking for JNI in $JAVA_HOME/include ... found"
   else
     JNI=""
-    AC_MSG_ERROR([
-
-    *** Can't find JNI directory in JAVA_HOME=${JAVA_HOME}
-    *** JNI is expected in ${JAVA_HOME}/include/$jni_os
-  ])
   fi
 
   HAS_JVMTI=""
@@ -1091,8 +1086,6 @@
     echo "Using JVMTI for class reloading"
     
     CFLAGS="$CFLAGS -DHAS_JVMTI"
-  else
-    AC_MSG_ERROR([Can't find JVMTI directory ${JNI_INCLUDE_DIR}/jvmti.h])
   fi
 fi
 
