diff -Nurp boost_1_34_1/boost/regex/v4/cpp_regex_traits.hpp boost_1_34_1_new/boost/regex/v4/cpp_regex_traits.hpp
--- boost_1_34_1/boost/regex/v4/cpp_regex_traits.hpp	2007-01-15 12:09:44.000000000 +0100
+++ boost_1_34_1_new/boost/regex/v4/cpp_regex_traits.hpp	2007-12-21 21:01:33.000000000 +0100
@@ -39,6 +39,7 @@
 #include <boost/regex/pending/object_cache.hpp>
 #endif
 
+#include <climits>
 #include <istream>
 #include <ios>
 
