diff -ur comm-1.9.1.orig/configure.in comm-1.9.1/configure.in
--- comm-1.9.1.orig/configure.in	2010-07-10 17:19:33.000000000 +0000
+++ comm-1.9.1/configure.in	2010-07-10 17:20:54.000000000 +0000
@@ -1487,6 +1487,9 @@
 x86_64 | ia64)
     CPU_ARCH="$OS_TEST"
     ;;
+arm*)
+    CPU_ARCH="arm"
+    ;;
 esac
 
 if test -z "$OS_TARGET"; then
diff -ur comm-1.9.1.orig/mozilla/js/src/configure.in comm-1.9.1/mozilla/js/src/configure.in
--- comm-1.9.1.orig/mozilla/js/src/configure.in	2010-07-10 17:19:33.000000000 +0000
+++ comm-1.9.1/mozilla/js/src/configure.in	2010-07-10 17:21:31.000000000 +0000
@@ -1221,6 +1221,9 @@
 x86_64 | ia64)
     CPU_ARCH="$OS_TEST"
     ;;
+arm*)
+    CPU_ARCH="arm"
+    ;;
 esac
 
 if test -z "$OS_TARGET"; then