diff -uprN -X dontdiff linux-2.6.3/drivers/net/Kconfig linux-2.6.3-netdev-random/drivers/net/Kconfig
--- linux-2.6.3/drivers/net/Kconfig	2004-02-18 04:59:18.000000000 +0100
+++ linux-2.6.3-netdev-random/drivers/net/Kconfig	2004-02-26 19:14:33.162880272 +0100
@@ -108,6 +108,22 @@ config TUN
 
 	  If you don't know what to use this for, you don't need it.
 
+config NET_RANDOM
+	bool "Allow Net Devices to contribute to /dev/random"
+	depends on NETDEVICES && EXPERIMENTAL
+	---help---
+    	  If you say Y here, network device interrupts will contribute to the
+          kernel entropy pool. Normally, block devices and some other devices
+          feed the pool. Some systems, such as those that are headless or diskless,
+          need additional entropy sources.  Some people, however, feel that network
+          devices should not contribute to /dev/random because an external attacker
+          could observe incoming packets in an attempt to learn the entropy pool's
+          state.  If you say N, no network device will contribute entropy.
+
+          If you believe there is a chance of your network packets being observed
+          and you doubt the security of the entropy pool's one-way hash, do not
+	  enable this.  If unsure, say N.
+
 config ETHERTAP
 	tristate "Ethertap network tap"
 	depends on NETDEVICES && EXPERIMENTAL && NETLINK_DEV
diff -uprN -X dontdiff linux-2.6.2/include/asm-alpha/signal.h linux-2.6.2-netdev-random/include/asm-alpha/signal.h
--- linux-2.6.2/include/asm-alpha/signal.h	2004-02-04 04:43:49.000000000 +0100
+++ linux-2.6.2-netdev-random/include/asm-alpha/signal.h	2004-02-16 22:59:59.567750000 +0100
@@ -121,8 +121,15 @@ typedef unsigned long sigset_t;
 #define SA_PROBE		SA_ONESHOT
 #define SA_SAMPLE_RANDOM	SA_RESTART
 #define SA_SHIRQ		0x40000000
+
+#ifdef CONFIG_NET_RANDOM
+#define SA_NET_RANDOM	SA_SAMPLE_RANDOM
+#else
+#define SA_NET_RANDOM	0
 #endif
 
+#endif /* __KERNEL__ */
+
 #define SIG_BLOCK          1	/* for blocking signals */
 #define SIG_UNBLOCK        2	/* for unblocking signals */
 #define SIG_SETMASK        3	/* for setting the signal mask */
diff -uprN -X dontdiff linux-2.6.2/include/asm-arm/signal.h linux-2.6.2-netdev-random/include/asm-arm/signal.h
--- linux-2.6.2/include/asm-arm/signal.h	2004-02-04 04:43:46.000000000 +0100
+++ linux-2.6.2-netdev-random/include/asm-arm/signal.h	2004-02-16 22:59:59.577748480 +0100
@@ -126,8 +126,15 @@ typedef unsigned long sigset_t;
 #define SA_SAMPLE_RANDOM	0x10000000
 #define SA_IRQNOMASK		0x08000000
 #define SA_SHIRQ		0x04000000
+
+#ifdef CONFIG_NET_RANDOM
+#define SA_NET_RANDOM	SA_SAMPLE_RANDOM
+#else
+#define SA_NET_RANDOM	0
 #endif
 
+#endif /* __KERNEL__ */
+
 #define SIG_BLOCK          0	/* for blocking signals */
 #define SIG_UNBLOCK        1	/* for unblocking signals */
 #define SIG_SETMASK        2	/* for setting the signal mask */
diff -uprN -X dontdiff linux-2.6.2/include/asm-cris/signal.h linux-2.6.2-netdev-random/include/asm-cris/signal.h
--- linux-2.6.2/include/asm-cris/signal.h	2004-02-04 04:44:21.000000000 +0100
+++ linux-2.6.2-netdev-random/include/asm-cris/signal.h	2004-02-16 22:59:59.587746960 +0100
@@ -120,8 +120,15 @@ typedef unsigned long sigset_t;
 #define SA_PROBE		SA_ONESHOT
 #define SA_SAMPLE_RANDOM	SA_RESTART
 #define SA_SHIRQ		0x04000000
+
+#ifdef CONFIG_NET_RANDOM
+#define SA_NET_RANDOM	SA_SAMPLE_RANDOM
+#else
+#define SA_NET_RANDOM	0
 #endif
 
+#endif /* __KERNEL__ */
+
 #define SIG_BLOCK          0	/* for blocking signals */
 #define SIG_UNBLOCK        1	/* for unblocking signals */
 #define SIG_SETMASK        2	/* for setting the signal mask */
diff -uprN -X dontdiff linux-2.6.2/include/asm-i386/signal.h linux-2.6.2-netdev-random/include/asm-i386/signal.h
--- linux-2.6.2/include/asm-i386/signal.h	2004-02-04 04:43:15.000000000 +0100
+++ linux-2.6.2-netdev-random/include/asm-i386/signal.h	2004-02-16 22:59:59.708728568 +0100
@@ -121,8 +121,15 @@ typedef unsigned long sigset_t;
 #define SA_PROBE		SA_ONESHOT
 #define SA_SAMPLE_RANDOM	SA_RESTART
 #define SA_SHIRQ		0x04000000
+
+#ifdef CONFIG_NET_RANDOM
+#define SA_NET_RANDOM	SA_SAMPLE_RANDOM
+#else
+#define SA_NET_RANDOM	0
 #endif
 
+#endif /* __KERNEL__ */
+
 #define SIG_BLOCK          0	/* for blocking signals */
 #define SIG_UNBLOCK        1	/* for unblocking signals */
 #define SIG_SETMASK        2	/* for setting the signal mask */
diff -uprN -X dontdiff linux-2.6.2/include/asm-ia64/signal.h linux-2.6.2-netdev-random/include/asm-ia64/signal.h
--- linux-2.6.2/include/asm-ia64/signal.h	2004-02-04 04:43:11.000000000 +0100
+++ linux-2.6.2-netdev-random/include/asm-ia64/signal.h	2004-02-16 22:59:59.729725376 +0100
@@ -126,6 +126,12 @@
 #define SA_SHIRQ		0x04000000
 #define SA_PERCPU_IRQ		0x02000000
 
+#ifdef CONFIG_NET_RANDOM
+#define SA_NET_RANDOM	SA_SAMPLE_RANDOM
+#else
+#define SA_NET_RANDOM	0
+#endif
+
 #endif /* __KERNEL__ */
 
 #define SIG_BLOCK          0	/* for blocking signals */
diff -uprN -X dontdiff linux-2.6.2/include/asm-m68k/signal.h linux-2.6.2-netdev-random/include/asm-m68k/signal.h
--- linux-2.6.2/include/asm-m68k/signal.h	2004-02-04 04:43:15.000000000 +0100
+++ linux-2.6.2-netdev-random/include/asm-m68k/signal.h	2004-02-16 22:59:59.739723856 +0100
@@ -116,8 +116,15 @@ typedef unsigned long sigset_t;
 #define SA_PROBE		SA_ONESHOT
 #define SA_SAMPLE_RANDOM	SA_RESTART
 #define SA_SHIRQ		0x04000000
+
+#ifdef CONFIG_NET_RANDOM
+#define SA_NET_RANDOM	SA_SAMPLE_RANDOM
+#else
+#define SA_NET_RANDOM	0
 #endif
 
+#endif /* __KERNEL__ */
+
 #define SIG_BLOCK          0	/* for blocking signals */
 #define SIG_UNBLOCK        1	/* for unblocking signals */
 #define SIG_SETMASK        2	/* for setting the signal mask */
diff -uprN -X dontdiff linux-2.6.2/include/asm-mips/signal.h linux-2.6.2-netdev-random/include/asm-mips/signal.h
--- linux-2.6.2/include/asm-mips/signal.h	2004-02-04 04:43:41.000000000 +0100
+++ linux-2.6.2-netdev-random/include/asm-mips/signal.h	2004-02-16 22:59:59.748722488 +0100
@@ -111,6 +111,12 @@ typedef unsigned long old_sigset_t;		/* 
 #define SA_SAMPLE_RANDOM	SA_RESTART
 #define SA_SHIRQ		0x02000000
 
+#ifdef CONFIG_NET_RANDOM
+#define SA_NET_RANDOM	SA_SAMPLE_RANDOM
+#else
+#define SA_NET_RANDOM	0
+#endif
+
 #endif /* __KERNEL__ */
 
 #define SIG_BLOCK	1	/* for blocking signals */
diff -uprN -X dontdiff linux-2.6.2/include/asm-parisc/signal.h linux-2.6.2-netdev-random/include/asm-parisc/signal.h
--- linux-2.6.2/include/asm-parisc/signal.h	2004-02-04 04:44:55.000000000 +0100
+++ linux-2.6.2-netdev-random/include/asm-parisc/signal.h	2004-02-16 23:00:00.530603624 +0100
@@ -100,6 +100,12 @@
 #define SA_SAMPLE_RANDOM	SA_RESTART
 #define SA_SHIRQ		0x04000000
 
+#ifdef CONFIG_NET_RANDOM
+#define SA_NET_RANDOM	SA_SAMPLE_RANDOM
+#else
+#define SA_NET_RANDOM	0
+#endif
+
 #endif /* __KERNEL__ */
 
 #define SIG_BLOCK          0	/* for blocking signals */
diff -uprN -X dontdiff linux-2.6.2/include/asm-ppc64/signal.h linux-2.6.2-netdev-random/include/asm-ppc64/signal.h
--- linux-2.6.2/include/asm-ppc64/signal.h	2004-02-04 04:43:43.000000000 +0100
+++ linux-2.6.2-netdev-random/include/asm-ppc64/signal.h	2004-02-16 23:00:00.575596784 +0100
@@ -107,8 +107,15 @@ typedef struct {
 #define SA_PROBE		SA_ONESHOT
 #define SA_SAMPLE_RANDOM	SA_RESTART
 #define SA_SHIRQ		0x04000000
+
+#ifdef CONFIG_NET_RANDOM
+#define SA_NET_RANDOM	SA_SAMPLE_RANDOM
+#else
+#define SA_NET_RANDOM	0
 #endif
 
+#endif /* __KERNEL__ */
+
 #define SIG_BLOCK          0	/* for blocking signals */
 #define SIG_UNBLOCK        1	/* for unblocking signals */
 #define SIG_SETMASK        2	/* for setting the signal mask */
diff -uprN -X dontdiff linux-2.6.2/include/asm-s390/signal.h linux-2.6.2-netdev-random/include/asm-s390/signal.h
--- linux-2.6.2/include/asm-s390/signal.h	2004-02-04 04:44:05.000000000 +0100
+++ linux-2.6.2-netdev-random/include/asm-s390/signal.h	2004-02-16 23:00:00.593594048 +0100
@@ -129,8 +129,15 @@ typedef unsigned long sigset_t;
 #define SA_PROBE                SA_ONESHOT
 #define SA_SAMPLE_RANDOM        SA_RESTART
 #define SA_SHIRQ                0x04000000
+
+#ifdef CONFIG_NET_RANDOM
+#define SA_NET_RANDOM	SA_SAMPLE_RANDOM
+#else
+#define SA_NET_RANDOM	0
 #endif
 
+#endif /* __KERNEL__ */
+
 #define SIG_BLOCK          0    /* for blocking signals */
 #define SIG_UNBLOCK        1    /* for unblocking signals */
 #define SIG_SETMASK        2    /* for setting the signal mask */
diff -uprN -X dontdiff linux-2.6.2/include/asm-sh/signal.h linux-2.6.2-netdev-random/include/asm-sh/signal.h
--- linux-2.6.2/include/asm-sh/signal.h	2004-02-04 04:43:56.000000000 +0100
+++ linux-2.6.2-netdev-random/include/asm-sh/signal.h	2004-02-16 23:00:01.436465912 +0100
@@ -108,8 +108,15 @@ typedef struct {
 #define SA_PROBE		SA_ONESHOT
 #define SA_SAMPLE_RANDOM	SA_RESTART
 #define SA_SHIRQ		0x04000000
+
+#ifdef CONFIG_NET_RANDOM
+#define SA_NET_RANDOM	SA_SAMPLE_RANDOM
+#else
+#define SA_NET_RANDOM	0
 #endif
 
+#endif /* __KERNEL__ */
+
 #define SIG_BLOCK          0	/* for blocking signals */
 #define SIG_UNBLOCK        1	/* for unblocking signals */
 #define SIG_SETMASK        2	/* for setting the signal mask */
diff -uprN -X dontdiff linux-2.6.2/include/asm-sparc/signal.h linux-2.6.2-netdev-random/include/asm-sparc/signal.h
--- linux-2.6.2/include/asm-sparc/signal.h	2004-02-04 04:43:57.000000000 +0100
+++ linux-2.6.2-netdev-random/include/asm-sparc/signal.h	2004-02-16 23:00:01.447464240 +0100
@@ -176,8 +176,15 @@ struct sigstack {
 #define SA_PROBE SA_ONESHOT
 #define SA_SAMPLE_RANDOM SA_RESTART
 #define SA_STATIC_ALLOC		0x80
+
+#ifdef CONFIG_NET_RANDOM
+#define SA_NET_RANDOM	SA_SAMPLE_RANDOM
+#else
+#define SA_NET_RANDOM	0
 #endif
 
+#endif /* __KERNEL__ */
+
 /* Type of a signal handler.  */
 #ifdef __KERNEL__
 typedef void (*__sighandler_t)(int, int, struct sigcontext *, char *);
diff -uprN -X dontdiff linux-2.6.2/include/asm-sparc64/signal.h linux-2.6.2-netdev-random/include/asm-sparc64/signal.h
--- linux-2.6.2/include/asm-sparc64/signal.h	2004-02-04 04:43:20.000000000 +0100
+++ linux-2.6.2-netdev-random/include/asm-sparc64/signal.h	2004-02-16 23:00:01.465461504 +0100
@@ -182,8 +182,15 @@ struct sigstack {
 #define SA_PROBE SA_ONESHOT
 #define SA_SAMPLE_RANDOM SA_RESTART
 #define SA_STATIC_ALLOC		0x80
+
+#ifdef CONFIG_NET_RANDOM
+#define SA_NET_RANDOM	SA_SAMPLE_RANDOM
+#else
+#define SA_NET_RANDOM	0
 #endif
 
+#endif /* __KERNEL__ */
+
 /* Type of a signal handler.  */
 #ifdef __KERNEL__
 typedef void (*__sighandler_t)(int, struct sigcontext *);
diff -uprN -X dontdiff linux-2.6.2/include/asm-x86_64/signal.h linux-2.6.2-netdev-random/include/asm-x86_64/signal.h
--- linux-2.6.2/include/asm-x86_64/signal.h	2004-02-04 04:43:06.000000000 +0100
+++ linux-2.6.2-netdev-random/include/asm-x86_64/signal.h	2004-02-16 23:00:01.484458616 +0100
@@ -128,8 +128,15 @@ typedef unsigned long sigset_t;
 #define SA_PROBE		SA_ONESHOT
 #define SA_SAMPLE_RANDOM	SA_RESTART
 #define SA_SHIRQ		0x04000000
+
+#ifdef CONFIG_NET_RANDOM
+#define SA_NET_RANDOM	SA_SAMPLE_RANDOM
+#else
+#define SA_NET_RANDOM	0
 #endif
 
+#endif /* __KERNEL__ */
+
 #define SIG_BLOCK          0	/* for blocking signals */
 #define SIG_UNBLOCK        1	/* for unblocking signals */
 #define SIG_SETMASK        2	/* for setting the signal mask */
