diff -Nru resin-3.1.6.vanilla/configure.ac resin-3.1.6/configure.ac
--- resin-3.1.6.vanilla/configure.ac	2008-05-10 22:05:43.000000000 +0200
+++ resin-3.1.6/configure.ac	2008-05-10 22:07:41.000000000 +0200
@@ -1162,10 +1162,3 @@
 modules/c/src/resin_os/Makefile
 contrib/init.resin
 ], [])
-
-if test -n "${resin_pro_plugin}"; then
-  AC_OUTPUT([
-    modules/c/src/resin/Makefile
-    modules/c/src/resinssl/Makefile
-  ])
-fi
