--- mozilla/security/manager/Makefile.in.orig	2007-02-23 22:51:02.000000000 +0100
+++ mozilla/security/manager/Makefile.in	2007-02-23 22:53:56.000000000 +0100
@@ -117,13 +117,34 @@
 FREEBL_64FPU_LIB = libfreebl_64fpu_3$(DLL_SUFFIX)
 FREEBL_64FPU_CHK = libfreebl_64fpu_3.chk
 endif
+
+ABS_DIST := $(shell cd $(DIST) && pwd)
+ifeq ($(HOST_OS_ARCH),WINNT)
+ifdef CYGDRIVE_MOUNT
+ABS_DIST := $(shell cygpath -w $(ABS_DIST) | sed -e 's|\\\\|/|g')
+endif
+endif
+NSPR_INCLUDE_DIR = $(firstword $(filter -I%,$(NSPR_CFLAGS)))
+ifneq (,$(strip $(NSPR_INCLUDE_DIR)))
+NSPR_INCLUDE_DIR := $(subst -I,,$(subst -I$(DIST),-I$(ABS_DIST),$(NSPR_INCLUDE_DIR)))
+else
+NSPR_INCLUDE_DIR = $(ABS_DIST)/include/nspr
+endif
+NSPR_LIB_DIR = $(firstword $(filter -L%,$(NSPR_LIBS)))
+ifneq (,$(strip $(NSPR_LIB_DIR)))
+NSPR_LIB_DIR := $(subst -L,,$(subst -L$(DIST),-L$(ABS_DIST),$(NSPR_LIB_DIR)))
+else
+NSPR_LIB_DIR = $(ABS_DIST)/lib
+endif
  
 # NSS makefiles are not safe for parallel execution.
 DEFAULT_GMAKE_FLAGS = MAKE="$(MAKE) -j1" -j1
 DEFAULT_GMAKE_FLAGS += CC="$(CC)"
-DEFAULT_GMAKE_FLAGS += MOZILLA_INCLUDES="-I$(MOZ_BUILD_ROOT)/dist/include/nspr -I$(MOZ_BUILD_ROOT)/dist/include/dbm"
-DEFAULT_GMAKE_FLAGS += SOURCE_MD_DIR=$(MOZ_BUILD_ROOT)/dist
-DEFAULT_GMAKE_FLAGS += DIST=$(MOZ_BUILD_ROOT)/dist
+DEFAULT_GMAKE_FLAGS += MOZILLA_INCLUDES=-I$(ABS_DIST)/include/dbm
+DEFAULT_GMAKE_FLAGS += SOURCE_MD_DIR=$(ABS_DIST)
+DEFAULT_GMAKE_FLAGS += DIST=$(ABS_DIST)
+DEFAULT_GMAKE_FLAGS += NSPR_INCLUDE_DIR=$(NSPR_INCLUDE_DIR)
+DEFAULT_GMAKE_FLAGS += NSPR_LIB_DIR=$(NSPR_LIB_DIR)
 DEFAULT_GMAKE_FLAGS += MOZILLA_CLIENT=1
 DEFAULT_GMAKE_FLAGS += NO_MDUPDATE=1
 ABS_topsrcdir   := $(shell cd $(topsrcdir); pwd)
