diff -Nrup linux-2.6.4-hardened.orig/fs/proc/array.c linux-2.6.4-hardened/fs/proc/array.c
--- linux-2.6.4-hardened.orig/fs/proc/array.c	2004-03-14 21:05:08.000000000 -0500
+++ linux-2.6.4-hardened/fs/proc/array.c	2004-03-15 01:08:47.000000000 -0500
@@ -168,6 +168,17 @@ static inline char * task_state(struct t
 		p->pid && p->ptrace ? p->parent->pid : 0,
 		p->uid, p->euid, p->suid, p->fsuid,
 		p->gid, p->egid, p->sgid, p->fsgid);
+#ifdef CONFIG_GRKERNSEC
+	buffer += sprintf(buffer,
+		"PaX:\t%c%c%c%c%c%c\n",
+		p->flags & PF_PAX_PAGEEXEC ? 'P' : 'p',
+		p->flags & PF_PAX_EMUTRAMP ? 'E' : 'e',
+		p->flags & PF_PAX_MPROTECT ? 'M' : 'm',
+		p->flags & PF_PAX_RANDMMAP ? 'R' : 'r',
+		p->flags & PF_PAX_RANDEXEC ? 'X' : 'x',
+		p->flags & PF_PAX_SEGMEXEC ? 'S' : 's'
+	);
+#endif
 	read_unlock(&tasklist_lock);	
 	task_lock(p);
 	buffer += sprintf(buffer,
