--- configure.in.orig	2007-03-23 08:43:15.000000000 +0000
+++ configure.in	2007-03-23 08:47:04.000000000 +0000
@@ -2578,7 +2578,7 @@
                        if ! ${CXX-g++} ${CXXFLAGS} ${DSO_PIC_CFLAGS} ${DSO_LDOPTS} -S -o conftest.S conftest.c ; then
                          ac_cv_have_visibility_class_bug=yes
                        else
-                         if test `grep -c "@PLT" conftest.S` == 0; then
+                         if test `grep -c "@PLT" conftest.S` = 0; then
                            ac_cv_have_visibility_class_bug=yes
                          fi
                        fi
@@ -2602,7 +2602,7 @@
                        if ! ${CC-cc} ${CFLAGS} ${DSO_PIC_CFLAGS} ${DSO_LDOPTS} -O2 -S -o conftest.S conftest.c ; then
                          ac_cv_have_visibility_builtin_bug=yes
                        else
-                         if test `grep -c "@PLT" conftest.S` == 0; then
+                         if test `grep -c "@PLT" conftest.S` = 0; then
                            ac_cv_visibility_builtin_bug=yes
                          fi
                        fi
