# Upstream: https://bugzilla.mozilla.org/show_bug.cgi?id=454993

--- toolkit/components/passwordmgr/src/storage-Legacy.js.orig	2008-09-24 15:04:27.000000000 +0000
+++ toolkit/components/passwordmgr/src/storage-Legacy.js	2008-09-24 15:06:18.000000000 +0000
@@ -850,7 +850,11 @@
 
         do {
             var hasMore = lineStream.readLine(line);
-            line.value = this._utfConverter.ConvertToUnicode(line.value);
+            try {
+             line.value = this._utfConverter.ConvertToUnicode(line.value);
+            } catch (e) {
+             this.log("Bad UTF8 conversion: " + line.value);
+            }
 
             switch (parseState) {
                 // Check file header
