Index: vlc-0.8.6g/src/playlist/view.c
===================================================================
--- vlc-0.8.6g.orig/src/playlist/view.c
+++ vlc-0.8.6g/src/playlist/view.c
@@ -175,7 +175,7 @@ int playlist_ViewUpdate( playlist_t *p_p
     if( i_view == VIEW_ALL )
     {
         p_view->p_root->i_children = p_playlist->i_size;
-        free( p_view->p_root->pp_children );
+        /*free( p_view->p_root->pp_children );*/
         p_view->p_root->pp_children = p_playlist->pp_items;
     }
 
