Reworked patch for printf() formats by solar.

Index: vlc-0.8.4/modules/visualization/xosd.c
===================================================================
--- vlc-0.8.4.orig/modules/visualization/xosd.c
+++ vlc-0.8.4/modules/visualization/xosd.c
@@ -233,13 +233,13 @@ static void Run( intf_thread_t *p_intf )
             if( p_playlist->status.i_status == PLAYLIST_STOPPED )
             {
                 psz_display = (char *)malloc( sizeof(char )*strlen(_("Stop")));
-                sprintf( psz_display,_("Stop") );
+                sprintf( psz_display, "%s",_("Stop") );
                 vlc_object_release( p_playlist );
             }
             else if( p_playlist->status.i_status == PLAYLIST_PAUSED )
             {
                 psz_display = (char *)malloc( sizeof(char )*strlen(_("Pause")));
-                sprintf( psz_display,_("Pause") );
+                sprintf( psz_display, "%s", _("Pause") );
                 vlc_object_release( p_playlist );
             }
             else
Index: vlc-0.8.4/modules/control/http/macro.c
===================================================================
--- vlc-0.8.4.orig/modules/control/http/macro.c
+++ vlc-0.8.4/modules/control/http/macro.c
@@ -150,7 +150,7 @@ void E_(MacroDo)( httpd_file_sys_t *p_ar
     }
 #define PRINT( str ) \
     ALLOC( strlen( str ) + 1 ); \
-    *pp_dst += sprintf( *pp_dst, str );
+    *pp_dst += sprintf( *pp_dst, "%s", str );
 
 #define PRINTS( str, s ) \
     ALLOC( strlen( str ) + strlen( s ) + 1 ); \
