--- samba-vscan-0.3.6/global/vscan-fileaccesslog.c	2004/09/24 21:42:20
+++ samba-vscan-0.3.6/global/vscan-fileaccesslog.c	2005/07/13 14:58:43
@@ -128,7 +128,7 @@
  *
 */
 struct lrufiles_struct *lrufiles_add(pstring fname, time_t mtime, BOOL infected) {
-	struct lrufiles_struct *new, *tmp, *found = NULL;
+	struct lrufiles_struct *new_entry, *tmp, *found = NULL;
 
 	/* check if lru file access was disabled by setting the corresponding
 	   value in the configuration file to zero (or below zero) */
@@ -150,15 +150,15 @@
 		return found;
 	} else {
 		DEBUG(10, ("alloc space for file entry '%s'\n", fname));
-		new = (struct lrufiles_struct *)malloc(sizeof(*new));
-		if (!new) return NULL;
+		new_entry = (struct lrufiles_struct *)malloc(sizeof(*new_entry));
+		if (!new_entry) return NULL;
 
-		ZERO_STRUCTP(new);
+		ZERO_STRUCTP(new_entry);
 
-		pstrcpy(new->fname, fname);
-		new->mtime = mtime;
-		new->infected = infected;
-		new->time_added = time(NULL);
+		pstrcpy(new_entry->fname, fname);
+		new_entry->mtime = mtime;
+		new_entry->infected = infected;
+		new_entry->time_added = time(NULL);
 
 		/* reached maximum? */
 		if ( lrufiles_count == lrufiles_max_entries ) {
@@ -170,12 +170,12 @@
 		}
 		
 		DEBUG(10, ("adding new entry to list...\n"));
-		DLIST_ADD_END(Lrufiles, new, tmp);
-		LrufilesEnd = new;
+		DLIST_ADD_END(Lrufiles, new_entry, tmp);
+		LrufilesEnd = new_entry;
 		lrufiles_count++;
 		DEBUG(10, ("entry '%s' added, count '%d'\n", fname, lrufiles_count));
 
-		return new;
+		return new_entry;
 	}
 }
 
