--- samba-vscan-0.3.6/nai/vscan-mcdaemon_core.c	2004/09/25 12:27:06
+++ samba-vscan-0.3.6/nai/vscan-mcdaemon_core.c	2005/08/18 13:36:23
@@ -1,5 +1,5 @@
 /*
- * $Id: 016_all_samba-3.0.20-vscan-mc_daemon_core.patch,v 1.1 2005/08/30 11:55:39 satya Exp $
+ * $Id: 016_all_samba-3.0.20-vscan-mc_daemon_core.patch,v 1.1 2005/08/30 11:55:39 satya Exp $
  * 
  * Core Interface for McAfee Virus Scanner via McDaemon			
  *
@@ -28,7 +28,7 @@
 int vscan_mcdaemon_init(void)
 {
 
-	return vscan_init_socket_init("mcdaemon", mcdaemon_ip, mcdaemon_port); 
+	return vscan_inet_socket_init("mcdaemon", mcdaemon_ip, mcdaemon_port); 
 
 }
 
