diff -urN samba-3.0.14a.orig/examples/VFS/samba-vscan-0.3.6/antivir/vscan-antivir.c samba-3.0.14a/examples/VFS/samba-vscan-0.3.6/antivir/vscan-antivir.c
--- samba-3.0.14a.orig/examples/VFS/samba-vscan-0.3.6/antivir/vscan-antivir.c	2005-03-23 15:30:30.000000000 -0500
+++ samba-3.0.14a/examples/VFS/samba-vscan-0.3.6/antivir/vscan-antivir.c	2005-04-26 15:50:58.000000000 -0400
@@ -133,7 +133,7 @@
                                            PARAMCONF or as set via vfs options
                                         */
 
-	#if (SAMBA_VERSION_MAJOR==2 && SAMBA_VERSION_RELEASE>=4) || SAMBA_VERSION_MAJOR==3
+	#if  SAMBA_VERSION_MAJOR==3
 	 #if !(SMB_VFS_INTERFACE_VERSION >= 6)
 	  pstring opts_str;
 	  PROTOTYPE_CONST char *p;
@@ -171,7 +171,7 @@
 	vscan_syslog("INFO: connect to service %s by user %s",
 	       svc, user);
 
-	#if (SAMBA_VERSION_MAJOR==2 && SAMBA_VERSION_RELEASE>=4) || SAMBA_VERSION_MAJOR==3
+	#if  SAMBA_VERSION_MAJOR==3
 
           fstrcpy(config_file, get_configuration_file(conn, VSCAN_MODULE_STR, PARAMCONF));
           DEBUG(3, ("configuration file is: %s\n", config_file));
diff -urN samba-3.0.14a.orig/examples/VFS/samba-vscan-0.3.6/clamav/vscan-clamav.c samba-3.0.14a/examples/VFS/samba-vscan-0.3.6/clamav/vscan-clamav.c
--- samba-3.0.14a.orig/examples/VFS/samba-vscan-0.3.6/clamav/vscan-clamav.c	2005-03-23 15:30:33.000000000 -0500
+++ samba-3.0.14a/examples/VFS/samba-vscan-0.3.6/clamav/vscan-clamav.c	2005-04-26 15:50:58.000000000 -0400
@@ -121,7 +121,7 @@
                                            PARAMCONF or as set via vfs options
                                         */
 
-	#if (SAMBA_VERSION_MAJOR==2 && SAMBA_VERSION_RELEASE>=4) || SAMBA_VERSION_MAJOR==3
+	#if  SAMBA_VERSION_MAJOR==3
 	 #if !(SMB_VFS_INTERFACE_VERSION >= 6)
           pstring opts_str;
           PROTOTYPE_CONST char *p;
@@ -153,7 +153,7 @@
 	vscan_syslog("INFO: connect to service %s by user %s", 
 	       svc, user);
 
-	#if (SAMBA_VERSION_MAJOR==2 && SAMBA_VERSION_RELEASE>=4) || SAMBA_VERSION_MAJOR==3
+	#if  SAMBA_VERSION_MAJOR==3
 
           fstrcpy(config_file, get_configuration_file(conn, VSCAN_MODULE_STR, PARAMCONF));
           DEBUG(3, ("configuration file is: %s\n", config_file));
diff -urN samba-3.0.14a.orig/examples/VFS/samba-vscan-0.3.6/f-secure/vscan-fsav.c samba-3.0.14a/examples/VFS/samba-vscan-0.3.6/f-secure/vscan-fsav.c
--- samba-3.0.14a.orig/examples/VFS/samba-vscan-0.3.6/f-secure/vscan-fsav.c	2005-03-23 15:30:36.000000000 -0500
+++ samba-3.0.14a/examples/VFS/samba-vscan-0.3.6/f-secure/vscan-fsav.c	2005-04-26 15:50:59.000000000 -0400
@@ -121,7 +121,7 @@
                                            PARAMCONF or as set via vfs options
                                         */
 
-        #if (SAMBA_VERSION_MAJOR==2 && SAMBA_VERSION_RELEASE>=4) || SAMBA_VERSION_MAJOR==3
+        #if  SAMBA_VERSION_MAJOR==3
          #if !(SMB_VFS_INTERFACE_VERSION >= 6)
           pstring opts_str;
           PROTOTYPE_CONST char *p;
@@ -158,7 +158,7 @@
 
 	vscan_syslog("INFO: connect to service %s by user %s", svc, user);
 
-        #if (SAMBA_VERSION_MAJOR==2 && SAMBA_VERSION_RELEASE>=4) || SAMBA_VERSION_MAJOR==3
+        #if  SAMBA_VERSION_MAJOR==3
 
          fstrcpy(config_file, get_configuration_file(conn, VSCAN_MODULE_STR, PARAMCONF));
          DEBUG(3, ("configuration file is: %s\n", config_file));
diff -urN samba-3.0.14a.orig/examples/VFS/samba-vscan-0.3.6/fprot/vscan-fprotd.c samba-3.0.14a/examples/VFS/samba-vscan-0.3.6/fprot/vscan-fprotd.c
--- samba-3.0.14a.orig/examples/VFS/samba-vscan-0.3.6/fprot/vscan-fprotd.c	2005-03-23 15:30:47.000000000 -0500
+++ samba-3.0.14a/examples/VFS/samba-vscan-0.3.6/fprot/vscan-fprotd.c	2005-04-26 15:50:59.000000000 -0400
@@ -98,7 +98,7 @@
 
 
 
-	#if (SAMBA_VERSION_MAJOR==2 && SAMBA_VERSION_RELEASE>=4) || SAMBA_VERSION_MAJOR==3
+	#if  SAMBA_VERSION_MAJOR==3
 	 #if !(SMB_VFS_INTERFACE_VERSION >= 6)
           pstring opts_str;
           PROTOTYPE_CONST char *p;
@@ -126,7 +126,7 @@
 	vscan_syslog("INFO: connect to service %s by user %s", 
 	       svc, user);
 
-	#if (SAMBA_VERSION_MAJOR==2 && SAMBA_VERSION_RELEASE>=4) || SAMBA_VERSION_MAJOR==3
+	#if  SAMBA_VERSION_MAJOR==3
           fstrcpy(config_file, get_configuration_file(conn, VSCAN_MODULE_STR, PARAMCONF));
           DEBUG(3, ("configuration file is: %s\n", config_file));
 
diff -urN samba-3.0.14a.orig/examples/VFS/samba-vscan-0.3.6/global/vscan-parameter.c samba-3.0.14a/examples/VFS/samba-vscan-0.3.6/global/vscan-parameter.c
--- samba-3.0.14a.orig/examples/VFS/samba-vscan-0.3.6/global/vscan-parameter.c	2005-02-05 10:08:20.000000000 -0500
+++ samba-3.0.14a/examples/VFS/samba-vscan-0.3.6/global/vscan-parameter.c	2005-04-26 15:51:00.000000000 -0400
@@ -102,7 +102,7 @@
 {
 	static fstring config_file;
 
-        #if (SAMBA_VERSION_MAJOR==2 && SAMBA_VERSION_RELEASE>=4) || SAMBA_VERSION_MAJOR==3
+        #if  SAMBA_VERSION_MAJOR==3
          #if !(SMB_VFS_INTERFACE_VERSION >= 6)
           pstring opts_str;
           PROTOTYPE_CONST char *p;
diff -urN samba-3.0.14a.orig/examples/VFS/samba-vscan-0.3.6/icap/vscan-icap.c samba-3.0.14a/examples/VFS/samba-vscan-0.3.6/icap/vscan-icap.c
--- samba-3.0.14a.orig/examples/VFS/samba-vscan-0.3.6/icap/vscan-icap.c	2005-03-23 15:30:52.000000000 -0500
+++ samba-3.0.14a/examples/VFS/samba-vscan-0.3.6/icap/vscan-icap.c	2005-04-26 15:51:00.000000000 -0400
@@ -100,7 +100,7 @@
                                         */
 
 
-	#if (SAMBA_VERSION_MAJOR==2 && SAMBA_VERSION_RELEASE>=4) || SAMBA_VERSION_MAJOR==3
+	#if  SAMBA_VERSION_MAJOR==3
 	 #if !(SMB_VFS_INTERFACE_VERSION >= 6)
           pstring opts_str;
           PROTOTYPE_CONST char *p;
@@ -125,7 +125,7 @@
 	vscan_syslog("INFO: connect to service %s by user %s", 
 	       svc, user);
 
-	#if (SAMBA_VERSION_MAJOR==2 && SAMBA_VERSION_RELEASE>=4) || SAMBA_VERSION_MAJOR==3
+	#if  SAMBA_VERSION_MAJOR==3
 
           fstrcpy(config_file, get_configuration_file(conn, VSCAN_MODULE_STR, PARAMCONF));
           DEBUG(3, ("configuration file is: %s\n", config_file));
diff -urN samba-3.0.14a.orig/examples/VFS/samba-vscan-0.3.6/kaspersky/vscan-kavp.c samba-3.0.14a/examples/VFS/samba-vscan-0.3.6/kaspersky/vscan-kavp.c
--- samba-3.0.14a.orig/examples/VFS/samba-vscan-0.3.6/kaspersky/vscan-kavp.c	2005-03-23 15:30:57.000000000 -0500
+++ samba-3.0.14a/examples/VFS/samba-vscan-0.3.6/kaspersky/vscan-kavp.c	2005-04-26 15:51:00.000000000 -0400
@@ -99,7 +99,7 @@
                                            PARAMCONF or as set via vfs options
                                         */
 
-	#if (SAMBA_VERSION_MAJOR==2 && SAMBA_VERSION_RELEASE>=4) || SAMBA_VERSION_MAJOR==3
+	#if  SAMBA_VERSION_MAJOR==3
 	 #if !(SMB_VFS_INTERFACE_VERSION >= 6)
           pstring opts_str;
           PROTOTYPE_CONST char *p;
@@ -123,7 +123,7 @@
 	vscan_syslog("INFO: connect to service %s by user %s", 
 	       svc, user);
 
-	#if (SAMBA_VERSION_MAJOR==2 && SAMBA_VERSION_RELEASE>=4) || SAMBA_VERSION_MAJOR==3
+	#if  SAMBA_VERSION_MAJOR==3
           fstrcpy(config_file, get_configuration_file(conn, VSCAN_MODULE_STR, PARAMCONF));
           DEBUG(3, ("configuration file is: %s\n", config_file));
 
diff -urN samba-3.0.14a.orig/examples/VFS/samba-vscan-0.3.6/mks/vscan-mksd.c samba-3.0.14a/examples/VFS/samba-vscan-0.3.6/mks/vscan-mksd.c
--- samba-3.0.14a.orig/examples/VFS/samba-vscan-0.3.6/mks/vscan-mksd.c	2005-03-23 15:31:22.000000000 -0500
+++ samba-3.0.14a/examples/VFS/samba-vscan-0.3.6/mks/vscan-mksd.c	2005-04-26 15:51:01.000000000 -0400
@@ -86,7 +86,7 @@
                                            PARAMCONF or as set via vfs options
                                         */
 
-	#if (SAMBA_VERSION_MAJOR==2 && SAMBA_VERSION_RELEASE>=4) || SAMBA_VERSION_MAJOR==3
+	#if  SAMBA_VERSION_MAJOR==3
 	 #if !(SMB_VFS_INTERFACE_VERSION >= 6)
           pstring opts_str;
           PROTOTYPE_CONST char *p;
@@ -107,7 +107,7 @@
 	vscan_syslog("INFO: connect to service %s by user %s", 
 	       svc, user);
 
-	#if (SAMBA_VERSION_MAJOR==2 && SAMBA_VERSION_RELEASE>=4) || SAMBA_VERSION_MAJOR==3
+	#if  SAMBA_VERSION_MAJOR==3
 
           fstrcpy(config_file, get_configuration_file(conn, VSCAN_MODULE_STR, PARAMCONF));
           DEBUG(3, ("configuration file is: %s\n", config_file));
diff -urN samba-3.0.14a.orig/examples/VFS/samba-vscan-0.3.6/nai/vscan-mcdaemon.c samba-3.0.14a/examples/VFS/samba-vscan-0.3.6/nai/vscan-mcdaemon.c
--- samba-3.0.14a.orig/examples/VFS/samba-vscan-0.3.6/nai/vscan-mcdaemon.c	2005-03-23 15:31:27.000000000 -0500
+++ samba-3.0.14a/examples/VFS/samba-vscan-0.3.6/nai/vscan-mcdaemon.c	2005-04-26 15:51:01.000000000 -0400
@@ -89,7 +89,7 @@
                                            PARAMCONF or as set via vfs options
                                         */
 
-        #if (SAMBA_VERSION_MAJOR==2 && SAMBA_VERSION_RELEASE>=4) || SAMBA_VERSION_MAJOR==3
+        #if  SAMBA_VERSION_MAJOR==3
          #if !(SMB_VFS_INTERFACE_VERSION >= 6)
           pstring opts_str;
           PROTOTYPE_CONST char *p;
@@ -116,7 +116,7 @@
 	vscan_syslog("INFO: connect to service %s by user %s", 
 	       svc, user);
 
-        #if (SAMBA_VERSION_MAJOR==2 && SAMBA_VERSION_RELEASE>=4) || SAMBA_VERSION_MAJOR==3
+        #if  SAMBA_VERSION_MAJOR==3
           fstrcpy(config_file, get_configuration_file(conn, VSCAN_MODULE_STR, PARAMCONF));
           DEBUG(3, ("configuration file is: %s\n", config_file));
 
diff -urN samba-3.0.14a.orig/examples/VFS/samba-vscan-0.3.6/openantivirus/vscan-oav.c samba-3.0.14a/examples/VFS/samba-vscan-0.3.6/openantivirus/vscan-oav.c
--- samba-3.0.14a.orig/examples/VFS/samba-vscan-0.3.6/openantivirus/vscan-oav.c	2005-03-23 15:31:42.000000000 -0500
+++ samba-3.0.14a/examples/VFS/samba-vscan-0.3.6/openantivirus/vscan-oav.c	2005-04-26 15:51:02.000000000 -0400
@@ -98,7 +98,7 @@
         	                           PARAMCONF or as set via vfs options
                 	                */
 
-	#if (SAMBA_VERSION_MAJOR==2 && SAMBA_VERSION_RELEASE>=4) || SAMBA_VERSION_MAJOR==3
+	#if  SAMBA_VERSION_MAJOR==3
 	 #if !(SMB_VFS_INTERFACE_VERSION >= 6)
           pstring opts_str;
           PROTOTYPE_CONST char *p;
@@ -126,7 +126,7 @@
 	vscan_syslog("INFO: connect to service %s by user %s", 
 	       svc, user);
 
-	#if (SAMBA_VERSION_MAJOR==2 && SAMBA_VERSION_RELEASE>=4) || SAMBA_VERSION_MAJOR==3
+	#if  SAMBA_VERSION_MAJOR==3
           
 	  fstrcpy(config_file, get_configuration_file(conn, VSCAN_MODULE_STR, PARAMCONF));
 	  DEBUG(0, ("configuration file is: %s\n", config_file));
diff -urN samba-3.0.14a.orig/examples/VFS/samba-vscan-0.3.6/sophos/vscan-sophos.c samba-3.0.14a/examples/VFS/samba-vscan-0.3.6/sophos/vscan-sophos.c
--- samba-3.0.14a.orig/examples/VFS/samba-vscan-0.3.6/sophos/vscan-sophos.c	2005-03-23 15:31:45.000000000 -0500
+++ samba-3.0.14a/examples/VFS/samba-vscan-0.3.6/sophos/vscan-sophos.c	2005-04-26 15:51:02.000000000 -0400
@@ -94,7 +94,7 @@
                                            PARAMCONF or as set via vfs options
                                         */
 
-	#if (SAMBA_VERSION_MAJOR==2 && SAMBA_VERSION_RELEASE>=4) || SAMBA_VERSION_MAJOR==3
+	#if  SAMBA_VERSION_MAJOR==3
 	 #if !(SMB_VFS_INTERFACE_VERSION >= 6)
           pstring opts_str;
           PROTOTYPE_CONST char *p;
@@ -118,7 +118,7 @@
 	vscan_syslog("INFO: connect to service %s by user %s", 
 	       svc, user);
 
-	#if (SAMBA_VERSION_MAJOR==2 && SAMBA_VERSION_RELEASE>=4) || SAMBA_VERSION_MAJOR==3
+	#if  SAMBA_VERSION_MAJOR==3
           fstrcpy(config_file, get_configuration_file(conn, VSCAN_MODULE_STR, PARAMCONF));
           DEBUG(3, ("configuration file is: %s\n", config_file));
 
diff -urN samba-3.0.14a.orig/examples/VFS/samba-vscan-0.3.6/symantec/vscan-symantec.c samba-3.0.14a/examples/VFS/samba-vscan-0.3.6/symantec/vscan-symantec.c
--- samba-3.0.14a.orig/examples/VFS/samba-vscan-0.3.6/symantec/vscan-symantec.c	2005-03-23 15:31:49.000000000 -0500
+++ samba-3.0.14a/examples/VFS/samba-vscan-0.3.6/symantec/vscan-symantec.c	2005-04-26 15:51:03.000000000 -0400
@@ -94,7 +94,7 @@
                                            PARAMCONF or as set via vfs options
                                         */
 
-	#if (SAMBA_VERSION_MAJOR==2 && SAMBA_VERSION_RELEASE>=4) || SAMBA_VERSION_MAJOR==3
+	#if  SAMBA_VERSION_MAJOR==3
 	 #if !(SMB_VFS_INTERFACE_VERSION >= 6)
           pstring opts_str;
           PROTOTYPE_CONST char *p;
@@ -118,7 +118,7 @@
 	vscan_syslog("INFO: connect to service %s by user %s", 
 	       svc, user);
 
-	#if (SAMBA_VERSION_MAJOR==2 && SAMBA_VERSION_RELEASE>=4) || SAMBA_VERSION_MAJOR==3
+	#if  SAMBA_VERSION_MAJOR==3
           fstrcpy(config_file, get_configuration_file(conn, VSCAN_MODULE_STR, PARAMCONF));
           DEBUG(3, ("configuration file is: %s\n", config_file));
 
diff -urN samba-3.0.14a.orig/examples/VFS/samba-vscan-0.3.6/trend/vscan-trend.c samba-3.0.14a/examples/VFS/samba-vscan-0.3.6/trend/vscan-trend.c
--- samba-3.0.14a.orig/examples/VFS/samba-vscan-0.3.6/trend/vscan-trend.c	2005-03-23 15:31:52.000000000 -0500
+++ samba-3.0.14a/examples/VFS/samba-vscan-0.3.6/trend/vscan-trend.c	2005-04-26 15:51:03.000000000 -0400
@@ -92,7 +92,7 @@
                                            PARAMCONF or as set via vfs options
                                         */
 
-	#if (SAMBA_VERSION_MAJOR==2 && SAMBA_VERSION_RELEASE>=4) || SAMBA_VERSION_MAJOR==3
+	#if  SAMBA_VERSION_MAJOR==3
 	 #if !(SMB_VFS_INTERFACE_VERSION >= 6)
           pstring opts_str;
           PROTOTYPE_CONST char *p;
@@ -116,7 +116,7 @@
 	vscan_syslog("INFO: connect to service %s by user %s", 
 	       svc, user);
 
-	#if (SAMBA_VERSION_MAJOR==2 && SAMBA_VERSION_RELEASE>=4) || SAMBA_VERSION_MAJOR==3
+	#if  SAMBA_VERSION_MAJOR==3
 
           fstrcpy(config_file, get_configuration_file(conn, VSCAN_MODULE_STR, PARAMCONF));
           DEBUG(3, ("configuration file is: %s\n", config_file));
