diff -ur mozilla.orig/xpcom/reflect/xptcall/src/md/unix/xptcinvoke_arm.cpp mozilla/xpcom/reflect/xptcall/src/md/unix/xptcinvoke_arm.cpp
--- mozilla.orig/xpcom/reflect/xptcall/src/md/unix/xptcinvoke_arm.cpp	2008-09-25 10:58:54.000000000 +0200
+++ mozilla/xpcom/reflect/xptcall/src/md/unix/xptcinvoke_arm.cpp	2008-09-25 11:05:25.000000000 +0200
@@ -212,7 +212,7 @@
     "add	sp, sp, r4	\n\t"	/* restore stack pointer		*/
     "mov	%0, r0		\n\t"	/* the result...			*/
     : "=r" (result)
-    : "r" (&my_params)
+    : "r" (&my_params), "m" (my_params)
     : "r0", "r1", "r2", "r3", "r4", "ip", "lr", "sp"
     );
     
diff -ur mozilla.orig/xpcom/reflect/xptcall/src/md/unix/xptcstubs_arm.cpp mozilla/xpcom/reflect/xptcall/src/md/unix/xptcstubs_arm.cpp
--- mozilla.orig/xpcom/reflect/xptcall/src/md/unix/xptcstubs_arm.cpp	2008-09-25 10:58:54.000000000 +0200
+++ mozilla/xpcom/reflect/xptcall/src/md/unix/xptcstubs_arm.cpp	2008-09-25 11:04:51.000000000 +0200
@@ -57,7 +57,8 @@
 #endif
 
 /* Specify explicitly a symbol for this function, don't try to guess the c++ mangled symbol.  */
-static nsresult PrepareAndDispatch(nsXPTCStubBase* self, uint32 methodIndex, PRUint32* args) asm("_PrepareAndDispatch") 
+static nsresult PrepareAndDispatch(nsXPTCStubBase* self, uint32 methodIndex, PRUint32* args) asm("_PrepareAndDispatch")
+__attribute__((used))
 DONT_DROP_OR_WARN;
 
 static nsresult
