--- toolkit/mozapps/update/src/updater/updater.cpp.old	Tue Oct 21 16:36:29 2008 +1300
+++ toolkit/mozapps/update/src/updater/updater.cpp	Mon Oct 20 22:55:57 2008 -0500
@@ -118,7 +118,9 @@ void LaunchChild(int argc, char **argv);
 #endif
 
 #ifndef MAXPATHLEN
-# ifdef MAX_PATH
+# ifdef PATH_MAX
+#  define MAXPATHLEN PATH_MAX
+# elif defined(_MAX_PATH)
 #  define MAXPATHLEN MAX_PATH
 # elif defined(_MAX_PATH)
 #  define MAXPATHLEN _MAX_PATH
--- toolkit/xre/nsAppRunner.h.old	Tue Oct 21 16:36:29 2008 +1300
+++ toolkit/xre/nsAppRunner.h	Mon Oct 20 22:55:57 2008 -0500
@@ -40,10 +40,14 @@
 
 #ifdef XP_WIN
 #include <windows.h>
+#else
+#include <limits.h>
 #endif
 
 #ifndef MAXPATHLEN
-#ifdef _MAX_PATH
+#ifdef PATH_MAX
+#define MAXPATHLEN PATH_MAX
+#elif defined(_MAX_PATH)
 #define MAXPATHLEN _MAX_PATH
 #elif defined(CCHMAXPATH)
 #define MAXPATHLEN CCHMAXPATH
--- xpcom/build/nsXPCOMPrivate.h.old	Tue Oct 21 16:36:29 2008 +1300
+++ xpcom/build/nsXPCOMPrivate.h	Mon Oct 20 22:55:57 2008 -0500
@@ -231,6 +231,7 @@ NS_GetFrozenFunctions(XPCOMFunctions *en
 #define XUL_DLL   "libxul"MOZ_DLL_SUFFIX
 
 #else // Unix
+#include <limits.h> // for PATH_MAX
 
 #define XPCOM_DLL "libxpcom"MOZ_DLL_SUFFIX
 

