Index: gpac/modules/ffmpeg_in/ffmpeg_decode.c
===================================================================
--- gpac.orig/modules/ffmpeg_in/ffmpeg_decode.c
+++ gpac/modules/ffmpeg_in/ffmpeg_decode.c
@@ -241,7 +241,7 @@ static GF_Err FFDEC_AttachStream(GF_Base
 
 	/*setup audio streams*/
 	if (ffd->st==GF_STREAM_AUDIO) {
-		if ((ffd->codec->type == CODEC_ID_MP3LAME) || (ffd->codec->type == CODEC_ID_MP2)) {
+		if ((ffd->codec->type == CODEC_ID_MP3) || (ffd->codec->type == CODEC_ID_MP2)) {
 			ffd->ctx->frame_size = (ffd->ctx->sample_rate > 24000) ? 1152 : 576;
 		}
 		/*may be 0 (cfg not known yet)*/
