from doc/APIchanges in ffmpeg:
 * ae628ec - rename ByteIOContext to AVIOContext.

Index: gpac/modules/ffmpeg_in/ffmpeg_in.h
===================================================================
--- gpac.orig/modules/ffmpeg_in/ffmpeg_in.h
+++ gpac/modules/ffmpeg_in/ffmpeg_in.h
@@ -144,7 +144,7 @@ typedef struct
 	/*file downloader*/
 	GF_DownloadSession *dnload;
 
-	ByteIOContext   io;
+	AVIOContext   io;
 	char *buffer;
 	u32 buffer_size;
 
