Prereq: 2007-01-29
diff -ur emacs-18.59-orig/src/ChangeLog emacs-18.59/src/ChangeLog
--- emacs-18.59-orig/src/ChangeLog	2007-01-29 21:47:56.000000000 +0100
+++ emacs-18.59/src/ChangeLog	2007-01-30 01:00:00.000000000 +0100
@@ -1,3 +1,7 @@
+2007-01-30  Ulrich Mueller  <ulm@kph.uni-mainz.de>
+
+	* x11term.c (internal_socket_read): Handle XK_BackSpace key.
+
 2007-01-29  Ulrich Mueller  <ulm@kph.uni-mainz.de>
 
 	* callproc.c, doprnt.c, emacssignal.h, fns.c, lisp.h, lread.c,
diff -ur emacs-18.59-orig/src/x11term.c emacs-18.59/src/x11term.c
--- emacs-18.59-orig/src/x11term.c	2007-01-29 21:47:56.000000000 +0100
+++ emacs-18.59/src/x11term.c	2007-01-30 01:00:00.000000000 +0100
@@ -1482,6 +1482,10 @@
 	  strcpy(mapping_buf,"\016");
 	  nbytes = 1;
 	  break;
+	case XK_BackSpace:
+	  strcpy(mapping_buf,"\177");
+	  nbytes = 1;
+	  break;
 	}
 #ifndef AIX
       }
