Prereq: 2007-01-29
--- emacs-18.59-orig/etc/ChangeLog	2007-01-29 21:47:56.000000000 +0100
+++ emacs-18.59/etc/ChangeLog	2009-05-29 14:55:51.000000000 +0200
@@ -1,3 +1,9 @@
+2009-05-29  Ulrich Mueller  <ulm@gentoo.org>
+
+	* etags.c (etags_getline): Renamed from getline, fixes compilation
+	with glibc 2.10 headers.
+	(put_entries): Fix format specifier for long int.
+
 2007-01-29  Ulrich Mueller  <ulm@kph.uni-mainz.de>
 
 	* emacsclient.c, server.c: Check for HAVE_SOCKETS instead of BSD.
--- emacs-18.59-orig/etc/etags.c	2007-01-29 21:47:56.000000000 +0100
+++ emacs-18.59/etc/etags.c	2009-05-29 14:50:13.000000000 +0200
@@ -616,7 +616,7 @@
 
   if (eflag)
     {
-      fprintf (outf, "%s%c%d,%d\n",
+      fprintf (outf, "%s%c%d,%ld\n",
 	       node->pat, 0177, node->lno, node->cno);
     }
   else if (!xflag)
@@ -840,9 +840,9 @@
 		      if (linestart != linecharno)
 			{
 #ifdef VMS
-			  getline (vmslinestart);
+			  etags_getline (vmslinestart);
 #else
-			  getline (linestart);
+			  etags_getline (linestart);
 #endif
 			  strncpy (tok, token + (lb1.buffer - buf),
 				   tp-token+1);
@@ -999,7 +999,7 @@
   return !bad && win;
 }
 
-getline (atchar)
+etags_getline (atchar)
      long atchar;
 {
   long saveftell = ftell (inf);
