Support x32 ABI on x86_64.

--- emacs-18.59-orig/etc/fakemail.c
+++ emacs-18.59/etc/fakemail.c
@@ -272,7 +272,7 @@
 make_file_preface ()
 {
   char *the_string, *temp;
-  long idiotic_interface;
+  time_t idiotic_interface;
   long prefix_length;
   long user_length;
   long date_length;
--- emacs-18.59-orig/etc/wakeup.c
+++ emacs-18.59/etc/wakeup.c
@@ -11,7 +11,7 @@
      char **argv;
 {
   int period = 60;
-  long when;
+  time_t when;
   struct tm *tp;
 
   if (argc > 1)
--- emacs-18.59-orig/src/m-x86-x32.h
+++ emacs-18.59/src/m-x86-x32.h
@@ -0,0 +1,9 @@
+/* m- file for x32 ABI on x86_64. */
+
+#include "m-intel386.h"
+
+#define NO_ARG_ARRAY
+
+#undef ULIMIT_BREAK_VALUE
+#undef DATA_SEG_BITS
+#undef SEGMENT_MASK
