--- skey-1.1.5-orig/Makefile.in
+++ skey-1.1.5/Makefile.in
@@ -67,10 +67,10 @@
 	${CC} -o $@ ${SKEYOBJS} ${LDFLAGS} -lskey ${LIBS}
 
 skeyinit: libskey.so ${SKEYINITOBJS}
-	${CC} -o $@ ${SKEYINITOBJS} ${LDFLAGS} -lskey ${LIBS} 
+	${CC} -o $@ ${SKEYINITOBJS} ${LDFLAGS} -lskey ${LIBS} -Wl,-z,now
 
 skeyinfo: libskey.so ${SKEYINFOOBJS}
-	${CC} -o $@ ${SKEYINFOOBJS} ${LDFLAGS} -lskey ${LIBS}
+	${CC} -o $@ ${SKEYINFOOBJS} ${LDFLAGS} -lskey ${LIBS} -Wl,-z,now
 
 ${MANPAGES} ${SCRIPTS}::
 	${FIXPATHSCMD} ${srcdir}/$@
