--- skey-1.1.5-orig/skeyinit.c
+++ skey-1.1.5/skeyinit.c
@@ -111,7 +111,7 @@
 	defaultseed[SKEY_NAMELEN] = '\0';
 	time(&now);
 	snprintf(tbuf, sizeof(tbuf), "%05ld", (long) (now % 100000));
-	strncat(defaultseed, tbuf, sizeof(defaultseed));
+	strncat(defaultseed, tbuf, sizeof(defaultseed) - SKEY_NAMELEN - 1);
 
 	if ((pp = getpwuid(getuid())) == NULL)
 		err(1, "no user with uid %ld", (u_long)getuid());
