diff -ur flac-1.2.0.orig/configure.in flac-1.2.0/configure.in
--- flac-1.2.0.orig/configure.in	2007-07-07 09:10:35.000000000 +0300
+++ flac-1.2.0/configure.in	2007-08-05 20:37:27.000000000 +0300
@@ -280,7 +280,7 @@
 else
 	OUR_CFLAGS_HEAD="$OUR_CFLAGS_HEAD -DNDEBUG"
 	if test "x$GCC" = xyes; then
-		OUR_CFLAGS_HEAD="$OUR_CFLAGS_HEAD -O3 -funroll-loops -finline-functions -Wall -W -Winline -DFLaC__INLINE=__inline__"
+		OUR_CFLAGS_HEAD="$OUR_CFLAGS_HEAD -DFLaC__INLINE=__inline__"
 	fi
 fi
 CFLAGS="$OUR_CFLAGS_HEAD $CFLAGS"
