diff -ur flac-1.2.0.orig/src/flac/encode.c flac-1.2.0/src/flac/encode.c
--- flac-1.2.0.orig/src/flac/encode.c	2007-07-10 05:10:44.000000000 +0300
+++ flac-1.2.0/src/flac/encode.c	2007-08-06 14:04:05.000000000 +0300
@@ -2177,10 +2177,12 @@
 		return true;
 
 	if(num_requested_seek_points < 0) {
+#if FLAC__HAS_OGG
 		/*@@@@@@ workaround ogg bug: too many seekpoints makes table not fit in one page */
 		if(e->use_ogg && e->total_samples_to_encode > 0 && e->total_samples_to_encode / e->sample_rate / 10 > 230)
 			requested_seek_points = "230x;";
 		else 
+#endif
 			requested_seek_points = "10s;";
 		num_requested_seek_points = 1;
 	}
