--- comm-1.9.2/mozilla/js/src/nanojit/CodeAlloc.cpp
+++ comm-1.9.2/mozilla/js/src/nanojit/CodeAlloc.cpp
@@ -247,8 +247,20 @@
 #endif
 
 #ifdef AVMPLUS_SPARC
+#ifdef __linux__  // bugzilla 502369
+void sync_instruction_memory(caddr_t v, u_int len)
+{
+    caddr_t end = v + len;
+    caddr_t p = v;
+    while (p < end) {
+        asm("flush %0" : : "r" (p));
+        p += 32;
+    }
+}
+#else
 extern  "C" void sync_instruction_memory(caddr_t v, u_int len);
 #endif
+#endif
 
 #if defined NANOJIT_IA32 || defined NANOJIT_X64
     // intel chips have dcache/icache interlock