diff -Nru '--exclude=*.xml' resin-4.0.5.vanilla//Makefile.am resin-4.0.5//Makefile.am
--- resin-4.0.5.vanilla//Makefile.am	2010-04-02 13:43:38.444790278 +0100
+++ resin-4.0.5//Makefile.am	2010-04-02 15:19:16.012937963 +0100
@@ -20,35 +20,35 @@
 	- rm -r $(LIBEXEC)
 
 install	:
-	if test $(resin_home) != `pwd`; then \
-	  mkdir -p $(resin_home)/bin; \
-	  mkdir -p $(resin_home)/keys; \
-	  mkdir -p $(resin_home)/licenses; \
-	  mkdir -p $(resin_home)/lib; \
-	  mkdir -p $(resin_home)/$(LIBEXEC); \
-	  cp lib/*.jar $(resin_home)/lib; \
-	  cp bin/* $(resin_home)/bin; \
+	if test $(DESTDIR)/$(resin_home) != `pwd`; then \
+	  mkdir -p $(DESTDIR)/$(resin_home)/bin; \
+	  mkdir -p $(DESTDIR)/$(resin_home)/keys; \
+	  mkdir -p $(DESTDIR)/$(resin_home)/licenses; \
+	  mkdir -p $(DESTDIR)/$(resin_home)/lib; \
+	  mkdir -p $(DESTDIR)/$(resin_home)/$(LIBEXEC); \
+	  cp lib/*.jar $(DESTDIR)/$(resin_home)/lib; \
+	  cp bin/* $(DESTDIR)/$(resin_home)/bin; \
 	fi; \
-	if test $(resin_root) != `pwd`; then \
-	  mkdir -p $(resin_root)/webapps; \
-	  cp -r $(CP_U_FLAG) webapps/* $(resin_root)/webapps; \
-	  mkdir -p $(resin_root)/doc; \
-	  cp -r $(CP_U_FLAG) doc/* $(resin_root)/doc; \
-	  mkdir -p $(resin_log); \
+	if test $(DESTDIR)/$(resin_root) != `pwd`; then \
+	  mkdir -p $(DESTDIR)/$(resin_root)/webapps; \
+	  cp -r $(CP_U_FLAG) webapps/* $(DESTDIR)/$(resin_root)/webapps; \
+	  mkdir -p $(DESTDIR)/$(resin_root)/doc; \
+	  cp -r $(CP_U_FLAG) doc/* $(DESTDIR)/$(resin_root)/doc; \
+	  mkdir -p $(DESTDIR)/$(resin_log); \
 	fi; \
-	if test $(resin_conf) != `pwd`/conf; then \
-	  mkdir -p $(resin_conf); \
-	  cp $(CP_U_FLAG) conf/app-default.xml $(resin_conf)/app-default.xml; \
-	  cp $(CP_U_FLAG) conf/resin.xml $(resin_conf)/resin.xml; \
+	if test $(DESTDIR)/$(resin_conf) != `pwd`/conf; then \
+	  mkdir -p $(DESTDIR)/$(resin_conf); \
+	  cp $(CP_U_FLAG) conf/app-default.xml $(DESTDIR)/$(resin_conf)/app-default.xml; \
+	  cp $(CP_U_FLAG) conf/resin.xml $(DESTDIR)/$(resin_conf)/resin.xml; \
 	fi; \
 	(cd modules/c/src; $(MAKE) install)
-	if test -r $(resin_pro)/modules/c/src/Makefile; then \
-	  cd $(resin_pro)/modules/c/src; $(MAKE) install; \
+	if test -r $(DESTDIR)/$(resin_pro)/modules/c/src/Makefile; then \
+	  cd $(DESTDIR)/$(resin_pro)/modules/c/src; $(MAKE) install; \
 	fi; \
-	if test -n "${resin_init_d}"; then \
+	if test -n "${resin_init_d}" && false; then \
 		if test -w "${resin_init_d}"; then \
-			cp $(CP_U_FLAG) init.d/resin $(resin_init_d); \
-			chmod +x $(resin_init_d); \
+			cp $(CP_U_FLAG) init.d/resin $(DESTDIR)/$(resin_init_d); \
+			chmod +x $(DESTDIR)/$(resin_init_d); \
 		else \
 			echo ;\
 			echo ;\
diff -Nru '--exclude=*.xml' resin-4.0.5.vanilla//modules/c/src/resin_os/Makefile.in resin-4.0.5//modules/c/src/resin_os/Makefile.in
--- resin-4.0.5.vanilla//modules/c/src/resin_os/Makefile.in	2010-04-02 13:43:37.835791135 +0100
+++ resin-4.0.5//modules/c/src/resin_os/Makefile.in	2010-04-02 15:39:03.979847962 +0100
@@ -39,8 +39,8 @@
 
 install	: libresin_os.$(SO)
 #	cp resin $(resin_home)/bin
-	- mkdir -p $(resin_home)/$(LIBEXEC)
-	cp libresin_os.$(SO) $(resin_home)/$(LIBEXEC)
+	- mkdir -p $(DISTDIR)/$(resin_home)/$(LIBEXEC)
+	cp libresin_os.$(SO) $(DISTDIR)/$(resin_home)/$(LIBEXEC)
 
 
 
