--- mozilla-1.9.2/media/liboggz/include/oggz/oggz_off_t_generated.h.old	2010-01-11 01:52:46 +0100
+++ mozilla-1.9.2/media/liboggz/include/oggz/oggz_off_t_generated.h	2010-01-11 01:53:03 +0100
@@ -59,7 +59,7 @@
 
 #include <sys/types.h>
 
-#if defined(__APPLE__) || defined(SOLARIS) || defined(OS2)
+#if defined(__APPLE__) || defined(SOLARIS) || defined(OS2) || defined(__FreeBSD__)
 typedef off_t oggz_off_t;
 #else
 typedef loff_t oggz_off_t;
