Index: rezound-0.12.3beta/src/frontend_fox/CMainWindow.cpp
===================================================================
--- rezound-0.12.3beta.orig/src/frontend_fox/CMainWindow.cpp
+++ rezound-0.12.3beta/src/frontend_fox/CMainWindow.cpp
@@ -438,7 +438,7 @@ void CMainWindow::rebuildSoundWindowList
 
 long CMainWindow::onSoundListChange(FXObject *sender,FXSelector sel,void *ptr)
 {
-	FXint index=(FXint)ptr;
+	FXint index=(long)ptr;
 
 	if(index>=0 && index<soundList->getNumItems())
 	{
@@ -1853,13 +1853,13 @@ long CMainWindow::onKeyboardShuttle(FXOb
 
 	if(pos==0 && FXSELID(sel)==ID_SHUTTLE_BACKWARD)
 	{
-		if((unsigned)ptr > 1/*came from keyboard event*/) enableAutoRepeat(getApp()->getDisplay(),false); // QQQ
+		if((long)ptr > 1/*came from keyboard event*/) enableAutoRepeat(getApp()->getDisplay(),false); // QQQ
 		shuttleDial->setValue(pos-inc);
 		onShuttleChange(sender,sel,ptr);
 	}
 	else if(pos==0 && FXSELID(sel)==ID_SHUTTLE_FORWARD)
 	{
-		if((unsigned)ptr > 1/*came from keyboard event*/) enableAutoRepeat(getApp()->getDisplay(),false); // QQQ
+		if((long)ptr > 1/*came from keyboard event*/) enableAutoRepeat(getApp()->getDisplay(),false); // QQQ
 		shuttleDial->setValue(pos+inc);
 		onShuttleChange(sender,sel,ptr);
 	}
Index: rezound-0.12.3beta/src/frontend_fox/CMp3Dialog.cpp
===================================================================
--- rezound-0.12.3beta.orig/src/frontend_fox/CMp3Dialog.cpp
+++ rezound-0.12.3beta/src/frontend_fox/CMp3Dialog.cpp
@@ -170,7 +170,7 @@ template <class type> void setEnable(typ
 
 long CMp3Dialog::onRadioButton(FXObject *sender,FXSelector sel,void *ptr)
 {
-	if((int)ptr==0) // only act when ptr==1 when it's getting checked
+	if((long)ptr==0) // only act when ptr==1 when it's getting checked
 		return 1;
 
 	// turn off all buttons
Index: rezound-0.12.3beta/src/frontend_fox/COggDialog.cpp
===================================================================
--- rezound-0.12.3beta.orig/src/frontend_fox/COggDialog.cpp
+++ rezound-0.12.3beta/src/frontend_fox/COggDialog.cpp
@@ -136,7 +136,7 @@ template <class type> void setEnable(typ
 
 long COggDialog::onRadioButton(FXObject *sender,FXSelector sel,void *ptr)
 {
-	if((int)ptr==0) // only act when ptr==1 when it's getting checked
+	if((long)ptr==0) // only act when ptr==1 when it's getting checked
 		return 1;
 
 	// turn off all buttons
Index: rezound-0.12.3beta/src/frontend_fox/CPasteChannelsDialog.cpp
===================================================================
--- rezound-0.12.3beta.orig/src/frontend_fox/CPasteChannelsDialog.cpp
+++ rezound-0.12.3beta/src/frontend_fox/CPasteChannelsDialog.cpp
@@ -206,7 +206,7 @@ bool CPasteChannelsDialog::show(CActionS
 	{
 		pasteChannels.clear();
 
-		actionParameters->setValue<unsigned>(_("MixMethod"),(unsigned)(mixTypeComboBox->getItemData(mixTypeComboBox->getCurrentItem())));
+		actionParameters->setValue<unsigned>(_("MixMethod"),(long)(mixTypeComboBox->getItemData(mixTypeComboBox->getCurrentItem())));
 		
 		if(repeatTypeComboBox->getCurrentItem()==0)
 		{ // repeating it a given number of times
