Index: rezound-0.12.3beta/config/m4/cxx.m4
===================================================================
--- rezound-0.12.3beta.orig/config/m4/cxx.m4
+++ rezound-0.12.3beta/config/m4/cxx.m4
@@ -29,7 +29,7 @@ dnl Usage - Pass one argument- the Compi
 dnl found to be valid, it is included in CXXFLAGS otherwise, it's left out
 AC_DEFUN(ajv_CXX_FLAG,
 
-	[AC_MSG_CHECKING(if $CXX accepts $1 )] 
+	[AC_MSG_CHECKING(if $CXX accepts $1 )
 
 	cat > ajv_cxx_flag.cc << EOF
 	int main(){}
@@ -45,5 +45,5 @@ EOF
 
 	rm -rf ajv_cxx_flag.cc
 	rm -rf ajv_cxx_flag.err
-)
+])
 
