--- rezound-0.12.2beta.orig/src/backend/COSSSoundPlayer.cpp	2007-01-28 13:39:58.039273892 +0100
+++ rezound-0.12.2beta/src/backend/COSSSoundPlayer.cpp	2007-01-28 13:40:04.931273892 +0100
@@ -83,7 +83,7 @@
 			int sampleSize=0;
 			string sSampleFormat="none";
 #warning need to change this to try several formats for a supported implementation because float is being reduced to 16bit right now
-#ifndef WORDS_BIGENDIN
+#ifndef WORDS_BIGENDIAN
 			// little endian platform
 	#if defined(SAMPLE_TYPE_S16)
 			sampleFormat=AFMT_S16_LE;
