Index: rezound-0.12.3beta/configure.ac
===================================================================
--- rezound-0.12.3beta.orig/configure.ac
+++ rezound-0.12.3beta/configure.ac
@@ -24,7 +24,7 @@
 # Process this file with autoconf to produce a configure script.
 
 AC_INIT(rezound, 0.12.3beta, [http://sourceforge.net/tracker/?atid=105056&group_id=5056])
-AC_CONFIG_AUX_DIR(config)
+AC_CONFIG_AUX_DIR([config])
 AM_CONFIG_HEADER([config/config.h])
 AM_INIT_AUTOMAKE
 AC_DISABLE_SHARED	# This makes libtool only build static libs 
Index: rezound-0.12.3beta/config/am_include.mk
===================================================================
--- rezound-0.12.3beta.orig/config/am_include.mk
+++ rezound-0.12.3beta/config/am_include.mk
@@ -35,19 +35,6 @@ INCLUDES=\
 	 -I$(top_srcdir)/src/PoolFile
 
 
-## CXXFLAGS is also automatically added to the $(CXX) macro at compile time, and
-## is passed down to the children as well
-## I list these here in case there needs to always be some flags passed to the compiler
-CXXFLAGS=@CXXFLAGS@
-
-
-## LDFLAGS will be added at link time
-## note that the -l flags are set by the LIBS macro that is set by configure
-## This line doesn't need to be here, configure will add LDFLAGS on it's own,
-## but I'm leaving here as a place to add LDFLAGS if so desired
-LDFLAGS=@LDFLAGS@
-
-
 # doc directory
 pkgdocdir=$(prefix)/doc/@PACKAGE@
 
