Index: rezound-0.12.3beta/config/m4/cxx-lib.m4
===================================================================
--- rezound-0.12.3beta.orig/config/m4/cxx-lib.m4
+++ rezound-0.12.3beta/config/m4/cxx-lib.m4
@@ -59,7 +59,7 @@ dnl
 dnl What a mess with the AS_TR_CPP($1); now if I knew how to assign a variable
 dnl at the top that I could use everwhere else I would
 dnl
-AC_DEFUN(ajv_CXX_CHECK_LIB,
+AC_DEFUN([ajv_CXX_CHECK_LIB],
 
 	dnl prepare to beable to AC_DEFINE the HAVE_LIBXXX #define
 	[AH_TEMPLATE(AS_TR_CPP(HAVE_LIB$1),[defined by $0])]
@@ -117,18 +117,15 @@ AC_DEFUN(ajv_CXX_CHECK_LIB,
 
 		saved_CPPFLAGS="$CPPFLAGS"
 		saved_CXXFLAGS="$CXXFLAGS"
-		saved_LDFLAGS="$LDFLAGS"
+		saved_LIBS="$LIBS"
 		saved_CXX="$CXX"
 
 		[CPPFLAGS="$CPPFLAGS $ajv_inc@&t@AS_TR_CPP($1)_path"]
 		[CXXFLAGS="$CXXFLAGS $ajv_inc@&t@AS_TR_CPP($1)_path"]
-		[LDFLAGS="$LDFLAGS $ajv_lib@&t@AS_TR_CPP($1)_path -l$1 $5"]
+		[LIBS="$LIBS $ajv_lib@&t@AS_TR_CPP($1)_path -l$1 $5"]
 		
 		builddir=`pwd`  # I'm assuming that the pwd will always be what $(top_builddir) will be set to in AC_OUTPUT
 
-		# HACK: force AC_LINK_IFELSE to use libtool to link so it will get FOX's dependancies.  Personally, I think it should do this anyway because I've used AC_PROG_LIBTOOL to request libtool as the linker for the subsequent make
-		CXX="$builddir/libtool --mode=link $CXX"
-
 		[AC_LINK_IFELSE([
 				#include <$3>
 				int main() { $2 xxx; return 0; }
@@ -187,7 +184,7 @@ AC_DEFUN(ajv_CXX_CHECK_LIB,
 
 		CPPFLAGS="$saved_CPPFLAGS"
 		CXXFLAGS="$saved_CXXFLAGS"
-		LDFLAGS="$saved_LDFLAGS"
+		LIBS="$saved_LIBS"
 		CXX="$saved_CXX"
 
 		[AC_LANG_POP(C++)]
