Index: rezound-0.12.3beta/src/backend/Remaster/CAdaptiveNormalizeAction.cpp
===================================================================
--- rezound-0.12.3beta.orig/src/backend/Remaster/CAdaptiveNormalizeAction.cpp
+++ rezound-0.12.3beta/src/backend/Remaster/CAdaptiveNormalizeAction.cpp
@@ -18,6 +18,8 @@
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA
  */
 
+#include <cstdio>
+
 #include "CAdaptiveNormalizeAction.h"
 
 #include <utility>
