Fix compilation with ABI=x32.
https://bugs.gentoo.org/420257

--- emacs-23.4-orig/lib-src/fakemail.c
+++ emacs-23.4/lib-src/fakemail.c
@@ -363,7 +363,7 @@
 make_file_preface ()
 {
   char *the_string, *temp;
-  long idiotic_interface;
+  time_t idiotic_interface;
   struct tm *tm;
   long prefix_length;
   long user_length;
--- emacs-23.4-orig/src/m/amdx86-64.h
+++ emacs-23.4/src/m/amdx86-64.h
@@ -30,8 +30,10 @@
    operating system this machine is likely to run.
    USUAL-OPSYS="linux"  */
 
+#ifndef __ILP32__               /* Don't define for x32 ABI.  */
 #define BITS_PER_LONG           64
 #define BITS_PER_EMACS_INT      64
+#endif
 
 /* Define WORDS_BIG_ENDIAN if lowest-numbered byte in a word
    is the most significant byte.  */
@@ -50,8 +52,10 @@
 /* __x86_64 defined automatically.  */
 
 /* Define the type to use.  */
+#ifndef __ILP32__
 #define EMACS_INT               long
 #define EMACS_UINT              unsigned long
+#endif
 
 /* Define EXPLICIT_SIGN_EXTEND if XINT must explicitly sign-extend
    the 24-bit bit field into an int.  In other words, if bit fields
