Fix linking on FreeBSD.
http://debbugs.gnu.org/12047

--- emacs-23.4-orig/src/s/freebsd.h
+++ emacs-23.4/src/s/freebsd.h
@@ -76,9 +76,9 @@
    there, contrary to what a stock GCC would do.  */
 
 #define LD_SWITCH_SYSTEM  -L/usr/local/lib
-#define START_FILES pre-crt0.o /usr/lib/crt1.o /usr/lib/crti.o /usr/lib/crtbegin.o
+#define START_FILES pre-crt0.o /usr/lib/crt1.o /usr/lib/crti.o
 #define UNEXEC unexelf.o
-#define LIB_STANDARD -lgcc -lc -lgcc /usr/lib/crtend.o /usr/lib/crtn.o
+#define LIB_STANDARD -lgcc -lc -lgcc /usr/lib/crtn.o
 #undef LIB_GCC
 #define LIB_GCC
 
