Index: documentation/samples/config
===================================================================
RCS file: /home/wine/wine/documentation/samples/config,v
retrieving revision 1.72
diff -u -B -r1.72 config
--- documentation/samples/config	21 Oct 2004 22:20:47 -0000	1.72
+++ documentation/samples/config	27 Oct 2004 18:04:57 -0000
@@ -5,12 +5,49 @@
 ;; bug report, filter out empty lines and comments with
 ;; grep -v "^;" ~/.wine/config | grep '.'
 
+[Drive C]
+"Path" = "fake_windows"
+"Type" = "hd"
+"Label" = "MS-DOS"
+"Filesystem" = "win95"
+
+;;[Drive D]
+;;"Path" = "/path/to/drive_d"
+;;"Type" = "hd"
+;;"Label" = "LABEL"
+;;"Filesystem" = "win95"
+
+;;[Drive E]
+;;"Path" = "/path/to/cdrom"
+;;"Type" = "cdrom"
+;;"Label" = "CD-Rom"
+;;"Filesystem" = "win95"
+
+[Drive F]
+"Path" = "/tmp"
+"Type" = "hd"
+"Label" = "Tmp Drive"
+"Filesystem" = "win95"
+
+[Drive H]
+"Path" = "../"
+"Type" = "network"
+"Label" = "Home"
+"Filesystem" = "win95"
+
+[Drive Z]
+"Path" = "/"
+"Type" = "network"
+"Label" = "Root"
+"Filesystem" = "win95"
+
+
 [wine]
-"Windows" = "c:\\windows"
-"System" = "c:\\windows\\system"
-"Temp" = "e:\\"
-"Path" = "c:\\windows;c:\\windows\\system;e:\\;e:\\test;f:\\"
-"Profile" = "c:\\windows\\Profiles\\Administrator"
+"Windows" = "c:\\Windows"
+"System" = "c:\\Windows\\System"
+"Temp" = "f:\\"
+"Path" = "c:\\Windows;c:\\Windows\\System;f:\\;h:\\;z:\\"
+"Profile" = "c:\\Windows\\Profiles\\Administrator"
 "GraphicsDriver" = "x11drv"
 ;"ShowDirSymlinks" = "1"
 ;"ShowDotFiles" = "1"
@@ -246,5 +287,90 @@
 ;; You can add an AppDefault entry like this for such cases.
 ;[AppDefaults\\pickygame.exe\\dsound]
 ;"EmulDriver" = "N"
+
+;; Kazaa Lite
+[AppDefaults\\KazaaLite.kpp\\DllOverrides]
+"*" = "builtin, native, so"
+;"shdoclc" = "native"
+"shdocvw" = "native"
+"shlwapi" = "native"
+"commctrl" = "native"
+;"comdlg32" = "native"
+"oleaut32" = "native"
+"ole32" = "native"
+
+[AppDefaults\\msimn.exe\\DllOverrides]
+; For ie6
+"*comctl32" = "builtin"
+; For Outlook97
+"mapi" = "native, builtin"
+"mapi32" = "native, builtin"
+; Native dlls needed for various parts of the install
+"ole32" = "native, builtin"
+"compobj" = "native, builtin"
+"ole2" = "native, builtin"
+"ole2nls" = "native, builtin"
+"ole2conv" = "native, builtin"
+"ole2prox" = "native, builtin"
+"ole2thk" = "native, builtin"
+"storage" = "native, builtin"
+"olepro32" = "native, builtin"
+"rpcrt4" = "native, builtin"
+"oleaut32" = "native, builtin"
+"msvcrt" = "native, builtin"
+"wininet" = "native, builtin"
+; for the ie runonce
+"setupapi" = "native, builtin"
+"devenum" = "native, builtin"
+"quartz" = "native, builtin"
+"urlmon" = "native"
+"shdocvw" = "native, builtin"
+"jscript" = "native, builtin"
+"wintrust" = "native, builtin"
+"shlwapi" = "native, builtin"
+"secur32" = "native, builtin"
+"crypt32" = "native, builtin"
+"ddraw" = "native, builtin"
+; allow launching user.exe (for Remedy)
+"*user.exe" = "native,builtin"
+; default for all other dlls
+"*" = "builtin, native"
+
+[AppDefaults\\IEXPLORE.EXE\\DllOverrides]
+; For ie6
+"*comctl32" = "builtin"
+; For Outlook97
+"mapi" = "native, builtin"
+"mapi32" = "native, builtin"
+; Native dlls needed for various parts of the install
+"ole32" = "native, builtin"
+"compobj" = "native, builtin"
+"ole2" = "native, builtin"
+"ole2nls" = "native, builtin"
+"ole2conv" = "native, builtin"
+"ole2prox" = "native, builtin"
+"ole2thk" = "native, builtin"
+"storage" = "native, builtin"
+"olepro32" = "native, builtin"
+"rpcrt4" = "native, builtin"
+"oleaut32" = "native, builtin"
+"msvcrt" = "native, builtin"
+"wininet" = "native, builtin"
+; for the ie runonce
+"setupapi" = "native, builtin"
+"devenum" = "native, builtin"
+"quartz" = "native, builtin"
+"urlmon" = "native"
+"shdocvw" = "native, builtin"
+"jscript" = "native, builtin"
+"wintrust" = "native, builtin"
+"shlwapi" = "native, builtin"
+"secur32" = "native, builtin"
+"crypt32" = "native, builtin"
+"ddraw" = "native, builtin"
+; allow launching user.exe (for Remedy)
+"*user.exe" = "native,builtin"
+; default for all other dlls
+"*" = "builtin, native"
 
 # [/wineconf]
