--- vnc-4_1_2-unixsrc/unix/xorg-server-1.1.1/GL/mesa/main/Makefile.am.mesa-652	2006-12-12 16:21:11.000000000 +0100
+++ vnc-4_1_2-unixsrc/unix/xorg-server-1.1.1/GL/mesa/main/Makefile.am	2006-12-12 16:21:11.000000000 +0100
@@ -62,6 +62,7 @@
                     light.c \
                     lines.c \
                     matrix.c \
+                    mipmap.c \
                     mm.c \
                     occlude.c \
                     pixel.c \
--- vnc-4_1_2-unixsrc/unix/xorg-server-1.1.1/GL/symlink-mesa.sh.mesa-652	2006-12-12 16:21:11.000000000 +0100
+++ vnc-4_1_2-unixsrc/unix/xorg-server-1.1.1/GL/symlink-mesa.sh	2006-12-12 16:24:51.000000000 +0100
@@ -158,6 +158,8 @@
     action macros.h
     action matrix.c
     action matrix.h
+    action mipmap.c
+    action mipmap.h
     action mm.c
     action mm.h
     action mtypes.h
@@ -469,6 +471,8 @@
     action slang_fragment_builtin_gc.h
     action slang_shader_syn.h
     action slang_pp_version_syn.h
+    action slang_pp_directives_syn.h
+    action slang_pp_expression_syn.h
     action slang_vertex_builtin_gc.h
 }        
 
@@ -550,6 +554,11 @@
 
     action glapi.c
     action glthread.c
+    action dispatch.h
+
+    src_dir src/mesa/main
+
+    action dispatch.c
 }
 
 #########
--- vnc-4_1_2-unixsrc/unix/xorg-server-1.1.1/GL/glx/Makefile.am.mesa-652	2006-07-05 20:31:36.000000000 +0200
+++ vnc-4_1_2-unixsrc/unix/xorg-server-1.1.1/GL/glx/Makefile.am	2006-12-12 16:21:11.000000000 +0100
@@ -30,7 +30,9 @@
 	-I$(top_srcdir)/mi
 
 
-nodist_libglx_la_SOURCES = indirect_size.h
+nodist_libglx_la_SOURCES = indirect_size.h \
+        dispatch.h \
+        dispatch.c
 
 libglxdri_la_SOURCES = \
         glxdri.c
