--- vnc-4_1_2-unixsrc/unix/xc/programs/Xserver/vnc/XserverDesktop.cc.render-sigfault	2006-05-15 18:56:20.000000000 +0200
+++ vnc-4_1_2-unixsrc/unix/xc/programs/Xserver/vnc/XserverDesktop.cc	2007-01-15 15:52:40.000000000 +0100
@@ -741,15 +741,15 @@
   } else {
     EntryPtr pent;
     pent = (EntryPtr)&cmap->red[index];
-    if (pent->fShared) {
-      *r = pent->co.shco.red->color;
-      *g = pent->co.shco.green->color;
-      *b = pent->co.shco.blue->color;
-    } else {
+    //if (pent->fShared) {
+    //  *r = pent->co.shco.red->color;
+    //  *g = pent->co.shco.green->color;
+    //  *b = pent->co.shco.blue->color;
+    //} else {
       *r = pent->co.local.red;
       *g = pent->co.local.green;
       *b = pent->co.local.blue;
-    }
+    //}
   }
 }
 
