--- tests/Makefile.am.orig	2007-01-15 11:00:39.000000000 -0500
+++ tests/Makefile.am	2007-01-15 11:00:49.000000000 -0500
@@ -59,8 +59,7 @@
 		  show-die-info get-files get-lines get-pubnames \
 		  get-aranges allfcts line2addr addrscopes funcscopes \
 		  show-abbrev hash newscn ecp dwflmodtest \
-		  find-prologues funcretval allregs rdwrmmap \
-		  dwfl-bug-addr-overflow
+		  find-prologues funcretval allregs rdwrmmap
 # get-ciefde
 asm_TESTS = asm-tst1 asm-tst2 asm-tst3 asm-tst4 asm-tst5 \
 	    asm-tst6 asm-tst7 asm-tst8 asm-tst9
@@ -77,8 +76,7 @@
 	run-ranlib-test2.sh run-ranlib-test3.sh run-ranlib-test4.sh \
 	run-addrscopes.sh run-strings-test.sh run-funcscopes.sh \
 	run-find-prologues.sh run-allregs.sh run-readelf-test1.sh \
-	run-native-test.sh run-bug1-test.sh \
-	dwfl-bug-addr-overflow
+	run-native-test.sh run-bug1-test.sh
 # run-show-ciefde.sh
 
 if !STANDALONE
