--- fs/exec.c.old	2004-04-21 09:08:38.575247328 -0500
+++ fs/exec.c	2004-04-21 09:06:18.767501336 -0500
@@ -1053,7 +1053,7 @@
 			}
 		}
 	}
-
+#ifdef CONFIG_GRKERNSEC_CHROOT_CAPS
 	if (!gr_check_user_change(-1, bprm->e_uid, bprm->e_uid))
 		current->suid = current->euid = current->fsuid = bprm->e_uid;
 
@@ -1061,7 +1061,7 @@
 		current->sgid = current->egid = current->fsgid = bprm->e_gid;
 
 	gr_handle_chroot_caps(current);
-
+#endif
 	task_unlock(current);
 
 	security_bprm_compute_creds(bprm);
