Index: vlc-0.8.5-test2/configure.ac
===================================================================
--- vlc-0.8.5-test2.orig/configure.ac
+++ vlc-0.8.5-test2/configure.ac
@@ -3854,6 +3854,8 @@ dnl  DirectFB module
 dnl
 AC_ARG_ENABLE(directfb,
   [  --enable-directfb       DirectFB support (default disabled)])
+AC_ARG_WITH([directfb],
+  AC_HELP_STRING([--with-directfb=dir], [Base directory for DirectFB]))
 if test "${enable_directfb}" = "yes"
 then
   if test "${with_directfb}" = "no"
