Index: vlc-0.8.5/modules/access/http.c
===================================================================
--- vlc-0.8.5.orig/modules/access/http.c
+++ vlc-0.8.5/modules/access/http.c
@@ -639,7 +639,8 @@ static int ReadICYMeta( access_t *p_acce
         p += strlen( "StreamTitle=" );
         if( *p == '\'' || *p == '"' )
         {
-            char *psz = strchr( &p[1], p[0] );
+	    char closing[] = { p[0], ';', '\0' };
+            char *psz = strstr( &p[1], closing );
             if( !psz )
                 psz = strchr( &p[1], ';' );
 
