Index: vlc-0.8.5/modules/codec/ffmpeg/ffmpeg.c
===================================================================
--- vlc-0.8.5.orig/modules/codec/ffmpeg/ffmpeg.c
+++ vlc-0.8.5/modules/codec/ffmpeg/ffmpeg.c
@@ -232,14 +232,6 @@ static int OpenDecoder( vlc_object_t *p_
         return VLC_EGENERIC;
     }
 
-    /* Bail out if buggy decoder */
-    if( i_codec_id == CODEC_ID_AAC )
-    {
-        msg_Dbg( p_dec, "refusing to use ffmpeg's (%s) decoder which is buggy",
-                 psz_namecodec );
-        return VLC_EGENERIC;
-    }
-
     /* Initialization must be done before avcodec_find_decoder() */
     E_(InitLibavcodec)(p_this);
 
